---
title: "20210322_HEK_stress_draft2"
author: "Chris Katanski"
date: "3/22/2021"
output: html_document
---



#Set up packages
```{r, set up packages etc., echo=FALSE, results="hide", warning=FALSE, message=FALSE}
#Improt packages for plotting and data manipulation.
library(ggplot2) #Praise be to Hadley
library(scales)
library(reshape2)
library(tidyr)
library(dplyr) #its important that this is after tidyr I think

library(grid) #for faceting figures
library(gridExtra) #more grid
library(ggh4x) #For nested faceting
library(ggpubr) #pables and p-values
library(errors) #for error propegation
library(gtools) #Combinations and permutations

#Reading in svg files into r for ggplotting
library(grImport2)
library(grConvert)
#Reading and plotting jpeg and PNG
library(jpeg)
library(png) #save hiRes png figures
library(svglite) #for saving SVG figures with arbitrary resolution

library(ggrepel) #The fanciest figure labels
library(stringr) #playig games with strings for labels
library(forcats) #Categorical variables need love too
library(readxl) #Read in excel data

#For nice log labels
#library(cat.extras)
#library(plotly)

library(extrafont) #We want the fonts
font_import() #Gotta get those fonts
loadfonts() #oowww!

#Set the global figure theme now
theme_set(
  theme_bw() + theme(#legend.position = "None", 
        text = element_text(family = "Arial", size = 14),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_rect(color = "black"),
        strip.background = element_blank())
  )

#Define a function that excludes things
`%nin%` = Negate(`%in%`)

empty_graph <- ggplot() + theme_void()

#Probably change for your own implementation of this script
setwd("~/4SR/data/20210125_HEK_stress_v2/source/")




AA_GROUP_LIST = c("Leu"="hydrophobic",
                        "Ile"="hydrophobic",
                        "Phe"="hydrophobic",
                        "Val"="hydrophobic",
                        "Trp"="hydrophobic",
                        "Mete"="hydrophobic",
                        "Meti"="hydrophobic",
                        "mtLeu"="hydrophobic",
                        "mtIle"="hydrophobic",
                        "mtPhe"="hydrophobic",
                        "mtVal"="hydrophobic",
                        "mtTrp"="hydrophobic",
                        "mtMet"="hydrophobic",
                        
                        "Ala"="small",
                        "Cys"="small",
                        "Pro"="small",
                        "Gly"="small",
                        "mtAla"="small",
                        "mtCys"="small",
                        "mtPro"="small",
                        "mtGly"="small",
                        
                        "Arg"="charged",
                        "Lys"="charged",
                        "Asp"="charged",
                        "Glu"="charged",
                        "mtArg"="charged",
                        "mtLys"="charged",
                        "mtAsp"="charged",
                        "mtGlu"="charged",
                        
                        "Ser"="polar",
                        "Thr"="polar",
                        "Tyr"="polar",
                        "His"="polar",
                        "Asn"="polar",
                        "Gln"="polar",
                        "SeCe"="polar",
                        "mtSer"="polar",
                        "mtThr"="polar",
                        "mtTyr"="polar",
                        "mtHis"="polar",
                        "mtAsn"="polar",
                        "mtGln"="polar",
                        "mtSeCe"="polar")
```

#_
#=====================
#Main figures
#=====================
#_

#Figure 2
##preamble


###LongRT data (tRNA reference only)
####Base
```{r}
#Point to the processed data and get a list of files
PATH="../20210125_new_data/updated_tRNA_reference/SE_map/5_tsv/human_combine/"
FILES <-data.frame( file_name = list.files(PATH) ) %>%
  filter(!grepl("unassigned",file_name))

#Split generic file names into salient parts
FILES <- FILES %>%
  separate(file_name, sep="_", c("library","junk1", "junk2", "barcode", "junk4", "bin_start", "bin_stop", "junk3"), 
           fill="right", remove=FALSE) %>%
  select(-junk1, -junk2, -junk3, -junk4) 

FILES <- FILES %>%
  mutate(treatment = barcode,
         sample = library) %>%
  select(-library, -barcode)

#PROJECT SPECIFIC DECODING
FILES$file_name <- as.character(FILES$file_name)
FILES$treatment <- recode(FILES$treatment, 
                          "bc1"="control_1", "bc2"="control_2", "bc3"="control_3",
                          "bc4"="heat_1", "bc5"="heat_2", "bc6"="heat_3",
                          "bc7"="H2O2_1", "bc8"="H2O2_2", "bc9"="H2O2_3",
                          "bc10"="AsO2_1", "bc11"="AsO2_2", "bc12"="AsO2_3")
FILES$sample <- recode(FILES$sample, "TP-CK-4S-CW1"="input", "TP-CK-4S-CW2"="polysome")
FILES <- FILES %>%
  separate(treatment, sep="_", into=c("treatment", "rep"))

#=======================+
#For fragment analysis, bins are specified in the file name. Except for the all reads bin, which gets designation "-3"
#Note: bin "-1" is all antisense reads; "-2" is all sense reads. 
FILES <- FILES %>%
  mutate(bin_start = ifelse(is.na(bin_start), -3, bin_start),
         bin_stop = ifelse(is.na(bin_stop), -3, bin_stop))
#========================

#Specific to this dataset
FILES <- FILES %>%
  mutate(DM="longRT")

#A function to read in files, designated from the list of files, and attach the project specific identifiers. 
read_in_one <- function(row){
  output <- read.csv(paste0(PATH, row$file_name), header=T, sep="\t") %>%
    mutate(sample = row$sample,
           DM = row$DM,
           rep = row$rep,
           treatment = row$treatment,
           bin_start = row$bin_start,
           bin_stop  = row$bin_stop,)
  #specify data types since enpty dataframes get confused and sad
  output$gene <- as.character(output$gene)
  output$base <- as.character(output$base)
  return(output)
}

new_SE_data_HEK_FULL_base <- FILES %>%
  group_by(file_name) %>%
  do(read_in_one(.)) %>%
  ungroup()
```

#####Extra processing
```{r}
#Remove select snoRNAs, rRNAs, and control RNAs from subsequent analysis
temp <- new_SE_data_HEK_FULL_base %>%
  filter(gene %nin% c("NR_004394.1",
                      "NR_002716.3",
                      "NR_003925.1",
                      "NR_004430.2",
                      "NR_002756.2",
                      "NR_004391.1",
                      "NR_004392.1",
                      "NR_004393.1",
                      "NR_001571.2",
                      "Homo_sapiens_chrX.rRNA-5SR5SR",
                      "Homo_sapiens_chrX.rRNA-58S58S",
                      "Ecoli_Tyr",
                      "Yeast_Phe",
                      "Ecoli_Lys",
                      "spikein_SCC1",
                      "spikein_SCCA1"))

#fix mitochondrial names
#Then rearrange all names
temp <- temp %>%
  mutate(source="cytosolic")
temp <- temp %>%
  filter(grepl("mt", gene)) %>%
  mutate(gene = paste0("Homo_sapiens_chrMT.trna1-", gene)) %>%
  mutate(source="mitochondrial") %>%
  rbind(filter(temp, !grepl("mt",gene))) %>%
  select(-file_name)

#Simplify names
temp <- temp%>%
  mutate(gene2 = gene) %>%
  separate(gene2, sep="_", c("homo","sapiens","info"))%>%
  select(-homo, -sapiens)


temp <- temp%>%
  separate(info, sep="-", c("chr","AA")) %>%
  separate(AA, sep= -3, c("AA", "anticodon")) 

#Rename Met genes as elongator or initiator based on 5' A/C or G respectively
temp_iMet <- temp %>%
  filter(AA =="Met" ,
         bin_start=="-3") %>%
  #filter(position==1) %>%
  mutate(gene_symbol2 = ifelse(gene %in% c("Homo_sapiens_chr6.trna61-MetCAT",
                                           "Homo_sapiens_chr1.trna32-MetCAT",
                                           "Homo_sapiens_chr9.trna1-MetCAT"), 
                               paste0(AA, "i"), paste0(AA,"e")) ) %>%
  #https://www.ncbi.nlm.nih.gov/pmc/articles/PMC108860/#:~:text=A%20special%20methionine%20tRNA%20is,linkages%20(33%2C%2048).
  group_by(gene) %>%
  summarise(gene_symbol2 = gene_symbol2[1] )

#Resume fMet filtering
temp <- temp %>%
  full_join(., temp_iMet, by=c("gene")) %>%
  mutate(AA = ifelse(is.na(gene_symbol2), AA, gene_symbol2) ) %>%
  select(-gene_symbol2) 

temp <- temp %>%
  filter(AA %nin% c("Undet", "Sup"))

new_SE_HEK_FULL_base <- temp %>%
  mutate(DM="longRT") %>%
  filter(!grepl("AsnATT", gene),
         !grepl("SerACC", gene),
         !grepl("CysACA", gene),
         !grepl("TyrATA", gene),
         !(anticodon=="GAT" & AA=="Ile" & source=="cytosolic"),
         !grepl("SeCTCA", gene),
         !grepl("SerACT", gene))

```

####counts
```{r}
PATH="../20210125_new_data/updated_tRNA_reference/SE_map/6_sam_counter/human_combine/"
FILES <-data.frame( file_name = list.files(PATH) ) %>%
  filter(!grepl("unassigned", file_name),
         grepl("remove", file_name),
         file_name %nin% c("kallisto.out"))

#Split generic file names into salient parts
FILES <- FILES %>%
  separate(file_name, sep="_", c("library","junk1", "junk4","barcode", "junk2", "bin_start", "bin_stop", "junk3"), fill="right", remove=FALSE) %>%
  select(-junk1, -junk2, -junk3, -junk4) 

FILES <- FILES %>%
  mutate(treatment = barcode,
         sample = library) %>%
  select(-library, -barcode)

#PROJECT SPECIFIC DECODING
FILES$file_name <- as.character(FILES$file_name)
FILES$treatment <- recode(FILES$treatment, 
                          "bc1"="control_1", "bc2"="control_2", "bc3"="control_3",
                          "bc4"="heat_1", "bc5"="heat_2", "bc6"="heat_3",
                          "bc7"="H2O2_1", "bc8"="H2O2_2", "bc9"="H2O2_3",
                          "bc10"="AsO2_1", "bc11"="AsO2_2", "bc12"="AsO2_3")
FILES$sample <- recode(FILES$sample, "TP-CK-4S-CW1"="input", "TP-CK-4S-CW2"="polysome")
FILES <- FILES %>%
  separate(treatment, sep="_", into=c("treatment", "rep"))

#=======================+
FILES <- FILES %>%
  mutate(bin_start = ifelse(is.na(bin_start), -3, bin_start),
         bin_stop = ifelse(is.na(bin_stop), -3, bin_stop))
#========================

read_in_one <- function(row){
  output <- read.csv(paste0(PATH, row$file_name), header=T, sep="\t") %>%
    mutate(sample = row$sample,
           rep = row$rep,
           treatment = row$treatment,
           bin_start = row$bin_start,
           bin_stop  = row$bin_stop,)
  #specify data types since enpty dataframes get confused and sad
  output$name <- as.character(output$name)
  output <- output %>%
    mutate(gene=name) %>%
    select(-name)
  return(output)
}

new_SE_data_HEK_FULL_count <- FILES %>%
  group_by(file_name) %>%
  do(read_in_one(.)) %>%
  ungroup()
```


#####Extra processing
```{r}
temp <- new_SE_data_HEK_FULL_count %>%
  filter(gene %nin% c("NR_004394.1",
                      "NR_002716.3",
                      "NR_003925.1",
                      "NR_004430.2",
                      "NR_002756.2",
                      "NR_004391.1",
                      "NR_004392.1",
                      "NR_004393.1",
                      "NR_001571.2",
                      "Homo_sapiens_chrX.rRNA-5SR5SR",
                      "Homo_sapiens_chrX.rRNA-58S58S",
                      "Ecoli_Tyr",
                      "Yeast_Phe",
                      "Ecoli_Lys",
                      "spikein_SCC1",
                      "spikein_SCCA1"))

#fix mitochondrial names
#Then rearrange all names
temp <- temp %>%
  mutate(source="cytosolic")

temp <- temp %>%
  filter(grepl("mt", gene)) %>%
  mutate(gene = paste0("Homo_sapiens_chrMT.trna1-", gene)) %>%
  mutate(source="mitochondrial") %>%
  rbind(filter(temp, !grepl("mt",gene))) %>%
  select(-file_name)


temp <- temp%>%
  mutate(gene2 = gene) %>%
  separate(gene2, sep="_", c("homo","sapiens","info"))%>%
  select(-homo, -sapiens)

temp <- temp%>%
  separate(info, sep="-", c("chr","AA")) %>%
  separate(AA, sep= -3, c("AA", "anticodon")) 

#Resume fMet filtering
temp <- temp %>%
  full_join(., temp_iMet, by=c("gene")) %>%
  mutate(AA = ifelse(is.na(gene_symbol2), AA, gene_symbol2) ) %>%
  select(-gene_symbol2) 

temp <- temp %>%
  filter(AA %nin% c("Undet", "Sup")) 

new_SE_HEK_FULL_count <- temp %>%
  mutate(DM="longRT") %>%
  filter(!grepl("AsnATT", gene),
         !grepl("SerACC", gene),
         !grepl("CysACA", gene),
         !grepl("TyrATA", gene),
         !(anticodon=="GAT" & AA=="Ile" & source=="cytosolic"),
         !grepl("SeCTCA", gene),
         !grepl("SerACT", gene))

```

##Fig 2A - piechart
###Read in full transcriptome mapped data
```{r}
#Read in the full transcriptome data
PATH="../20210125_new_data/updated_tRNA_reference/SE_map/6_sam_counter/human_combine/"
FILES <-data.frame( file_name = list.files(PATH) ) %>%
  filter(!grepl("unassigned", file_name),
         !grepl("remove", file_name), #This gets the full data, not just the tRNA-only data
         file_name %nin% c("kallisto.out"))

#Split generic file names into salient parts
FILES <- FILES %>%
  separate(file_name, sep="_", c("library","junk1", "junk4","barcode", "junk2", "bin_start", "bin_stop", "junk3"), fill="right", remove=FALSE) %>%
  select(-junk1, -junk2, -junk3, -junk4) 

FILES <- FILES %>%
  mutate(treatment = barcode,
         sample = library) %>%
  select(-library, -barcode)

#PROJECT SPECIFIC DECODING
FILES$file_name <- as.character(FILES$file_name)
FILES$treatment <- recode(FILES$treatment, 
                          "bc1"="control_1", "bc2"="control_2", "bc3"="control_3",
                          "bc4"="heat_1", "bc5"="heat_2", "bc6"="heat_3",
                          "bc7"="H2O2_1", "bc8"="H2O2_2", "bc9"="H2O2_3",
                          "bc10"="AsO2_1", "bc11"="AsO2_2", "bc12"="AsO2_3")
FILES$sample <- recode(FILES$sample, "TP-CK-4S-CW1"="input", "TP-CK-4S-CW2"="polysome")
FILES <- FILES %>%
  separate(treatment, sep="_", into=c("treatment", "rep"))

#=======================+
FILES <- FILES %>%
  mutate(bin_start = ifelse(is.na(bin_start), -3, bin_start),
         bin_stop = ifelse(is.na(bin_stop), -3, bin_stop))
#========================

read_in_one <- function(row){
  output <- read.csv(paste0(PATH, row$file_name), header=T, sep="\t") %>%
    mutate(sample = row$sample,
           rep = row$rep,
           treatment = row$treatment,
           bin_start = row$bin_start,
           bin_stop  = row$bin_stop,)
  #specify data types since enpty dataframes get confused and sad
  output$name <- as.character(output$name)
  output <- output %>%
    mutate(gene=name) %>%
    select(-name)
  return(output)
}

temp <- FILES %>%
  group_by(file_name) %>%
  do(read_in_one(.)) %>%
  ungroup()




temp2 <- read.csv("./human_combine_name_dict.txt",sep="\t")
temp3 <- temp2 %>%
  filter(gene_biotype=="tRNA") %>%
  filter(grepl("mt", gene_symbol)) %>%
  mutate(gene_biotype="mt_tRNA") 
temp2 <- temp2 %>%
  filter(gene_biotype=="tRNA") %>%
  filter(!grepl("mt", gene_symbol)) %>%
  filter(gene_symbol %nin% c("spikein_SCC1","spikein_SCCA1","spikein_SCCA2",
                             "spikein_SCCA3","Ecoli_Lys","Ecoli_Tyr", "Yeast_Phe")) %>%
  rbind(.,temp3) %>%
  rbind(.,filter(temp2, gene_symbol!="tRNA"))

new_PE_data_HEK_full_transcriptome_count <- temp %>%
  filter(bin_start == "-3") %>%
  full_join(., temp2, by=c("gene"))




```

###Stacked bar graph
```{r}

temp <- new_PE_data_HEK_full_transcriptome_count %>%
  group_by(sample, rep, treatment, gene_symbol, gene_biotype) %>%
  summarise(count = sum(count, na.rm=T)) %>%
  filter(count > 0)
temp$gene_biotype = as.factor(temp$gene_biotype)

temp$treatment = factor(temp$treatment, levels=c("control","heat","H2O2","AsO2"))

#Tao's figure edits
temp <- temp %>%
  filter(gene_biotype != "lncRNA")
list_of_other_types <- c("IG_C_gene", "IG_D_gene","IG_J_gene",
                         "IG_V_gene","misc_RNA","polymorphic_pseudogene",
                         "ribozyme","TR_C_gene","TR_D_gene","TR_J_gene",
                         "TR_V_gene", "scaRNA","scRNA","sRNA","Mt_rRNA",
                         "vaultRNA", "miRNA", "snoRNA")
temp <- temp %>%
  filter(gene_biotype %in% list_of_other_types) %>%
  mutate(gene_biotype="other") %>%
  rbind(filter(temp, gene_biotype %nin% list_of_other_types))
temp$gene_biotype <- recode(temp$gene_biotype, "mt_tRNA"="mt-tRNA",
                            "other"="other","protein_coding"="mRNA","rRNA"="rRNA",
                            "snRNA"="snRNA","tRNA"="tRNA")
temp$gene_biotype <- factor(temp$gene_biotype,
                            levels=rev(c("tRNA","rRNA","mRNA","mt-tRNA","snRNA","other")))

#Make pretty label for figure
temp <- temp %>%
  mutate(x_label=paste0(treatment, " ", "rep"," ", rep))
temp <- temp[order( temp$treatment, temp$rep ),]
temp$x_label <- factor(temp$x_label, levels = unique(temp$x_label))




plot1 <- ggplot(filter(temp, sample=="input"), 
       aes(x= x_label, y=count,fill=gene_biotype)) +
  geom_bar(position="stack", stat="identity") +
  scale_y_continuous(breaks=c(5000000,10000000, 15000000, 20000000,30000000,40000000)  ) +
  scale_fill_manual(values = c("tRNA"="#99CCFF", "rRNA"="#CC99CC", "mRNA"="#FFCC33",
                                "other"="grey", "mt-tRNA"="#99CC99", "snRNA"="#CC6666"),
                    guide = guide_legend(reverse = TRUE)) +
  ylab("Read count") +
  labs(fill="Gene type") +
  # facet_wrap(~sample) +
  theme(legend.position = "none",
        strip.background = element_blank(),
        axis.text.x = element_text(angle=90, hjust=1, vjust=0.5),
        axis.title.x = element_blank())

# plot2 <- ggplot(filter(temp, sample=="polysome"), 
#        aes(x=interaction(rep, treatment),y=count,fill=gene_biotype)) +
#   geom_bar(position="stack", stat="identity") +
#   scale_y_continuous(breaks=c(2000000,4000000,6000000) ) +
#   scale_fill_manual(values = c("tRNA"="#99CCFF", "rRNA"="#CC99CC", "mRNA"="#FFCC33",
#                                 "other"="grey", "mt-tRNA"="#99CC99", "snRNA"="#CC6666"),
#                     guide = guide_legend(reverse = TRUE)) +
#   ylab("") +
#   labs(fill="Gene type") +
#   facet_wrap(~sample) +
#   theme(legend.position = "none",
#         strip.background = element_blank(),
#         axis.text.x = element_text(angle=90, hjust=1, vjust=0.5),
#         axis.title.x = element_blank())

figure_legend <- get_legend(plot1+theme(legend.position = "right",
                                       legend.background = element_blank()) )


layout <- rbind(c(1,1,1,3))
figure <- grid.arrange(plot1, figure_legend,  layout_matrix = layout)

path="../figures/draft2/"
figure_name="panel_2A_total_mapping_stacked_bars"
width=3
height=3
scale=1.5
dpi=600
# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```



##Fig 2B - Bulk abundance change; isoaccepter

###Work with counts data, make the plot
```{r}
#Start with the tRNA-only counts data
temp <- new_SE_HEK_FULL_count %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, sample, anticodon, AA, DM, rep) %>%
  summarise(sum_count = sum(count) ) %>%
  group_by(treatment, sample, DM, rep) %>%
  mutate(rpm = sum_count / sum(sum_count)) %>% #Note; rpm isn't rpm. Its reads per mapped tRNA read.
  filter(rpm >= 0.001) %>%
  group_by(treatment, sample, anticodon, AA, DM) %>%
  summarise(mean_rpm = mean(rpm, na.rm=T))

#Add a column for mean abundance of control
temp <- temp %>%
  filter(treatment == "control") %>%
  group_by() %>%
  mutate(control_rpm_mean = mean_rpm) %>%
  select(sample, anticodon, AA, DM, control_rpm_mean)  %>%
  full_join(temp, ., by=c("sample","anticodon", "AA","DM") )

#Beautify labels
temp$treatment <- factor(temp$treatment, levels=c("heat", "H2O2", "AsO2"))

#label outlier points; semi-manually
# temp <- temp %>%
#   mutate(point_label=NA)
temp1 <- temp %>%
  filter(   (treatment=="heat" & mean_rpm / control_rpm_mean <=0.85) |
            #(treatment=="H2O2" & mean_rpm / control_rpm_mean <=0) |
            (treatment=="AsO2" & (mean_rpm / control_rpm_mean <=0.9 | mean_rpm / control_rpm_mean >= 1.12)) ) %>%
  mutate(point_label=paste0(AA, anticodon)) %>%
  select(treatment, sample, AA, anticodon, DM, point_label)
temp <- temp %>%
  full_join(.,temp1, by=c("treatment","sample","AA","anticodon","DM"))

#Make the figure
plot1 <- ggplot(filter(temp,
              sample=="input",
              !grepl("mt", AA) ,
              treatment !="control"
              ),
       aes(x=treatment, y= mean_rpm / control_rpm_mean )) +
  geom_hline(yintercept = 1, linetype=2, alpha=0.3) +
  geom_boxplot() +
  geom_text_repel(aes(label=point_label)) +
  scale_y_log10(breaks=c(0.8,1,1.25), limits=c(0.74,1.35)) + #Maybe indicate that its a log scale in the label
  ylab("tRNA abundance change (stress / control)") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14, color="black"),
        axis.title.y = element_text(size=16),
        axis.text.y = element_text(size=12, color="black"))

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_2B_total_isoacceptor_abundance"
width=2
height=3
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```

###Write table S1
```{r}
#Start with the tRNA-only counts data
tempS <- new_SE_HEK_FULL_count %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, sample, anticodon, AA, DM, rep) %>%
  summarise(sum_count = sum(count) ) %>% #Pool iso acceptors
  group_by(treatment, sample, DM, rep) %>%
  mutate(rpm = sum_count / sum(sum_count)) #%>% #Note; rpm isn't rpm. Its reads per mapped tRNA read.
  # filter(rpm >= 0.001)

asdf2 <- tempS %>%
  ungroup() %>%
  filter(sample=="input") %>%
  select(-DM) %>%
  group_by(treatment, anticodon, AA) %>%
  pivot_wider(names_from=c("rep"), values_from=c("rpm", "sum_count") )

write.table(asdf2,
            file='/Users/ckatanski/4SR/data/20210125_HEK_stress_v2/figures/draft2/supplementry_tables/table_S1.csv',
            quote=FALSE, sep=',', col.names = NA)

```


##Fig 2C - Bulk charging change; iso acceptor
###Calculate charging, and sum
```{r}
#Filter for abundance =======================================================
abundance_list <- new_SE_HEK_FULL_count %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, sample, anticodon, AA, DM, rep) %>%
  summarise(sum_count = sum(count) ) %>%
  group_by(treatment, sample, DM, rep) %>%
  mutate(rpm = sum_count / sum(sum_count)) %>%
  filter(rpm >= 0.001) %>%
  group_by(treatment, sample, anticodon, AA, DM) %>%
  summarise(mean_rpm = mean(rpm, na.rm=T)) %>%
  select(sample, treatment, DM, anticodon, AA, mean_rpm)


#Work with the base-wise data, calculate charging for each gene
temp <- new_SE_HEK_FULL_base %>%
  select(-bin_stop) %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, rep, bin_start, DM, sample, source, gene, AA, anticodon) %>%
  mutate(max_position = max(position)) %>%
  filter(position %in% c(max_position, max_position-1)) %>%
  mutate(charging = ifelse(position==max_position, "end", "penultimate")) %>%
  filter((charging=="end" & base=="A")  |
         (charging=="penultimate" & base=="C")) %>%
  select(treatment, rep, bin_start, DM, sample, source, gene, pileup, charging, AA, anticodon) %>%
  group_by(treatment, rep, bin_start, DM, sample, source, gene, anticodon, AA) %>%
  pivot_wider(names_from = c("charging"), values_from = c("pileup")) %>%
  mutate(charging= end/penultimate)

#Apply abundance filter (don't calculate charging on spurious low read genes)
tem <- temp %>%
  full_join(., abundance_list, by=c("sample","treatment","DM","anticodon","AA")) %>%
  filter(mean_rpm >= 0.001)

#Sum up isoacceptors (sum of penultimate, sum up ultimate, ratio)
temp <- temp %>%
  group_by(sample, source, treatment, DM, rep, AA, anticodon) %>%
  summarise(penultimate = sum(penultimate, na.rm=T),
            end = sum(end, na.rm=T)) %>%
  mutate(charging= end/penultimate)

#Average replicates
temp <- temp %>%
  group_by(sample, source, treatment, DM, AA, anticodon) %>%
  summarise(charging = mean(charging, na.rm=T))

#Normalize to average of control
temp <- temp %>%
  filter(treatment=="control") %>%
  group_by(AA, anticodon, sample) %>%
  summarise(control_charging_average = mean(charging, na.rm=T)) %>%
  select(sample, AA, anticodon, control_charging_average) %>%
  full_join(temp, ., by=c("AA","anticodon","sample"))



#Beautify labels
temp$treatment <- factor(temp$treatment, levels=c("heat", "H2O2", "AsO2"))

#label outlier points; semi-manually
# temp <- temp %>%
#   mutate(point_label=NA)
temp1 <- temp %>%
  filter(   (treatment=="heat" & (charging / control_charging_average <= 0.92 | charging / control_charging_average >= 1.05)) |
            (treatment=="H2O2" & charging / control_charging_average <= 0.94) |
            (treatment=="AsO2" & (charging / control_charging_average <=0.95 | charging / control_charging_average >= 1.03)) ) %>%
  mutate(point_label=paste0(AA, anticodon)) %>%
  select(treatment, sample, AA, anticodon, DM, point_label)
temp <- temp %>%
  full_join(.,temp1, by=c("treatment","sample","AA","anticodon","DM"))



plot1 <- ggplot(filter(temp,
              treatment !="control",
              !grepl("mt", AA),
              sample=="input"),
       aes(x=treatment, y=charging / control_charging_average)) +
  geom_hline(yintercept = 1, linetype=2, alpha=0.4) +
  geom_boxplot() +
  geom_text_repel(aes(label=point_label)) +
  scale_y_log10(breaks=c(0.8,1,1.25), limits=c(0.74,1.35)) + #Maybe indicate that its a log scale in the label
  ylab("tRNA charging change (stress / control)") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14, color="black"),
        axis.title.y = element_text(size=16),
        axis.text.y = element_text(size=12, color="black"))

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_2C_total_isoacceptor_charging"
width=2
height=3
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```

###Write table S2
```{r}
#Filter for abundance =======================================================
abundance_list <- new_SE_HEK_FULL_count %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, sample, anticodon, AA, DM, rep) %>%
  summarise(sum_count = sum(count) ) %>%
  group_by(treatment, sample, DM, rep) %>%
  mutate(rpm = sum_count / sum(sum_count)) %>%
  # filter(rpm >= 0.001) %>%
  group_by(treatment, sample, anticodon, AA, DM) %>%
  summarise(mean_rpm = mean(rpm, na.rm=T)) %>%
  select(sample, treatment, DM, anticodon, AA, mean_rpm)


#Work with the base-wise data, calculate charging for each gene
tempS <- new_SE_HEK_FULL_base %>%
  select(-bin_stop) %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, rep, bin_start, DM, sample, source, gene, AA, anticodon) %>%
  mutate(max_position = max(position)) %>%
  filter(position %in% c(max_position, max_position-1)) %>%
  mutate(charging = ifelse(position==max_position, "end", "penultimate")) %>%
  filter((charging=="end" & base=="A")  |
         (charging=="penultimate" & base=="C")) %>%
  select(treatment, rep, bin_start, DM, sample, source, gene, pileup, charging, AA, anticodon) %>%
  group_by(treatment, rep, bin_start, DM, sample, source, gene, anticodon, AA) %>%
  pivot_wider(names_from = c("charging"), values_from = c("pileup")) %>%
  mutate(charging= end/penultimate)

#Apply abundance filter (don't calculate charging on spurious low read genes)
temS <- tempS %>%
  full_join(., abundance_list, by=c("sample","treatment","DM","anticodon","AA")) %>%
  filter(mean_rpm >= 0.001)

#Sum up isoacceptors (sum of penultimate, sum up ultimate, ratio)
tempS <- tempS %>%
  group_by(sample, source, treatment, DM, rep, AA, anticodon) %>%
  summarise(penultimate = sum(penultimate, na.rm=T),
            end = sum(end, na.rm=T)) %>%
  mutate(charging= end/penultimate)
tempS$rep <- recode(tempS$rep, "1"="rep_1_charging", "2"="rep_2_charging", "3"="rep_3_charging")

asdf2 <- tempS %>%
  ungroup() %>%
  filter(sample=="input") %>%
  select(-DM, -penultimate, -end) %>%
  group_by(treatment, anticodon, AA) %>%
  pivot_wider(names_from=c("rep"), values_from=c("charging") )

write.table(asdf2,
            file='/Users/ckatanski/4SR/data/20210125_HEK_stress_v2/figures/draft2/supplementry_tables/table_S2.csv',
            quote=FALSE, sep=',', col.names = NA)

```


##Fig 2D - m3C47D SerGCT trace
###Add up all GCT isodecoder mutation rate
```{r}

temp <- new_SE_HEK_FULL_base %>%
  filter(AA=="Ser",
         anticodon =="GCT",
         source=="cytosolic",
         treatment=="control",
         bin_start=="-2",
         sample=="input") %>%
  filter(!grepl("chr11.trna8", gene)) #This one has T 47d


#5 isodecoders remain. 4 of them have m3C47d at nucleotide 50. One is at 52. I will shift the position information over for that one gene
temp <- temp %>%
  filter(grepl("chr6.trna175", gene)) %>%
  mutate(position = position -2) %>%
  rbind(filter(temp, !grepl("chr6.trna175",gene)))

#Sum isodecoders and calculate a new mutation rate (weighted by abundance)
temp <- temp %>%
  group_by(position, rep) %>%
  mutate(t_nucleotide = T) %>% #"T" is 'true' in R. best to change the name to avoid confustion
  mutate(mutated_reads = pileup * mutation) %>%
  summarise(mutated_reads = sum(mutated_reads, na.rm=T),
            pileup = sum(pileup, na.rm=T)) %>%
  mutate(mutation = mutated_reads / pileup)


plot1 <- ggplot(temp, aes(x=position, y=mutation, group=interaction(rep))) +
  geom_line(color="#FFAD0A") +
  coord_cartesian(xlim = c(40,60), ylim=c(0,0.8)) +
  ylab("Mutation rate") +
  theme(axis.title.x = element_text(size=12),
        axis.text.x = element_text(size=8, color="black"),
        axis.title.y = element_text(size=12),
        axis.text.y = element_text(size=8, color="black"))

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_2D_SerGCT_trace"
width=2
height=1
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```


##Fig 2E - m3C47D change in stress
###Wrangle basewise data and plot
```{r}
#Ser m3C47
#C at m3C47
Ser_C_list <- c("Homo_sapiens_chr6.trna172-SerTGA",
"Homo_sapiens_chr10.trna2-SerTGA",
"Homo_sapiens_chr6.trna145-SerAGA",
"Homo_sapiens_chr17.trna35-SerAGA",
"Homo_sapiens_chr6.trna35-SerCGA",
"Homo_sapiens_chr17.trna41-SerCGA",
"Homo_sapiens_chr6.trna175-SerGCT",
"Homo_sapiens_chr6.trna62-SerGCT",
"Homo_sapiens_chr15.trna10-SerGCT",
"Homo_sapiens_chr6.trna43-SerGCT",
"Homo_sapiens_chr6.trna31-SerGCT")

#T at m3C47
Ser_T_list <- c("Homo_sapiens_chr6.trna148-SerTGA",
"Homo_sapiens_chr6.trna51-SerTGA",
"Homo_sapiens_chr6.trna50-SerAGA",
"Homo_sapiens_chr6.trna147-SerAGA",
"Homo_sapiens_chr12.trna2-SerCGA",
"Homo_sapiens_chr6.trna137-SerCGA",
"Homo_sapiens_chr11.trna8-SerGCT")


temp <- new_SE_HEK_FULL_base %>%
  filter(pileup >1000) %>%
  select(-bin_stop) %>%
  filter(bin_start==-2) %>%
  filter((AA=="Ser")|(AA=="Leu" & anticodon=="CAG") )

temp <- temp %>%
  mutate(m3C47 = ifelse(gene %in%Ser_C_list,"C",ifelse(gene %in% Ser_T_list,"T","C"))) 

#Beautify labels
temp$treatment <- factor(temp$treatment, levels=rev(c("control","AsO2","H2O2","heat")))
temp$chr <- str_replace_all(temp$chr, "chr","c")
temp$chr <- str_replace_all(temp$chr, "trna","t")
temp <- temp %>%
  filter(treatment =="control") %>%
  group_by(sample, gene, position, DM) %>%
  summarise(control_mean_mut = mean(mutation, na.rm=T)) %>%
  select(sample, gene, position, DM, control_mean_mut) %>%
  full_join(temp, ., by=c("sample","gene","DM","position")) %>%
  mutate(gene_label=paste0(chr,"   ",AA,anticodon) )
#Re order
temp <- temp[order( temp$AA, temp$anticodon, temp$chr ),]
temp$gene_label <- factor(temp$gene_label, levels = unique(temp$gene_label))


# 
# plot1 <- ggplot(filter(temp,
#               (AA=="Ser")|(anticodon=="CAG"),
#               m3C47=="C",
#               treatment !="control",
#               position >=45,
#               position <=56,
#               sample=="input"),
#        aes(x=position, y=mutation - control_mean_mut, 
#            group=interaction(treatment, rep))) +
#   geom_line(color="black") +
#   scale_y_continuous(breaks=c(-0.1,0,0.1))+
#   scale_x_continuous(breaks=c(45,50,55)) +
#   ylab("Δ mutation rate (stress - control)") +
#   # geom_point(size=0.75, color="black")+
#   facet_grid(rows = vars(gene_label),
#              cols = vars(treatment)) +
#   theme(legend.position = "right",
#         strip.background = element_blank(),
#         strip.text.y = element_text(angle=0),
#         strip.text.x = element_text(size=14),
#         axis.text.y = element_blank(),
#         axis.title = element_text(size=16))

#Save the figure
# layout <- rbind(c(1))
# figure <- grid.arrange(plot1,  layout_matrix = layout)
# path="../figures/draft2/"
# figure_name="panel_2E_m3C47_traces"
# width=4
# height=5
# scale=1.5
# dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

# ggsave(paste0(path, figure_name,".png"),
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width,
#        height = height)
```


##Fig 2E - m3C32 AND m3C47
###Separate A's and T's and normalize by stress
```{r}
Ser_C_list <- c("Homo_sapiens_chr6.trna172-SerTGA",
"Homo_sapiens_chr10.trna2-SerTGA",
"Homo_sapiens_chr6.trna145-SerAGA",
"Homo_sapiens_chr17.trna35-SerAGA",
"Homo_sapiens_chr6.trna35-SerCGA",
"Homo_sapiens_chr17.trna41-SerCGA",
"Homo_sapiens_chr6.trna175-SerGCT",
"Homo_sapiens_chr6.trna62-SerGCT",
"Homo_sapiens_chr15.trna10-SerGCT",
"Homo_sapiens_chr6.trna43-SerGCT",
"Homo_sapiens_chr6.trna31-SerGCT")

#T at m3C47
Ser_T_list <- c("Homo_sapiens_chr6.trna148-SerTGA",
"Homo_sapiens_chr6.trna51-SerTGA",
"Homo_sapiens_chr6.trna50-SerAGA",
"Homo_sapiens_chr6.trna147-SerAGA",
"Homo_sapiens_chr12.trna2-SerCGA",
"Homo_sapiens_chr6.trna137-SerCGA",
"Homo_sapiens_chr11.trna8-SerGCT")

temp <- new_SE_HEK_FULL_base %>%
  filter(pileup >1000) %>%
  select(-bin_stop) %>%
  filter(bin_start==-2) %>%
  filter((AA=="Ser") )

temp$chr <- str_replace_all(temp$chr, "chr","c")
temp$chr <- str_replace_all(temp$chr, "trna","t")

temp <- temp %>%
  filter(treatment =="control") %>%
  group_by(sample, gene, position, DM) %>%
  summarise(control_mean_mut = mean(mutation, na.rm=T)) %>%
  select(sample, gene, position, DM, control_mean_mut) %>%
  full_join(temp, ., by=c("sample","gene","DM","position")) %>%
  mutate(gene_label=paste0(AA,anticodon,chr))

temp$treatment <- factor(temp$treatment, levels=rev(c("control","AsO2","H2O2","heat")))
temp <- temp %>%
  mutate(m3C47 = ifelse(gene %in%Ser_C_list,"C",ifelse(gene %in% Ser_T_list,"T","C")))

temp <- temp %>%
  mutate(gene_label=paste0(chr, "    ",AA, anticodon))
temp <- temp[order( temp$m3C47, temp$AA, temp$anticodon, temp$chr ),]
temp$gene_label <- factor(temp$gene_label, levels = unique(temp$gene_label))


plot1 <- ggplot(filter(temp,
              treatment !="control",
              AA=="Ser",
              position >=25,
              position <=57,
              sample=="input"),
       aes(x=position, y=mutation -  control_mean_mut, 
           group=interaction(treatment, rep),
           color=m3C47
           )) +
  geom_line() +
  scale_y_continuous()+
  ylab("Δ mutation rate (stress - control)") +
  scale_color_manual(values=c("C"="#FFAD0A", "T"="#1BB6AF")) +
  facet_grid(rows = vars(gene_label),
             cols = vars(treatment)) +
  facet_nested(m3C47+ anticodon +chr ~ treatment) +
  theme(legend.position = "none",
        strip.background.x = element_blank(),
        strip.background.y =element_rect(color="grey30", fill="white"),
        strip.text.y = element_text(angle=0),
        strip.text.x = element_text(size=14),
        axis.text.y = element_blank(),
        axis.title = element_text(size=16))

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_2E_m3C47_traces"
width=4
height=5
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```


##Fig 2F - box and whisker summary of m3C32 and m3C47 change
###Adjust position for one isodecoder, then take mutation rate change at position 50
```{r}
temp <- new_SE_HEK_FULL_base %>%
  filter(pileup >1000) %>%
  select(-bin_stop) %>%
  filter(bin_start==-2) %>%
  filter((AA=="Ser") )

temp <- temp %>%
  filter(treatment =="control") %>%
  group_by(sample, gene, position, DM) %>%
  summarise(control_mean_mut = mean(mutation, na.rm=T)) %>%
  select(sample, gene, position, DM, control_mean_mut) %>%
  full_join(temp, ., by=c("sample","gene","DM","position")) %>%
  mutate(gene_label=paste0(AA,anticodon,chr))

temp$treatment <- factor(temp$treatment, levels=rev(c("AsO2","H2O2","heat","control")))
temp <- temp %>%
  mutate(m3C47 = ifelse(gene %in%Ser_C_list,"C",ifelse(gene %in% Ser_T_list,"T","C")))

temp <- temp %>%
  mutate(gene_label=paste0(chr, "    ",AA, anticodon))
temp <- temp[order( temp$m3C47, temp$AA, temp$anticodon, temp$chr ),]
temp$gene_label <- factor(temp$gene_label, levels = unique(temp$gene_label))

#I guess there's only one gene where the positions are off, but this is a robust, scalable solution
position_adjust <- data.frame(
  chr=c("chr6.trna175"),
  anticodon=c("GCT"),
  adjust=c(-2)
)

temp <- temp %>%
  full_join(., position_adjust, by=c("anticodon", "chr")) %>%
  mutate(adjust = ifelse(is.na(adjust), 0, adjust)) %>%
  mutate(position = position + adjust) %>%
  filter(position %in% c(32, 50)) %>%
  mutate(position = ifelse(position==32, "position 32", "position 47d"))


plot1 <- ggplot(filter(temp, 
              sample=="input",
              treatment != "control"
              ), 
       aes(y = mutation - control_mean_mut , x=treatment, color=m3C47 )) +
  geom_boxplot() +
  scale_color_manual(values=c("C"="#FFAD0A", "T"="#1BB6AF")) +
  ylab("Δ mutation rate (stress - control)") +
  facet_grid(cols=vars(position) ) +
  theme(legend.position = "none",
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0),
        strip.text.x = element_text(size=14),
        axis.text = element_text(size=12),
        axis.title.y = element_text(size=16),
        axis.title.x = element_blank())


#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_2F_m3C_summary"
width=4
height=2.5
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```


##Fig 2G - abundance of Serine
###Wrangle and plot
```{r}

temp <- new_SE_HEK_FULL_count %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, sample, anticodon, AA, chr, DM, gene, rep) %>%
  summarise(sum_count = sum(count) ) %>%
  group_by(treatment, sample, DM, rep) %>%
  mutate(rpm = sum_count / sum(sum_count)) %>%
  filter(rpm >= 0.0001) #%>%
  # group_by(treatment, sample, anticodon, AA, chr, gene, DM) %>%
  # summarise(mean_rpm = mean(rpm, na.rm=T))

temp <- temp %>%
  filter(treatment == "control") %>%
  group_by(gene, sample, DM) %>%
  summarise(control_rpm_mean = mean(rpm, na.rm=T)) %>%
  select(sample, gene, DM, control_rpm_mean)  %>%
  full_join(temp, ., by=c("sample","DM","gene") )

#Ser only ===========================================================
temp <- temp %>%
  mutate(m3C47 = ifelse(gene %in% Ser_C_list,"C",ifelse(gene %in% Ser_T_list,"T","C"))) 

temp$treatment <- factor(temp$treatment, levels=c("control","heat","H2O2","AsO2"))
my_comparisons <- list( c("C", "T"))

plot1 <- ggplot(filter(temp,
              AA=="Ser",
              sample=="input",
              !grepl("mt", AA) ,
              treatment !="control"
              ),
       aes(x=m3C47, y=rpm / control_rpm_mean, color=m3C47 )) +
  geom_hline(yintercept = 1, linetype=2, alpha=0.3) +
  geom_boxplot() +
  scale_color_manual(values=c("C"="#FFAD0A", "T"="#1BB6AF")) +
  scale_y_log10(breaks=c(0.8, 1, 1.25, 2), 
                limits=c(0.75, 2.1)) +
  stat_compare_means(comparisons = my_comparisons, size=4,  method="t.test",label="p.signif") + #how does this compute p-value?
  facet_grid(cols=vars(treatment)) +
  xlab("Position 47d") +
  ylab("tRNA abundance change\n(stress / control)") +
  theme(legend.position = "none",
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0),
        strip.text.x = element_text(size=14),
        axis.text = element_text(size=12),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16) )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_2G_m3C_abundance_change"
width=3
height=2.5
scale=1.5
dpi=600
# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```


##Fig 2H - charging of Serine
###Calculate charging, just do Serine
```{r}
#Filter for abundance =======================================================
abundance_list <- new_SE_HEK_FULL_count %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, sample, anticodon, AA, DM, rep) %>%
  summarise(sum_count = sum(count) ) %>%
  group_by(treatment, sample, DM, rep) %>%
  mutate(rpm = sum_count / sum(sum_count)) %>%
  filter(rpm >= 0.001) %>%
  group_by(treatment, sample, anticodon, AA, DM) %>%
  summarise(mean_rpm = mean(rpm, na.rm=T)) %>%
  select(sample, treatment, DM, anticodon, AA, mean_rpm)


#Work with the base-wise data, calculate charging for each gene
temp <- new_SE_HEK_FULL_base %>%
  select(-bin_stop) %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, rep, bin_start, DM, sample, source, gene, AA, anticodon) %>%
  mutate(max_position = max(position)) %>%
  filter(position %in% c(max_position, max_position-1)) %>%
  mutate(charging = ifelse(position==max_position, "end", "penultimate")) %>%
  filter((charging=="end" & base=="A")  |
         (charging=="penultimate" & base=="C")) %>%
  select(treatment, rep, bin_start, DM, sample, source, gene, pileup, charging, AA, anticodon) %>%
  group_by(treatment, rep, bin_start, DM, sample, source, gene, anticodon, AA) %>%
  pivot_wider(names_from = c("charging"), values_from = c("pileup")) %>%
  mutate(charging= end/penultimate)

#Apply abundance filter (don't calculate charging on spurious low read genes)
tem <- temp %>%
  full_join(., abundance_list, by=c("sample","treatment","DM","anticodon","AA")) %>%
  filter(mean_rpm >= 0.001)

#Get fold change to control
temp <- temp %>%
  filter(treatment=="control") %>%
  group_by(gene, treatment, sample, DM) %>%
  mutate(mean_control_charging = mean(charging, na.rm=T)) %>%
  ungroup() %>%
  select(gene, sample, DM, mean_control_charging) %>%
  full_join(temp, ., by=c("gene","sample","DM"))

#Make this a Serine only
temp <- temp %>%
  filter(AA=="Ser")
#Ser only ===========================================================
temp <- temp %>%
  mutate(m3C47 = ifelse(gene %in% Ser_C_list,"C",ifelse(gene %in% Ser_T_list,"T","C"))) 

temp$treatment <- factor(temp$treatment, levels=c("control","heat","H2O2","AsO2"))
my_comparisons <- list( c("C", "T"))


plot1 <- ggplot(filter(temp,
              AA=="Ser",
              sample=="input",
              !grepl("mt", AA) ,
              treatment !="control"
              ),
       aes(x=m3C47, y= charging / mean_control_charging, color=m3C47 )) +
  geom_hline(yintercept = 1, linetype=2, alpha=0.3) +
  geom_boxplot() +
  scale_color_manual(values=c("C"="#FFAD0A", "T"="#1BB6AF")) +
  scale_y_log10(breaks=c(0.8, 1, 1.25, 2), 
                limits=c(0.7, 1.25)
                ) +
  stat_compare_means(comparisons = my_comparisons, size=4,  method="t.test", label="p.signif") + #how does this compute p-value?
  facet_grid(cols=vars(treatment)) +
  xlab("Position 47d") +
  ylab("tRNA charging change\n(stress / control") +
  theme(legend.position = "none",
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0),
        strip.text.x = element_text(size=14),
        axis.text = element_text(size=12),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16) )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_2H_m3C_charging_change"
width=3
height=2.5
scale=1.5
dpi=600
# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```

#_


#Figure 3

##Fig 3B - pie chart
###Read in full transcriptome mapped data
```{r}
#Read in the full transcriptome data
PATH="../20210125_new_data/updated_tRNA_reference/SE_map/6_sam_counter/human_combine/"
FILES <-data.frame( file_name = list.files(PATH) ) %>%
  filter(!grepl("unassigned", file_name),
         !grepl("remove", file_name), #This gets the full data, not just the tRNA-only data
         file_name %nin% c("kallisto.out"))

#Split generic file names into salient parts
FILES <- FILES %>%
  separate(file_name, sep="_", c("library","junk1", "junk4","barcode", "junk2", "bin_start", "bin_stop", "junk3"), fill="right", remove=FALSE) %>%
  select(-junk1, -junk2, -junk3, -junk4) 

FILES <- FILES %>%
  mutate(treatment = barcode,
         sample = library) %>%
  select(-library, -barcode)

#PROJECT SPECIFIC DECODING
FILES$file_name <- as.character(FILES$file_name)
FILES$treatment <- recode(FILES$treatment, 
                          "bc1"="control_1", "bc2"="control_2", "bc3"="control_3",
                          "bc4"="heat_1", "bc5"="heat_2", "bc6"="heat_3",
                          "bc7"="H2O2_1", "bc8"="H2O2_2", "bc9"="H2O2_3",
                          "bc10"="AsO2_1", "bc11"="AsO2_2", "bc12"="AsO2_3")
FILES$sample <- recode(FILES$sample, "TP-CK-4S-CW1"="input", "TP-CK-4S-CW2"="polysome")
FILES <- FILES %>%
  separate(treatment, sep="_", into=c("treatment", "rep"))

#=======================+
FILES <- FILES %>%
  mutate(bin_start = ifelse(is.na(bin_start), -3, bin_start),
         bin_stop = ifelse(is.na(bin_stop), -3, bin_stop))
#========================

read_in_one <- function(row){
  output <- read.csv(paste0(PATH, row$file_name), header=T, sep="\t") %>%
    mutate(sample = row$sample,
           rep = row$rep,
           treatment = row$treatment,
           bin_start = row$bin_start,
           bin_stop  = row$bin_stop,)
  #specify data types since enpty dataframes get confused and sad
  output$name <- as.character(output$name)
  output <- output %>%
    mutate(gene=name) %>%
    select(-name)
  return(output)
}

temp <- FILES %>%
  group_by(file_name) %>%
  do(read_in_one(.)) %>%
  ungroup()




temp2 <- read.csv("./human_combine_name_dict.txt",sep="\t")
temp3 <- temp2 %>%
  filter(gene_biotype=="tRNA") %>%
  filter(grepl("mt", gene_symbol)) %>%
  mutate(gene_biotype="mt_tRNA") 
temp2 <- temp2 %>%
  filter(gene_biotype=="tRNA") %>%
  filter(!grepl("mt", gene_symbol)) %>%
  filter(gene_symbol %nin% c("spikein_SCC1","spikein_SCCA1","spikein_SCCA2",
                             "spikein_SCCA3","Ecoli_Lys","Ecoli_Tyr", "Yeast_Phe")) %>%
  rbind(.,temp3) %>%
  rbind(.,filter(temp2, gene_symbol!="tRNA"))

new_PE_data_HEK_full_transcriptome_count <- temp %>%
  filter(bin_start == "-3") %>%
  full_join(., temp2, by=c("gene"))


```

###Stacked bar graph
```{r}

temp <- new_PE_data_HEK_full_transcriptome_count %>%
  group_by(sample, rep, treatment, gene_symbol, gene_biotype) %>%
  summarise(count = sum(count, na.rm=T)) %>%
  filter(count > 0)
temp$gene_biotype = as.factor(temp$gene_biotype)

temp$treatment = factor(temp$treatment, levels=c("control","heat","H2O2","AsO2"))

#Tao's figure edits
temp <- temp %>%
  filter(gene_biotype != "lncRNA")
list_of_other_types <- c("IG_C_gene", "IG_D_gene","IG_J_gene",
                         "IG_V_gene","misc_RNA","polymorphic_pseudogene",
                         "ribozyme","TR_C_gene","TR_D_gene","TR_J_gene",
                         "TR_V_gene", "scaRNA","scRNA","sRNA","Mt_rRNA",
                         "vaultRNA", "miRNA", "snoRNA")
temp <- temp %>%
  filter(gene_biotype %in% list_of_other_types) %>%
  mutate(gene_biotype="other") %>%
  rbind(filter(temp, gene_biotype %nin% list_of_other_types))
temp$gene_biotype <- recode(temp$gene_biotype, "mt_tRNA"="mt-tRNA",
                            "other"="other","protein_coding"="mRNA","rRNA"="rRNA",
                            "snRNA"="snRNA","tRNA"="tRNA")
temp$gene_biotype <- factor(temp$gene_biotype,
                            levels=rev(c("tRNA","rRNA","mRNA","mt-tRNA","snRNA","other")))

#Make pretty label for figure
temp <- temp %>%
  mutate(x_label=paste0(treatment, " ", "rep"," ", rep))
temp <- temp[order( temp$treatment, temp$rep ),]
temp$x_label <- factor(temp$x_label, levels = unique(temp$x_label))




plot1 <- ggplot(filter(temp, sample=="polysome"), 
       aes(x= x_label, y=count,fill=gene_biotype)) +
  geom_bar(position="stack", stat="identity") +
  scale_y_continuous(breaks=c(1000000, 2000000,3000000,4000000,5000000)  ) +
  scale_fill_manual(values = c("tRNA"="#99CCFF", "rRNA"="#CC99CC", "mRNA"="#FFCC33",
                                "other"="grey", "mt-tRNA"="#99CC99", "snRNA"="#CC6666"),
                    guide = guide_legend(reverse = TRUE)) +
  ylab("Read count") +
  labs(fill="Gene type") +
  # facet_wrap(~sample) +
  theme(legend.position = "none",
        strip.background = element_blank(),
        axis.text.x = element_text(angle=90, hjust=1, vjust=0.5),
        axis.title.x = element_blank())

# plot2 <- ggplot(filter(temp, sample=="polysome"), 
#        aes(x=interaction(rep, treatment),y=count,fill=gene_biotype)) +
#   geom_bar(position="stack", stat="identity") +
#   scale_y_continuous(breaks=c(2000000,4000000,6000000) ) +
#   scale_fill_manual(values = c("tRNA"="#99CCFF", "rRNA"="#CC99CC", "mRNA"="#FFCC33",
#                                 "other"="grey", "mt-tRNA"="#99CC99", "snRNA"="#CC6666"),
#                     guide = guide_legend(reverse = TRUE)) +
#   ylab("") +
#   labs(fill="Gene type") +
#   facet_wrap(~sample) +
#   theme(legend.position = "none",
#         strip.background = element_blank(),
#         axis.text.x = element_text(angle=90, hjust=1, vjust=0.5),
#         axis.title.x = element_blank())

figure_legend <- get_legend(plot1+theme(legend.position = "right",
                                       legend.background = element_blank()) )


layout <- rbind(c(1,1,1,3))
figure <- grid.arrange(plot1, figure_legend,  layout_matrix = layout)

path="../figures/draft2/"
figure_name="panel_3B_total_mapping_stacked_bars"
width=3
height=3
scale=1.5
dpi=600
# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```


##Fig 3C - abundance change
###Work with counts data, make the plot
```{r}
#Start with the tRNA-only counts data
temp <- new_SE_HEK_FULL_count %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, sample, anticodon, AA, DM, rep) %>%
  summarise(sum_count = sum(count) ) %>%
  group_by(treatment, sample, DM, rep) %>%
  mutate(rpm = sum_count / sum(sum_count)) %>% #Note; rpm isn't rpm. Its reads per mapped tRNA read.
  filter(rpm >= 0.001) %>%
  group_by(treatment, sample, anticodon, AA, DM) %>%
  summarise(mean_rpm = mean(rpm, na.rm=T))

#Add a column for mean abundance of control
temp <- temp %>%
  filter(treatment == "control") %>%
  group_by() %>%
  mutate(control_rpm_mean = mean_rpm) %>%
  select(sample, anticodon, AA, DM, control_rpm_mean)  %>%
  full_join(temp, ., by=c("sample","anticodon", "AA","DM") )

#Beautify labels
temp$treatment <- factor(temp$treatment, levels=c("heat", "H2O2", "AsO2"))

#label outlier points; semi-manually
# temp <- temp %>%
#   mutate(point_label=NA)
temp1 <- temp %>%
  filter(   (treatment=="heat" & mean_rpm / control_rpm_mean <=0.8) |
            (treatment=="H2O2" & mean_rpm / control_rpm_mean >= 1.2) |
            (treatment=="AsO2" & mean_rpm / control_rpm_mean >= 1.3 ) ) %>%
  mutate(point_label=paste0(AA, anticodon)) %>%
  select(treatment, sample, AA, anticodon, DM, point_label)
temp <- temp %>%
  full_join(.,temp1, by=c("treatment","sample","AA","anticodon","DM"))

#Make the figure
plot1 <- ggplot(filter(temp,
              sample=="polysome",
              !grepl("mt", AA) ,
              treatment !="control"
              ),
       aes(x=treatment, y= mean_rpm / control_rpm_mean )) +
  geom_hline(yintercept = 1, linetype=2, alpha=0.3) +
  geom_boxplot() +
  geom_text_repel(aes(label=point_label)) +
  scale_y_log10(breaks=c(0.5,1,2, 3), 
                #limits=c(0.74,1.35)
                ) + #Maybe indicate that its a log scale in the label
  ylab("tRNA abundance change (stress / control)") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14, color="black"),
        axis.title.y = element_text(size=16),
        axis.text.y = element_text(size=12, color="black"))

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_3C_polysome_isoacceptor_abundance"
width=2
height=3
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)


```

###Write table S3
```{r}
#Start with the tRNA-only counts data
tempS <- new_SE_HEK_FULL_count %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, sample, anticodon, AA, DM, rep) %>%
  summarise(sum_count = sum(count) ) %>% #Pool iso acceptors
  group_by(treatment, sample, DM, rep) %>%
  mutate(rpm = sum_count / sum(sum_count)) #%>% #Note; rpm isn't rpm. Its reads per mapped tRNA read.
  # filter(rpm >= 0.001)

asdf2 <- tempS %>%
  ungroup() %>%
  filter(sample=="polysome") %>%
  select(-DM) %>%
  group_by(treatment, anticodon, AA) %>%
  pivot_wider(names_from=c("rep"), values_from=c("rpm", "sum_count") )

write.table(asdf2,
            file='/Users/ckatanski/4SR/data/20210125_HEK_stress_v2/figures/draft2/supplementry_tables/table_S3.csv',
            quote=FALSE, sep=',', col.names = NA)

```

##Fig 3D - Ile abundance
###Ile summed isoaccepter
```{r}
temp <- new_SE_HEK_FULL_count %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, sample, anticodon, AA, DM, rep) %>%
  summarise(sum_count = sum(count) ) %>%
  group_by(treatment, sample, DM, rep) %>%
  mutate(rpm = sum_count / sum(sum_count) *1000000) %>%
  filter(rpm >= 0.0001*1000000) 

temp <- temp %>%
  group_by(treatment, sample, anticodon, AA, DM) %>%
  summarise(mean_rpm = mean(rpm, na.rm=T)) %>%
  full_join(temp,., by=c("treatment", "sample", "anticodon", "AA", "DM"))

#Prettier label
temp <- temp %>%
  mutate(strip_label = paste0(AA, anticodon))

temp$treatment <- factor(temp$treatment, levels=c("control","heat","H2O2","AsO2"))

plot1 <- ggplot(filter(temp,
              AA=="Ile",
              sample=="polysome"),
       aes(x=treatment, y=rpm)) +
  geom_point() +
  ylab("Reads per million") +
  scale_y_log10(breaks=c(12500,25000, 50000,100000)) +
  facet_grid(cols=vars(strip_label)) +
  geom_point(aes(y=mean_rpm), shape=95, size=5) +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14, color="black"),
        axis.title.y = element_text(size=16),
        axis.text.y = element_text(size=12, color="black"))

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_3D_polysome_Ile_abundance"
width=4
height=2
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```


##Fig 3F - charging fold change
###Calculate charging, and sum
```{r}
#Filter for abundance =======================================================
abundance_list <- new_SE_HEK_FULL_count %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, sample, anticodon, AA, DM, rep) %>%
  summarise(sum_count = sum(count) ) %>%
  group_by(treatment, sample, DM, rep) %>%
  mutate(rpm = sum_count / sum(sum_count)) %>%
  filter(rpm >= 0.001) %>%
  group_by(treatment, sample, anticodon, AA, DM) %>%
  summarise(mean_rpm = mean(rpm, na.rm=T)) %>%
  select(sample, treatment, DM, anticodon, AA, mean_rpm)


#Work with the base-wise data, calculate charging for each gene
temp <- new_SE_HEK_FULL_base %>%
  select(-bin_stop) %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, rep, bin_start, DM, sample, source, gene, AA, anticodon) %>%
  mutate(max_position = max(position)) %>%
  filter(position %in% c(max_position, max_position-1)) %>%
  mutate(charging = ifelse(position==max_position, "end", "penultimate")) %>%
  filter((charging=="end" & base=="A")  |
         (charging=="penultimate" & base=="C")) %>%
  select(treatment, rep, bin_start, DM, sample, source, gene, pileup, charging, AA, anticodon) %>%
  group_by(treatment, rep, bin_start, DM, sample, source, gene, anticodon, AA) %>%
  pivot_wider(names_from = c("charging"), values_from = c("pileup")) %>%
  mutate(charging= end/penultimate)

#Apply abundance filter (don't calculate charging on spurious low read genes)
tem <- temp %>%
  full_join(., abundance_list, by=c("sample","treatment","DM","anticodon","AA")) %>%
  filter(mean_rpm >= 0.001)

#Sum up isoacceptors (sum of penultimate, sum up ultimate, ratio)
temp <- temp %>%
  group_by(sample, source, treatment, DM, rep, AA, anticodon) %>%
  summarise(penultimate = sum(penultimate, na.rm=T),
            end = sum(end, na.rm=T)) %>%
  mutate(charging= end/penultimate)

#Average replicates
temp <- temp %>%
  group_by(sample, source, treatment, DM, AA, anticodon) %>%
  summarise(charging = mean(charging, na.rm=T))

#Normalize to average of control
temp <- temp %>%
  filter(treatment=="control") %>%
  group_by(AA, anticodon, sample) %>%
  summarise(control_charging_average = mean(charging, na.rm=T)) %>%
  select(sample, AA, anticodon, control_charging_average) %>%
  full_join(temp, ., by=c("AA","anticodon","sample"))



#Beautify labels
temp$treatment <- factor(temp$treatment, levels=c("heat", "H2O2", "AsO2"))

#label outlier points; semi-manually
# temp <- temp %>%
#   mutate(point_label=NA)
temp1 <- temp %>%
  filter(   (treatment=="heat" & (charging / control_charging_average <= 0.83 | charging / control_charging_average >= 1.05)) |
            (treatment=="H2O2" & charging / control_charging_average <= 0.9) |
            (treatment=="AsO2" & (charging / control_charging_average <=0.9 | charging / control_charging_average >= 1.3)) ) %>%
  mutate(point_label=paste0(AA, anticodon)) %>%
  select(treatment, sample, AA, anticodon, DM, point_label)
temp <- temp %>%
  full_join(.,temp1, by=c("treatment","sample","AA","anticodon","DM"))



plot1 <- ggplot(filter(temp,
              treatment !="control",
              !grepl("mt", AA),
              sample=="polysome"),
       aes(x=treatment, y=charging / control_charging_average)) +
  geom_hline(yintercept = 1, linetype=2, alpha=0.4) +
  geom_boxplot() +
  geom_text_repel(aes(label=point_label)) +
  scale_y_log10(breaks=c(0.8,1,1.25), 
                limits=c(0.65,1.3)
                ) + #Maybe indicate that its a log scale in the label
  ylab("tRNA charging change (stress / control)") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14, color="black"),
        axis.title.y = element_text(size=16),
        axis.text.y = element_text(size=12, color="black"))

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_3F_polysome_isoacceptor_charging"
width=2
height=3
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```

###Write table S4
```{r}
#Filter for abundance =======================================================
abundance_list <- new_SE_HEK_FULL_count %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, sample, anticodon, AA, DM, rep) %>%
  summarise(sum_count = sum(count) ) %>%
  group_by(treatment, sample, DM, rep) %>%
  mutate(rpm = sum_count / sum(sum_count)) %>%
  # filter(rpm >= 0.001) %>%
  group_by(treatment, sample, anticodon, AA, DM) %>%
  summarise(mean_rpm = mean(rpm, na.rm=T)) %>%
  select(sample, treatment, DM, anticodon, AA, mean_rpm)


#Work with the base-wise data, calculate charging for each gene
tempS <- new_SE_HEK_FULL_base %>%
  select(-bin_stop) %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, rep, bin_start, DM, sample, source, gene, AA, anticodon) %>%
  mutate(max_position = max(position)) %>%
  filter(position %in% c(max_position, max_position-1)) %>%
  mutate(charging = ifelse(position==max_position, "end", "penultimate")) %>%
  filter((charging=="end" & base=="A")  |
         (charging=="penultimate" & base=="C")) %>%
  select(treatment, rep, bin_start, DM, sample, source, gene, pileup, charging, AA, anticodon) %>%
  group_by(treatment, rep, bin_start, DM, sample, source, gene, anticodon, AA) %>%
  pivot_wider(names_from = c("charging"), values_from = c("pileup")) %>%
  mutate(charging= end/penultimate)

#Apply abundance filter (don't calculate charging on spurious low read genes)
temS <- tempS %>%
  full_join(., abundance_list, by=c("sample","treatment","DM","anticodon","AA")) %>%
  filter(mean_rpm >= 0.001)

#Sum up isoacceptors (sum of penultimate, sum up ultimate, ratio)
tempS <- tempS %>%
  group_by(sample, source, treatment, DM, rep, AA, anticodon) %>%
  summarise(penultimate = sum(penultimate, na.rm=T),
            end = sum(end, na.rm=T)) %>%
  mutate(charging= end/penultimate)
tempS$rep <- recode(tempS$rep, "1"="rep_1_charging", "2"="rep_2_charging", "3"="rep_3_charging")

asdf2 <- tempS %>%
  ungroup() %>%
  filter(sample=="polysome") %>%
  select(-DM, -penultimate, -end) %>%
  group_by(treatment, anticodon, AA) %>%
  pivot_wider(names_from=c("rep"), values_from=c("charging") )

write.table(asdf2,
            file='/Users/ckatanski/4SR/data/20210125_HEK_stress_v2/figures/draft2/supplementry_tables/table_S4.csv',
            quote=FALSE, sep=',', col.names = NA)

```

##Fig 3G - Serine isodecoder abundance
###Wrangle and plot
```{r}

temp <- new_SE_HEK_FULL_count %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, sample, anticodon, AA, chr, DM, gene, rep) %>%
  summarise(sum_count = sum(count) ) %>%
  group_by(treatment, sample, DM, rep) %>%
  mutate(rpm = sum_count / sum(sum_count)) %>%
  filter(rpm >= 0.0001) #%>%
  # group_by(treatment, sample, anticodon, AA, chr, gene, DM) %>%
  # summarise(mean_rpm = mean(rpm, na.rm=T))

temp <- temp %>%
  filter(treatment == "control") %>%
  group_by(gene, sample, DM) %>%
  summarise(control_rpm_mean = mean(rpm, na.rm=T)) %>%
  select(sample, gene, DM, control_rpm_mean)  %>%
  full_join(temp, ., by=c("sample","DM","gene") )

#Ser only ===========================================================
temp <- temp %>%
  mutate(m3C47 = ifelse(gene %in% Ser_C_list,"C",ifelse(gene %in% Ser_T_list,"T","C"))) 

temp$treatment <- factor(temp$treatment, levels=c("control","heat","H2O2","AsO2"))
my_comparisons <- list( c("C", "T"))

plot1 <- ggplot(filter(temp,
              AA=="Ser",
              sample=="polysome",
              !grepl("mt", AA) ,
              treatment !="control"
              ),
       aes(x=m3C47, y=rpm / control_rpm_mean, color=m3C47 )) +
  geom_hline(yintercept = 1, linetype=2, alpha=0.3) +
  geom_boxplot() +
  scale_color_manual(values=c("C"="#FFAD0A", "T"="#1BB6AF")) +
  scale_y_log10(breaks=c(0.5, 0.8, 1, 1.25, 2), 
                limits=c(0.6, 2.5)
                ) +
  stat_compare_means(comparisons = my_comparisons, size=4,  method="t.test", label="p.signif") + #how does this compute p-value?
  facet_grid(cols=vars(treatment)) +
  xlab("Position 47d") +
  ylab("tRNA abundance change\n(stress / control)") +
  theme(legend.position = "none",
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0),
        strip.text.x = element_text(size=14),
        axis.text = element_text(size=12),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16) )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_3G_m3C_polysome_abundance_change"
width=3
height=2.5
scale=1.5
dpi=600
# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```


##Fig 3H - m3C polysome vs input
###Plot input and polysome mutation rates
```{r}
temp <- new_SE_HEK_FULL_base %>%
  filter( (sample=="input"&pileup >=1000)|
            ( sample=="polysome"&pileup >=100) #polysome coverage is lower, less stringent cutoff
         ) %>%
  select(-bin_stop) %>%
  filter(bin_start==-2) %>%
  filter((AA=="Ser") )

#I guess there's only one gene where the positions are off, but this is a robust, scalable solution
position_adjust <- data.frame(
  chr=c("chr6.trna175"),
  anticodon=c("GCT"),
  adjust=c(-2)  )
temp <- temp %>%
  full_join(., position_adjust, by=c("anticodon", "chr")) %>%
  mutate(adjust = ifelse(is.na(adjust), 0, adjust)) %>%
  mutate(position = position + adjust) %>%
  filter(position %in% c(32, 50)) %>%
  mutate(position = ifelse(position==32, "position 32", "position 47d"))


#Beautify
temp$treatment <- factor(temp$treatment, levels=rev(c("AsO2","H2O2","heat","control")))
#Separate isodecoders by position 47d
temp <- temp %>%
  mutate(m3C47 = ifelse(gene %in%Ser_C_list,"C47d",ifelse(gene %in% Ser_T_list,"T47d","C47d")))
#P-values
my_comparisons <- list( c("input", "polysome"))

#Plot the good stuff
plot1 <- ggplot(filter(temp,
              treatment=="control"),
       aes(x=sample, y=mutation, color=m3C47 )) +
  geom_boxplot() +
  ylab("Mutation rate") +
  scale_y_continuous(breaks=c(0,0.5, 1), limits=c(0, 1.15)) +
  scale_color_manual(values=c("C47d"="#FFAD0A", "T47d"="#1BB6AF")) +
  stat_compare_means(comparisons = my_comparisons, size=4,  method="t.test", label="p.signif") + #how does this compute p-value?
  facet_nested(. ~ position + m3C47) +
  theme(legend.position = "none",
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0),
        strip.text.x = element_text(size=14),
        axis.text = element_text(size=12),
        axis.title.y = element_text(size=16),
        axis.title.x = element_blank() )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_3H_m3C_mutation_polysome_and_input_control"
width=4.5
height=2
scale=1.5
dpi=600
# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)










```

##Fig 3I - polysome change with stress
###wrangel polysome - poysome_control
```{r}
temp <- new_SE_HEK_FULL_base %>%
  filter( (sample=="input"&pileup >=1000)|
            ( sample=="polysome"&pileup >=100) #polysome coverage is lower, less stringent cutoff
         ) %>%
  select(-bin_stop) %>%
  filter(bin_start==-2) %>%
  filter((AA=="Ser") )

temp <- temp %>%
  filter(treatment =="control") %>%
  group_by(sample, gene, position, DM) %>%
  summarise(control_mean_mut = mean(mutation, na.rm=T)) %>%
  select(sample, gene, position, DM, control_mean_mut) %>%
  full_join(temp, ., by=c("sample","gene","DM","position")) %>%
  mutate(gene_label=paste0(AA,anticodon,chr))

temp$treatment <- factor(temp$treatment, levels=rev(c("AsO2","H2O2","heat","control")))
temp <- temp %>%
  mutate(m3C47 = ifelse(gene %in%Ser_C_list,"C",ifelse(gene %in% Ser_T_list,"T","C")))

temp <- temp %>%
  mutate(gene_label=paste0(chr, "    ",AA, anticodon))
temp <- temp[order( temp$m3C47, temp$AA, temp$anticodon, temp$chr ),]
temp$gene_label <- factor(temp$gene_label, levels = unique(temp$gene_label))

#I guess there's only one gene where the positions are off, but this is a robust, scalable solution
position_adjust <- data.frame(
  chr=c("chr6.trna175"),
  anticodon=c("GCT"),
  adjust=c(-2)
)

temp <- temp %>%
  full_join(., position_adjust, by=c("anticodon", "chr")) %>%
  mutate(adjust = ifelse(is.na(adjust), 0, adjust)) %>%
  mutate(position = position + adjust) %>%
  filter(position %in% c(32, 50)) %>%
  mutate(position = ifelse(position==32, "position 32", "position 47d"))


plot1 <- ggplot(filter(temp, 
              sample=="polysome",
              treatment != "control"
              ), 
       aes(y = mutation - control_mean_mut , x=treatment, color=m3C47 )) +
  geom_boxplot() +
  scale_color_manual(values=c("C"="#FFAD0A", "T"="#1BB6AF")) +
  ylab("Δ mutation rate (stress - control)") +
  facet_grid(cols=vars(position) ) +
  theme(legend.position = "none",
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0),
        strip.text.x = element_text(size=14),
        axis.text = element_text(size=12),
        axis.title.y = element_text(size=16),
        axis.title.x = element_blank())


#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_3I_m3C_polysome_mutation_change"
width=4
height=2.5
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```



#_
#Figure 4
##Preamble
###Read in codon dict
```{r}
#Add in some codons
gene_id_dict <- read.csv("../20210125_new_data/mRNA_mapping/human_combine_name_dict.txt",sep="\t")

PATH = "/Users/ckatanski/4SR/data/20210125_HEK_stress_v2/source/Codon_counter/human_combine_codon.tsv"
codon_dict <- read.csv(PATH, header=T, sep="\t") %>%
  mutate(seq = nchar(seq)/3) %>%
  full_join(., gene_id_dict, by=c("gene")) %>%
  filter(gene_biotype=="protein_coding") %>%
  select(-feature, -gene_biotype, -chromosome, -transcript_biotype, -gene)
#temp$gene <- factor(temp$gene)
codon_dict$gene_symbol <- factor(codon_dict$gene_symbol)

#Normalize codon usage by gene length
temp <- codon_dict[-ncol(codon_dict)] / codon_dict[,(ncol(codon_dict)-1)] 
temp <- cbind(codon_dict[ncol(codon_dict)], temp) %>%
  select(-seq)
#Average transcript varients
#https://dplyr.tidyverse.org/reference/summarise_all.html
codon_dict <- temp %>%
  group_by(gene_symbol) %>%
  summarise(across(where(is.numeric), ~ median(.x, na.rm = TRUE))) 

```

###Human codon usage
```{r}
PATH="/Users/ckatanski/4SR/data/20210125_HEK_stress_v2/source/human_codon_usage.tsv"
human_codon <- read.csv(PATH, sep="\t", header=T) %>%
  mutate(genome_usage = usage/1000) %>%
  select(-usage)
```

###Combine mRNA with codon usage
```{r}
#combine transcript varianets
temp <- mRNA_data %>%
  group_by(gene_symbol, treatment, sample, rep, gene_biotype) %>%
  summarise(count = sum(count)) 
#Make factors
temp$sample <- factor(temp$sample)
temp$treatment <- factor(temp$treatment, levels=c("control","heat","H2O2","AsO2"))
temp$rep <- factor(temp$rep)

#summarise to RPM; filter to abundnat mRNAs;
temp1 <- temp %>%
  group_by(rep, treatment, sample) %>%
  filter(gene_biotype=="protein_coding") %>%
  mutate(rpm = count / sum(count)) %>% #Not a true RPM. Its fraction of mRNA mapped reads
  filter(count >= 100, #Mostly a filter on polysome fraction, which has 10X lower coverage
         !is.na(gene_biotype))

#pivot wider by sample; calculate ratio polysome to input
temp1 <- temp1 %>%
  filter(gene_biotype=="protein_coding") %>%
  select(-count) %>% #Esential
  group_by(treatment, gene_symbol, gene_biotype, rep) %>%
  pivot_wider(names_from = c("sample"), values_from=c("rpm")) %>%
  ungroup() %>%
  mutate(TE = polysome / input)

#Limit analysis to a set of well detected genes
common_detection_list <- temp1 %>%
  select(gene_symbol, rep, treatment, TE) %>%
  filter(rep==1) %>%
  pivot_wider(names_from = c("treatment"), values_from=c("TE")) %>%
  filter(!is.na(control) &
           !is.na(heat) &
           !is.na(H2O2) &
           !is.na(AsO2))
common_detection_list <- unique(common_detection_list$gene_symbol)

#Confirm distribution of TE values
ggplot(filter(temp1,
              !(treatment=="control" & rep==2),
              !(treatment=="H2O2" & rep==3),
              gene_symbol %in% common_detection_list
              ),
       aes(x=TE, color=treatment, group=interaction(treatment, rep))) +
  geom_freqpoly() +
  # geom_density() +
  scale_x_log10()

#Compute Zscore for each replicate
temp2 <- temp1 %>%
  filter(gene_symbol %in% common_detection_list) %>%
  group_by(rep, treatment) %>%
  mutate(log_TE = log10(TE)) %>% #The data are LOG normally distributed, take that log
  mutate(zscore = (log_TE - mean(log_TE, na.rm=T))/sd(log_TE, na.rm=T) )

# #Confirm log-normal zscore shape
# ggplot(filter(temp2,
#               # !(treatment=="control" & rep==2),
#               # !(treatment=="H2O2" & rep==3),
#               # gene_symbol %in% common_detection_list
#               ),
#        aes(x=zscore, color=treatment, group=interaction(treatment, rep))) +
#   # geom_freqpoly() +
#   geom_density() +
#   geom_vline(xintercept = 0, linetype=2, alpha=0.3) 


#make a column for control ratio
temp3 <- temp2 %>%
  filter(treatment=="control") %>%
  filter(!(treatment=="control" & rep==2)) %>% #This replicates is too poor coverage to be useful
  group_by(gene_symbol) %>%
  summarise(control_TE = mean(TE, na.rm=T),
            control_log_TE = mean(log_TE, na.rm=T),
            control_zscore = mean(zscore, na.rm=T) ) %>% #take mean of control reps
  select(gene_symbol, control_TE, control_log_TE, control_zscore) %>%
  full_join(temp2, by=c("gene_symbol")) %>%
  mutate(change_zscore = zscore - control_zscore)

# #Plot change in z-score
# ggplot(filter(temp3,
#               # !(treatment=="control" & rep==2),
#               # !(treatment=="H2O2" & rep==3),
#               # gene_symbol %in% common_detection_list
#               ),
#        aes(x=zscore - control_zscore, color=treatment, group=interaction(treatment, rep))) +
#   geom_freqpoly() +
#   # geom_density() +
#   geom_vline(xintercept = c(-1, 0, 1 ), linetype=2, alpha=0.3) 

```

##Fig 4C - distribution of TE
###plot the figure old bean
```{r}
#Confirm distribution of TE values
temp1$treatment <- factor(temp1$treatment, levels = rev(c("control", "heat", "H2O2", "AsO2")) )
 
plot1 <- ggplot(filter(temp1,
              !(treatment=="control" & rep==2),
              !(treatment=="H2O2" & rep==3),
              gene_symbol %in% common_detection_list,
              rep ==1
              ),
       aes(x=TE, color=treatment, group=interaction(treatment, rep))) +
  # geom_vline(xintercept = 1, linetype=2, alpha=0.3) +
  geom_freqpoly(size=1) +
  scale_x_log10() +
  scale_color_manual(values= alpha(c("heat"="#f76f73", "H2O2"="#07c4c5", "AsO2"="#027fdc", "control"="grey70"), 1), 
                     guide = guide_legend(reverse = TRUE) )+
  scale_x_log10(limits=c(0.1, 10),
                breaks=c(0.1,0.5,1,2,10)) +
  scale_y_continuous(breaks=c(0,100,200)) +
  xlab("Translational efficency\n(polysome / input)") +
  ylab("Number of genes") +
  theme(legend.position = "right",
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0),
        strip.text.x = element_text(size=14),
        axis.text = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16))


#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_4Z_change_in_TE"
width=3.5
height=2.5
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```


###TE period; percentile; codon
```{r}
#Calculate percentile; take top and bottom 20% for each rep, stress
HIGH_percentile = 0.66
LOW_percentile = 0.33

temp4 <- temp3 %>%
  group_by(rep, treatment ) %>%
  mutate( TE_percent_rank = percent_rank(TE)) 

#Merge with codons 
temp5 <- temp4 %>% 
  mutate(percentile_grouping = ifelse(TE_percent_rank <= LOW_percentile, "low", 
                                      ifelse( TE_percent_rank >= HIGH_percentile, "high", "average") ) ) %>%
  filter(!is.na(TE)) %>% #Some genes with poor polysome detection result in NA percentiles
  select(rep, treatment, gene_symbol, percentile_grouping) %>%
  full_join(., codon_dict, by=c("gene_symbol")) %>%
  filter(!is.na(rep)) %>% #Codon_dict has more genes than are in the TE dataset
  pivot_longer(col=c(-rep, -treatment, -gene_symbol, -percentile_grouping),
               names_to = c("codon"), values_to=c("frequency")) %>%
  full_join(., AA_table, by=c("codon")) %>%
  group_by(treatment, rep, codon) %>%
  mutate(frequency_percentile = percent_rank(frequency))

temp5$percentile_grouping <- factor(temp5$percentile_grouping, levels=c("low", "average","high"))
my_comparisons <- list( c("average", "high"),
                        c("average", "low"),
                        c("high", "low") )

ggplot(filter(temp5, 
              # AA == "Ile",
              codon=="GAG",
              !(rep==2 & treatment=="control"),
              # frequency > 0 
              ),
       aes(x=percentile_grouping, y=frequency_percentile, color=percentile_grouping)) +
  geom_boxplot() +
  coord_cartesian(ylim=c(-0.01, 1.5)) +
  # scale_y_log10() +
  stat_compare_means(comparisons = my_comparisons, size=4,  
                     method="wilcox.test",
                     p.adjust.method = "none",
                     #label="p.signif"
                     ) +
  facet_nested(treatment ~ AA + codon + rep ) +
  theme(strip.background.x = element_rect(fill="white", color="black"),
        strip.text.x = element_text(angle=0))

##Show the actual distributions
# ggplot(filter(temp5, 
#               AA == "Lys",
#               codon=="AAA",
#               !(rep==2 & treatment=="control"),
#               # frequency > 0 
#               ),
#        aes( x=frequency_percentile, color=percentile_grouping)) +
#   geom_freqpoly() +
#   # coord_cartesian(ylim=c(0, 1.5)) +
#   facet_nested(treatment ~ AA + codon + rep ) +
#   theme(strip.background.x = element_rect(fill="white", color="black"),
#         strip.text.x = element_text(angle=0))

```
##Fig 4E - Third codon position usage

###P-vlaues (wilcox) for each codon on TE
```{r}

get_codon_pvalue <- function(df){
  asdf <- df %>%
    ungroup() %>%
    mutate(metric = frequency_percentile ) %>%
    filter(!is.na(metric),
           !is.infinite(metric))
  
  side = mean(filter(asdf, percentile_grouping=="high")$metric) / 
          mean(filter(asdf, percentile_grouping=="low")$metric)
  side = ifelse(side >= 1, 1, -1) #enriched in high metric genes
    
  
  A <- pairwise.wilcox.test(asdf$metric, asdf$percentile_grouping, 
                   p.adjust.method = "none", 
                   alternative = "two.sided", 
                   pool.sd = F #Not sure if this function even accepts this parameter
                   )$p.value
  #A[1] is 1 vs 2; A[2] is 1 vs 3; A[3] is 2 vs 2; A[4] is 2 vs 3
  df=data.frame(low.average=A[1],
                low.high=A[2],
                average.low=A[4],
                enrichment = side, 
                N_equals.group1= nrow(filter(asdf, percentile_grouping=="low")),
                N_equals.group2= nrow(filter(asdf, percentile_grouping=="average")),
                N_equals.group3= nrow(filter(asdf, percentile_grouping=="high")) )
  } 


temp6 <- temp5 %>% 
  filter( !(rep==2 & treatment=="control") ) %>%
  group_by(rep, treatment, codon) %>%
  do(get_codon_pvalue(.))

#prety plot
temp6 <- temp6 %>%
  separate(codon, sep=c(1,2), into=c("first","second", "third"), remove=F)
temp6$third <- factor(temp6$third, levels=c("A","T","C","G"))

ggplot(filter(temp6,
              codon %nin% c("TAA","TGA","TAG","ATG","TGG") #STOP, Trp, Met
              ),
       aes(x=interaction(codon, third), y=rep, 
           fill= log(low.high) *enrichment*-1 #Remember, the log gives another negative -1 
           ))+
  geom_tile() +
  scale_fill_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="grey70", 
                       # limits=c(0,1),
                       # breaks=c(0,.5,1)
                       ) +
  facet_nested( treatment ~ .) +
  theme(axis.text.x=element_text(angle=90, vjust=0.5) )

```


##TE movers
###delta-TE percentile; codon
```{r}
#Calculate percentile; take top and bottom 20% for each rep, stress
HIGH_percentile = 0.66
LOW_percentile = 0.33

temp4 <- temp3 %>%
  group_by(rep, treatment ) %>%
  mutate( change_Z_percent_rank = percent_rank(change_zscore)) 

#Merge with codons 
temp5 <- temp4 %>% 
  mutate(percentile_grouping = ifelse(change_Z_percent_rank <= LOW_percentile, "low", 
                                      ifelse( change_Z_percent_rank >= HIGH_percentile, "high", "average") ) ) %>%
  filter(!is.na(change_Z_percent_rank)) %>% #Some genes with poor polysome detection result in NA percentiles
  select(rep, treatment, gene_symbol, percentile_grouping) %>%
  full_join(., codon_dict, by=c("gene_symbol")) %>%
  filter(!is.na(rep)) %>% #Codon_dict has more genes than are in the TE dataset
  pivot_longer(col=c(-rep, -treatment, -gene_symbol, -percentile_grouping),
               names_to = c("codon"), values_to=c("frequency")) %>%
  full_join(., AA_table, by=c("codon")) %>%
  group_by(treatment, rep, codon) %>%
  mutate(frequency_percentile = percent_rank(frequency))

temp5$percentile_grouping <- factor(temp5$percentile_grouping, levels=c("low", "average","high"))
my_comparisons <- list( c("average", "high"),
                        c("average", "low"),
                        c("high", "low") )

ggplot(filter(temp5, 
              # AA == "Ile",
              codon=="AAA",
              !(rep==2 & treatment=="control"),
              treatment != "control", #replicates are necessarily symetric
              # frequency > 0 
              ),
       aes(x=percentile_grouping, y=frequency_percentile, color=percentile_grouping)) +
  geom_boxplot() +
  coord_cartesian(ylim=c(-0.01, 1.5)) +
  # scale_y_log10() +
  stat_compare_means(comparisons = my_comparisons, size=4,  
                     method="wilcox.test",
                     p.adjust.method = "none",
                     #label="p.signif"
                     ) +
  facet_nested(treatment ~ AA + codon + rep ) +
  theme(strip.background.x = element_rect(fill="white", color="black"),
        strip.text.x = element_text(angle=0))

##Show the actual distributions
# ggplot(filter(temp5, 
#               AA == "Lys",
#               codon=="AAA",
#               !(rep==2 & treatment=="control"),
#               # frequency > 0 
#               ),
#        aes( x=frequency_percentile, color=percentile_grouping)) +
#   geom_freqpoly() +
#   # coord_cartesian(ylim=c(0, 1.5)) +
#   facet_nested(treatment ~ AA + codon + rep ) +
#   theme(strip.background.x = element_rect(fill="white", color="black"),
#         strip.text.x = element_text(angle=0))

```

###P-vlaues (wilcox) for each codon on change in Zscore
```{r}

get_codon_pvalue <- function(df){
  asdf <- df %>%
    ungroup() %>%
    mutate(metric = frequency_percentile ) %>%
    filter(!is.na(metric),
           !is.infinite(metric))
  
  side = mean(filter(asdf, percentile_grouping=="high")$metric) / 
          mean(filter(asdf, percentile_grouping=="low")$metric)
  side = ifelse(side >= 1, 1, -1) #enriched in high metric genes
    
  
  A <- pairwise.wilcox.test(asdf$metric, asdf$percentile_grouping, 
                   p.adjust.method = "none", 
                   alternative = "two.sided", 
                   pool.sd = F #Not sure if this function even accepts this parameter
                   )$p.value
  #A[1] is 1 vs 2; A[2] is 1 vs 3; A[3] is 2 vs 2; A[4] is 2 vs 3
  df=data.frame(low.average=A[1],
                low.high=A[2],
                average.low=A[4],
                enrichment = side, 
                N_equals.group1= nrow(filter(asdf, percentile_grouping=="low")),
                N_equals.group2= nrow(filter(asdf, percentile_grouping=="average")),
                N_equals.group3= nrow(filter(asdf, percentile_grouping=="high")) )
  } 


temp6 <- temp5 %>% 
  filter( !(rep==2 & treatment=="control") ) %>%
  group_by(rep, treatment, codon, AA) %>%
  do(get_codon_pvalue(.))

#prety plot
temp6 <- temp6 %>%
  separate(codon, sep=c(1,2), into=c("first","second", "third"), remove=F)
temp6$third <- factor(temp6$third, levels=c("A","T","C","G"))
temp6 <- temp6 %>%
  mutate(p_value_color = ifelse(low.high >= PVALUE_CUTOFF, "No enrichment", ifelse(enrichment >0, "Enriched in genes\nwith increasing TE", "Depleted in genes\nwith increasing TE")) )

#Averge p-values to simplify plotting - gross
temp7 <- temp6 %>%
  group_by(codon, AA, treatment, first, second, third) %>%
  summarise(median_pvalue = median(low.high, na.rm=T),
            enrichment = mean(enrichment, na.rm=T))

PVALUE_CUTOFF = 0.05
temp7 <- temp7 %>%
  mutate(p_value_color = ifelse(median_pvalue >= PVALUE_CUTOFF, "No enrichment", ifelse(enrichment > 0, "Enriched codons", "Depleted codons")) )
#Beaturify order
temp7 <- temp7 %>%
  mutate(x_label=codon)
temp7$third <- factor(temp7$third, levels = c("A","T","C","G"))
temp7 <- temp7[order(temp7$third, temp7$codon ),]
temp7$x_label <- factor(temp7$x_label, levels = unique(temp7$x_label))
#Further beauty
temp7$treatment <- factor(temp7$treatment, levels = rev(c("control","heat","H2O2","AsO2")))

# #Colored by p-value value
# ggplot(filter(temp6,
#               treatment != "control", #replicates are necessarily symetric
#               # codon %nin% c("TAA","TGA","TAG","ATG","TGG") #STOP, Trp, Met
#               ),
#        aes(x=interaction(codon, third), y=rep, 
#            fill= log(low.high) *enrichment*-1 #Remember, the log gives another negative -1 
#            ))+
#   geom_tile() +
#   geom_vline(xintercept = c(15.5), color="black", size=1) +
#   scale_fill_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="grey70", 
#                        # limits=c(0,1),
#                        # breaks=c(0,.5,1)
#                        ) +
#   facet_nested( treatment ~ .) +
#   theme(axis.text.x=element_text(angle=90, vjust=0.5) )


#Chris' preferred plot
# #Simplified colors
# ggplot(filter(temp6,
#               treatment != "control", #replicates are necessarily symetric
#               codon %nin% c("TAA","TGA","TAG","ATG","TGG") #STOP, Trp, Met
#               ),
#        aes(x=interaction(codon, third), y=rep,
#            fill= p_value_color #Remember, the log gives another negative -1
#            ))+
#   geom_tile() +
#   geom_vline(xintercept = c(13.5, 30.5, 46.5 ), color="black", size=1) +
#   scale_fill_manual(values=c("No enrichment"="grey70", 
#                              "Enriched in genes\nwith increasing TE"="red3", 
#                              "Depleted in genes\nwith increasing TE"="blue3")) +
#   facet_nested( treatment ~ .) +
#   theme(legend.position = "top",
#         legend.title = element_blank(),
#         strip.background = element_blank(),
#         strip.text.y = element_text(angle=0, size=16),
#         axis.text = element_text(size=10, color="black"),
#         axis.title.y = element_blank(),
#         axis.title.x = element_blank() )


#Simplified colors
plot1 <- ggplot(filter(temp7,
              treatment != "control", #replicates are necessarily symetric
              codon %nin% c("TAA","TGA","TAG","ATG","TGG") #STOP, Trp, Met
              ),
       aes(x=x_label, y=treatment, 
           fill= p_value_color #Remember, the log gives another negative -1 
           ))+
  geom_tile() +
  geom_vline(xintercept = c(13.5, 30.5, 46.5 ), color="black", size=1) +
  scale_fill_manual(values=c("No enrichment"="grey70", 
                             "Enriched codons"="red3", 
                             "Depleted codons"="blue3")) +
  # facet_nested( treatment ~ .) +
  theme(legend.position = "top",
        legend.title = element_blank(),
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0, size=16),
        axis.text.y = element_text(size=12, color="black"),
        axis.text.x = element_text(size=12, color="black", angle=90, vjust=0.5),
        axis.title.y = element_blank(),
        axis.title.x = element_blank() )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_4X_codon_use_andTE"
width=6
height=2
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```

##Fig 4F - Specific codon
###Lys codons change in Z-score
```{r}
#Calculate percentile; take top and bottom 20% for each rep, stress
HIGH_percentile = 0.66
LOW_percentile = 0.33

temp4 <- temp3 %>%
  group_by(rep, treatment ) %>%
  mutate( change_Z_percent_rank = percent_rank(change_zscore)) 

#Merge with codons 
temp5 <- temp4 %>% 
  mutate(percentile_grouping = ifelse(change_Z_percent_rank <= LOW_percentile, "decreased TE", 
                                      ifelse( change_Z_percent_rank >= HIGH_percentile, "increased TE", "no change") ) ) %>%
  filter(!is.na(change_Z_percent_rank)) %>% #Some genes with poor polysome detection result in NA percentiles
  select(rep, treatment, gene_symbol, percentile_grouping) %>%
  full_join(., codon_dict, by=c("gene_symbol")) %>%
  filter(!is.na(rep)) %>% #Codon_dict has more genes than are in the TE dataset
  pivot_longer(col=c(-rep, -treatment, -gene_symbol, -percentile_grouping),
               names_to = c("codon"), values_to=c("frequency")) %>%
  full_join(., AA_table, by=c("codon")) %>%
  group_by(treatment, rep, codon) %>%
  mutate(frequency_percentile = percent_rank(frequency))

temp5$percentile_grouping <- factor(temp5$percentile_grouping, levels=c("decreased TE","no change","increased TE"))
my_comparisons <- list( c("decreased TE", "no change"),
                        c("no change", "increased TE"),
                        c("increased TE", "decreased TE") )

plot1 <- ggplot(filter(temp5,
              AA =="Lys",
              # codon %in% c("AGC","AGT"),
              # rep==1,
              treatment != "control",
              treatment == "AsO2",
              ),
       aes(y=frequency*1000, x=interaction(percentile_grouping), 
           # color=rep
           ) ) +
  geom_boxplot() +
  scale_y_log10() +
  coord_cartesian(ylim = c(0.5, 2500)) +
  stat_compare_means(comparisons = my_comparisons, size=4,  
                     method="wilcox.test",
                     p.adjust.method = "none",
                     label="p.signif"
                     ) +
  facet_nested(. ~ codon  ) +
  ylab("Codon use per thousand") +
  theme(#legend.position = "top",
        #legend.title = element_blank(),
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0, size=16),
        axis.text.y = element_text(size=12, color="black"),
        axis.text.x = element_text(size=12, color="black", angle=90, vjust=0.5, hjust=1),
        axis.title.y = element_text(size=16),
        axis.title.x = element_blank() )


#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_4Y_Lys_codon_TE"
width=3
height=3
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)


```




##Fig 4G - abundance of isoacceptors
###Wrangle and go
```{r}
#Filter bin, rpm, average replicates
temp <-new_SE_HEK_FULL_count %>%
  filter(bin_start =="-2") %>%
  select(-bin_stop) %>%
  group_by(rep, sample, treatment, AA, anticodon, DM) %>%
  summarise(sum_count = sum(count)) %>%
  group_by(treatment, sample, rep, DM) %>%
  mutate(rpm = sum_count / sum(sum_count)) %>%
  group_by(treatment, sample, DM, AA, anticodon) %>%
  mutate(anticodon_mean = mean(rpm, na.rm=T) ) %>%
  filter(AA != "SeCe") #Word from on high

#add a column for control
temp <- temp %>%
  filter(treatment =="control") %>%
  group_by(sample, DM, AA, anticodon) %>%
  summarise(control_mean = mean(anticodon_mean, na.rm = T)) %>% #Use summarise to collapse rows
  select(AA, anticodon, sample, DM, control_mean) %>%
  full_join(temp,., by=c("AA", "anticodon", "sample", "DM"))

temp <- temp %>%
  separate(anticodon, sep=c(1,2), into=c("first","second","third"), remove=F)


temp$treatment <- factor(temp$treatment, levels=c("control","heat","H2O2", "AsO2"))
temp$first <- factor(temp$first, levels=c("T","A","G","C"))

my_comparisons <- list( c("A", "G"),
                        c("T","C")
                        )



plot1 <- ggplot(filter(temp,
              treatment !="control",
              sample=="polysome",
              # AA %nin% c("Meti", "SeCe"),
              !grepl("mt", AA),
              ),
       aes(x=first, y=rpm / control_mean
           )) +
  geom_hline(yintercept = 1, linetype=2, alpha=0.3) +
  geom_boxplot() +
  # geom_text_repel(data=filter(temp, sample=="polysome", treatment !="control",!grepl("mt", AA), rpm / control_mean>1.3), aes(label=paste0(AA, anticodon)),color="red") +
  xlab("tRNA gene wobble position") +
  ylab("Change in isoaccepter abundance\n(stress/ control)") +
  scale_y_log10() +
  stat_compare_means(comparisons = my_comparisons, size=3,  method="t.test", label="p.signif")  +
  facet_nested(.~  treatment) +
  theme(legend.position = "top",
        legend.title = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_text(angle=0, size=16),
        axis.text = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16) )


#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_4G_tRNA_wobble_abundance"
width=3
height=3
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```


##Fig 4H - T/C ratio among amino acids
###Start from above wrangling
```{r}

temp1 <- temp
temp1$first <- recode(temp1$first, "T"="t", "C"="c", "A"="a","G"="g")
temp1 <- temp1 %>%
  # filter(first %in% c("t","c")) %>% 
  mutate(rpm = rpm /control_mean ) %>%
  ungroup() %>%
  group_by(AA, first, treatment, rep, sample) %>%
  summarise(rpm = sum(rpm) ) %>%
  select(treatment, rep, AA, rpm, first, sample) %>%
  pivot_wider(names_from=c("first"), values_from=c("rpm")) 

my_comparisons <- list( c("control", "heat"),
                        c("control","H2O2"),
                        c("control","AsO2")
                        )

plot1 <- ggplot(filter(temp1,
                       !grepl("m1",AA),
                       ), 
       aes(x=treatment, y=t / c )) +
  geom_hline(yintercept = 1, linetype=2, alpha=0.3) +
  geom_boxplot() +
  # geom_point(data=filter(temp1, !grepl("mt", AA), AA=="Lys"), color="red") +
  stat_compare_means(comparisons = my_comparisons, size=4,  method="t.test",  label="p.signif") + #how does this compute p-value?
  scale_y_log10() +
  ylab("Ratio of T34 to C34 isoacceptors\nfor each amino acid") +
  facet_nested(.~sample) +
  theme(legend.position = "top",
        legend.title = element_blank(),
        strip.background = element_blank(),
        strip.text.x = element_text(angle=0, size=16),
        axis.text = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_blank() )


#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_4H_tRNA_TC_ratio"
width=4
height=3
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```



#_
#Figure 5

##Fig 5A total fragment abundance
###Wrangle, group by bin and stress
```{r}
#Make things factors for fast math and easy plotting
temp <- new_SE_HEK_FULL_count
temp$treatment = factor(temp$treatment, levels=c("control","heat","H2O2","AsO2"))
temp$AA = factor(temp$AA)
temp$AA = factor(temp$AA, levels=rev(levels(temp$AA)))
temp$anticodon = factor(temp$anticodon)
temp$anticodon = factor(temp$anticodon, levels=rev(levels(temp$anticodon)))
temp$source = factor(temp$source)
temp$DM = factor(temp$DM)
temp <- temp %>%
  mutate(AA_group=AA)

#I don't think there is a better way to do this
temp$AA_group <- recode(temp$AA_group, 
                        "Leu"="hydrophobic",
                        "Ile"="hydrophobic",
                        "Phe"="hydrophobic",
                        "Val"="hydrophobic",
                        "Trp"="hydrophobic",
                        "Mete"="hydrophobic",
                        "Meti"="hydrophobic",
                        "mtLeu"="hydrophobic",
                        "mtIle"="hydrophobic",
                        "mtPhe"="hydrophobic",
                        "mtVal"="hydrophobic",
                        "mtTrp"="hydrophobic",
                        "mtMet"="hydrophobic",
                        
                        "Ala"="small",
                        "Cys"="small",
                        "Pro"="small",
                        "Gly"="small",
                        "mtAla"="small",
                        "mtCys"="small",
                        "mtPro"="small",
                        "mtGly"="small",
                        
                        "Arg"="charged",
                        "Lys"="charged",
                        "Asp"="charged",
                        "Glu"="charged",
                        "mtArg"="charged",
                        "mtLys"="charged",
                        "mtAsp"="charged",
                        "mtGlu"="charged",
                        
                        "Ser"="polar",
                        "Thr"="polar",
                        "Tyr"="polar",
                        "His"="polar",
                        "Asn"="polar",
                        "Gln"="polar",
                        "SeCe"="polar",
                        "mtSer"="polar",
                        "mtThr"="polar",
                        "mtTyr"="polar",
                        "mtHis"="polar",
                        "mtAsn"="polar",
                        "mtGln"="polar",
                        "mtSeCe"="polar")

#More factors for easy
temp$AA_group <- factor(temp$AA_group, levels=rev(c("hydrophobic","small","charged","polar")))
temp$gene = factor(temp$gene)
temp$gene = factor(temp$gene, levels=rev(levels(temp$gene)))

#Normalize abundance to sum in all bins (not -1,-2,-3) for all tRNAs per rep
temp <- temp %>%
  #filter(source !="mitochondrial") %>%
  group_by(treatment, DM, sample, rep, bin_start) %>%
  summarise(bin_sum = sum(count)) %>% #Sum all tRNAs together
  filter(bin_start %nin% c("-1","-2","-3")) %>%
  group_by(treatment, DM, sample, rep) %>%
  mutate(bin_fraction = bin_sum / sum(bin_sum)) #Scale by total number of mapped tRNA reads

#Add a column for mean
temp <- temp %>%
  group_by(treatment, DM, sample, bin_start) %>%
  mutate(mean_bin_fraction = mean(bin_fraction, na.rm=T))

temp$bin_start <- recode(temp$bin_start, 
                         "20"="20-29",
                         "30"="30-39",
                         "40"="40-49",
                         "50"="50-59",
                         "60"="60+")

plot1 <- ggplot(filter(temp, 
              DM !="minus",
              bin_start %nin% c("20-29"),
              #treatment=="control",
              sample=="input"),
       aes(x=bin_start, y=bin_fraction, 
           color=treatment, group=interaction(treatment, bin_start)
           )) +
  geom_point(position=position_dodge(width=0.75)) +
  geom_point(aes(y=mean_bin_fraction), shape=95, size=5, position=position_dodge(width=0.75)) +
  scale_color_manual(values= alpha(c("heat"="#f76f73", "H2O2"="#07c4c5", "AsO2"="#027fdc", "control"="grey70"), 1), 
                     guide = guide_legend(reverse = FALSE) )+
  scale_y_log10(labels=trans_format('log10',math_format(10^.x)),
                breaks=c(0.0001, 0.001, 0.01, 0.1, 1)) +
  ylab("Fraction of tRNA reads") +
  xlab("Location of read 3' end") +
  theme(legend.position = "right",
        strip.background = element_blank(),
        strip.text.x = element_text(angle=0, size=16),
        axis.text = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16) )



#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_5A_fragment_abundance"
width=3
height=2
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)


```


##Fig 5B - Gly fragment traces
###Need a new color palette
```{r}

temp <- new_SE_HEK_FULL_base %>%
  filter(pileup >=10,
         treatment=="control",
         bin_start %nin% c("-1","-2","-3"),
         sample=="input",
         DM !="minus")
temp$bin_start <- recode(temp$bin_start, 
                         "20"="20-29",
                         "30"="30-39",
                         "40"="40-49",
                         "50"="50-59",
                         "60"="60+")

plot1 <- ggplot(filter(temp,
              anticodon=="CCC",
              position <= 72,
              grepl("chr16.trna34",gene),
              bin_start %nin%c("20-29")
              ),
       aes(x=position, y= pileup, color=bin_start, group=interaction(rep, bin_start)) ) +
  geom_line(size=1) +
  # coord_cartesian(xlim = c(0,73)) +
  scale_y_log10(labels=trans_format('log10',math_format(10^.x)) ) +
  scale_color_manual(values=c("60+"="#1BB6AF",
                              "50-59"="#E9A17C",
                              "40-49"="#0076BB",
                              "30-39"="#dad371" ), guide = guide_legend(reverse = TRUE)) +
  geom_vline(xintercept = c(30,40,50,60), linetype=2, alpha=0.2) +
  labs(color="Location of\n3' end") +
  ylab("Read pileup") +
  xlab("position") +
  theme(legend.position = "right",
        strip.background = element_blank(),
        strip.text.x = element_text(angle=0, size=16),
        axis.text = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16) )


#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_5B_GlyCCC_fragment_trace"
width=3
height=2
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```


##Fig 5C - m3C47d modification in fragments
###Keep the same colors
```{r}
temp <- new_SE_HEK_FULL_base 
temp$bin_start <- recode(temp$bin_start, 
                         "20"="20-29",
                         "30"="30-39",
                         "40"="40-49",
                         "50"="50-59",
                         "60"="60+")
temp$chr <- str_replace_all(temp$chr, "chr","c")
temp$chr <- str_replace_all(temp$chr, "trna","t")


temp <- temp %>%
  filter(bin_start %nin% c("-1","-2","-3")) %>%
  filter( (gene == "Homo_sapiens_chr1.trna34-LeuCAG"  & position == 51) |
          (gene == "Homo_sapiens_chr16.trna17-LeuCAG" & position == 51) | #pos ???
          (gene == "Homo_sapiens_chr17.trna35-SerAGA"  & position == 50) |
          (gene == "Homo_sapiens_chr17.trna41-SerCGA"  & position == 50) |
          (gene == "Homo_sapiens_chr6.trna35-SerCGA"  & position == 50) |
          (gene == "Homo_sapiens_chr15.trna10-SerGCT"  & position == 50) |
          (gene == "Homo_sapiens_chr6.trna175-SerGCT"  & position == 52) |
          (gene == "Homo_sapiens_chr6.trna31-SerGCT"  & position == 50) |
          (gene == "Homo_sapiens_chr6.trna43-SerGCT"  & position == 50) |
          (gene == "Homo_sapiens_chr6.trna62-SerGCT"  & position == 50) |
          (gene == "Homo_sapiens_chr10.trna2-SerTGA"  & position == 50) |
          (gene == "Homo_sapiens_chr6.trna172-SerTGA"  & position == 50) 
          ) %>%
  group_by(sample, gene, treatment, position, bin_start) %>%
  mutate(mean_mutation = mean(mutation, na.rm=T)) %>%
  mutate(m3C_site = mutation,
         x_label=paste0(AA,anticodon, " ", chr))


plot1 <- ggplot(filter(temp,
              sample=="input",
              treatment=="control"),
       aes(x=x_label, y=m3C_site, color=bin_start)) +
  geom_point() +
  geom_vline(xintercept = 2.5, linetype=2, alpha=0.6) +
  geom_point(aes(y=mean_mutation), shape=95, size=5) +
  scale_y_continuous(breaks=c(0,0.5,1), limits=c(0,1)) +
  scale_color_manual(values=c("60+"="#1BB6AF",
                              "50-59"="#E9A17C"
                              ), guide = guide_legend(reverse = TRUE)) +
  labs(color="Mapping of\n3' end") +
  ylab("Mutation at m3C47d") +
  theme(legend.position = "right",
        strip.background = element_blank(),
        axis.text.x = element_text(angle=90, size=12, color="black", vjust=0.5, hjust=0),
        axis.text.y = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_blank() )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_5C_m3C47d_fragments"
width=3.5
height=3
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)



```


## Fig 5D - m1G37 modification fragments
###Make the nice plot
```{r}
temp <- new_SE_HEK_FULL_base %>%
  filter(pileup >=30,
         treatment=="control",
         bin_start %nin% c("-1","-2","-3"),
         sample=="input",
         DM !="minus")

temp$bin_start <- recode(temp$bin_start, 
                         "20"="20-29",
                         "30"="30-39",
                         "40"="40-49",
                         "50"="50-59",
                         "60"="60+")
temp$chr <- str_replace_all(temp$chr, "chr","c")
temp$chr <- str_replace_all(temp$chr, ".trna","t")

temp <- temp %>%
  filter(bin_start %nin% c("-1","-2","-3")) %>%
  filter( #(gene == "Homo_sapiens_chr1.trna34-LeuCAG"  & position == 51) |
          AA=="Leu", anticodon=="CAA", position==38,
          gene %nin% c("Homo_sapiens_chr1.trna66-LeuCAA",
                       "Homo_sapiens_chr11.trna1-LeuCAA")
          ) %>%
  group_by(sample, gene, treatment, position, bin_start) %>%
  mutate(mean_mutation = mean(mutation, na.rm=T)) %>%
  mutate(m3C_site = mutation,
         x_label=paste0(AA,anticodon, " ", chr))


plot1 <- ggplot(filter(temp,
              sample=="input",
              treatment=="control",
              bin_start %in% c("60+","50-59", "40-49")
              ),
       aes(x=x_label, y=m3C_site, color=bin_start)) +
  geom_point() +
  geom_point(aes(y=mean_mutation), shape=95, size=5) +
  scale_color_manual(values=c("60+"="#1BB6AF",
                              "50-59"="#E9A17C",
                              "40-49"="#0076BB",
                              "30-39"="#dad371" ), guide = guide_legend(reverse = TRUE)) +
  labs(color="Mapping of\n3' end") +
  scale_y_continuous(breaks=c(0, .10, .20, .30),
                     limits=c(0, .31)) +
  #scale_color_manual(values = c("50-59"="#00BFC4","60+"="#C77CFF")) +
  ylab("Mutation at m1G37") +
  theme(legend.position = "right",
        strip.background = element_blank(),
        axis.text.x = element_text(angle=90, size=12, color="black", vjust=0.5, hjust=0),
        axis.text.y = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_blank() )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_5D_m1G37_fragments"
width=2.5
height=3
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)


```

###double check the coverage on 30-39 bin
```{r}
temp <- new_SE_HEK_FULL_base %>%
  filter(pileup >=10,
         treatment=="control",
         bin_start %nin% c("-1","-2","-3"),
         sample=="input",
         DM !="minus")
temp$bin_start <- recode(temp$bin_start, 
                         "20"="20-29",
                         "30"="30-39",
                         "40"="40-49",
                         "50"="50-59",
                         "60"="60+")

# plot1 <- 
ggplot(filter(temp,
              AA=="Leu",
              anticodon=="CAA",
              gene %nin% c("Homo_sapiens_chr1.trna66-LeuCAA",
                       "Homo_sapiens_chr11.trna1-LeuCAA"),
              position <= 72,
              bin_start %nin%c("20-29")
              ),
       aes(x=position, y= pileup, color=bin_start, group=interaction(rep, bin_start, gene)) ) +
  geom_line(size=1) +
  geom_vline(xintercept = 38, linetype=2, color="red", alpha=0.3) +
  # coord_cartesian(xlim = c(0,73)) +
  scale_y_log10(labels=trans_format('log10',math_format(10^.x)) ) +
  scale_color_manual(values=c("60+"="#1BB6AF",
                              "50-59"="#E9A17C",
                              "40-49"="#0076BB",
                              "30-39"="#dad371" ), guide = guide_legend(reverse = TRUE)) +
  geom_vline(xintercept = c(30,40,50,60), linetype=2, alpha=0.2) +
  labs(color="Location of\n3' end") +
  ylab("Read pileup") +
  xlab("position") +
  facet_wrap(~gene) +
  theme(legend.position = "right",
        strip.background = element_blank(),
        strip.text.x = element_text(angle=0, size=16),
        axis.text = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16) )

```


##Fig 5E - m22G26
###copy paste heavily
```{r}

temp <- new_SE_HEK_FULL_base %>%
  filter(pileup >=30,
         treatment=="control",
         bin_start %nin% c("-1","-2","-3"),
         sample=="input",
         DM !="minus")

temp$bin_start <- recode(temp$bin_start, 
                         "20"="20-29",
                         "30"="30-39",
                         "40"="40-49",
                         "50"="50-59",
                         "60"="60+")
temp$chr <- str_replace_all(temp$chr, "chr","c")
temp$chr <- str_replace_all(temp$chr, ".trna","t")

temp <- temp %>%
  filter( #(gene == "Homo_sapiens_chr1.trna34-LeuCAG"  & position == 51) |
          AA=="Trp", anticodon=="CCA", position==25,
          ) %>%
  group_by(sample, gene, treatment, position, bin_start) %>%
  mutate(mean_mutation = mean(mutation, na.rm=T)) %>%
  mutate(m3C_site = mutation,
         x_label=paste0(AA,anticodon, " ", chr))


plot1 <- ggplot(filter(temp,
              sample=="input",
              treatment=="control",
              bin_start %in% c("60+","50-59", "30-39")
              ),
       aes(x=x_label, y=m3C_site, color=bin_start)) +
  geom_point() +
  geom_point(aes(y=mean_mutation), shape=95, size=5) +
  scale_color_manual(values=c("60+"="#1BB6AF",
                              "50-59"="#E9A17C",
                              "40-49"="#0076BB",
                              "30-39"="#dad371" ), guide = guide_legend(reverse = TRUE)) +
  labs(color="Mapping of\n3' end") +
  scale_y_continuous(#breaks=c(0, .10, .20, .30),
                     # limits=c(0, .31)
                     ) +
  #scale_color_manual(values = c("50-59"="#00BFC4","60+"="#C77CFF")) +
  ylab("Mutation at m22G26") +
  theme(legend.position = "right",
        strip.background = element_blank(),
        axis.text.x = element_text(angle=90, size=12, color="black", vjust=0.5, hjust=0),
        axis.text.y = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_blank() )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_5E_m22G26_fragments"
width=2.5
height=3
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)
```

###double check the coverage 
```{r}
temp <- new_SE_HEK_FULL_base %>%
  filter(pileup >=10,
         treatment=="control",
         bin_start %nin% c("-1","-2","-3"),
         sample=="input",
         DM !="minus")

temp$bin_start <- recode(temp$bin_start, 
                         "20"="20-29",
                         "30"="30-39",
                         "40"="40-49",
                         "50"="50-59",
                         "60"="60+")
temp$chr <- str_replace_all(temp$chr, "chr","c")
temp$chr <- str_replace_all(temp$chr, ".trna","t")

# plot1 <- 
ggplot(filter(temp,
              AA=="Trp",
              anticodon=="CCA",
              position <= 72,
              bin_start %nin%c("20-29")
              ),
       aes(x=position, y= pileup, color=bin_start, group=interaction(rep, bin_start, gene)) ) +
  geom_line(size=1) +
  geom_vline(xintercept = 38, linetype=2, color="red", alpha=0.3) +
  # coord_cartesian(xlim = c(0,73)) +
  scale_y_log10(labels=trans_format('log10',math_format(10^.x)) ) +
  scale_color_manual(values=c("60+"="#1BB6AF",
                              "50-59"="#E9A17C",
                              "40-49"="#0076BB",
                              "30-39"="#dad371" ), guide = guide_legend(reverse = TRUE)) +
  geom_vline(xintercept = c(30,40,50,60), linetype=2, alpha=0.2) +
  labs(color="Location of\n3' end") +
  ylab("Read pileup") +
  xlab("position") +
  facet_wrap(~gene) +
  theme(legend.position = "right",
        strip.background = element_blank(),
        strip.text.x = element_text(angle=0, size=16),
        axis.text = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16) )

```




#_
#Figure 6
##Preamble
###SE mapped; 5S rRNA RF00001
####count
```{r}
PATH="../microbe_oral/6_sam_counter/rRNA_5S_combine/"
FILES <-data.frame( file_name = list.files(PATH) ) %>%
  filter(!grepl("unassigned",file_name),
         !grepl("kallisto", file_name))

#Split generic file names into salient parts
FILES <- FILES %>%
  separate(file_name, sep="_", c("library","junk1", "junk2", "junk3", "barcode", "junk4"), fill="right", remove=FALSE) %>%
  select(-junk1, -junk2, -junk3, -junk4) 

FILES <- FILES %>%
  mutate(volunteer = barcode,
         day = barcode,
         rep = library,
         DM = library) %>%
  select(-library, -barcode)

#PROJECT SPECIFIC DECODING
FILES$file_name <- as.character(FILES$file_name)
FILES$volunteer <- recode(FILES$volunteer, 
                          "bc1"="v03",
                          "bc2"="v03",
                          "bc3"="v04",
                          "bc4"="v04",
                          "bc5"="v05",
                          "bc6"="v05",
                          "bc7"="v06",
                          "bc8"="v06"
                          )
FILES$day <- recode(FILES$day, 
                          "bc1"="a",
                          "bc2"="b",
                          "bc3"="a",
                          "bc4"="b",
                          "bc5"="a",
                          "bc6"="b",
                          "bc7"="a",
                          "bc8"="b"
                          )
FILES$DM <- recode(FILES$DM, 
                    "TP-CK-2S-CW1"="minus", 
                    "TP-CK-2S-CW2"="plus"
                    )

#=======================+
# FILES <- FILES %>%
#   mutate(bin_start = ifelse(is.na(bin_start), -3, bin_start),
#          bin_stop = ifelse(is.na(bin_stop), -3, bin_stop))
#========================

FILES <- FILES %>%
  mutate(sample="oral") %>%
  filter(!is.na(rep))

read_in_one <- function(row){
  output <- read.csv(paste0(PATH, row$file_name), header=T, sep="\t") %>%
    mutate(sample = row$sample,
           volunteer = row$volunteer,
           DM = row$DM,
           day = row$day,
           volunteer = row$volunteer,
           bin_start = row$bin_start
           )
  #specify data types since enpty dataframes get confused and sad
  output$gene <- as.character(output$name)
  return(output)
}

oral_SE_5S_count_data <- FILES %>%
  group_by(file_name) %>%
  do(read_in_one(.)) %>%
  select(-name) %>%
  ungroup()
```

####Extra processing
```{r}
#Re-read in rRNA 5S taxon info
PATH = "./rRNA_5s_combine_lineage.tsv"
rRNA_5S_taxon <- read.csv(#"~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_SRP/6_sam_counter/rRNA_5S_combine/rRNA_5s_combine_lineage.tsv", 
  PATH,
                          sep="\t", na.strings = "NA") %>%
  mutate(type="rRNA_5S", sub_type="rRNA_5S") %>%
  separate(name, sep="_", c("gene", "other")) %>%
  select(-other)

oral_SE_5S_count <- oral_SE_5S_count_data %>%
  ungroup() %>%
  full_join(., rRNA_5S_taxon, by=c("gene"))

```

###SE mapped; 5S rRNA RF00001
####count
```{r}
PATH="../microbe_oral/6_sam_counter/SRP_combine/"
FILES <-data.frame( file_name = list.files(PATH) ) %>%
  filter(!grepl("unassigned",file_name),
         !grepl("kallisto", file_name))

#Split generic file names into salient parts
FILES <- FILES %>%
  separate(file_name, sep="_", c("library","junk1", "junk2", "junk3", "barcode", "junk4"), fill="right", remove=FALSE) %>%
  select(-junk1, -junk2, -junk3, -junk4) 

FILES <- FILES %>%
  mutate(volunteer = barcode,
         day = barcode,
         rep = library,
         DM = library) %>%
  select(-library, -barcode)

#PROJECT SPECIFIC DECODING
FILES$file_name <- as.character(FILES$file_name)
FILES$volunteer <- recode(FILES$volunteer, 
                          "bc1"="v03",
                          "bc2"="v03",
                          "bc3"="v04",
                          "bc4"="v04",
                          "bc5"="v05",
                          "bc6"="v05",
                          "bc7"="v06",
                          "bc8"="v06"
                          )
FILES$day <- recode(FILES$day, 
                          "bc1"="a",
                          "bc2"="b",
                          "bc3"="a",
                          "bc4"="b",
                          "bc5"="a",
                          "bc6"="b",
                          "bc7"="a",
                          "bc8"="b"
                          )
FILES$DM <- recode(FILES$DM, 
                    "TP-CK-2S-CW1"="minus", 
                    "TP-CK-2S-CW2"="plus"
                    )

#=======================+
# FILES <- FILES %>%
#   mutate(bin_start = ifelse(is.na(bin_start), -3, bin_start),
#          bin_stop = ifelse(is.na(bin_stop), -3, bin_stop))
#========================

FILES <- FILES %>%
  mutate(sample="oral") %>%
  filter(!is.na(rep))

read_in_one <- function(row){
  output <- read.csv(paste0(PATH, row$file_name), header=T, sep="\t") %>%
    mutate(sample = row$sample,
           volunteer = row$volunteer,
           DM = row$DM,
           day = row$day,
           volunteer = row$volunteer,
           bin_start = row$bin_start
           )
  #specify data types since enpty dataframes get confused and sad
  output$gene <- as.character(output$name)
  return(output)
}

oral_SE_SRP_count_data <- FILES %>%
  group_by(file_name) %>%
  do(read_in_one(.)) %>%
  select(-name) %>%
  ungroup()
```

####Extra processing
```{r}
#Re-read in rRNA 5S taxon info
SRP_short <- read.csv("./RF00169_lineage.tsv", sep="\t", na.strings = "NA") %>%
  mutate(sub_type="short_SRP")
SRP_long <- read.csv("./RF01854_lineage.tsv", sep="\t", na.strings = "NA") %>%
  mutate(sub_type="long_SRP")

rfam_SRP_taxon <- rbind(SRP_short, SRP_long) %>%
  mutate(type="rfam_SRP") %>%
  mutate(gene = name) %>%
  select(-name)

temp <- oral_SE_SRP_count_data %>%
  ungroup() %>%
  separate(gene, sep="\\.", into=c("gene","junk")) %>%
  select(-junk) %>%
  full_join(., rfam_SRP_taxon, by=c("gene"))

oral_SE_SRP_count <- temp
```

###Anvio mapped; tRNA  
####count
```{r}
PATH="../microbe_oral/new_oral/ORAL_TAXONOMY/"
Folders <-list.files(PATH)
FILES <-data.frame( file_name = paste(PATH, Folders,"/", Folders, "_order_aa_counts.tsv", sep="") )



#_order_aa_counts.tsv

#Split generic file names into salient parts
FILES <- FILES %>%
  separate(file_name, sep="/", remove=F, into=c("junk1", "junk2", "junk3", "junk4", "good_stuff"), extra="drop") %>%
  select(-junk1, -junk2, -junk3, -junk4) %>%
  separate(good_stuff, sep="_", into=c("volunteer", "day", "DM"))



#PROJECT SPECIFIC DECODING
FILES$file_name <- as.character(FILES$file_name)
FILES$volunteer <- recode(FILES$volunteer, 
                          "tongue1"="v03",
                          "tongue2"="v04",
                          "tongue3"="v05",
                          "tongue4"="v06",
                          )
FILES$day <- recode(FILES$day, 
                          "day1"="a",
                          "day2"="b"
                          )
FILES$DM <- recode(FILES$DM, 
                    "untreated"="minus", 
                    "demethylase"="plus"
                    )

#=======================+
# FILES <- FILES %>%
#   mutate(bin_start = ifelse(is.na(bin_start), -3, bin_start),
#          bin_stop = ifelse(is.na(bin_stop), -3, bin_stop))
#========================

FILES <- FILES %>%
  mutate(sample="oral") %>%
  filter(!is.na(rep))

read_in_one <- function(row){
  output <- read.csv(row$file_name, header=T, sep="\t") %>%
    mutate(sample = row$sample,
           volunteer = row$volunteer,
           DM = row$DM,
           day = row$day,
           volunteer = row$volunteer,
           #bin_start = row$bin_start
           )
  # #specify data types since enpty dataframes get confused and sad
  # output$gene <- as.character(output$name)
  return(output)
}

oral_tRNA_data <- FILES %>%
  group_by(file_name) %>%
  do(read_in_one(.)) %>%
  ungroup() %>%
  mutate(order=X,
         total=X.1,
         type="tRNA",
         sub_type="tRNA_2020") %>%
  select(-X, -X.1) %>%
  pivot_longer(col=c(-volunteer, -day,-sample, -DM, -type, -sub_type, -order, -file_name),
               names_to = c("AA"), values_to=c("count") ) #%>%
  #mutate(gene=NA, taxonid=NA, kingdom=NA, phylum=NA, class=NA, family=NA, genus=NA, species=NA)
```



####Combine 5S SRP tRNA
```{r}

oral_SE_count <- rbind(oral_SE_5S_count, oral_SE_SRP_count)

temp <- oral_SE_count %>%
  group_by(volunteer, day, sample, DM, type) %>%
  mutate(rpm = count / sum(count)) %>%
  filter(!is.na(type)) %>%
  select(-file_name, -gene, -sub_type, taxonid) %>%
  group_by(volunteer, day, sample, DM, type, kingdom, phylum, class, order) %>%
  summarise(sum_rpm = sum(rpm)) %>%
  pivot_wider(names_from = c("type"), values_from = c("sum_rpm")) %>%
  filter(!grepl("Candidatus", phylum))

#c(-volunteer, -day,-sample, -DM, -type, -sub_type, -order, -file_name)
oral_tRNA_count <- oral_tRNA_data %>%
  #normalize to fraction of mapped tRNAs
  filter(AA=="total") %>%
  group_by(volunteer, day, sample, DM, type) %>%
  summarise(sum_count = sum(count, na.rm=T)) %>%
  select(volunteer, day, sample, DM, type, sum_count) %>%
  full_join(., oral_tRNA_data, by=c("volunteer","day","sample","DM","type")) %>%
  mutate(tRNA = count / sum_count) %>%
  select(-count, -sum_count, -file_name, -type, -sub_type) %>%
  full_join(.,temp, by=c("volunteer","day","sample","DM", "order")) %>%
  filter(!is.na(order), 
         order !="")

```



##Fig 6C - SRP, 5S, tRNA
###SRP vs. 5S
```{r}

temp <- oral_SE_count %>%
  group_by(volunteer, day, sample, DM, type) %>%
  mutate(rpm = count / sum(count) *1000000) %>%
  filter(!is.na(type)) %>%
  select(-file_name, -gene, -sub_type, taxonid) %>%
  group_by(volunteer, day, sample, DM, type, kingdom, phylum, class, order) %>%
  summarise(sum_rpm = sum(rpm)) %>%
  pivot_wider(names_from = c("type"), values_from = c("sum_rpm")) %>%
  filter(!grepl("Candidatus", phylum))


plot1 <- ggplot(filter(temp,
              DM=="plus",
              #volunteer=="v03",
              #day=="a"
              phylum %in% c("Actinobacteria", "Bacteroidetes","Firmicutes","Proteobacteria","Spirochaetes","Tenericutes")
              ), 
       aes(x=rfam_SRP, y=rRNA_5S, color=phylum)) +
  geom_point() +
  scale_color_manual(values=c("Actinobacteria"="grey20", "Bacteroidetes"="red", "Firmicutes"="dodgerblue2",
                              "Proteobacteria"="olivedrab3", "Spirochaetes"="mediumorchid1", "Tenericutes"="goldenrod1")) +
  scale_x_log10(labels=trans_format('log10',math_format(10^.x)), breaks=c(100000, 1000, 1), limits=c(0.5, 400000) ) +
  scale_y_log10(labels=trans_format('log10',math_format(10^.x)), breaks=c(100000, 1000, 1), limits=c(0.5, 400000) ) +
  geom_abline(slope=1, linetype=2, alpha=0.3) +
  theme(legend.position = "right",
        strip.background = element_blank(),
        axis.text = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16) )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_6C_SRP_vs_5S"
width=4
height=3
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```


##Fig 6D - SRP vs tRNA
###SRP vs tRNA total
```{r}

temp <- oral_tRNA_count

ggplot(filter(temp,
              DM=="plus",
              AA=="total",
              phylum %in% c("Actinobacteria", "Bacteroidetes","Firmicutes","Proteobacteria","Spirochaetes","Tenericutes")
              ), 
       aes(x=rfam_SRP, y=tRNA, color=phylum)) +
  geom_point() +
  scale_x_log10() +
  scale_y_log10() +
  geom_abline(slope=1, linetype=2, alpha=0.3) 
  #theme(legend.position = "None")


```

##Fig 6E - 5S vs. tRNA
###rRNA5S vs tRNA total
```{r}

temp <- oral_tRNA_count

ggplot(filter(temp,
              DM=="plus",
              AA=="total",
              phylum %in% c("Actinobacteria", "Bacteroidetes","Firmicutes","Proteobacteria","Spirochaetes","Tenericutes")
              ), 
       aes(x=rfam_SRP, y=tRNA, color=phylum)) +
  geom_point() +
  scale_x_log10() +
  scale_y_log10() +
  geom_abline(slope=1, linetype=2, alpha=0.3) 
  #theme(legend.position = "None")


```


##Fig 6F - AA with A34 codons
###Search all bacterial tRNAs in gtRNAdb for "A" 34
```{r}
PATH <- "../microbe_oral/all_bacterial_tRNA/gtrnadb-search134145.fa"
temp <- read.csv(PATH, header=F) %>%
  mutate(row = (row_number()-1) %/% 2,
         feature = ifelse((row_number()-1) %% 2 ==0, "gene", "sequence")) %>%
  ungroup() %>%
  mutate(rest = V1)

# asdf <- temp %>%
#   filter(row==181604)

temp1 <- temp %>%
  select(-V1) %>% 
  #filter(row  < 10000) %>%
  group_by(row) %>%
  pivot_wider(names_from = c("feature"), values_from=c("rest") )

#Extract the score from the name
temp1 <- temp1 %>%
  separate(gene, sep=c("Sc: "), into=c("junk1", "score"), remove=F) %>%
  select(-junk1)

temp2 <- temp1 %>%
  ungroup() %>%
  select(gene, sequence, score) %>%
  separate(gene, sep=" ", into=c("first"),extra = "drop") %>%
  separate(first, sep="_tRNA-", into=c("organism","tRNA")) %>%
  separate(tRNA, sep="-", into=c("AA", "anticodon", "n1", "n2")) %>%
  mutate(number = paste0(n1, "-", n2)) %>%
  select(-n1, -n2) %>%
  separate(anticodon, remove=F, sep=c(1,2), into=c("first", "second", "third"))

temp3 <- temp2 %>%
  filter(first=="A",
         score >= 50)

temp4 <- temp3 %>%
  group_by(AA) %>%
  summarise(count = n() + 0.1)

#Figure!
plot1 <- ggplot(temp4, 
       aes(x=AA, y=count)) +
  geom_bar(stat="identity") +
  scale_y_log10(breaks=c(1,100,10000)) +
  # geom_hline(yintercept = 75, linetype=2, alpha=0.3) +
  theme(axis.text.x = element_text(angle=90, vjust=0.5)) +
  ylab("Number of species with \nwobble A34 tRNA gene") +
  theme(legend.position = "right",
        strip.background = element_blank(),
        axis.text.x = element_text(angle=90, size=12, color="black", vjust=0.5, hjust=0),
        axis.text.y = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_blank() )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_6F_A34_anticodons"
width=2.5
height=2
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```

##Fig 6X - which genus have AAG Leu?
###continue from above
```{r}
temp4 <- temp3 %>%
  filter(AA=="Leu")
temp4 <- temp4$organism

temp5 <- temp2 %>%
  filter(organism %in% temp4) %>%
  group_by(organism, anticodon) %>%
  summarise(present = T) %>%
  group_by(organism) %>%
  pivot_wider(names_from = c("anticodon"), values_from=c("present")) %>%
  mutate(sum_anticodons = sum(AAG, CAA, CAG, GAG, TAA, TAG, na.rm=T))

tempZ <- filter(temp5,
                AAG==T, GAG==T)

temp6 <- temp5 %>%
  filter(sum_anticodons %in% c(4))

temp7 <- temp5 %>%
  filter(AAG==T &  is.na(GAG))

#Get genus and species
tempX <- temp7 %>%
  separate(organism, sep="_", into=c("genus","species"), remove=F, extra="drop") %>%
  separate(genus, sep=c(1), into=c("junk", "genus")) %>%
  select(-junk)

plot1 <- ggplot(tempX, aes(x=interaction(genus))) +
  geom_histogram(stat="count") +
  ylab("LeuAAG tRNA\ngene count") +
  theme(legend.position = "right",
        strip.background = element_blank(),
        axis.text.x = element_text(angle=90, size=12, color="black", vjust=0.5, hjust=0),
        axis.text.y = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_blank() )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_6X_species_LeuAAG"
width=3.5
height=2
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```


##Fig 6G - LeuAAG inosine trace
###SE mapped; select abundant reference
####Base
```{r}
PATH="../microbe_oral/5_tsv/abundant_oral_microbes+human_Ser_Leu/"
FILES <-data.frame( file_name = list.files(PATH) ) %>%
  filter(!grepl("unassigned",file_name))

#Split generic file names into salient parts
FILES <- FILES %>%
  separate(file_name, sep="_", c("library","junk1", "junk2", "junk3", "barcode", "junk4"), fill="right", remove=FALSE) %>%
  select(-junk1, -junk2, -junk3, -junk4) 

FILES <- FILES %>%
  mutate(volunteer = barcode,
         day = barcode,
         rep = library,
         DM = library) %>%
  select(-library, -barcode)

#PROJECT SPECIFIC DECODING
FILES$file_name <- as.character(FILES$file_name)
FILES$volunteer <- recode(FILES$volunteer, 
                          "bc1"="v03",
                          "bc2"="v03",
                          "bc3"="v04",
                          "bc4"="v04",
                          "bc5"="v05",
                          "bc6"="v05",
                          "bc7"="v06",
                          "bc8"="v06"
                          )
FILES$day <- recode(FILES$day, 
                          "bc1"="a",
                          "bc2"="b",
                          "bc3"="a",
                          "bc4"="b",
                          "bc5"="a",
                          "bc6"="b",
                          "bc7"="a",
                          "bc8"="b"
                          )
FILES$DM <- recode(FILES$DM, 
                    "TP-CK-2S-CW1"="minus", 
                    "TP-CK-2S-CW2"="plus"
                    )

#=======================+
# FILES <- FILES %>%
#   mutate(bin_start = ifelse(is.na(bin_start), -3, bin_start),
#          bin_stop = ifelse(is.na(bin_stop), -3, bin_stop))
# #========================

FILES <- FILES %>%
  mutate(sample="oral") %>%
  filter(!is.na(rep))

read_in_one <- function(row){
  output <- read.csv(paste0(PATH, row$file_name), header=T, sep="\t") %>%
    mutate(sample = row$sample,
           volunteer = row$volunteer,
           DM = row$DM,
           day = row$day,
           volunteer = row$volunteer,
           bin_start = row$bin_start
           )
  #specify data types since enpty dataframes get confused and sad
  output$gene <- as.character(output$gene)
  output$base <- as.character(output$base)
  return(output)
}

oral_SE_abundant_base_data <- FILES %>%
  group_by(file_name) %>%
  do(read_in_one(.)) %>%
  ungroup()

```

####Extra processing
```{r}
temp <- oral_SE_abundant_base_data %>%
  separate(gene, sep="_", remove=F, into=c("genus","species", extra="drop" )) %>%
  separate(gene, remove=F, sep="_tRNA-", into=c("other", "tRNA")) %>%
  separate(tRNA, sep="-", into=c("AA", "anticodon", "n1", "n2"))  %>%
  mutate(number = paste0(n1, "-", n2)) %>%
  select(-n1, -n2, -other) 

oral_SE_abundant_base <- temp

```

###Jump write into plotting
```{r}

temp <- oral_SE_abundant_base %>%
  filter(AA=="Leu",
         genus=="Streptococcus",
         anticodon %in% c("AAG"),
         DM =="plus") %>%
  mutate(color_label = volunteer)
temp$color_label <- str_replace_all(temp$color_label, "v0","")
temp <- temp %>%
  mutate(color_label=paste0(color_label,day))

temp$gene <- str_replace_all(temp$gene, "us_","us ")
temp$gene <- str_replace_all(temp$gene, "_tRNA","\ntRNA")



plot1 <- ggplot(filter(temp),
       aes(x=position, y=mutation, color=color_label) ) +
  # geom_text(data=filter(temp), aes(label=base), y=0, color="black") +
  geom_line() + 
  # facet_grid(cols=vars(gene)) +
  ylab("Mutation rate") +
  labs(color="Sample ID") +
  theme(legend.position = "right",
        strip.background = element_blank(),
        axis.text.x = element_text(angle=0, size=12, color="black", vjust=0.5, hjust=0),
        axis.text.y = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_blank() )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_6G_Strep_LeuAAG_trace"
width=3.5
height=2
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```





##Fig 6H - Leu codon usage
###Read in NCBI taxon info
```{r}

PATH="./Organismal_taxonomy/taxdump 2/nodes.dmp"
Nodes = read.csv(PATH, sep="\t", header=F) 
Nodes <- Nodes %>%
  select(V1, V3, V5)
colnames(Nodes) <- c("taxonID", "parentID","Rank")
Nodes <- Nodes %>%
  filter(Rank=="no rank") %>%
  mutate(Rank="no_rank") %>%
  rbind(filter(Nodes, Rank!="no rank")) %>%
  mutate(taxonID = as.numeric(taxonID),
         parentID = as.numeric(parentID),
         Rank = as.factor(Rank))

PATH="./Organismal_taxonomy/taxdump 2/names.dmp"
Names = read.csv(PATH, sep="\t", header=F) 
Names <- Names %>%
  select(V1, V3, V7) %>%
  filter(V7 =="scientific name") %>%
  select(-V7) %>%
  mutate(V1 = as.numeric(V1))
colnames(Names) <- c("taxonID","name")
```

###Make big taxon table
```{r}

recursive_taxon3 <- function(df, Nodes, Names, rank="species"){
  link = filter(Nodes, taxonID %in% df$taxonID)
  temp_data <- df %>%
    full_join(., link, by=c("taxonID"))
  #Split into matches and mismatches
  matches = filter(temp_data, Rank==rank)
  mismatches = filter(temp_data, Rank != rank) %>%
    mutate(taxonID=parentID) %>%
    select(-parentID, -Rank)

  #Check for root; update mismatches
  if(identical(unique(link$taxonID), c(1) )  ){
    print(paste0("root"," ", rank))
    mismatches[rank] = "skip"
    mismatches[paste0(rank, "_ID")] = 0
    mismatches$taxonID = mismatches$OG_ID
    }

  #For non-matches, update taxonID, recursive
  else{ if(nrow(df >0 )){
    mismatches <- recursive_taxon3(mismatches, Nodes, Names, rank) } }
  
  #If matches is NOT empty, update it
  if(nrow(matches != 0 )){
    #Update matches
    return_df = matches %>%
      full_join(., Names, by=c("taxonID")) %>%
      filter(taxonID %in% matches$taxonID)
    
    return_df[rank] = return_df["name"]
    return_df[paste0(rank, "_ID")] = as.numeric(return_df$taxonID)
    return_df$OG_ID = return_df$taxonID
    return_df$taxonID = return_df$parentID
    return_df <- return_df %>%
      select(-parentID, -Rank, -name)
    #Merge matches and returned results
    return_df <- rbind(return_df, mismatches) 
  }
  
  #If matches is empty, skip it
  else{
    return_df = mismatches }
  
  #Return matches++
  return(return_df)
  }


# test <- Nodes %>%
Big_taxon_table <- Nodes %>%
  filter(Rank=="species") %>%
  # filter(taxonID  %in% c(1,2, 9606)) %>%
  select(taxonID) %>%
  mutate(OG_ID = taxonID) %>%
  do(recursive_taxon3(., Nodes, Names, "species")) %>%
  do(recursive_taxon3(., Nodes, Names, "genus")) %>%
  do(recursive_taxon3(., Nodes, Names, "family")) %>%
  do(recursive_taxon3(., Nodes, Names, "order")) %>%
  do(recursive_taxon3(., Nodes, Names, "class")) %>%
  do(recursive_taxon3(., Nodes, Names, "phylum")) %>%
  do(recursive_taxon3(., Nodes, Names, "kingdom")) %>%
  do(recursive_taxon3(., Nodes, Names, "superkingdom")) %>%
  do(recursive_taxon3(., Nodes, Names, "no_rank")) %>%
  filter(OG_ID !=1) %>%
  select(-taxonID, - OG_ID)

# write.table(filter(Big_taxon_table, no_rank=="cellular organisms"), file='Taxonomy_table.csv',
#             quote=FALSE, sep=',', col.names = NA)

```

###Species codon usage
```{r}
PATH="../microbe_oral/all_bacterial_tRNA/CoCoPUTs/o586358-genbank_species.tsv"
temp <- read.csv(PATH, header=F, sep="\t", 
                 # nrows=10, 
                 skip = 1)%>%
  select(-V77)
temp_header <- read.csv(PATH, header=F, nrows=1, sep="\t") %>%
  mutate(V7 = "CDS", V8="total_codons", V9="GC_fraction")
colnames(temp) <- as.vector(temp_header[1,])
temp <- temp %>%
  mutate(taxonID = Taxid) %>%
  select(-Taxid,
         #-"GC%",
         -"GC1%", -"GC2%", -"GC3%",
         -"Translation Table", -Organelle, -Assembly, -Division )

```

###Species taxon lookup
```{r}

#Get species for each strain
# temp1 <- temp[1:1000,] %>%
temp1 <- temp %>%
  mutate(OG_ID = taxonID) %>%
  #filter(taxonID %in% c(102890, 100177,1,2)) %>%
  do(recursive_taxon3(., Nodes, Names, "species"))

```

###Combine codon with taxon
```{r}
#combine with taxon lineage data
temp2 <- Big_taxon_table %>%
  mutate(species_ID = as.numeric(species_ID))
temp3 <- temp1 %>%
  ungroup() %>%
  full_join(.,temp2, by=c("species_ID")) %>%
  filter(species_ID %in% temp1$species_ID)

#Side bar to get a list of codons
# codon_list <- combinations(n=4, r=3, v=c("A","T","C","G") )
# codon_list <- paste0(codon_list[,1], codon_list[,2], codon_list[,3])
codon_list1 <- permutations(n=3, r=3, v=c("A","T","C"), repeats.allowed=T) 
codon_list1 <- paste0(codon_list1[,1], codon_list1[,2], codon_list1[,3])
codon_list2 <- permutations(n=3, r=3, v=c("A","T","G"), repeats.allowed=T) 
codon_list2 <- paste0(codon_list2[,1], codon_list2[,2], codon_list2[,3])
codon_list3 <- permutations(n=3, r=3, v=c("A","C","G"), repeats.allowed=T) 
codon_list3 <- paste0(codon_list3[,1], codon_list3[,2], codon_list3[,3])
codon_list4 <- permutations(n=3, r=3, v=c("T","C","G"), repeats.allowed=T) 
codon_list4 <- paste0(codon_list4[,1], codon_list4[,2], codon_list4[,3])
codon_list <- unique(unlist(c(codon_list1, codon_list2, codon_list3, codon_list4)))

AA_table = data.frame(AA = c("Ile","Ile","Ile",
                            "Leu","Leu","Leu","Leu","Leu","Leu",
                            "Val","Val","Val","Val",
                            "Phe","Phe",
                            "Met",
                            "Cys","Cys",
                            "Ala","Ala","Ala","Ala",
                            "Gly","Gly","Gly","Gly",
                            "Pro","Pro","Pro","Pro",
                            "Thr","Thr","Thr","Thr",
                            "Ser","Ser","Ser","Ser","Ser","Ser",
                            "Tyr","Tyr",
                            "Trp",
                            "Gln","Gln",
                            "Asn","Asn",
                            "His","His",
                            "Glu","Glu",
                            "Asp","Asp",
                            "Lys","Lys",
                            "Arg","Arg","Arg","Arg","Arg","Arg",
                            "Stop","Stop","Stop"),
                      codon = c("ATT", "ATC", "ATA", #Ile
                               "CTT", "CTC", "CTA", "CTG", "TTA", "TTG", #Leu
                               "GTT", "GTC", "GTA", "GTG", #Val
                               "TTT","TTC", #Phe
                               "ATG", #Met
                               "TGT", "TGC", #Cys
                               "GCT", "GCC","GCA", "GCG", #Ala
                               "GGT", "GGC", "GGA", "GGG", #Gly
                               "CCT", "CCC","CCA", "CCG", #Pro
                               "ACT", "ACC", "ACA", "ACG",
                               "TCT", "TCC", "TCA", "TCG", "AGT", "AGC",
                               "TAT","TAC",
                               "TGG",
                               "CAA", "CAG",
                               "AAT", "AAC",
                               "CAT", "CAC",
                               "GAA", "GAG",
                               "GAT", "GAC",
                               "AAA","AAG",
                               "CGT", "CGC", "CGA", "CGG", "AGA", "AGG",
                               "TAA", "TAG", "TGA") )


#Mean for strains within species
temp4 <- temp3 %>%
  filter(CDS >1000)

temp5 <- filter(temp4, 
                # class=="Bacilli",
                order=="Lactobacillales"
                )

#Make codon usage a frequency
temp6 <- temp5 %>%
  pivot_longer(cols=c(all_of(codon_list)), names_to=c("codon"), values_to=c("frequency")) %>%
  mutate(frequency = frequency / total_codons) %>%
  full_join(., AA_table, by=c("codon"))


temp7 <- temp6 %>%
  filter(AA %in% c("Leu"),
         # GC_fraction <= 40
         ) %>%
  mutate(frequency_per_1000= frequency*1000) %>%
  mutate(facet_label = paste0(AA,codon))

# temp7A <- temp7 %>%
#   group_by(family) %>%
#   summarise(nspecies = length(unique(Species))) %>%
#   filter(nspecies >= 100)

plot1 <- ggplot(filter(temp7,
                       family %nin% c("Aerococcaceae"),
              ), 
       aes(x=frequency_per_1000, group=interaction(family), 
           fill=family,
           color=family
           )) +
  stat_ecdf() +
  scale_x_continuous(breaks=c(0,25,50)) +
  scale_y_continuous(breaks=c(0,0.5,1)) +
  ylab("Fraction of Data") +
  xlab("Codon use per 1000") +
  facet_wrap(~facet_label) +
  theme(legend.position = "right",
        strip.background = element_blank(),
        axis.text.x = element_text(angle=0, size=12, color="black"),
        axis.text.y = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16) )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_6Y_Leu_codon_usage"
width=5
height=3
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

# ggplot(filter(temp7
#               ), 
#        aes(x=GC_fraction, group=interaction(family), 
#            fill=family,
#            color=family
#            )) +
#   # geom_histogram(alpha=0.8) +
#   stat_ecdf() +
#   coord_cartesian(xlim = c(0,100))
#   # geom_density(alpha=0.6)+
#   # scale_x_log10() +
#   # facet_wrap(~interaction(AA, codon))

```

###A diferent vision of the plot
```{r}
temp8 <- temp7 %>%
  group_by(Species, family) %>%
  mutate(sum_Leu = sum(frequency_per_1000)) %>%
  group_by(Species, facet_label, family, codon) %>%
  mutate(fraction_Leu = frequency_per_1000 / sum_Leu)


plot1 <- ggplot(filter(temp8,
              family %nin% c("Aerococcaceae"),
              codon %in% c("CTT", "CTC", "TTA"),
              ), 
       aes(x=family, y=fraction_Leu, 
           # fill=family,
           # color=family
           )) +
  geom_boxplot(position = position_dodge(width=1)) +
  scale_y_continuous(
    breaks=c(0, .25, .5)
    ) +
  ylab("Fraction of Leu codons") +
  facet_grid(rows=vars(codon)) +
  # ylab("Codon use per 1000") +
  theme(legend.position = "right",
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0, size=16),
        axis.text.x = element_text(angle=90, size=16, color="black", hjust=1, vjust=0.5),
        axis.text.y = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_blank() ) 

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_6H_Species_Leu_codon_comparision"
width=2
height=4
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)
  

```




#_
#=====================
#Supplement
#=====================
#_
#Figure S2
##Fig S2D comparision with DM-tRNA-seq
###read in SE-mapped DM-tRNA-seq
```{r}
#Another day

```

##Fig S2E comparison of input RNA
###read in SE-mapped titration data
```{r}
#Another day

```

##Fig S2F mutaiton rate in minus DM
###read in SE mapped +/- DM data
```{r}
#Big oof

```

##Fig S2G change in mutation with DM
###pivot wider and subtract
```{r}
#Saturday is for the supplement

```

##Fig S2H trace of Arg ACG DM sensitivity
###filter, plot, E-Z
```{r}
#Keep it simple

```

##Fig S2I mutation rate with long RT
###read in longRT data
####Base
```{r}
#Point to the processed data and get a list of files
PATH="../20210125_new_data/updated_tRNA_reference/SE_map/5_tsv/human_combine/"
FILES <-data.frame( file_name = list.files(PATH) ) %>%
  filter(!grepl("unassigned",file_name))

#Split generic file names into salient parts
FILES <- FILES %>%
  separate(file_name, sep="_", c("library","junk1", "junk2", "barcode", "junk4", "bin_start", "bin_stop", "junk3"), 
           fill="right", remove=FALSE) %>%
  select(-junk1, -junk2, -junk3, -junk4) 

FILES <- FILES %>%
  mutate(treatment = barcode,
         sample = library) %>%
  select(-library, -barcode)

#PROJECT SPECIFIC DECODING
FILES$file_name <- as.character(FILES$file_name)
FILES$treatment <- recode(FILES$treatment, 
                          "bc1"="control_1", "bc2"="control_2", "bc3"="control_3",
                          "bc4"="heat_1", "bc5"="heat_2", "bc6"="heat_3",
                          "bc7"="H2O2_1", "bc8"="H2O2_2", "bc9"="H2O2_3",
                          "bc10"="AsO2_1", "bc11"="AsO2_2", "bc12"="AsO2_3")
FILES$sample <- recode(FILES$sample, "TP-CK-4S-CW1"="input", "TP-CK-4S-CW2"="polysome")
FILES <- FILES %>%
  separate(treatment, sep="_", into=c("treatment", "rep"))

#=======================+
#For fragment analysis, bins are specified in the file name. Except for the all reads bin, which gets designation "-3"
#Note: bin "-1" is all antisense reads; "-2" is all sense reads. 
FILES <- FILES %>%
  mutate(bin_start = ifelse(is.na(bin_start), -3, bin_start),
         bin_stop = ifelse(is.na(bin_stop), -3, bin_stop))
#========================

#Specific to this dataset
FILES <- FILES %>%
  mutate(DM="longRT")

#A function to read in files, designated from the list of files, and attach the project specific identifiers. 
read_in_one <- function(row){
  output <- read.csv(paste0(PATH, row$file_name), header=T, sep="\t") %>%
    mutate(sample = row$sample,
           DM = row$DM,
           rep = row$rep,
           treatment = row$treatment,
           bin_start = row$bin_start,
           bin_stop  = row$bin_stop,)
  #specify data types since enpty dataframes get confused and sad
  output$gene <- as.character(output$gene)
  output$base <- as.character(output$base)
  return(output)
}

new_SE_data_HEK_FULL_base <- FILES %>%
  group_by(file_name) %>%
  do(read_in_one(.)) %>%
  ungroup()
```

####Extra processing
```{r}
#Remove select snoRNAs, rRNAs, and control RNAs from subsequent analysis
temp <- new_SE_data_HEK_FULL_base %>%
  filter(gene %nin% c("NR_004394.1",
                      "NR_002716.3",
                      "NR_003925.1",
                      "NR_004430.2",
                      "NR_002756.2",
                      "NR_004391.1",
                      "NR_004392.1",
                      "NR_004393.1",
                      "NR_001571.2",
                      "Homo_sapiens_chrX.rRNA-5SR5SR",
                      "Homo_sapiens_chrX.rRNA-58S58S",
                      "Ecoli_Tyr",
                      "Yeast_Phe",
                      "Ecoli_Lys",
                      "spikein_SCC1",
                      "spikein_SCCA1"))

#fix mitochondrial names
#Then rearrange all names
temp <- temp %>%
  mutate(source="cytosolic")
temp <- temp %>%
  filter(grepl("mt", gene)) %>%
  mutate(gene = paste0("Homo_sapiens_chrMT.trna1-", gene)) %>%
  mutate(source="mitochondrial") %>%
  rbind(filter(temp, !grepl("mt",gene))) %>%
  select(-file_name)

#Simplify names
temp <- temp%>%
  mutate(gene2 = gene) %>%
  separate(gene2, sep="_", c("homo","sapiens","info"))%>%
  select(-homo, -sapiens)


temp <- temp%>%
  separate(info, sep="-", c("chr","AA")) %>%
  separate(AA, sep= -3, c("AA", "anticodon")) 

#Rename Met genes as elongator or initiator based on 5' A/C or G respectively
temp_iMet <- temp %>%
  filter(AA =="Met" ,
         bin_start=="-3") %>%
  #filter(position==1) %>%
  mutate(gene_symbol2 = ifelse(gene %in% c("Homo_sapiens_chr6.trna61-MetCAT",
                                           "Homo_sapiens_chr1.trna32-MetCAT",
                                           "Homo_sapiens_chr9.trna1-MetCAT"), 
                               paste0(AA, "i"), paste0(AA,"e")) ) %>%
  #https://www.ncbi.nlm.nih.gov/pmc/articles/PMC108860/#:~:text=A%20special%20methionine%20tRNA%20is,linkages%20(33%2C%2048).
  group_by(gene) %>%
  summarise(gene_symbol2 = gene_symbol2[1] )

#Resume fMet filtering
temp <- temp %>%
  full_join(., temp_iMet, by=c("gene")) %>%
  mutate(AA = ifelse(is.na(gene_symbol2), AA, gene_symbol2) ) %>%
  select(-gene_symbol2) 

temp <- temp %>%
  filter(AA %nin% c("Undet", "Sup"))

new_SE_HEK_FULL_base <- temp %>%
  mutate(DM="longRT") %>%
  filter(!grepl("AsnATT", gene),
         !grepl("SerACC", gene),
         !grepl("CysACA", gene),
         !grepl("TyrATA", gene),
         !(anticodon=="GAT" & AA=="Ile" & source=="cytosolic"),
         !grepl("SeCTCA", gene),
         !grepl("SerACT", gene))

```

####counts
```{r}
PATH="../20210125_new_data/updated_tRNA_reference/SE_map/6_sam_counter/human_combine/"
FILES <-data.frame( file_name = list.files(PATH) ) %>%
  filter(!grepl("unassigned", file_name),
         grepl("remove", file_name),
         file_name %nin% c("kallisto.out"))

#Split generic file names into salient parts
FILES <- FILES %>%
  separate(file_name, sep="_", c("library","junk1", "junk4","barcode", "junk2", "bin_start", "bin_stop", "junk3"), fill="right", remove=FALSE) %>%
  select(-junk1, -junk2, -junk3, -junk4) 

FILES <- FILES %>%
  mutate(treatment = barcode,
         sample = library) %>%
  select(-library, -barcode)

#PROJECT SPECIFIC DECODING
FILES$file_name <- as.character(FILES$file_name)
FILES$treatment <- recode(FILES$treatment, 
                          "bc1"="control_1", "bc2"="control_2", "bc3"="control_3",
                          "bc4"="heat_1", "bc5"="heat_2", "bc6"="heat_3",
                          "bc7"="H2O2_1", "bc8"="H2O2_2", "bc9"="H2O2_3",
                          "bc10"="AsO2_1", "bc11"="AsO2_2", "bc12"="AsO2_3")
FILES$sample <- recode(FILES$sample, "TP-CK-4S-CW1"="input", "TP-CK-4S-CW2"="polysome")
FILES <- FILES %>%
  separate(treatment, sep="_", into=c("treatment", "rep"))

#=======================+
FILES <- FILES %>%
  mutate(bin_start = ifelse(is.na(bin_start), -3, bin_start),
         bin_stop = ifelse(is.na(bin_stop), -3, bin_stop))
#========================

read_in_one <- function(row){
  output <- read.csv(paste0(PATH, row$file_name), header=T, sep="\t") %>%
    mutate(sample = row$sample,
           rep = row$rep,
           treatment = row$treatment,
           bin_start = row$bin_start,
           bin_stop  = row$bin_stop,)
  #specify data types since enpty dataframes get confused and sad
  output$name <- as.character(output$name)
  output <- output %>%
    mutate(gene=name) %>%
    select(-name)
  return(output)
}

new_SE_data_HEK_FULL_count <- FILES %>%
  group_by(file_name) %>%
  do(read_in_one(.)) %>%
  ungroup()
```


####Extra processing
```{r}
temp <- new_SE_data_HEK_FULL_count %>%
  filter(gene %nin% c("NR_004394.1",
                      "NR_002716.3",
                      "NR_003925.1",
                      "NR_004430.2",
                      "NR_002756.2",
                      "NR_004391.1",
                      "NR_004392.1",
                      "NR_004393.1",
                      "NR_001571.2",
                      "Homo_sapiens_chrX.rRNA-5SR5SR",
                      "Homo_sapiens_chrX.rRNA-58S58S",
                      "Ecoli_Tyr",
                      "Yeast_Phe",
                      "Ecoli_Lys",
                      "spikein_SCC1",
                      "spikein_SCCA1"))

#fix mitochondrial names
#Then rearrange all names
temp <- temp %>%
  mutate(source="cytosolic")

temp <- temp %>%
  filter(grepl("mt", gene)) %>%
  mutate(gene = paste0("Homo_sapiens_chrMT.trna1-", gene)) %>%
  mutate(source="mitochondrial") %>%
  rbind(filter(temp, !grepl("mt",gene))) %>%
  select(-file_name)


temp <- temp%>%
  mutate(gene2 = gene) %>%
  separate(gene2, sep="_", c("homo","sapiens","info"))%>%
  select(-homo, -sapiens)

temp <- temp%>%
  separate(info, sep="-", c("chr","AA")) %>%
  separate(AA, sep= -3, c("AA", "anticodon")) 

#Resume fMet filtering
temp <- temp %>%
  full_join(., temp_iMet, by=c("gene")) %>%
  mutate(AA = ifelse(is.na(gene_symbol2), AA, gene_symbol2) ) %>%
  select(-gene_symbol2) 

temp <- temp %>%
  filter(AA %nin% c("Undet", "Sup")) 

new_SE_HEK_FULL_count <- temp %>%
  mutate(DM="longRT") %>%
  filter(!grepl("AsnATT", gene),
         !grepl("SerACC", gene),
         !grepl("CysACA", gene),
         !grepl("TyrATA", gene),
         !(anticodon=="GAT" & AA=="Ile" & source=="cytosolic"),
         !grepl("SeCTCA", gene),
         !grepl("SerACT", gene))

```

###Plot
```{r}
temp <- new_SE_HEK_FULL_base %>%
  filter(bin_start=="-2",
         sample=="input",
         treatment=="control",
         source !="mitochondrial") %>%
  group_by(position, gene, chr, anticodon, AA, treatment, DM, bin_start, source) %>%
  summarise(mean_mutation = mean(mutation, na.rm=T))#average mutation for replicates

# #Test plot to see LysTTT isodecoder abundance
# ggplot(filter(new_SE_HEK_FULL_base, anticodon=="TTT",
#               sample=="input",
#               bin_start=="-2",
#               rep==1,
#               pileup > 20000,
#               treatment=="control"),
#        aes(x=position, y=mutation)) +
#   geom_line()+
#   facet_wrap(~gene)
#Filter for most abundant isodecoder for reach anticodon (use counts data)
# temp2 <- new_PE_HEK_count %>%
#   group_by(treatment, DM, bin_start, AA, anticodon) %>%
#   filter((bin_start=="-2"),
#          rep==1,
#          treatment=="control",
#          sample=="input") %>%
#   mutate(max_count = max(count)) %>%
#   filter(count==max_count) %>%
#   ungroup() %>%
#   select(gene)

temp2 <- new_SE_HEK_FULL_base %>%
  filter(bin_start=="-2",
         sample=="input",
         treatment=="control",
         source !="mitochondrial") %>%
  filter(position==70) %>%
  group_by(treatment, AA, anticodon, DM, bin_start, source) %>%
  mutate(max_read = max(pileup)) %>%
  filter(pileup == max_read) %>%
  ungroup() %>%
  select(gene)
temp <- temp %>%
  filter(gene %in% temp2$gene)

#Pretty labels
temp <- temp %>%
  mutate(x_label = paste0(AA, anticodon))


plot1 <- ggplot(filter(temp),
       aes(x=x_label, y=position, fill=mean_mutation)) +
  geom_tile() +
  scale_fill_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="grey70", 
                       limits=c(0,1),breaks=c(0,.5,1)) +
  labs(fill="mutation\nrate") +
  scale_y_reverse() +
  theme(legend.position = "right",
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0, size=16),
        axis.text.x = element_text(angle=90, size=12, color="black", hjust=1, vjust=0.5),
        axis.text.y = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_blank() ) 

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_S2I_longRT_mutation"
width=6
height=3
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```



#_
#Figure S3
##Fig S3B - Lys polysome abundance
###Just like 3D but Lys
```{r}
temp <- new_SE_HEK_FULL_count %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, sample, anticodon, AA, DM, rep) %>%
  summarise(sum_count = sum(count) ) %>%
  group_by(treatment, sample, DM, rep) %>%
  mutate(rpm = sum_count / sum(sum_count) *1000000) %>%
  filter(rpm >= 0.0001*1000000) 

temp <- temp %>%
  group_by(treatment, sample, anticodon, AA, DM) %>%
  summarise(mean_rpm = mean(rpm, na.rm=T)) %>%
  full_join(temp,., by=c("treatment", "sample", "anticodon", "AA", "DM"))

#Prettier label
temp <- temp %>%
  mutate(strip_label = paste0(AA, anticodon))

temp$treatment <- factor(temp$treatment, levels=c("control","heat","H2O2","AsO2"))

plot1 <- ggplot(filter(temp,
              AA=="Lys",
              sample=="polysome"),
       aes(x=treatment, y=rpm)) +
  geom_point() +
  ylab("Reads per million") +
  scale_y_log10(breaks=c(12500,25000, 50000,100000)) +
  facet_grid(cols=vars(strip_label)) +
  geom_point(aes(y=mean_rpm), shape=95, size=5) +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_text(size=14, color="black"),
        axis.title.y = element_text(size=16),
        axis.text.y = element_text(size=12, color="black"))

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_S3B_polysome_Lys_abundance"
width=4
height=2
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```

##Fig S3C - Ser isodecoder charging poysome
###Just like 2H but polysome
```{r}
#Filter for abundance =======================================================
abundance_list <- new_SE_HEK_FULL_count %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, sample, anticodon, AA, DM, rep) %>%
  summarise(sum_count = sum(count) ) %>%
  group_by(treatment, sample, DM, rep) %>%
  mutate(rpm = sum_count / sum(sum_count)) %>%
  filter(rpm >= 0.001) %>%
  group_by(treatment, sample, anticodon, AA, DM) %>%
  summarise(mean_rpm = mean(rpm, na.rm=T)) %>%
  select(sample, treatment, DM, anticodon, AA, mean_rpm)


#Work with the base-wise data, calculate charging for each gene
temp <- new_SE_HEK_FULL_base %>%
  select(-bin_stop) %>%
  filter(bin_start=="-2") %>%
  group_by(treatment, rep, bin_start, DM, sample, source, gene, AA, anticodon) %>%
  mutate(max_position = max(position)) %>%
  filter(position %in% c(max_position, max_position-1)) %>%
  mutate(charging = ifelse(position==max_position, "end", "penultimate")) %>%
  filter((charging=="end" & base=="A")  |
         (charging=="penultimate" & base=="C")) %>%
  select(treatment, rep, bin_start, DM, sample, source, gene, pileup, charging, AA, anticodon) %>%
  group_by(treatment, rep, bin_start, DM, sample, source, gene, anticodon, AA) %>%
  pivot_wider(names_from = c("charging"), values_from = c("pileup")) %>%
  mutate(charging= end/penultimate)

#Apply abundance filter (don't calculate charging on spurious low read genes)
tem <- temp %>%
  full_join(., abundance_list, by=c("sample","treatment","DM","anticodon","AA")) %>%
  filter(mean_rpm >= 0.001)

#Get fold change to control
temp <- temp %>%
  filter(treatment=="control") %>%
  group_by(gene, treatment, sample, DM) %>%
  mutate(mean_control_charging = mean(charging, na.rm=T)) %>%
  ungroup() %>%
  select(gene, sample, DM, mean_control_charging) %>%
  full_join(temp, ., by=c("gene","sample","DM"))

#Make this a Serine only
temp <- temp %>%
  filter(AA=="Ser")
#Ser only ===========================================================
temp <- temp %>%
  mutate(m3C47 = ifelse(gene %in% Ser_C_list,"C",ifelse(gene %in% Ser_T_list,"T","C"))) 

temp$treatment <- factor(temp$treatment, levels=c("control","heat","H2O2","AsO2"))
my_comparisons <- list( c("C", "T"))


plot1 <- ggplot(filter(temp,
              AA=="Ser",
              sample=="polysome",
              !grepl("mt", AA) ,
              treatment !="control"
              ),
       aes(x=m3C47, y= charging / mean_control_charging, color=m3C47 )) +
  geom_hline(yintercept = 1, linetype=2, alpha=0.3) +
  geom_boxplot() +
  scale_color_manual(values=c("C"="#FFAD0A", "T"="#1BB6AF")) +
  scale_y_log10(breaks=c(0.25, 0.5, 1, 2), 
                limits=c(0.25, 2.3)
                ) +
  stat_compare_means(comparisons = my_comparisons, size=4,  method="t.test", label="p.signif") + #how does this compute p-value?
  facet_grid(cols=vars(treatment)) +
  xlab("Position 47d") +
  ylab("tRNA charging change\n(stress / control") +
  theme(legend.position = "none",
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0),
        strip.text.x = element_text(size=14),
        axis.text = element_text(size=12),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16) )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_S3C_m3C_charging_change_polysome"
width=3
height=2.5
scale=1.5
dpi=600
# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```

##Fig S3D - Isodecoder level, Ser mutations
###polysome minus input traces
```{r}
temp <- new_SE_HEK_FULL_base %>%
  filter( (sample=="input"&pileup >=1000)|
            ( sample=="polysome"&pileup >=10)  ) %>%#polysome coverage is lower, less stringent cutoff
  select(-bin_stop) %>%
  filter(bin_start==-2) %>%
  filter(AA=="Ser",
         position >= 29, position <= 54)
  # filter((AA=="Ser")|(AA=="Leu" & anticodon=="CAG") )
temp <- temp %>%
  select(gene, position, mutation, sample, DM, rep, treatment, AA, chr, anticodon) %>%
  pivot_wider(names_from = c("sample"), values_from = c("mutation"))
temp <- temp %>%
  mutate(m3C47 = ifelse(gene %in%Ser_C_list,"C",ifelse(gene %in% Ser_T_list,"T","C"))) 
temp$chr <- str_replace_all(temp$chr, "chr","c")
temp$chr <- str_replace_all(temp$chr, "trna","t")

plot1 <- ggplot(filter(temp, 
              treatment=="control",
              m3C47 =="C",
              chr %nin% c("c11.t10", "c17.t32", "c6.t145", "c6.t35", "c6.t31", "c6.t50")
              ),
       aes(x=position, y= polysome - input, color=m3C47, group=rep )) +
  geom_vline(xintercept = c(32, 50), linetype=2, alpha=0.3) +
  geom_hline(yintercept = c(0), linetype=2, alpha=0.3) +
  geom_line() +
  scale_x_continuous(breaks=c(30,40,50)) +
  ylab("Δ mutation (polysome - input)") +
  coord_cartesian(ylim=c(-0.3, 0.3)) +
  scale_color_manual(values=c("C"="#FFAD0A", "T"="#1BB6AF")) +
  facet_nested( anticodon +chr ~ m3C47) +
  theme(legend.position = "none",
        strip.background.x = element_blank(),
        strip.background.y =element_rect(color="grey30", fill="white"),
        strip.text.y = element_text(angle=0),
        strip.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        axis.title = element_text(size=16))

plot2 <- ggplot(filter(temp, 
              treatment=="control",
              m3C47 =="T",
              chr %nin% c("c11.t10", "c17.t32", "c6.t145", "c6.t35", "c6.t31", "c6.t50")
              ),
       aes(x=position, y= polysome - input, color=m3C47, group=rep )) +
  geom_vline(xintercept = c(32, 50), linetype=2, alpha=0.3) +
  geom_hline(yintercept = c(0), linetype=2, alpha=0.3) +
  geom_line() +
  scale_x_continuous(breaks=c(30,40,50)) +
  ylab("Δ mutation (polysome - input)") +
  coord_cartesian(ylim=c(-0.3, 0.3)) +
  scale_color_manual(values=c("C"="#FFAD0A", "T"="#1BB6AF")) +
  facet_nested( anticodon +chr ~ m3C47) +
  theme(legend.position = "none",
        strip.background.x = element_blank(),
        strip.background.y =element_rect(color="grey30", fill="white"),
        strip.text.y = element_text(angle=0),
        strip.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        axis.title = element_text(size=16))

#Save the figure
layout <- rbind(c(1, 2))
figure <- grid.arrange(plot1, plot2,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_S3D_m3C47_traces_polysome_minus_input"
width=4
height=6
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```


##Fig S3E - Isodecoder level, Ser mutations
###polysome stress minus polysome control
```{r}
#Filter down to genes with m3C47d
temp <- new_SE_HEK_FULL_base %>%
  filter( (sample=="input"&pileup >=1000)|
            ( sample=="polysome"&pileup >=10)  ) %>%#polysome coverage is lower, less stringent cutoff
  select(-bin_stop) %>%
  filter(bin_start== -2) %>%
  filter(position >= 45, position <= 54,
         (AA=="Ser")|(AA=="Leu" & anticodon=="CAG") )

#Add a column for control
temp <- temp %>%
  filter(treatment =="control") %>%
  group_by(gene, position, sample, treatment, DM) %>%
  summarise(control_mean_mutation = mean(mutation, na.rm=T)) %>%
  ungroup() %>%
  select(gene, position, sample, control_mean_mutation, DM) %>%
  full_join(temp, ., by=c("gene", "position", "sample", "DM"))

#Beautify
temp <- temp %>%
  mutate(m3C47 = ifelse(gene %in%Ser_C_list,"C",ifelse(gene %in% Ser_T_list,"T","C"))) 
temp$chr <- str_replace_all(temp$chr, "chr","c")
temp$chr <- str_replace_all(temp$chr, "trna","t")
temp$treatment <- factor(temp$treatment, levels=c("control","heat","H2O2","AsO2"))

#Good plot
plot1 <- ggplot(filter(temp, 
              m3C47 =="C",
              sample =="polysome",
              treatment !="control",
              chr %nin% c("c7.t12","c11.t10", "c17.t32", "c6.t145", "c6.t35", "c6.t31", "c6.t50")
              ),
       aes(x=position, y= mutation - control_mean_mutation, color=m3C47, group=rep )) +
  geom_vline(xintercept = c(50), linetype=2, alpha=0.3) +
  geom_hline(yintercept = c(0), linetype=2, alpha=0.3) +
  geom_line() +
  scale_x_continuous(breaks=c(30,40,50)) +
  ylab("Δ mutation (stress - control)") +
  coord_cartesian(ylim=c(-0.3, 0.3)) +
  scale_color_manual(values=c("C"="#FFAD0A", "T"="#1BB6AF")) +
  facet_nested( AA + anticodon + chr ~  treatment) +
  theme(legend.position = "none",
        strip.background.x = element_blank(),
        strip.background.y =element_rect(color="grey30", fill="white"),
        strip.text.y = element_text(angle=0),
        strip.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        axis.title = element_text(size=16))

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_S3E_m3C47_traces_polysome_stress"
width=3
height=6
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)


```




#_
#Figure S4

##Fig S4B - transcription of stress genes
###Read in the data
```{r}
PATH="../20210125_new_data/mRNA_Wen/HEK_stress_polysome_read_count_group_name_dict.txt"

mRNA_data <- read.table(PATH, header=T) %>%
  pivot_longer(cols=c(-gene_id, -feature, -chromosome, -gene_biotype, -transcript_biotype, -gene_symbol), 
               names_to="experiment", values_to="count") %>%
  separate(experiment, sep=c("_"), into=c("sample","etc")) %>%
  separate(etc, sep="\\.", into=c("treatment","rep"))

mRNA_data$feature <- factor(mRNA_data$feature)
mRNA_data$gene_biotype <- factor(mRNA_data$gene_biotype)
mRNA_data$transcript_biotype <- factor(mRNA_data$transcript_biotype)
mRNA_data$treatment <- recode(mRNA_data$treatment, "Heat"="heat", "H2O2"="H2O2", "NaAsO2"="AsO2", "ctrl"="control")


```

###control vs stress
```{r}

#Pool transcript varients
temp <- mRNA_data %>%
  group_by(gene_symbol, treatment, sample, rep, gene_biotype) %>%
  summarise(count = sum(count)) 

temp1 <- temp

#make categorical variables
temp1$sample <- factor(temp1$sample)
temp1$treatment <- factor(temp1$treatment, levels=c("control","heat","H2O2","AsO2"))
temp1$rep <- factor(temp1$rep)

#Normalize reps to RPM, average reps
temp1 <- temp1 %>%
  group_by(rep, treatment, sample) %>%
  mutate(rpm = count / sum(count)*1000000) %>%
  filter(count >= 100,
         !is.na(gene_biotype)) %>%
  group_by(treatment, sample, gene_symbol, gene_biotype) %>%
  summarise(rpm_mean = mean(rpm, na.rm=T))

#add a column for control; 
temp1 <- temp1 %>%
  ungroup() %>%
  filter(treatment=="control") %>%
  mutate(ctrl_rpm = rpm_mean)%>%
  select(gene_symbol, sample, ctrl_rpm) %>%
  full_join(temp1, ., by=c("gene_symbol","sample")) %>%
  mutate(ratio = rpm_mean / ctrl_rpm)

temp1 <- temp1 %>%
  mutate(label_name = NA) %>%
  mutate(label_name = ifelse( ( (ratio >5 | ratio <0.2) & (sample=="input" & ctrl_rpm >=50) ) |
                              ( (ratio >5 | ratio <0.2) & (sample=="polysome" & rpm_mean >=100) ),
                             gene_symbol, NA))

plot1 <- ggplot(filter(temp1,
              treatment !="control",
              ctrl_rpm > 0.00001,
              gene_biotype=="protein_coding",
              #sample=="polysome"
              ),
       aes(x=ctrl_rpm, y=rpm_mean)) +
  geom_point() +
  geom_text_repel(aes(label=label_name), color="grey40") +
  scale_y_log10() +
  scale_x_log10() +
  geom_abline(slope=1, linetype=2, alpha=0.3) +
  facet_grid(cols=vars(treatment), rows=vars(sample)) +
  xlab("control expression (read per million)") +
  ylab("stress expression (read per million)") +
  theme(legend.position = "none",
        strip.background.x = element_blank(),
        strip.background.y = element_blank(),
        strip.text.y = element_text(angle=0),
        strip.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        axis.title = element_text(size=16))

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_S4B_stress_transcription"
width=6
height=4
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```


##Fig S4C - TE vs. TE
###Calculate TE and plot
```{r}
#combine transcript varianets
temp <- mRNA_data %>%
  group_by(gene_symbol, treatment, sample, rep, gene_biotype) %>%
  summarise(count = sum(count)) 
#Make factors
temp$sample <- factor(temp$sample)
temp$treatment <- factor(temp$treatment, levels=c("control","heat","H2O2","AsO2"))
temp$rep <- factor(temp$rep)
#summarise to RPM; filter to abundnat mRNAs; average replicates
temp <- temp %>%
  group_by(rep, treatment, sample) %>%
  mutate(rpm = count / sum(count)) %>%
  filter(count >= 100,
         !is.na(gene_biotype)) %>%
  group_by(treatment, sample, gene_symbol, gene_biotype) %>%
  summarise(rpm_mean = mean(rpm, na.rm=T))
#pivot wider by sample; calculate ratio polysome to input
temp <- temp %>%
  group_by(treatment, gene_symbol, gene_biotype) %>%
  pivot_wider(names_from = c("sample"), values_from=c("rpm_mean")) %>%
  mutate(ratio = polysome / input)

#Pivot wider by treatment
temp2 <- temp %>%
  ungroup() %>%
  select(treatment, gene_symbol, gene_biotype, ratio) 
temp2 <- temp2 %>%
  filter(treatment=="control") %>%
  mutate(ctrl_ratio = ratio) %>%
  select(gene_symbol, ctrl_ratio) %>%
  full_join(temp2, by=c("gene_symbol"))

#Make the good plot
plot1 <- ggplot(filter(temp2,
              gene_biotype=="protein_coding",
              treatment !="control"),
       aes(x=ctrl_ratio, y=ratio)) +
  geom_point(alpha=0.1) +
  scale_y_log10() +
  scale_x_log10() +
  xlab("control TE (polysome / input)") +
  ylab("stress TE (polysome / input)") +
  geom_abline(slope=1, linetype=2, alpha=1, color="grey70", size=1) +
  # stat_smooth(method="lm") +
  facet_grid(cols=vars(treatment)) +
  theme(legend.position = "none",
        strip.background.x = element_blank(),
        strip.background.y =element_rect(color="grey30", fill="white"),
        strip.text.x = element_text(size=16),
        axis.text = element_text(size=12),
        axis.title = element_text(size=16))

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_S4C_TE_plots"
width=4
height=2
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)


```



#_
#Figure S5

##Fig S5B - Gly isodecoder fragments
###wrangle the same as above, just facet
```{r}

temp <- new_SE_HEK_FULL_base %>%
  filter(pileup >=10,
         # treatment=="control",
         bin_start %nin% c("-1","-2","-3"),
         sample=="input",
         DM !="minus")
temp$bin_start <- recode(temp$bin_start, 
                         "20"="20-29",
                         "30"="30-39",
                         "40"="40-49",
                         "50"="50-59",
                         "60"="60+")

#PRetty label
temp$chr <- str_replace_all(temp$chr, "chr","c")
temp$chr <- str_replace_all(temp$chr, "trna","t")
temp$treatment <- factor(temp$treatment, levels=c("control","heat","H2O2","AsO2"))
temp <- temp %>%
  mutate(gene_label= paste0(AA, anticodon,"\n",chr))

plot1 <- ggplot(filter(temp,
              anticodon=="CCC",
              position <= 72,
              # grepl("chr16.trna34",gene),
              bin_start %nin%c("20-29")
              ),
       aes(x=position, y= pileup, color=bin_start, group=interaction(rep, bin_start)) ) +
  geom_line(size=1) +
  # coord_cartesian(xlim = c(0,73)) +
  scale_y_log10(labels=trans_format('log10',math_format(10^.x)), breaks=c(10,1000,100000) ) +
  scale_color_manual(values=c("60+"="#1BB6AF",
                              "50-59"="#E9A17C",
                              "40-49"="#0076BB",
                              "30-39"="#dad371" ), guide = guide_legend(reverse = TRUE)) +
  geom_vline(xintercept = c(30,40,50,60), linetype=2, alpha=0.2) +
  scale_x_continuous(breaks=c(0,30,60)) +
  labs(color="Location of\n3' end") +
  ylab("Read pileup") +
  xlab("position") +
  facet_grid(cols=vars(gene_label), rows=vars(treatment)) + 
  theme(legend.position = "right",
        strip.background = element_blank(),
        strip.text.x = element_text(angle=0, size=14),
        axis.text = element_text(size=10, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16) )


#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_S5B_GlyCCC_fragment_trace"
width=6
height=3
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```


##Fig S5C - Ser m3C47D fragment trace
###Focus on 47D
```{r}
temp <- new_SE_HEK_FULL_base 
temp$bin_start <- recode(temp$bin_start, 
                         "20"="20-29",
                         "30"="30-39",
                         "40"="40-49",
                         "50"="50-59",
                         "60"="60+")
temp$chr <- str_replace_all(temp$chr, "chr","c")
temp$chr <- str_replace_all(temp$chr, "trna","t")


temp <- temp %>%
  filter(bin_start %nin% c("-1","-2","-3")) %>%
  filter(AA=="Leu") %>%
  filter(grepl("c1.t34", chr)) %>%
  group_by(sample, gene, treatment, position, bin_start) #%>%
  # mutate(mean_mutation = mean(mutation, na.rm=T)) %>%
  # mutate(x_label=paste0(AA,anticodon, " ", chr))

#Nice label
temp <- temp %>%
  mutate(gene_label = paste0(AA, anticodon, "\n", chr))

plot1 <- ggplot(filter(temp,
              sample=="input",
              treatment=="control",
              position <55,  position >45,
              bin_start %in% c("60+", "50-59")),
       aes(x=position, y=mutation, color=bin_start, group=interaction(bin_start, rep))) +
  geom_line() +
  facet_wrap(~gene_label) +
  scale_y_continuous(breaks=c(0,0.5,1), limits=c(0,1)) +
  scale_color_manual(values=c("60+"="#1BB6AF",
                              "50-59"="#E9A17C",
                              "40-49"="#0076BB",
                              "30-39"="#dad371" ), guide = guide_legend(reverse = TRUE)) +
  labs(color="Mapping of\n3' end") +
  ylab("Mutation rate") +
  theme(legend.position = "right",
        strip.background = element_blank(),
        axis.text.x = element_text(angle=0, size=12, color="black", vjust=0.5, hjust=0.5),
        axis.text.y = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16) )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_S5C_m3C47D_trace_fragment"
width=2.5
height=2
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)
```

##Fig S5D - Leu M1G37 fragment trace
###Focus on m1G37
```{r}
temp <- new_SE_HEK_FULL_base 
temp$bin_start <- recode(temp$bin_start, 
                         "20"="20-29",
                         "30"="30-39",
                         "40"="40-49",
                         "50"="50-59",
                         "60"="60+")
temp$chr <- str_replace_all(temp$chr, "chr","c")
temp$chr <- str_replace_all(temp$chr, "trna","t")


temp <- temp %>%
  filter(bin_start %nin% c("-1","-2","-3")) %>%
  filter(AA=="Leu") %>%
  filter(grepl("c6.t141", chr)) %>%
  group_by(sample, gene, treatment, position, bin_start) #%>%
  # mutate(mean_mutation = mean(mutation, na.rm=T)) %>%
  # mutate(x_label=paste0(AA,anticodon, " ", chr))

#Nice label
temp <- temp %>%
  mutate(gene_label = paste0(AA, anticodon, "\n", chr))

plot1 <- ggplot(filter(temp,
              sample=="input",
              treatment=="control",
              position > 35,  position < 42,
              bin_start %in% c("60+", "50-59", "40-49")),
       aes(x=position, y=mutation, color=bin_start, group=interaction(bin_start, rep))) +
  geom_line() +
  facet_wrap(~gene_label) +
  scale_y_continuous(breaks=c(0,0.25, 0.5, 1), limits=c(0,0.5)) +
  scale_color_manual(values=c("60+"="#1BB6AF",
                              "50-59"="#E9A17C",
                              "40-49"="#0076BB",
                              "30-39"="#dad371" ), guide = guide_legend(reverse = TRUE)) +
  labs(color="Mapping of\n3' end") +
  ylab("Mutation rate") +
  theme(legend.position = "right",
        strip.background = element_blank(),
        axis.text.x = element_text(angle=0, size=12, color="black", vjust=0.5, hjust=0.5),
        axis.text.y = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16) )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_S5D_m1G37C_trace_fragment"
width=2.5
height=2
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)
```

##Fig S5E - Leu M22G26 fragment trace
###Focus on m22G26
```{r}
temp <- new_SE_HEK_FULL_base 
temp$bin_start <- recode(temp$bin_start, 
                         "20"="20-29",
                         "30"="30-39",
                         "40"="40-49",
                         "50"="50-59",
                         "60"="60+")
temp$chr <- str_replace_all(temp$chr, "chr","c")
temp$chr <- str_replace_all(temp$chr, "trna","t")


temp <- temp %>%
  filter(bin_start %nin% c("-1","-2","-3")) %>%
  filter(AA=="Trp") %>%
  filter(grepl("c12.t6", chr)) %>%
  group_by(sample, gene, treatment, position, bin_start) #%>%
  # mutate(mean_mutation = mean(mutation, na.rm=T)) %>%
  # mutate(x_label=paste0(AA,anticodon, " ", chr))

#Nice label
temp <- temp %>%
  mutate(gene_label = paste0(AA, anticodon, "\n", chr))

plot1 <- ggplot(filter(temp,
              sample=="input",
              treatment=="control",
              position >= 20,  position <= 30,
              bin_start %in% c("60+", "50-59", "30-39")),
       aes(x=position, y=mutation, color=bin_start, group=interaction(bin_start, rep))) +
  geom_line() +
  facet_wrap(~gene_label) +
  scale_y_continuous(breaks=c(0,0.25, 0.5,1), limits=c(0,0.5)) +
  scale_x_continuous(breaks=c(20,25,30)) +
  scale_color_manual(values=c("60+"="#1BB6AF",
                              "50-59"="#E9A17C",
                              "40-49"="#0076BB",
                              "30-39"="#dad371" ), guide = guide_legend(reverse = TRUE)) +
  labs(color="Mapping of\n3' end") +
  ylab("Mutation rate") +
  theme(legend.position = "right",
        strip.background = element_blank(),
        axis.text.x = element_text(angle=0, size=12, color="black", vjust=0.5, hjust=0.5),
        axis.text.y = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16) )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_S5E_m22G26_trace_fragment"
width=2.5
height=2
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)
```










#_
#Figure S6

##Fig S6B - Compare taxonomy differnt methods
###read in 16S data
```{r}
#Lol where is this?
```

###Read in SRP 
```{r}
#The saga
```

##Fig S6C - m1A58 in bacteria
###Read in mapped data
```{r}
#re-map these data I guess
```

###Plot a grid with mutation rate
```{r}
#asdfasdfasd
```


##Fig S6D - which genus have AAG Leu?
###continue from above
```{r}
temp4 <- temp3 %>%
  filter(AA=="Leu")
temp4 <- temp4$organism

temp5 <- temp2 %>%
  filter(organism %in% temp4) %>%
  group_by(organism, anticodon) %>%
  summarise(present = T) %>%
  group_by(organism) %>%
  pivot_wider(names_from = c("anticodon"), values_from=c("present")) %>%
  mutate(sum_anticodons = sum(AAG, CAA, CAG, GAG, TAA, TAG, na.rm=T))

tempZ <- filter(temp5,
                AAG==T, GAG==T)

temp6 <- temp5 %>%
  filter(sum_anticodons %in% c(4))

temp7 <- temp5 %>%
  filter(AAG==T &  is.na(GAG))

#Get genus and species
tempX <- temp7 %>%
  separate(organism, sep="_", into=c("genus","species"), remove=F, extra="drop") %>%
  separate(genus, sep=c(1), into=c("junk", "genus")) %>%
  select(-junk)

plot1 <- ggplot(tempX, aes(x=interaction(genus))) +
  geom_histogram(stat="count") +
  ylab("LeuAAG tRNA\ngene count") +
  theme(legend.position = "right",
        strip.background = element_blank(),
        axis.text.x = element_text(angle=90, size=12, color="black", vjust=0.5, hjust=1),
        axis.text.y = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_blank() )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_S6D_species_LeuAAG"
width=3.5
height=2
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```
###

##Fig 6F - GC content of Lactobacillae families
###Read in data above; wrangle here
```{r}
#combine with taxon lineage data
temp2 <- Big_taxon_table %>%
  mutate(species_ID = as.numeric(species_ID))
temp3 <- temp1 %>%
  ungroup() %>%
  full_join(.,temp2, by=c("species_ID")) %>%
  filter(species_ID %in% temp1$species_ID)

#Side bar to get a list of codons
# codon_list <- combinations(n=4, r=3, v=c("A","T","C","G") )
# codon_list <- paste0(codon_list[,1], codon_list[,2], codon_list[,3])
codon_list1 <- permutations(n=3, r=3, v=c("A","T","C"), repeats.allowed=T) 
codon_list1 <- paste0(codon_list1[,1], codon_list1[,2], codon_list1[,3])
codon_list2 <- permutations(n=3, r=3, v=c("A","T","G"), repeats.allowed=T) 
codon_list2 <- paste0(codon_list2[,1], codon_list2[,2], codon_list2[,3])
codon_list3 <- permutations(n=3, r=3, v=c("A","C","G"), repeats.allowed=T) 
codon_list3 <- paste0(codon_list3[,1], codon_list3[,2], codon_list3[,3])
codon_list4 <- permutations(n=3, r=3, v=c("T","C","G"), repeats.allowed=T) 
codon_list4 <- paste0(codon_list4[,1], codon_list4[,2], codon_list4[,3])
codon_list <- unique(unlist(c(codon_list1, codon_list2, codon_list3, codon_list4)))

AA_table = data.frame(AA = c("Ile","Ile","Ile",
                            "Leu","Leu","Leu","Leu","Leu","Leu",
                            "Val","Val","Val","Val",
                            "Phe","Phe",
                            "Met",
                            "Cys","Cys",
                            "Ala","Ala","Ala","Ala",
                            "Gly","Gly","Gly","Gly",
                            "Pro","Pro","Pro","Pro",
                            "Thr","Thr","Thr","Thr",
                            "Ser","Ser","Ser","Ser","Ser","Ser",
                            "Tyr","Tyr",
                            "Trp",
                            "Gln","Gln",
                            "Asn","Asn",
                            "His","His",
                            "Glu","Glu",
                            "Asp","Asp",
                            "Lys","Lys",
                            "Arg","Arg","Arg","Arg","Arg","Arg",
                            "Stop","Stop","Stop"),
                      codon = c("ATT", "ATC", "ATA", #Ile
                               "CTT", "CTC", "CTA", "CTG", "TTA", "TTG", #Leu
                               "GTT", "GTC", "GTA", "GTG", #Val
                               "TTT","TTC", #Phe
                               "ATG", #Met
                               "TGT", "TGC", #Cys
                               "GCT", "GCC","GCA", "GCG", #Ala
                               "GGT", "GGC", "GGA", "GGG", #Gly
                               "CCT", "CCC","CCA", "CCG", #Pro
                               "ACT", "ACC", "ACA", "ACG",
                               "TCT", "TCC", "TCA", "TCG", "AGT", "AGC",
                               "TAT","TAC",
                               "TGG",
                               "CAA", "CAG",
                               "AAT", "AAC",
                               "CAT", "CAC",
                               "GAA", "GAG",
                               "GAT", "GAC",
                               "AAA","AAG",
                               "CGT", "CGC", "CGA", "CGG", "AGA", "AGG",
                               "TAA", "TAG", "TGA") )


#Mean for strains within species
temp4 <- temp3 %>%
  filter(CDS >1000)

temp5 <- filter(temp4, 
                # class=="Bacilli",
                order=="Lactobacillales"
                )

#Make codon usage a frequency
temp6 <- temp5 %>%
  pivot_longer(cols=c(all_of(codon_list)), names_to=c("codon"), values_to=c("frequency")) %>%
  mutate(frequency = frequency / total_codons) %>%
  full_join(., AA_table, by=c("codon"))


temp7 <- temp6 %>%
  filter(AA %in% c("Leu"),
         # GC_fraction <= 40
         ) %>%
  mutate(frequency_per_1000= frequency*1000) %>%
  mutate(facet_label = paste0(AA,codon))

# temp7A <- temp7 %>%
#   group_by(family) %>%
#   summarise(nspecies = length(unique(Species))) %>%
#   filter(nspecies >= 100)

plot1 <- ggplot(filter(temp7,
              family %nin% c("Aerococcaceae") ),
       aes(x=GC_fraction, group=interaction(family),
           fill=family,
           color=family
           )) +
  # geom_histogram(alpha=0.8) +
  stat_ecdf(size=1) +
  ylab("Fraction of Data") +
  xlab("%GC in genome") +
  coord_cartesian(xlim = c(0,100)) +
  # geom_density(alpha=0.6)+
  # scale_x_log10() +
  # facet_wrap(~interaction(AA, codon))
  theme(legend.position = "right",
        strip.background = element_blank(),
        axis.text.x = element_text(angle=0, size=12, color="black"),
        axis.text.y = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16) )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_S6F_GC_content"
width=3.5
height=2
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)
```


##Fig S6G - distribution of Leu codons
###Wrange the old way
```{r}
#combine with taxon lineage data
temp2 <- Big_taxon_table %>%
  mutate(species_ID = as.numeric(species_ID))
temp3 <- temp1 %>%
  ungroup() %>%
  full_join(.,temp2, by=c("species_ID")) %>%
  filter(species_ID %in% temp1$species_ID)

#Side bar to get a list of codons
# codon_list <- combinations(n=4, r=3, v=c("A","T","C","G") )
# codon_list <- paste0(codon_list[,1], codon_list[,2], codon_list[,3])
codon_list1 <- permutations(n=3, r=3, v=c("A","T","C"), repeats.allowed=T) 
codon_list1 <- paste0(codon_list1[,1], codon_list1[,2], codon_list1[,3])
codon_list2 <- permutations(n=3, r=3, v=c("A","T","G"), repeats.allowed=T) 
codon_list2 <- paste0(codon_list2[,1], codon_list2[,2], codon_list2[,3])
codon_list3 <- permutations(n=3, r=3, v=c("A","C","G"), repeats.allowed=T) 
codon_list3 <- paste0(codon_list3[,1], codon_list3[,2], codon_list3[,3])
codon_list4 <- permutations(n=3, r=3, v=c("T","C","G"), repeats.allowed=T) 
codon_list4 <- paste0(codon_list4[,1], codon_list4[,2], codon_list4[,3])
codon_list <- unique(unlist(c(codon_list1, codon_list2, codon_list3, codon_list4)))

AA_table = data.frame(AA = c("Ile","Ile","Ile",
                            "Leu","Leu","Leu","Leu","Leu","Leu",
                            "Val","Val","Val","Val",
                            "Phe","Phe",
                            "Met",
                            "Cys","Cys",
                            "Ala","Ala","Ala","Ala",
                            "Gly","Gly","Gly","Gly",
                            "Pro","Pro","Pro","Pro",
                            "Thr","Thr","Thr","Thr",
                            "Ser","Ser","Ser","Ser","Ser","Ser",
                            "Tyr","Tyr",
                            "Trp",
                            "Gln","Gln",
                            "Asn","Asn",
                            "His","His",
                            "Glu","Glu",
                            "Asp","Asp",
                            "Lys","Lys",
                            "Arg","Arg","Arg","Arg","Arg","Arg",
                            "Stop","Stop","Stop"),
                      codon = c("ATT", "ATC", "ATA", #Ile
                               "CTT", "CTC", "CTA", "CTG", "TTA", "TTG", #Leu
                               "GTT", "GTC", "GTA", "GTG", #Val
                               "TTT","TTC", #Phe
                               "ATG", #Met
                               "TGT", "TGC", #Cys
                               "GCT", "GCC","GCA", "GCG", #Ala
                               "GGT", "GGC", "GGA", "GGG", #Gly
                               "CCT", "CCC","CCA", "CCG", #Pro
                               "ACT", "ACC", "ACA", "ACG",
                               "TCT", "TCC", "TCA", "TCG", "AGT", "AGC",
                               "TAT","TAC",
                               "TGG",
                               "CAA", "CAG",
                               "AAT", "AAC",
                               "CAT", "CAC",
                               "GAA", "GAG",
                               "GAT", "GAC",
                               "AAA","AAG",
                               "CGT", "CGC", "CGA", "CGG", "AGA", "AGG",
                               "TAA", "TAG", "TGA") )


#Mean for strains within species
temp4 <- temp3 %>%
  filter(CDS >1000)

temp5 <- filter(temp4, 
                # class=="Bacilli",
                order=="Lactobacillales"
                )

#Make codon usage a frequency
temp6 <- temp5 %>%
  pivot_longer(cols=c(all_of(codon_list)), names_to=c("codon"), values_to=c("frequency")) %>%
  mutate(frequency = frequency / total_codons) %>%
  full_join(., AA_table, by=c("codon"))


temp7 <- temp6 %>%
  filter(AA %in% c("Leu"),
         # GC_fraction <= 40
         ) %>%
  mutate(frequency_per_1000= frequency*1000) %>%
  mutate(facet_label = paste0(AA,codon))

# temp7A <- temp7 %>%
#   group_by(family) %>%
#   summarise(nspecies = length(unique(Species))) %>%
#   filter(nspecies >= 100)

plot1 <- ggplot(filter(temp7,
                       family %nin% c("Aerococcaceae"),
              ), 
       aes(x=frequency_per_1000, group=interaction(family), 
           fill=family,
           color=family
           )) +
  stat_ecdf(size=1) +
  scale_x_continuous(breaks=c(0,25,50)) +
  scale_y_continuous(breaks=c(0,0.5,1)) +
  ylab("Fraction of Data") +
  xlab("Codon use per 1000") +
  facet_wrap(~facet_label) +
  theme(legend.position = "right",
        strip.background = element_blank(),
        axis.text.x = element_text(angle=0, size=12, color="black"),
        axis.text.y = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16) )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_S6G_Leu_codon_usage"
width=5
height=3
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```




















#_
#_
#_
#_
#_
#Redo
##Figure 4
###Read in codon dict
```{r}
#Add in some codons
gene_id_dict <- read.csv("../20210125_new_data/mRNA_mapping/human_combine_name_dict.txt",sep="\t")

PATH = "/Users/ckatanski/4SR/data/20210125_HEK_stress_v2/source/Codon_counter/human_combine_codon.tsv"
codon_dict <- read.csv(PATH, header=T, sep="\t") %>%
  mutate(seq = nchar(seq)/3) %>%
  full_join(., gene_id_dict, by=c("gene")) %>%
  filter(gene_biotype=="protein_coding") %>%
  select(-feature, -gene_biotype, -chromosome, -transcript_biotype, -gene)
#temp$gene <- factor(temp$gene)
codon_dict$gene_symbol <- factor(codon_dict$gene_symbol)

#Normalize codon usage by gene length
temp <- codon_dict[-ncol(codon_dict)] / codon_dict[,(ncol(codon_dict)-1)] 
temp <- cbind(codon_dict[ncol(codon_dict)], temp) %>%
  select(-seq)
#Average transcript varients
#https://dplyr.tidyverse.org/reference/summarise_all.html
codon_dict <- temp %>%
  group_by(gene_symbol) %>%
  summarise(across(where(is.numeric), ~ median(.x, na.rm = TRUE))) 

```

###Human codon usage
```{r}
PATH="/Users/ckatanski/4SR/data/20210125_HEK_stress_v2/source/human_codon_usage.tsv"
human_codon <- read.csv(PATH, sep="\t", header=T) %>%
  mutate(genome_usage = usage/1000) %>%
  select(-usage)
```

###Combine mRNA with codon usage
```{r}
#combine transcript varianets
temp <- mRNA_data %>%
  group_by(gene_symbol, treatment, sample, rep, gene_biotype) %>%
  summarise(count = sum(count)) 
#Make factors
temp$sample <- factor(temp$sample)
temp$treatment <- factor(temp$treatment, levels=c("control","heat","H2O2","AsO2"))
temp$rep <- factor(temp$rep)

#summarise to RPM; filter to abundnat mRNAs;
temp1 <- temp %>%
  group_by(rep, treatment, sample) %>%
  filter(gene_biotype=="protein_coding") %>%
  mutate(rpm = count / sum(count)) %>% #Not a true RPM. Its fraction of mRNA mapped reads
  filter(count >= 100, #Mostly a filter on polysome fraction, which has 10X lower coverage
         !is.na(gene_biotype))

#pivot wider by sample; calculate ratio polysome to input
temp1 <- temp1 %>%
  filter(gene_biotype=="protein_coding") %>%
  select(-count) %>% #Esential
  group_by(treatment, gene_symbol, gene_biotype, rep) %>%
  pivot_wider(names_from = c("sample"), values_from=c("rpm")) %>%
  ungroup() %>%
  mutate(TE = polysome / input)

#Limit analysis to a set of well detected genes
common_detection_list <- temp1 %>%
  select(gene_symbol, rep, treatment, TE) %>%
  filter(rep==1) %>%
  pivot_wider(names_from = c("treatment"), values_from=c("TE")) %>%
  filter(!is.na(control) &
           !is.na(heat) &
           !is.na(H2O2) &
           !is.na(AsO2))
common_detection_list <- unique(common_detection_list$gene_symbol)

#Confirm distribution of TE values
ggplot(filter(temp1,
              !(treatment=="control" & rep==2),
              !(treatment=="H2O2" & rep==3),
              gene_symbol %in% common_detection_list
              ),
       aes(x=TE, color=treatment, group=interaction(treatment, rep))) +
  geom_freqpoly() +
  # geom_density() +
  scale_x_log10()

#Compute Zscore for each replicate
temp2 <- temp1 %>%
  filter(gene_symbol %in% common_detection_list) %>%
  group_by(rep, treatment) %>%
  mutate(log_TE = log10(TE)) %>% #The data are LOG normally distributed, take that log
  mutate(zscore = (log_TE - mean(log_TE, na.rm=T))/sd(log_TE, na.rm=T) )

# #Confirm log-normal zscore shape
# ggplot(filter(temp2,
#               # !(treatment=="control" & rep==2),
#               # !(treatment=="H2O2" & rep==3),
#               # gene_symbol %in% common_detection_list
#               ),
#        aes(x=zscore, color=treatment, group=interaction(treatment, rep))) +
#   # geom_freqpoly() +
#   geom_density() +
#   geom_vline(xintercept = 0, linetype=2, alpha=0.3) 


#make a column for control ratio
temp3 <- temp2 %>%
  filter(treatment=="control") %>%
  filter(!(treatment=="control" & rep==2)) %>% #This replicates is too poor coverage to be useful
  group_by(gene_symbol) %>%
  summarise(control_TE = mean(TE, na.rm=T),
            control_log_TE = mean(log_TE, na.rm=T),
            control_zscore = mean(zscore, na.rm=T) ) %>% #take mean of control reps
  select(gene_symbol, control_TE, control_log_TE, control_zscore) %>%
  full_join(temp2, by=c("gene_symbol")) %>%
  mutate(change_zscore = zscore - control_zscore)

# #Plot change in z-score
# ggplot(filter(temp3,
#               # !(treatment=="control" & rep==2),
#               # !(treatment=="H2O2" & rep==3),
#               # gene_symbol %in% common_detection_list
#               ),
#        aes(x=zscore - control_zscore, color=treatment, group=interaction(treatment, rep))) +
#   geom_freqpoly() +
#   # geom_density() +
#   geom_vline(xintercept = c(-1, 0, 1 ), linetype=2, alpha=0.3) 

```

##Fig 4C - distribution of TE
###plot the figure old bean
```{r}
#Confirm distribution of TE values
temp1$treatment <- factor(temp1$treatment, levels = rev(c("control", "heat", "H2O2", "AsO2")) )
 
plot1 <- ggplot(filter(temp1,
              !(treatment=="control" & rep==2),
              !(treatment=="H2O2" & rep==3),
              gene_symbol %in% common_detection_list,
              rep ==1
              ),
       aes(x=TE, color=treatment, group=interaction(treatment, rep))) +
  # geom_vline(xintercept = 1, linetype=2, alpha=0.3) +
  geom_freqpoly(size=1) +
  scale_x_log10() +
  scale_color_manual(values= alpha(c("heat"="#f76f73", "H2O2"="#07c4c5", "AsO2"="#027fdc", "control"="grey70"), 1), 
                     guide = guide_legend(reverse = TRUE) )+
  scale_x_log10(limits=c(0.1, 10),
                breaks=c(0.1,0.5,1,2,10)) +
  scale_y_continuous(breaks=c(0,100,200)) +
  xlab("Translational efficency\n(polysome / input)") +
  ylab("Number of genes") +
  theme(legend.position = "right",
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0),
        strip.text.x = element_text(size=14),
        axis.text = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16))


#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_4Z_change_in_TE"
width=3.5
height=2.5
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```


###TE period; percentile; codon
```{r}
#Calculate percentile; take top and bottom 20% for each rep, stress
HIGH_percentile = 0.66
LOW_percentile = 0.33

temp4 <- temp3 %>%
  group_by(rep, treatment ) %>%
  mutate( TE_percent_rank = percent_rank(TE)) 

#Merge with codons 
temp5 <- temp4 %>% 
  mutate(percentile_grouping = ifelse(TE_percent_rank <= LOW_percentile, "low", 
                                      ifelse( TE_percent_rank >= HIGH_percentile, "high", "average") ) ) %>%
  filter(!is.na(TE)) %>% #Some genes with poor polysome detection result in NA percentiles
  select(rep, treatment, gene_symbol, percentile_grouping) %>%
  full_join(., codon_dict, by=c("gene_symbol")) %>%
  filter(!is.na(rep)) %>% #Codon_dict has more genes than are in the TE dataset
  pivot_longer(col=c(-rep, -treatment, -gene_symbol, -percentile_grouping),
               names_to = c("codon"), values_to=c("frequency")) %>%
  full_join(., AA_table, by=c("codon")) %>%
  group_by(treatment, rep, codon) %>%
  mutate(frequency_percentile = percent_rank(frequency))

temp5$percentile_grouping <- factor(temp5$percentile_grouping, levels=c("low", "average","high"))
my_comparisons <- list( c("average", "high"),
                        c("average", "low"),
                        c("high", "low") )

ggplot(filter(temp5, 
              # AA == "Ile",
              codon=="GAG",
              !(rep==2 & treatment=="control"),
              # frequency > 0 
              ),
       aes(x=percentile_grouping, y=frequency_percentile, color=percentile_grouping)) +
  geom_boxplot() +
  coord_cartesian(ylim=c(-0.01, 1.5)) +
  # scale_y_log10() +
  stat_compare_means(comparisons = my_comparisons, size=4,  
                     method="wilcox.test",
                     p.adjust.method = "none",
                     #label="p.signif"
                     ) +
  facet_nested(treatment ~ AA + codon + rep ) +
  theme(strip.background.x = element_rect(fill="white", color="black"),
        strip.text.x = element_text(angle=0))

##Show the actual distributions
# ggplot(filter(temp5, 
#               AA == "Lys",
#               codon=="AAA",
#               !(rep==2 & treatment=="control"),
#               # frequency > 0 
#               ),
#        aes( x=frequency_percentile, color=percentile_grouping)) +
#   geom_freqpoly() +
#   # coord_cartesian(ylim=c(0, 1.5)) +
#   facet_nested(treatment ~ AA + codon + rep ) +
#   theme(strip.background.x = element_rect(fill="white", color="black"),
#         strip.text.x = element_text(angle=0))

```
##Fig 4E - Third codon position usage

###P-vlaues (wilcox) for each codon on TE
```{r}

get_codon_pvalue <- function(df){
  asdf <- df %>%
    ungroup() %>%
    mutate(metric = frequency_percentile ) %>%
    filter(!is.na(metric),
           !is.infinite(metric))
  
  side = mean(filter(asdf, percentile_grouping=="high")$metric) / 
          mean(filter(asdf, percentile_grouping=="low")$metric)
  side = ifelse(side >= 1, 1, -1) #enriched in high metric genes
    
  
  A <- pairwise.wilcox.test(asdf$metric, asdf$percentile_grouping, 
                   p.adjust.method = "none", 
                   alternative = "two.sided", 
                   pool.sd = F #Not sure if this function even accepts this parameter
                   )$p.value
  #A[1] is 1 vs 2; A[2] is 1 vs 3; A[3] is 2 vs 2; A[4] is 2 vs 3
  df=data.frame(low.average=A[1],
                low.high=A[2],
                average.low=A[4],
                enrichment = side, 
                N_equals.group1= nrow(filter(asdf, percentile_grouping=="low")),
                N_equals.group2= nrow(filter(asdf, percentile_grouping=="average")),
                N_equals.group3= nrow(filter(asdf, percentile_grouping=="high")) )
  } 


temp6 <- temp5 %>% 
  filter( !(rep==2 & treatment=="control") ) %>%
  group_by(rep, treatment, codon) %>%
  do(get_codon_pvalue(.))

#prety plot
temp6 <- temp6 %>%
  separate(codon, sep=c(1,2), into=c("first","second", "third"), remove=F)
temp6$third <- factor(temp6$third, levels=c("A","T","C","G"))

ggplot(filter(temp6,
              codon %nin% c("TAA","TGA","TAG","ATG","TGG") #STOP, Trp, Met
              ),
       aes(x=interaction(codon, third), y=rep, 
           fill= log(low.high) *enrichment*-1 #Remember, the log gives another negative -1 
           ))+
  geom_tile() +
  scale_fill_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="grey70", 
                       # limits=c(0,1),
                       # breaks=c(0,.5,1)
                       ) +
  facet_nested( treatment ~ .) +
  theme(axis.text.x=element_text(angle=90, vjust=0.5) )

```


##TE movers
###delta-TE percentile; codon
```{r}
#Calculate percentile; take top and bottom 20% for each rep, stress
HIGH_percentile = 0.66
LOW_percentile = 0.33

temp4 <- temp3 %>%
  group_by(rep, treatment ) %>%
  mutate( change_Z_percent_rank = percent_rank(change_zscore)) 

#Merge with codons 
temp5 <- temp4 %>% 
  mutate(percentile_grouping = ifelse(change_Z_percent_rank <= LOW_percentile, "low", 
                                      ifelse( change_Z_percent_rank >= HIGH_percentile, "high", "average") ) ) %>%
  filter(!is.na(change_Z_percent_rank)) %>% #Some genes with poor polysome detection result in NA percentiles
  select(rep, treatment, gene_symbol, percentile_grouping) %>%
  full_join(., codon_dict, by=c("gene_symbol")) %>%
  filter(!is.na(rep)) %>% #Codon_dict has more genes than are in the TE dataset
  pivot_longer(col=c(-rep, -treatment, -gene_symbol, -percentile_grouping),
               names_to = c("codon"), values_to=c("frequency")) %>%
  full_join(., AA_table, by=c("codon")) %>%
  group_by(treatment, rep, codon) %>%
  mutate(frequency_percentile = percent_rank(frequency))

temp5$percentile_grouping <- factor(temp5$percentile_grouping, levels=c("low", "average","high"))
my_comparisons <- list( c("average", "high"),
                        c("average", "low"),
                        c("high", "low") )

ggplot(filter(temp5, 
              # AA == "Ile",
              codon=="AAA",
              !(rep==2 & treatment=="control"),
              treatment != "control", #replicates are necessarily symetric
              # frequency > 0 
              ),
       aes(x=percentile_grouping, y=frequency_percentile, color=percentile_grouping)) +
  geom_boxplot() +
  coord_cartesian(ylim=c(-0.01, 1.5)) +
  # scale_y_log10() +
  stat_compare_means(comparisons = my_comparisons, size=4,  
                     method="wilcox.test",
                     p.adjust.method = "none",
                     #label="p.signif"
                     ) +
  facet_nested(treatment ~ AA + codon + rep ) +
  theme(strip.background.x = element_rect(fill="white", color="black"),
        strip.text.x = element_text(angle=0))

##Show the actual distributions
# ggplot(filter(temp5, 
#               AA == "Lys",
#               codon=="AAA",
#               !(rep==2 & treatment=="control"),
#               # frequency > 0 
#               ),
#        aes( x=frequency_percentile, color=percentile_grouping)) +
#   geom_freqpoly() +
#   # coord_cartesian(ylim=c(0, 1.5)) +
#   facet_nested(treatment ~ AA + codon + rep ) +
#   theme(strip.background.x = element_rect(fill="white", color="black"),
#         strip.text.x = element_text(angle=0))

```

###P-vlaues (wilcox) for each codon on change in Zscore
```{r}

get_codon_pvalue <- function(df){
  asdf <- df %>%
    ungroup() %>%
    mutate(metric = frequency_percentile ) %>%
    filter(!is.na(metric),
           !is.infinite(metric))
  
  side = mean(filter(asdf, percentile_grouping=="high")$metric) / 
          mean(filter(asdf, percentile_grouping=="low")$metric)
  side = ifelse(side >= 1, 1, -1) #enriched in high metric genes
    
  
  A <- pairwise.wilcox.test(asdf$metric, asdf$percentile_grouping, 
                   p.adjust.method = "none", 
                   alternative = "two.sided", 
                   pool.sd = F #Not sure if this function even accepts this parameter
                   )$p.value
  #A[1] is 1 vs 2; A[2] is 1 vs 3; A[3] is 2 vs 2; A[4] is 2 vs 3
  df=data.frame(low.average=A[1],
                low.high=A[2],
                average.low=A[4],
                enrichment = side, 
                N_equals.group1= nrow(filter(asdf, percentile_grouping=="low")),
                N_equals.group2= nrow(filter(asdf, percentile_grouping=="average")),
                N_equals.group3= nrow(filter(asdf, percentile_grouping=="high")) )
  } 


temp6 <- temp5 %>% 
  filter( !(rep==2 & treatment=="control") ) %>%
  group_by(rep, treatment, codon, AA) %>%
  do(get_codon_pvalue(.))

#prety plot
temp6 <- temp6 %>%
  separate(codon, sep=c(1,2), into=c("first","second", "third"), remove=F)
temp6$third <- factor(temp6$third, levels=c("A","T","C","G"))
temp6 <- temp6 %>%
  mutate(p_value_color = ifelse(low.high >= PVALUE_CUTOFF, "No enrichment", ifelse(enrichment >0, "Enriched in genes\nwith increasing TE", "Depleted in genes\nwith increasing TE")) )

#Averge p-values to simplify plotting - gross
temp7 <- temp6 %>%
  group_by(codon, AA, treatment, first, second, third) %>%
  summarise(median_pvalue = median(low.high, na.rm=T),
            enrichment = mean(enrichment, na.rm=T))

PVALUE_CUTOFF = 0.05
temp7 <- temp7 %>%
  mutate(p_value_color = ifelse(median_pvalue >= PVALUE_CUTOFF, "No enrichment", ifelse(enrichment > 0, "Enriched codons", "Depleted codons")) )
#Beaturify order
temp7 <- temp7 %>%
  mutate(x_label=codon)
temp7$third <- factor(temp7$third, levels = c("A","T","C","G"))
temp7 <- temp7[order(temp7$third, temp7$codon ),]
temp7$x_label <- factor(temp7$x_label, levels = unique(temp7$x_label))
#Further beauty
temp7$treatment <- factor(temp7$treatment, levels = rev(c("control","heat","H2O2","AsO2")))

# #Colored by p-value value
# ggplot(filter(temp6,
#               treatment != "control", #replicates are necessarily symetric
#               # codon %nin% c("TAA","TGA","TAG","ATG","TGG") #STOP, Trp, Met
#               ),
#        aes(x=interaction(codon, third), y=rep, 
#            fill= log(low.high) *enrichment*-1 #Remember, the log gives another negative -1 
#            ))+
#   geom_tile() +
#   geom_vline(xintercept = c(15.5), color="black", size=1) +
#   scale_fill_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="grey70", 
#                        # limits=c(0,1),
#                        # breaks=c(0,.5,1)
#                        ) +
#   facet_nested( treatment ~ .) +
#   theme(axis.text.x=element_text(angle=90, vjust=0.5) )


#Chris' preferred plot
# #Simplified colors
# ggplot(filter(temp6,
#               treatment != "control", #replicates are necessarily symetric
#               codon %nin% c("TAA","TGA","TAG","ATG","TGG") #STOP, Trp, Met
#               ),
#        aes(x=interaction(codon, third), y=rep,
#            fill= p_value_color #Remember, the log gives another negative -1
#            ))+
#   geom_tile() +
#   geom_vline(xintercept = c(13.5, 30.5, 46.5 ), color="black", size=1) +
#   scale_fill_manual(values=c("No enrichment"="grey70", 
#                              "Enriched in genes\nwith increasing TE"="red3", 
#                              "Depleted in genes\nwith increasing TE"="blue3")) +
#   facet_nested( treatment ~ .) +
#   theme(legend.position = "top",
#         legend.title = element_blank(),
#         strip.background = element_blank(),
#         strip.text.y = element_text(angle=0, size=16),
#         axis.text = element_text(size=10, color="black"),
#         axis.title.y = element_blank(),
#         axis.title.x = element_blank() )


#Simplified colors
plot1 <- ggplot(filter(temp7,
              treatment != "control", #replicates are necessarily symetric
              codon %nin% c("TAA","TGA","TAG","ATG","TGG") #STOP, Trp, Met
              ),
       aes(x=x_label, y=treatment, 
           fill= p_value_color #Remember, the log gives another negative -1 
           ))+
  geom_tile() +
  geom_vline(xintercept = c(13.5, 30.5, 46.5 ), color="black", size=1) +
  scale_fill_manual(values=c("No enrichment"="grey70", 
                             "Enriched codons"="red3", 
                             "Depleted codons"="blue3")) +
  # facet_nested( treatment ~ .) +
  theme(legend.position = "top",
        legend.title = element_blank(),
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0, size=16),
        axis.text.y = element_text(size=12, color="black"),
        axis.text.x = element_text(size=12, color="black", angle=90, vjust=0.5),
        axis.title.y = element_blank(),
        axis.title.x = element_blank() )

#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_4X_codon_use_andTE"
width=6
height=2
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```

##Fig 4F - Specific codon
###Lys codons change in Z-score
```{r}
#Calculate percentile; take top and bottom 20% for each rep, stress
HIGH_percentile = 0.66
LOW_percentile = 0.33

temp4 <- temp3 %>%
  group_by(rep, treatment ) %>%
  mutate( change_Z_percent_rank = percent_rank(change_zscore)) 

#Merge with codons 
temp5 <- temp4 %>% 
  mutate(percentile_grouping = ifelse(change_Z_percent_rank <= LOW_percentile, "decreased TE", 
                                      ifelse( change_Z_percent_rank >= HIGH_percentile, "increased TE", "no change") ) ) %>%
  filter(!is.na(change_Z_percent_rank)) %>% #Some genes with poor polysome detection result in NA percentiles
  select(rep, treatment, gene_symbol, percentile_grouping) %>%
  full_join(., codon_dict, by=c("gene_symbol")) %>%
  filter(!is.na(rep)) %>% #Codon_dict has more genes than are in the TE dataset
  pivot_longer(col=c(-rep, -treatment, -gene_symbol, -percentile_grouping),
               names_to = c("codon"), values_to=c("frequency")) %>%
  full_join(., AA_table, by=c("codon")) %>%
  group_by(treatment, rep, codon) %>%
  mutate(frequency_percentile = percent_rank(frequency))

temp5$percentile_grouping <- factor(temp5$percentile_grouping, levels=c("decreased TE","no change","increased TE"))
my_comparisons <- list( c("decreased TE", "no change"),
                        c("no change", "increased TE"),
                        c("increased TE", "decreased TE") )

plot1 <- ggplot(filter(temp5,
              AA =="Lys",
              # codon %in% c("AGC","AGT"),
              # rep==1,
              treatment != "control",
              treatment == "AsO2",
              ),
       aes(y=frequency*1000, x=interaction(percentile_grouping), 
           # color=rep
           ) ) +
  geom_boxplot() +
  scale_y_log10() +
  coord_cartesian(ylim = c(0.5, 2500)) +
  stat_compare_means(comparisons = my_comparisons, size=4,  
                     method="wilcox.test",
                     p.adjust.method = "none",
                     label="p.signif"
                     ) +
  facet_nested(. ~ codon  ) +
  ylab("Codon use per thousand") +
  theme(#legend.position = "top",
        #legend.title = element_blank(),
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0, size=16),
        axis.text.y = element_text(size=12, color="black"),
        axis.text.x = element_text(size=12, color="black", angle=90, vjust=0.5, hjust=1),
        axis.title.y = element_text(size=16),
        axis.title.x = element_blank() )


#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_4Y_Lys_codon_TE"
width=3
height=3
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)


```







#++++++++++++++++
#Old figure 4
##Fig 4C - TE distributions among well detected genes
###Wrangle mRNA data 
```{r}
PATH="../20210125_new_data/mRNA_Wen/HEK_stress_polysome_read_count_group_name_dict.txt"

mRNA_data <- read.table(PATH, header=T) %>%
  pivot_longer(cols=c(-gene_id, -feature, -chromosome, -gene_biotype, -transcript_biotype, -gene_symbol), 
               names_to="experiment", values_to="count") %>%
  separate(experiment, sep=c("_"), into=c("sample","etc")) %>%
  separate(etc, sep="\\.", into=c("treatment","rep"))

mRNA_data$feature <- factor(mRNA_data$feature)
mRNA_data$gene_biotype <- factor(mRNA_data$gene_biotype)
mRNA_data$transcript_biotype <- factor(mRNA_data$transcript_biotype)
mRNA_data$treatment <- recode(mRNA_data$treatment, "Heat"="heat", "H2O2"="H2O2", "NaAsO2"="AsO2", "ctrl"="control")

```

###Calculate TE
```{r}
#combine transcript varianets
temp <- mRNA_data %>%
  group_by(gene_symbol, treatment, sample, rep, gene_biotype) %>%
  summarise(count = sum(count)) 
#Make factors
temp$sample <- factor(temp$sample)
temp$treatment <- factor(temp$treatment, levels=c("control","heat","H2O2","AsO2"))
temp$rep <- factor(temp$rep)
#summarise to RPM; filter to abundnat mRNAs; average replicates
temp <- temp %>%
  group_by(rep, treatment, sample) %>%
  mutate(rpm = count / sum(count)) %>%
  filter(count >= 100,
         !is.na(gene_biotype)) %>%
  group_by(treatment, sample, gene_symbol, gene_biotype) %>%
  summarise(rpm_mean = mean(rpm, na.rm=T))
#pivot wider by sample; calculate ratio polysome to input
temp <- temp %>%
  group_by(treatment, gene_symbol, gene_biotype) %>%
  pivot_wider(names_from = c("sample"), values_from=c("rpm_mean")) %>%
  mutate(ratio = polysome / input)
#Make useful x label
temp <- temp %>%
  mutate(label_name = NA) %>%
  mutate(label_name = ifelse( ( (ratio >2 #| ratio <0.5
                                 ) & ( input    >=0.0001 & polysome >=0.001) ) |
                                input > 0.005,
                             gene_symbol, NA))



temp2 <- temp %>%
  ungroup() %>%
  select(treatment, gene_symbol, gene_biotype, ratio) 
temp2 <- temp2 %>%
  filter(treatment=="control") %>%
  mutate(ctrl_ratio = ratio) %>%
  select(gene_symbol, ctrl_ratio) %>%
  full_join(temp2, by=c("gene_symbol"))

temp3 <- temp2 %>%
  mutate(ratio_ratio = ratio / ctrl_ratio) %>%
  filter(!is.na(ratio_ratio))

#Filter for only the ~2500 genes detected in all conditions
temp4 <- temp3 %>%
  filter(gene_biotype=="protein_coding") %>%
  select(gene_symbol, treatment, ratio_ratio) %>%
  pivot_wider(names_from=c("treatment"), values_from=c("ratio_ratio")) %>%
  filter(control >0 & heat >0 & H2O2 >0 & AsO2 >0 ) %>%
  pivot_longer(cols=c(-gene_symbol), names_to=c("treatment"), values_to=c("ratio_ratio"))

plot1 <- ggplot(filter(temp4,
              treatment !="control",
              # gene_biotype=="protein_coding"
              ),
       aes(x=ratio_ratio, color=treatment)) +
  geom_vline(xintercept = 1, linetype=2) +
  geom_freqpoly(size=2) +
  # geom_density() +
  scale_color_manual(values= alpha(c("heat"="#f76f73", "H2O2"="#07c4c5", "AsO2"="#027fdc", "control"="grey70"), 1), 
                     guide = guide_legend(reverse = TRUE) )+
  scale_x_log10(limits=c(0.1, 10),
                breaks=c(0.1,0.5,1,2,10)) +
  scale_y_continuous(breaks=c(0,250,500)) +
  xlab("Change in translational efficency\n(stress / control)") +
  ylab("Number of genes") +
  theme(legend.position = "right",
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0),
        strip.text.x = element_text(size=14),
        axis.text = element_text(size=12, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_text(size=16))


#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_4C_change_in_TE"
width=3.5
height=2.5
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```


##Fig 4E - Third codon position usage
###Read in codon dict
```{r}
#Add in some codons
gene_id_dict <- read.csv("../20210125_new_data/mRNA_mapping/human_combine_name_dict.txt",sep="\t")

PATH = "/Users/ckatanski/4SR/data/20210125_HEK_stress_v2/source/Codon_counter/human_combine_codon.tsv"
codon_dict <- read.csv(PATH, header=T, sep="\t") %>%
  mutate(seq = nchar(seq)/3) %>%
  full_join(., gene_id_dict, by=c("gene")) %>%
  filter(gene_biotype=="protein_coding") %>%
  select(-feature, -gene_biotype, -chromosome, -transcript_biotype, -gene)
#temp$gene <- factor(temp$gene)
codon_dict$gene_symbol <- factor(codon_dict$gene_symbol)

#Normalize codon usage by gene length
temp <- codon_dict[-ncol(codon_dict)] / codon_dict[,(ncol(codon_dict)-1)] 
temp <- cbind(codon_dict[ncol(codon_dict)], temp) %>%
  select(-seq)
#Average transcript varients
#https://dplyr.tidyverse.org/reference/summarise_all.html
codon_dict <- temp %>%
  group_by(gene_symbol) %>%
  summarise(across(where(is.numeric), ~ median(.x, na.rm = TRUE))) 
```

###Human codon usage
```{r}
PATH="/Users/ckatanski/4SR/data/20210125_HEK_stress_v2/source/human_codon_usage.tsv"
human_codon <- read.csv(PATH, sep="\t", header=T) %>%
  mutate(genome_usage = usage/1000) %>%
  select(-usage)
```

###Combine mRNA with codon usage
```{r}
#combine transcript varianets
temp <- mRNA_data %>%
  group_by(gene_symbol, treatment, sample, rep, gene_biotype) %>%
  summarise(count = sum(count)) 
#Make factors
temp$sample <- factor(temp$sample)
temp$treatment <- factor(temp$treatment, levels=c("control","heat","H2O2","AsO2"))
temp$rep <- factor(temp$rep)

#summarise to RPM; filter to abundnat mRNAs; average replicates
temp <- temp %>%
  group_by(rep, treatment, sample) %>%
  mutate(rpm = count / sum(count)) %>%
  filter(count >= 100,
         !is.na(gene_biotype)) %>%
  group_by(treatment, sample, gene_symbol, gene_biotype) %>%
  summarise(rpm_mean = mean(rpm, na.rm=T))

#pivot wider by sample; calculate ratio polysome to input
temp <- temp %>%
  group_by(treatment, gene_symbol, gene_biotype) %>%
  pivot_wider(names_from = c("sample"), values_from=c("rpm_mean")) %>%
  mutate(TE = polysome / input)

#clean it up a bit
temp2 <- temp %>%
  ungroup() %>%
  select(treatment, gene_symbol, gene_biotype, TE)

#make a column for control ratio
temp2 <- temp2 %>%
  filter(treatment=="control") %>%
  mutate(ctrl_TE = TE) %>%
  select(gene_symbol, ctrl_TE) %>%
  full_join(temp2, by=c("gene_symbol"))
#Calculate change in ratio
temp3 <- temp2 %>%
  mutate(change_TE = TE / ctrl_TE) %>%
  filter(!is.na(change_TE))

#calculate a Z-score
temp4 <- temp3 %>%
  filter(gene_biotype=="protein_coding") %>%
  group_by(treatment) %>%
  mutate(change_TE = log(change_TE)) %>%
  mutate(zscore = (change_TE - mean(change_TE) )/sd(change_TE) )

#Verify that this is log-normal distributed and zscore is appropriate
ggplot(temp4, aes(x=zscore, color=treatment)) +
  geom_density() #Lookin' pretty good

# Take genes that are 1.9 sd up regulated
ZSCORE_THRESHOLD = 1

temp5 <- temp4 %>%
  filter(zscore >= ZSCORE_THRESHOLD | zscore <= (-1*ZSCORE_THRESHOLD),
         treatment !="control" #Zscore for control is NA anyways
         ) %>%
  select(gene_symbol, treatment, zscore) %>%
  pivot_wider(names_from = c("treatment"), values_from=c("zscore"))

#Combine codons
temp6 <- temp5 %>%
  full_join(., codon_dict, by=c("gene_symbol")) 
#Pivot longer codons
temp6 <- temp6 %>%
  pivot_longer(cols=c("AsO2","heat","H2O2"), names_to=c("treatment"), values_to=c("zscore")) %>%
  filter(!is.na(zscore)) %>%
  pivot_longer(cols=c(-"gene_symbol", -"treatment",-"zscore"),
               names_to=c("codon"), values_to=c("fraction")) 

#Designate genes as TE up or TE down for each stress 
genes_TE_stress <- temp6 %>%
  mutate(zscore=ifelse(zscore>0, "TE_up", "TE_down")) %>%
  separate(codon, sep=c(1,2), into=c("first","second","third"), remove = F)

```

###Calculate lots of p-valeus
```{r}

#Calculate a p-vlaue based on the TE enriched and TE delepted distributions
get_codon_pvalue <- function(df){
  asdf <- df %>%
    ungroup() %>%
    mutate(metric = fraction ) %>%
    filter(!is.na(metric),
           !is.infinite(metric))
  B <- length(filter(asdf, zscore=="TE_up")$metric)
  C <- length(filter(asdf, zscore=="TE_down")$metric)

  if( B<1 | C<1){
    return(data.frame(N_equals.group1=NA))
  }

  A <- pairwise.t.test(asdf$metric, asdf$zscore, p.adjust.method = "none", alternative = "two.sided", pool.sd = F)$p.value
  #print(A)
  #A[1] is ctrl vs mild; A[2] is ctrl severe; A[3] is mild vs mild; A[4] is mild vs severe
  ratio = mean(filter(asdf, zscore=="TE_up")$metric) / 
          mean(filter(asdf, zscore=="TE_down")$metric)
  
  df=data.frame(pvalue=c(A[1]),
                ratio=c(ratio))
  return(df)
}




temp8 <- genes_TE_stress %>%
  filter() %>%
  group_by(treatment, codon) %>%
  do(get_codon_pvalue(.) ) %>%
  mutate(sign=ifelse(ratio>1, "enriched_in_high_TE", "enriched_in_low_TE")) %>%
  separate(codon, sep=c(1,2), into=c("first","second","third"), remove=F)

#Designate a p-value cut off
PVALUE_CUTOFF = 0.01

temp8$treatment <- factor(temp8$treatment, levels=rev(c("heat","H2O2","AsO2")))
temp8 <- temp8 %>%
  mutate(plot_pvalue = ifelse(pvalue <= PVALUE_CUTOFF & sign=="enriched_in_high_TE", "Enriched in genes\nwith increased TE", 
                              ifelse(pvalue <= PVALUE_CUTOFF & sign=="enriched_in_low_TE", "Enriched in genes\nwith decreased TE", "No significant\nenrichment")))


#Beaturify order
temp8 <- temp8 %>%
  mutate(x_label=codon)
temp8$third <- factor(temp8$third, levels = c("A","T","C","G"))
temp8 <- temp8[order(temp8$third, temp8$codon ),]
temp8$x_label <- factor(temp8$x_label, levels = unique(temp8$x_label))

plot1 <- ggplot(filter(temp8,
                       codon %nin% c("TAA","TGA","TAG","ATG","TGG")), 
       aes(x=x_label, y=treatment, fill=plot_pvalue ) ) +
  geom_tile() +
  scale_fill_manual(values=c("Enriched in genes\nwith increased TE"="red3", 
                             "Enriched in genes\nwith decreased TE"="blue3", 
                             "No significant\nenrichment"="grey70")) +
  theme(axis.text.x = element_text(angle=90, vjust=0.5)) +
  geom_vline(xintercept = c(14.5, 30.5, 46.5), size=1, color="black", linetype=1) +
  theme(legend.position = "top",
        legend.title = element_blank(),
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0, size=16),
        axis.text = element_text(size=10, color="black"),
        axis.title.y = element_blank(),
        axis.title.x = element_blank() )


#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_4E_codon_use_andTE"
width=6
height=2
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)

```


##Fig 4F - Serine codons TE level
###Start from TE calculations; plot fractions
```{r}
temp <- genes_TE_stress %>%
  full_join(., human_codon, by=c("codon")) %>%
  select(-genome_usage)

temp$treatment <- factor(temp$treatment, levels=c("heat","H2O2","AsO2"))
temp$third <- factor(temp$third, levels=c("A","T","C","G"))
temp$zscore <- recode(temp$zscore, "TE_up"="Increased\nTE", "TE_down"="Decreased\nTE")
my_comparisons <- list( c("Increased\nTE", "Decreased\nTE"))

#Pretty label
temp <- temp %>%
  mutate(plot_label=NA) %>%
  mutate(plot_label = ifelse(treatment=="heat"&fraction*1000 > 50 & zscore=="Increased\nTE", gene_symbol, NA))


#serine, each codon
plot1 <- ggplot(filter(temp, 
              AA=="Ser",
              codon %in% c("AGT","AGC")
              ),
       aes(x=zscore, y= fraction*1000  )) +
  geom_boxplot() +
  scale_y_continuous(breaks= c(0,30,60, 90),
                     limits = c(0, 85)
                     ) +
  # geom_text_repel(aes(label=plot_label)) +
  stat_compare_means(comparisons = my_comparisons, size=5,  method="t.test", label="p.signif") + #how does this compute p-value?
  ylab("Codon usage per 1000") +
  facet_nested(treatment  ~  codon ) +
  theme(legend.position = "top",
        legend.title = element_blank(),
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0, size=16),
        axis.text = element_text(size=10, color="black"),
        axis.title.y = element_text(size=16),
        axis.title.x = element_blank() )


#Save the figure
layout <- rbind(c(1))
figure <- grid.arrange(plot1,  layout_matrix = layout)
path="../figures/draft2/"
figure_name="panel_4F_Ser_codon_TE"
width=3
height=4
scale=1.5
dpi=600

# ggsave(paste0(path, figure_name,".svg"), 
#        plot = figure,
#        scale = scale,
#        dpi = dpi,
#        width = width, 
#        height = height)

ggsave(paste0(path, figure_name,".png"),
       plot = figure,
       scale = scale,
       dpi = dpi,
       width = width,
       height = height)


```










