---
title: "Nat_meth_2020_v2_figures"
author: "Chris Katanski"
date: "3/10/2020"
output: html_document
---

#Set up
```{r, set up packages etc., echo=FALSE, results="hide", warning=FALSE, message=FALSE}
library(ggplot2)
library(scales)
library(reshape2)
library(tidyr)
library(dplyr)

library(grid)
library(gridExtra)
library(ggpubr)

#Reading in svg files into r for ggplotting
library(grImport2)
library(grConvert)
#Reading and plotting jpeg and PNG
library(jpeg)
library(png)

library(ggrepel)
library(stringr)
library(forcats)
library(readxl)

#For nice log labels
#library(cat.extras)
library(plotly)

library(extrafont)
font_import()
loadfonts()

#Set the global figure theme now
theme_set(
  theme_bw() + theme(legend.position = "None", 
        text = element_text(family = "Arial", size = 14),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.background = element_rect(color = "black"))
  )

#Define a function that excludes things
`%nin%` = Negate(`%in%`)

empty_graph <- ggplot() + theme_void()


setwd("~/4SR/data/20200317_Nature_methods_data/2020 02 24 ecoli figures/source/")

```

#Figure 2: E. coli no stress
##Read in Ecoli catagory data
```{r, echo=FALSE}

#There are two datasets that I made from the sequencing reads.
#The "catagory data" ends in ".counts.tsv". This just counts the 
#number of reads that map to each gene, without any location information

#The "base data" has every nucleotide for every gene all in one massive 
#tsv file. Its an easy format for R to parse though.


#Rep1 minus DM
data1 = read.csv("../counts/TP-CK4-RPL7_S7bc6.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep=1, stress="None")

data2 = read.csv("../counts/TP-CK4-RPL7_S7bc7.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep = 1, stress="DIP")

data3 = read.csv("../counts/TP-CK4-RPL7_S7bc10.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep = 1, stress="aMG")

data4 = read.csv("../counts/TP-CK4-RPL7_S7bc12.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep = 1, stress="H2O2")

#Rep1 plus DM
data5 = read.csv("../counts/TP-CK4-RPL10_S10bc6.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep=1, stress="None")

data6 = read.csv("../counts/TP-CK4-RPL10_S10bc7.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep = 1, stress="DIP")

data7 = read.csv("../counts/TP-CK4-RPL10_S10bc10.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep = 1, stress="aMG")

data8 = read.csv("../counts/TP-CK4-RPL10_S10bc12.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep = 1, stress="H2O2")



#Rep2 minus DM
data9 = read.csv("../counts/TP-CK4-RPL11_S11bc6.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep=2, stress="None")

data10 = read.csv("../counts/TP-CK4-RPL11_S11bc7.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep = 2, stress="DIP")

data11 = read.csv("../counts/TP-CK4-RPL11_S11bc10.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep = 2, stress="aMG")

data12 = read.csv("../counts/TP-CK4-RPL11_S11bc12.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep = 2, stress="H2O2")

#Rep2 plus DM
data13 = read.csv("../counts/TP-CK4-RPL12_S12bc6.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep=2, stress="None")

data14 = read.csv("../counts/TP-CK4-RPL12_S12bc7.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep = 2, stress="DIP")

data15 = read.csv("../counts/TP-CK4-RPL12_S12bc10.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep = 2, stress="aMG")

data16 = read.csv("../counts/TP-CK4-RPL12_S12bc12.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep = 2, stress="H2O2")

Ecoli_counts_data <- rbind(data1, data2, data3, data4, 
                      data5, data6, data7, data8, 
                      data9, data10,data11,data12,
                      data13,data14,data15,data16)
```

##Read in Ecoli base data
```{r, read in basewise data, echo=FALSE}
#Rep1 minus DM
data1 = read.csv("../basewise/trna_only/TP-CK4-RPL7_S7bc6.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep=1, stress="None")

data2 = read.csv("../basewise/trna_only/TP-CK4-RPL7_S7bc7.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep = 1, stress="DIP")

data3 = read.csv("../basewise/trna_only/TP-CK4-RPL7_S7bc10.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep = 1, stress="aMG")

data4 = read.csv("../basewise/trna_only/TP-CK4-RPL7_S7bc12.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep = 1, stress="H2O2")

#Rep1 plus DM
data5 = read.csv("../basewise/trna_only/TP-CK4-RPL10_S10bc6.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep=1, stress="None")

data6 = read.csv("../basewise/trna_only/TP-CK4-RPL10_S10bc7.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep = 1, stress="DIP")

data7 = read.csv("../basewise/trna_only/TP-CK4-RPL10_S10bc10.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep = 1, stress="aMG")

data8 = read.csv("../basewise/trna_only/TP-CK4-RPL10_S10bc12.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep = 1, stress="H2O2")


#Rep2 minus DM
data9 = read.csv("../basewise/trna_only/TP-CK4-RPL11_S11bc6.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep=2, stress="None")

data10 = read.csv("../basewise/trna_only/TP-CK4-RPL11_S11bc7.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep = 2, stress="DIP")

data11 = read.csv("../basewise/trna_only/TP-CK4-RPL11_S11bc10.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep = 2, stress="aMG")

data12 = read.csv("../basewise/trna_only/TP-CK4-RPL11_S11bc12.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep = 2, stress="H2O2")

#Rep2 plus DM
data13 = read.csv("../basewise/trna_only/TP-CK4-RPL12_S12bc6.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep=2, stress="None")

data14 = read.csv("../basewise/trna_only/TP-CK4-RPL12_S12bc7.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep = 2, stress="DIP")

data15 = read.csv("../basewise/trna_only/TP-CK4-RPL12_S12bc10.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep = 2, stress="aMG")

data16 = read.csv("../basewise/trna_only/TP-CK4-RPL12_S12bc12.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep = 2, stress="H2O2")

#combine everything
data_ec_base <- rbind(data1, data2, data3, data4, 
                      data5, data6, data7, data8, 
                      data9, data10,data11,data12,
                      data13,data14,data15,data16)

#reorganize the names of genes into something sensible
data <- data_ec_base %>%
  separate(gene, sep="_", into=c("asdf1","asdf2","asdf3",
                                       "asdf4","asdf5","asdf6","Isodecoder")) %>%
  select(-asdf1, -asdf2, -asdf3, -asdf4, -asdf5, -asdf6)

#IDK what these isodecoders are, so I'm cutting them for now. They do have some intersting behavior
data <- filter(data, Isodecoder %nin% c("tRNA-Und-NNN-1-1","tRNA-Und-NNN-2-1"))

#Fix names further
Ecoli_base_data <- data %>%
  separate(Isodecoder, sep="tRNA-", c("wasted","Isodecoder"))%>%
  select(-wasted)

#Reorder the stresses
Ecoli_base_data$stress <-factor(Ecoli_base_data$stress, levels=c("None","DIP","aMG","H2O2"))

#Clean up names just a bit more
Ecoli_base_data$Isodecoder <- str_replace_all(Ecoli_base_data$Isodecoder, "-1-1","-1")
Ecoli_base_data$Isodecoder <- str_replace_all(Ecoli_base_data$Isodecoder, "-2-1","-2")
```




##Fig 2A1: Piechart Ecoli all RNAs
```{r, more different text, echo=FALSE}

#Normalize coutns to RPM
#Then average reps
temp <- Ecoli_counts_data %>%
  select(-type) %>% #get rid of unused data column
  filter(DM=="+" & stress=="None") %>% #filter data set for piechart
  group_by(DM, stress, rep) %>% #normalize counts to RPM
  mutate(counts = counts / sum(counts) *1000000) %>% #normalize counts to RPM
  group_by(Gene, gene_biotype, gene_symbol, DM, stress) %>%
  summarise(counts = mean(counts)) %>%
  group_by()

#combine biotype of less abundant catagories into "ncRNA"
temp <- temp %>%
  filter(gene_biotype %nin% c("tRNA", "rRNA", "protein_coding")) %>%
  mutate(gene_biotype = "ncRNA") %>%
  rbind(., filter(temp, gene_biotype %in% c("tRNA", "rRNA", "protein_coding"))) %>%
  group_by(gene_biotype) %>%
  summarise(counts = sum(counts)) %>%
  mutate(counts = round((counts / sum(.$counts))*100,1))

#rename "protein_coding" to "mRNA"
temp <- temp %>%
  filter(gene_biotype=="protein_coding") %>%
  mutate(gene_biotype="mRNA") %>%
  rbind(., filter(temp, gene_biotype != "protein_coding"))


#transform value to be compatable with piechart - idk how this works, I copied it from stack exchange
temp <- temp %>% 
  arrange(desc(gene_biotype)) %>%
  mutate(ypos = cumsum(counts)- 0.5*counts )

fig2A1 <-ggplot(temp, aes(x="", y=counts, fill=gene_biotype)) +
  geom_bar(stat="identity", width=1, color="white") +
  coord_polar("y", start=0) +
  geom_text_repel(aes(y=ypos, label=paste0(gene_biotype, "\n", counts, "%") ), box.padding = 1)+
  theme_void() +
  scale_fill_manual(values = c("tRNA"="#99CCFF", 
                                "rRNA"="#FFCC33", 
                               "mRNA"="#99CC99",
                                "ncRNA"="#CC99CC"))+
  labs(fill="Gene type") +
  guides(fill = guide_legend(reverse=TRUE)) + 
  theme(legend.position = "None",
        plot.title = element_text(hjust = 0.5)) +
  labs(title="All reads")

fig2A1
```

##Fig 2A2: Piechart ncRNA
```{r, abundance of ncRNAs, echo=FALSE}

#Filter and simplify, minus DM, no stress
temp <- Ecoli_counts_data %>%
  filter(DM=="+" & stress=="None") %>%
  group_by() %>%
  group_by(Gene, type, gene_biotype, gene_symbol, DM, stress) %>%
  summarise(counts = mean(counts)) %>%
  group_by()

#filter for just ncRNAs
temp <- temp %>%
  filter(gene_biotype %in% c("ncRNA")) 

#summing catagory counts
temp <- temp %>%
  filter(counts < 100) %>% #take all the low expression genes
  mutate(gene_symbol = "Other") %>% #and bin them together in the "other" catagory
  rbind(., filter(temp, counts >= 100)) %>% #combine this with the rest of the data
  group_by() %>%
  group_by(gene_symbol) %>%
  summarise(counts = sum(counts)) %>% #add up all the counts
  mutate(counts = round((counts / sum(.$counts))*100,1)) #round to 1 decimal

#transform value to be compatable with piechart
temp <- temp %>% 
  arrange(desc(gene_symbol)) %>%
  mutate(ypos = cumsum(counts)- 0.5*counts )

#Nice easy piechart
fig2A2 <-ggplot(temp, aes(x="", y=counts, fill=gene_symbol)) +
  geom_bar(stat="identity", width=1, color="white") +
  coord_polar("y", start=0) +
  geom_text_repel(aes(y=ypos, label=paste0(gene_symbol, "\n", counts, "%")), box.padding = 1)+
  theme_void() +
  # scale_fill_manual(values = c("tRNA"="maroon", 
  #                               "rRNA"="grey40", 
  #                              "protein_coding"="forestgreen",
  #                               "ncRNA"="purple2"))+
  labs(fill="Gene type") +
  guides(fill = guide_legend(reverse=TRUE)) + 
  theme(legend.position = "None",
        plot.title = element_text(hjust = 0.5)) +
  labs(title="ncRNA reads")

fig2A2

myplotb1 <- arrangeGrob(fig2A1)
myplotb2 <- arrangeGrob(fig2A2)
lay = rbind(c(1,2))
fig2A <- grid.arrange(myplotb1, myplotb2, layout_matrix = lay)
fig2A

```


##Fig 2B1: abundance of Leu isoacceptors
```{r}

microarray_data <- read.csv("../microarray_data.txt", sep="\t", header=TRUE)
microarray_data$gene_symbol <- as.factor(microarray_data$gene_symbol)
microarray_data <- microarray_data %>%
  group_by(gene_symbol) %>%
  mutate(method="Microarray") %>%
  filter(grepl("Leu", gene_symbol)) %>%
  group_by() 
microarray_data$rep1 <- as.numeric(microarray_data$rep1)
microarray_data$rep2 <- as.numeric(microarray_data$rep2)

temp <- Ecoli_counts_data %>%
  filter(gene_biotype=="tRNA") %>%
  group_by(DM, rep, stress) %>%
  mutate(RPM = counts / sum(counts)*1000000) %>% #normalize counts to rpm
  group_by(Gene, type, gene_biotype, gene_symbol, DM, stress) %>%
  mutate(mean_RPM = mean(RPM)) %>%
  filter(stress=="None", DM=="+") %>%#filter for the plot
  group_by() %>%
  filter(grepl("leu", gene_symbol))

#Fix the stupid tRNA names
temp$rep <- recode(temp$rep, "1"="rep1", "2"="rep2", "3"="rep3")
temp$gene_symbol <- recode(temp$gene_symbol,  "leuP"="LeuCAG",
                                              "leuQ"="LeuCAG",
                                              "leuT"="LeuCAG",
                                              "leuU"="LeuGAG",
                                              "leuV"="LeuCAG",
                                              "leuW"="LeuUAG",
                                              "leuX"="LeuCAA",
                                              "leuZ"="LeuUAA")


temp <- temp %>%
  filter(DM=="+", stress=="None") %>%
  select(gene_symbol, rep, RPM) %>%
  group_by(gene_symbol, rep) %>%
  summarise(RPM = sum(RPM)) %>%
  group_by(rep) %>%
  mutate(RPM = RPM/ sum(RPM)) %>%
  pivot_wider(names_from = rep, values_from = RPM) %>%
  mutate(method = "CHRIS-seq") %>%
  rbind(microarray_data) %>%
  pivot_longer(cols =c("rep1", "rep2") ,names_to ="rep", values_to = "RPM") %>%
  group_by(method, gene_symbol) %>%
  mutate(mean_RPM = mean(RPM))



fig2B1 <- ggplot(filter(temp), aes(x=gene_symbol, y=mean_RPM, color=method)) +
  geom_point(shape=95, size=5, position= position_dodge(width=1) ) +
  geom_point(aes(y=RPM), position=position_dodge(width=1))+
  scale_y_continuous(breaks=c(0,.5,1), limits = c(0,1))+
  theme(#aspect.ratio = 2,
        axis.text.x = element_text(angle = 90, hjust = 0, vjust=.5),
        axis.title.x = element_blank(),
        plot.title = element_text(hjust = 0.5),
        axis.title.y = element_blank()
        #legend.position = "bottom"
        )+
  ylab("Fraction of total signal") +
  annotate(geom="text", label="Leu", y=0.9, x="LeuCAA") +
  scale_color_manual(values=c("CHRIS-seq"="red", "Microarray"="grey50")) 

fig2B1
```

##Fig 2B2: abundance of Arg isoacceptors
```{r}

microarray_data <- read.csv("../microarray_data.txt", sep="\t", header=TRUE)
microarray_data$gene_symbol <- as.factor(microarray_data$gene_symbol)
microarray_data <- microarray_data %>%
  group_by(gene_symbol) %>%
  mutate(method="Microarray") %>%
  filter(grepl("Arg", gene_symbol)) %>%
  group_by() 
microarray_data$rep1 <- as.numeric(microarray_data$rep1)
microarray_data$rep2 <- as.numeric(microarray_data$rep2)

temp <- Ecoli_counts_data %>%
  filter(gene_biotype=="tRNA") %>%
  group_by(DM, rep, stress) %>%
  mutate(RPM = counts / sum(counts)*1000000) %>% #normalize counts to rpm
  group_by(Gene, type, gene_biotype, gene_symbol, DM, stress) %>%
  mutate(mean_RPM = mean(RPM)) %>%
  filter(stress=="None", DM=="+") %>%#filter for the plot
  group_by() %>%
  filter(grepl("arg", gene_symbol))

#Fix the stupid tRNA names
temp$rep <- recode(temp$rep, "1"="rep1", "2"="rep2", "3"="rep3")
temp$gene_symbol <- recode(temp$gene_symbol,  "argQ"="ArgACG",
                                              "argU"="ArgUCU",
                                              "argV"="ArgACG",
                                              "argW"="ArgCCU",
                                              "argX"="ArgCCG",
                                              "argY"="ArgACG",
                                              "argZ"="ArgACG")

temp <- temp %>%
  filter(DM=="+", stress=="None") %>%
  select(gene_symbol, rep, RPM) %>%
  group_by(gene_symbol, rep) %>%
  summarise(RPM = sum(RPM)) %>%
  group_by(rep) %>%
  mutate(RPM = RPM/ sum(RPM)) %>%
  pivot_wider(names_from = rep, values_from = RPM) %>%
  mutate(method = "CHRIS-seq") %>%
  rbind(microarray_data) %>%
  pivot_longer(cols =c("rep1", "rep2") ,names_to ="rep", values_to = "RPM") %>%
  group_by(method, gene_symbol) %>%
  mutate(mean_RPM = mean(RPM))



fig2B2 <- ggplot(filter(temp), aes(x=gene_symbol, y=mean_RPM, color=method)) +
  geom_point(shape=95, size=5, position= position_dodge(width=1) ) +
  geom_point(aes(y=RPM), position=position_dodge(width=1))+
  scale_y_continuous(breaks=c(0,.5,1), limits = c(0,1) )+
  theme(axis.text.x = element_text(angle = 90, hjust = 0, vjust=.5),
        axis.title.x = element_blank(),
        plot.title = element_text(hjust = 0.5),
        legend.position = "top"
        #legend.text = element_text(size=8),
        #legend.title = element_text(size=10),
        #legend.background = element_blank())+
        )+
  ylab("Fraction of total signal") +
  labs(color="") +
  annotate(geom="text", label="Arg", y=.9, x="ArgCCU") +
  scale_color_manual(values=c("CHRIS-seq"="red", "Microarray"="grey50")) +
  theme(#aspect.ratio = 2,
        #legend.key = element_rect(fill = "white", colour = "black"),
        legend.background = element_blank()
        ) 

fig2B2







myplotb1 <- arrangeGrob(fig2B1)
myplotb2 <- arrangeGrob(fig2B2 + theme(legend.position = "None"))
plt_legend <- get_legend(fig2B2)

lay = rbind(c(3,3),
            c(1,2),
            c(1,2),
            c(1,2))

fig2B <- grid.arrange(myplotb2, myplotb1, plt_legend, layout_matrix = lay)
#
fig2B
```


##Fig 2C: Effect of DM abundance correlation
```{r, echo=FALSE}

#Don't average by replicate
temp <- Ecoli_counts_data %>%
  filter(stress=="None") %>%
  select(-type) %>%
  group_by()

#rename "protein_coding" to "mRNA"
temp <- temp %>%
  filter(gene_biotype=="protein_coding") %>%
  mutate(gene_biotype="mRNA") %>%
  rbind(., filter(temp, gene_biotype != "protein_coding"))

#combine biotype of less abundant catagories into "ncRNA"
temp <- temp %>%
  filter(gene_biotype %nin% c("tRNA", "rRNA", "mRNA")) %>%
  mutate(gene_biotype = "ncRNA") %>%
  rbind(., filter(temp, gene_biotype %in% c("tRNA", "rRNA", "mRNA"))) %>%
  group_by(DM, stress, rep) %>%
  mutate(counts = (counts / sum(.$counts)))

#rename the reps and DM to make ggplot easier
temp$rep <- recode(temp$rep, "1"="rep 1", "2"="rep 2")
temp$DM <- recode(temp$DM, "-"="minusDM", "+"="plusDM")
temp$gene_biotype <- factor(temp$gene_biotype, levels = c("mRNA","ncRNA","rRNA","tRNA"))

#Make columns of counts for DM and rep
temp <- temp %>%
  spread(DM, counts)

temp <- temp %>%
  group_by() %>%
  group_by(gene_biotype, rep) %>%
  summarise(correlation = cor(minusDM, plusDM, method=c("spearman"))**2  )


fig2C <- ggplot(temp, aes(x=rep, y=gene_biotype, fill=correlation)) +
  geom_tile() +
  geom_text(aes(label = round(correlation, 2))) +
  scale_fill_gradient(low="grey70", high="red", limits = c(0,1))+
  theme_void() +
  theme(legend.position = "right",
        plot.title = element_text(hjust = 0.5, size=12),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.text = element_text())+
  labs(title="±DM",
       fill=bquote(~r^2)) +
  theme(aspect.ratio = 2) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))

fig2C
```

##Fig 2D: Modification heatmap (-DM, no stress)
```{r}

temp <- Ecoli_base_data %>%
  mutate(mutation = ifelse(pileup<10, NA, mutation)) %>%
  filter(stress=="None" & DM=="-") %>%
  group_by(Isodecoder, position, stress, DM) %>%
  mutate(mutation = mean(mutation) ) %>%
  group_by()
  # mutate(mutation = round(mutation/5, 2)*5 )


fig2D <- ggplot(temp, aes(Isodecoder, y=position, fill=mutation))+
  geom_tile()+
  scale_y_reverse(breaks = c(8, 37, 47, 76)
                  #sec.axis = dup_axis()
                  ) +
  scale_fill_gradient(low="grey70", high="red", limits=c(0,.25), na.value = "grey70" , breaks=c(0, .12, .25)) +
  geom_tile(data=filter(temp, mutation>=0.25), fill="red4") +
  theme(legend.position = "right",
        legend.text = element_text(angle=0, vjust=.5),
        legend.background = element_blank(),
        axis.text.x = element_text(angle = 90, hjust = 0, vjust=.5),
        axis.title.x = element_blank(),
        axis.ticks.y = element_line(size=1),
        axis.ticks.length.y = unit(.2, "cm"),
        plot.title = element_blank()) +
  labs(fill="Mutation") + #element_text(hjust = 0.5)) +
  #labs(title="E. coli mutation rate\nNo stress, -DM, average of replicates") +
  ylab("Position (nt)") +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))


#Wonkey legend games
plt_legend <- get_legend(fig2D + theme(legend.title = element_blank() ) )
plt_legend <- plt_legend #+ theme(legend.title = element_blank())
#make a >0.25 legend
legend_data <-data.frame(Mutation=c(">0.25",">0.25"), x=c(0,1), y=c(0,1))
legend_legend <- ggplot(legend_data, aes(x=y,y=y,fill=Mutation))+
  geom_tile() +
  scale_fill_manual(values= c(">0.25"="red4")) +
  theme(legend.position = "right", legend.background = element_blank(),
        legend.key.height = unit(c(0.2),"cm") )
legend_legend <- get_legend(legend_legend)

lay=rbind(c(1),
          c(1),
          c(2),
          c(2),
          c(3),
          c(3))

legend_legend <- arrangeGrob(empty_graph,left=legend_legend)
plt_legend <- arrangeGrob(empty_graph, left=plt_legend)

plt_legend <- arrangeGrob(legend_legend, plt_legend, empty_graph, layout_matrix = lay  )

#plt_legend <- arrangeGrob(legend_legend, plt_legend, empty_graph,layout_matrix=lay )


fig2D <- arrangeGrob(fig2D + theme(legend.position = "None") )

lay = rbind(c(1,1,1,1,2))
fig2D <- grid.arrange(fig2D, plt_legend, layout_matrix = lay)

fig2D



```


##Fig 2E: sRNA abundance
```{r, echo=FALSE}

temp <- Ecoli_counts_data %>%
  filter(counts >1) %>%
  group_by(DM, rep, stress) %>%
  mutate(counts = counts/sum(counts)*1000000 ) %>%
  filter(gene_biotype %nin% c("tRNA","rRNA","protein_coding")) %>%
  group_by(Gene, gene_symbol, stress, DM) %>%
  mutate(std_count = sd(counts),
            mean_count = mean(counts)) %>%
  group_by() %>%
  filter(stress=="None") %>%
  filter(counts >1) #This is a rpm filter now, I just didn't rename it to RPM

#Arrange by -DM value
temp <- temp %>%
  filter(DM=="-") %>%
  mutate(minus_mc = mean_count) %>%
  select(gene_symbol, minus_mc) %>%
  full_join(.,temp, by=c("gene_symbol")) %>%
  mutate(gene_symbol = fct_reorder(gene_symbol, desc(minus_mc)))


fig2E <- ggplot(filter(temp, stress=="None"), aes(x=gene_symbol, y=mean_count, gourp=DM, color=DM))+
  geom_point(shape=95, size=5) +
  geom_point(aes(y=counts), size=.4) +
  scale_color_manual(values = c("-"="black", "+"="red")) +
  #geom_errorbar(aes(ymin=mean_count-std_count, ymax=mean_count+std_count)) +
  scale_y_log10(labels=trans_format('log10',math_format(10^.x))) +
  theme(axis.text.x = element_text(angle=90, vjust=.5), 
        axis.title.x = element_blank(),
        legend.position=c(0.9,0.6))+
  #labs(title="Abundance of ncRNA")+
  ylab("Reads per million") +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")) +
  #Add stuff
  facet_grid(rows=vars(stress)) +
  theme(strip.background = element_blank(),
        strip.text.y = element_text(color="white", angle = 0))

fig2E
```

##Fig 2F: Summary change in sRNA abundance
```{r, echo=FALSE}
temp <- Ecoli_counts_data %>%
  filter(counts >1) %>%
  group_by(DM, rep, stress) %>%
  mutate(counts = counts/sum(counts)*1000000 ) %>%
  filter(gene_biotype %nin% c("tRNA","rRNA","protein_coding")) %>%
  group_by(Gene, gene_symbol, stress, DM) %>%
  mutate(mean_count = mean(counts)) %>%
  arrange(-mean_count) %>%
  group_by() %>%
  mutate(gene_symbol = fct_reorder(gene_symbol, desc(mean_count) ))

#Normalize to None stress
temp <- temp %>%
  filter(stress=="None") %>%
  mutate(ns_mc = mean_count) %>%
  select(gene_symbol, DM, ns_mc) %>%
  full_join(., temp, by=c("gene_symbol","DM")) %>%
  mutate(norm_mean_count = mean_count / ns_mc,
         norm_counts = counts / ns_mc )

temp$stress <- recode(temp$stress, "None"="None","DIP"="DIP","aMG"="αMG","H2O2"="H2O2" )
temp$stress <- factor(temp$stress, levels=c("None","DIP","αMG","H2O2")) 

fig2F <- ggplot(filter(temp, 
                       gene_symbol %in% c("ffs","ryhB","oxyS","sgrS"),
                       DM=="-"), 
                aes(x= gene_symbol, 
                    y= mean_count, 
                    color=stress, 
                    group=gene_symbol))+
  geom_point(shape=95, size=5)+
  geom_point(aes(y= counts))+
  #geom_line() +
  scale_y_continuous(trans="log2", breaks=c(5, 50, 500, 5000))+
  scale_color_manual(values = c("None"="#ffa900",
                                "DIP" ="#008f00",
                                "αMG" ="#021eaa",
                                "H2O2"="#b6192b")) +
  theme(axis.text.x = element_text(angle=0, vjust=.5), 
        axis.title.x = element_blank(),
        plot.title = element_text(hjust = 0.5),
        legend.position="bottom")+
  #labs(title="Change in abundance of ncRNA")+
  labs(color="Stress") +
  ylab("Reads per million") +
  theme(#aspect.ratio = 1.5,
        legend.text = element_text(size=8),
        legend.title = element_text(size=10),
        legend.box.background = element_rect()) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))
  #ylab("Fold change from non-stressed")
fig2F


```

##Fig 2G: sRNA traces with stress
###read in full data
```{r, echo=FALSE}
#Rep1 minus DM
data1 = read.csv("../basewise/full_map/TP-CK4-RPL7_S7bc6.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep=1, stress="None")

data2 = read.csv("../basewise/full_map/TP-CK4-RPL7_S7bc7.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep = 1, stress="DIP")

data3 = read.csv("../basewise/full_map/TP-CK4-RPL7_S7bc10.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep = 1, stress="aMG")

data4 = read.csv("../basewise/full_map/TP-CK4-RPL7_S7bc12.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep = 1, stress="H2O2")

#Rep1 plus DM
data5 = read.csv("../basewise/full_map/TP-CK4-RPL10_S10bc6.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep=1, stress="None")

data6 = read.csv("../basewise/full_map/TP-CK4-RPL10_S10bc7.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep = 1, stress="DIP")

data7 = read.csv("../basewise/full_map/TP-CK4-RPL10_S10bc10.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep = 1, stress="aMG")

data8 = read.csv("../basewise/full_map/TP-CK4-RPL10_S10bc12.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep = 1, stress="H2O2")


#Rep2 minus DM
data9 = read.csv("../basewise/full_map/TP-CK4-RPL11_S11bc6.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep=2, stress="None")

data10 = read.csv("../basewise/full_map/TP-CK4-RPL11_S11bc7.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep = 2, stress="DIP")

data11 = read.csv("../basewise/full_map/TP-CK4-RPL11_S11bc10.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep = 2, stress="aMG")

data12 = read.csv("../basewise/full_map/TP-CK4-RPL11_S11bc12.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep = 2, stress="H2O2")

#Rep2 plus DM
data13 = read.csv("../basewise/full_map/TP-CK4-RPL12_S12bc6.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep=2, stress="None")

data14 = read.csv("../basewise/full_map/TP-CK4-RPL12_S12bc7.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep = 2, stress="DIP")

data15 = read.csv("../basewise/full_map/TP-CK4-RPL12_S12bc10.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep = 2, stress="aMG")

data16 = read.csv("../basewise/full_map/TP-CK4-RPL12_S12bc12.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep = 2, stress="H2O2")

data <- rbind(data1, data2, data3, data4, 
                      data5, data6, data7, data8, 
                      data9, data10,data11,data12,
                      data13,data14,data15,data16)

#Cull the data heard
Ecoli_full_base_data <- data %>%
  select(-Gene,-gene, -type, -A,-C,-G,-"T",-N,-deletion,-insertion)

Ecoli_full_base_data$stress <- recode(Ecoli_full_base_data$stress, "None"="None","DIP"="DIP","aMG"="αMG","H2O2"="H2O2" )
Ecoli_full_base_data$stress <- factor(Ecoli_full_base_data$stress, levels=c("None","DIP","αMG","H2O2")) 

```

###make the plot
```{r, echo=FALSE}
temp <- Ecoli_full_base_data %>%
  #group_by(gene_symbol, DM,rep,stress, position) %>%
  #mutate(pileup = pileup/(sum(pileup)+1) ) %>% #pseudo count
  filter(gene_symbol %in% c("ffs","ryhB","oxyS","sgrS")) %>%
  group_by(gene_symbol, DM, stress, position, rep) 

temp$rep <- recode(temp$rep, "1"="Rep 1", "2"= "Rep 2" )

fig2G <- ggplot(filter(temp, DM=="-"), aes(x=position, y=pileup, color=stress, group=interaction(stress, rep)))+
  geom_line()+
  scale_y_log10(breaks=c(1,100,10000), limits = c(1,10000), labels=trans_format('log10',math_format(10^.x)) ) +
  #scale_y_continuous(limits=c(0,1)) +
  scale_x_continuous(breaks=c(0,100,200)) +
  scale_color_manual(values = c("None"="#ffa900",
                                "DIP" ="#008f00",
                                "αMG" ="#021eaa",
                                "H2O2"="#b6192b")) +
  theme(plot.title = element_text(hjust=0.5),
        legend.position="none")+
  labs(color="Stress")+
  ylab("Mapped reads") +
  xlab("Position along transcript (nt)")+
  facet_grid(rows = vars(gene_symbol),
             cols = vars(rep)) +
  theme(strip.background = element_blank(), 
        legend.text = element_text(size=8),
        legend.title = element_text(size=10),
        axis.text.y = element_text(size=8),
        axis.title.x = element_text(size=12)) +
  theme(strip.text.y = element_text(angle=0)) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))
fig2G
```


##Fig 2H: change in abundance box-and-whisker
```{r}
temp <- Ecoli_counts_data 
temp$gene_symbol <- recode(temp$gene_symbol, 
                           "alaT"="AlaUGC",
                           "alaU"="AlaUCG",
                           "alaV"="AlaUGC",
                           "alaW"="AlaGGC",
                           "alaX"="AlaGGC",
                           
                           "argQ"="ArgACG",
                           "argU"="ArgUCU",
                           "argV"="ArgACG",
                           "argW"="ArgCCU",
                           "argX"="ArgCCG",
                           "argY"="ArgACG",
                           "argZ"="ArgACG",
                           
                           "asnT"="AsnGUU",
                           "asnU"="AsnGUU",
                           "asnV"="AsnGUU",
                           "asnW"="AsnGUU",
                           
                           "aspT"="AspGUC",
                           "aspU"="AspGUC",
                           "aspV"="AspGUC",
                           
                           "cysT"="CysGCA",
                           
                           "glnU"="GlnUUG",
                           "glnV"="GlnCUG",
                           "glnW"="GlnUUG",
                           "glnX"="GlnCUG",
                           
                           "gltT"="GluUUC",
                           "gltU"="GluUUC",
                           "gltV"="GluUUC",
                           "gltW"="GluUUC",
                           
                           "glyT"="GlyUCC",
                           "glyU"="GlyUCC",
                           "glyV"="GlyGCC",
                           "glyW"="GlyGCC",
                           "glyX"="GlyGCC",
                           "glyY"="GlyGCC",
                           
                           "hisR"="HisGUG",
                           
                           "ileT"="IleGAU",
                           "ileU"="IleGAU",
                           "ileV"="IleGAU",
                           "ileX"="IleCAU",
                           "ileY"="IleCAU",
                           
                           "leuP"="LeuCAG",
                           "leuQ"="LeuCAG",
                           "leuT"="LeuCAG",
                           "leuU"="LeuGAG",
                           "leuV"="LeuCAG",
                           "leuW"="LeuUAG",
                           "leuX"="LeuCAA",
                           "leuZ"="LeuUAA",
                           
                           "lysQ"="LysUUU",
                           "lysT"="LysUUU",
                           "lysV"="LysUUU",
                           "lysW"="LysUUU",
                           "lysY"="LysUUU",
                           "lysZ"="LysUUU",
                           
                           "metT"="MetCAU",
                           "metU"="MetCAU",
                           "metV"="MetCAU",
                           "metW"="MetCAU",
                           "metY"="MetCAU",
                           "metZ"="MetCAU",
                           
                           "pheU"="PheGAA",
                           "pheV"="PheGAA",
                           
                           "proK"="ProCGG",
                           "proL"="ProGGG",
                           "proM"="ProUGG",
                           
                           "selC"="SelUCA",
                           
                           "serT"="SerUGA",
                           "serU"="SerUGA",
                           "serV"="SerGCU",
                           "serW"="SerGGA",
                           "serX"="SerGGA",
                           
                           "thrT"="ThrGGU",
                           "thrU"="ThrUGU",
                           "thrV"="ThrGGU",
                           "thrW"="ThrCGU",
                           
                           "trpT"="TrpCCA",
                           
                           "tyrT"="TyrGUA",
                           "tyrU"="TyrGUA",
                           "tyrV"="TyrGUA",
                           
                           "valT"="ValUAC",
                           "valU"="ValUAC",
                           "valV"="ValGAC",
                           "valW"="ValGAC",
                           "valX"="ValUAC",
                           "valY"="ValUAC",
                           "valZ"="ValUAC"
                           )

#Add codon and AA fields
temp <- temp %>%
  separate(gene_symbol, sep=3, c("AA","codon")) %>%
  mutate(gene_symbol = paste0(AA, codon))

#filter to just tRNA
temp <-temp %>%
  filter(gene_biotype =="tRNA") %>%
  select(-type)

#Normalize by total counts in library
temp <- temp %>%
  group_by(DM, stress, rep) %>%
  mutate(RPM = counts / sum(counts) *1000000) %>%
  group_by()

#Average reps
temp <- temp %>%
  group_by(stress, AA, codon, DM, gene_symbol) %>%
  summarise(mean_RPM = mean(RPM))

# #Collapse codons into AA 
# temp <- temp %>%
#   group_by(AA, stress, DM) %>%
#   summarise(combine_RPM = sum(mean_RPM))

#Normalize to unstressed
temp <- temp %>%
  filter(stress=="None") %>%
  mutate(ns_RPM = mean_RPM) %>%
  group_by(gene_symbol, DM) %>%
  select(gene_symbol, DM, ns_RPM) %>%
  full_join(., filter(temp, stress!="None"), by=c("gene_symbol","DM")) %>%
  mutate(norm_RPM = mean_RPM / ns_RPM) %>%
  filter(DM=="+")

#A function to identify boxplot outliers
is_outlier <- function(x) {
  return(x < quantile(x, 0.25) - 1.5 * IQR(x)  | x > quantile(x, 0.75) + 1.5 * IQR(x) )
}
#Archieval version
# is_outlier <- function(x) {
#   return(x < quantile(x, 0.25) - 1.5 * IQR(x) | x > quantile(x, 0.75) + 1.5 * IQR(x))
# }


#label outliar points
temp <- temp %>%
  group_by(stress) %>%
  mutate(outlier = ifelse(is_outlier(norm_RPM), gene_symbol, as.character(NA) ) )

temp$stress <- recode(temp$stress, "None"="None","DIP"="DIP","aMG"="αMG","H2O2"="H2O2" )
temp$stress <- factor(temp$stress, levels=c("None","DIP","αMG","H2O2")) 

#Figure time
fig2H <- ggplot(temp, aes(x=stress, y=norm_RPM )) +
  #geom_jitter(position = position_jitter(seed = 1, width = .1) ) +
  geom_boxplot(width=0.4) +
  geom_text_repel(aes(label=outlier), size=2) + #position = position_jitter(seed = 1, width = .1),
  #scale_y_continuous(trans="log2", breaks=c(0.85, 1, 1.1, 1.2)) +
  geom_hline(yintercept = 1, linetype=2, alpha=0.2) +
  theme(axis.text.x = element_text(angle=0, vjust=.5),
        axis.title.x = element_blank(),
        legend.position="right") +
  ylab("Fold change in\nabundance") +
  #coord_equal() +
  #theme(aspect.ratio = 1) +
  theme(strip.background = element_blank() ) +
  theme(strip.text.y = element_text(angle=0)) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))

fig2H
```


##**Fig 2I: change in charging box-and-whisker
```{r}
#Get the last base pileup
temp <- Ecoli_base_data %>%
  group_by(Isodecoder) %>%
  filter(position %in% c(max(position)) ) %>%
  group_by(Isodecoder, DM, stress, rep) %>%
  mutate(last_pileup = pileup) %>%
  select(last_pileup, Isodecoder, rep, DM, stress)

#calculate charging
temp <- Ecoli_base_data %>%
  group_by(Isodecoder) %>%
  filter(position %in% c(max(position)-1) ) %>%
  group_by(Isodecoder, DM, stress, rep) %>%
  full_join(., temp, by=c("Isodecoder","DM","rep","stress")) %>%
  mutate(charging = last_pileup / ( pileup) )

#average replicates
temp <- temp %>%
  group_by(Isodecoder, DM, stress)  %>%
  summarise(mean_charging = mean(charging) )


# ggplot(filter(temp, DM=="+", stress=="H2O2"), aes(x=Isodecoder, y=mean_charging)) +
#   geom_point() +
#   theme(axis.text.x = element_text(angle=90))

#normalize to none stress
temp <- temp %>%
  group_by(Isodecoder, DM) %>%
  filter(stress=="None") %>%
  mutate(ns_charging = mean_charging) %>%
  select(DM, Isodecoder, ns_charging) %>%
  full_join(., temp, by=c("DM", "Isodecoder")) %>%
  mutate(norm_charging = mean_charging / ns_charging)

#Get AA and codon
temp <- temp %>%
  separate(Isodecoder, sep="-", c("AA","codon","number")) %>%
  mutate(Isodecoder = paste0(AA, codon, number),
         gene_symbol = paste0(AA,codon))

#Collapse by AA
temp <- temp %>%
  group_by(stress, AA, DM, gene_symbol) %>%
  summarise(norm_charging = mean(norm_charging))

#A function to identify boxplot outliers
is_outlier <- function(x) {
  return(x < quantile(x, 0.25) - 1.5 * IQR(x)  | x > quantile(x, 0.75) + 1.5 * IQR(x) )
}

#label outliar points
#https://stackoverflow.com/questions/33524669/labeling-outliers-of-boxplots-in-r
temp <- temp %>%
  group_by(stress) %>%
  mutate(outlier = ifelse(is_outlier(norm_charging), gene_symbol, as.character(NA) ) )


temp$stress <- recode(temp$stress, "None"="None","DIP"="DIP","aMG"="αMG","H2O2"="H2O2" )
temp$stress <- factor(temp$stress, levels=c("None","DIP","αMG","H2O2")) 

fig2I <- ggplot(filter(temp, DM=="-", stress!="None"), aes(x=stress, y=norm_charging)) +
  #geom_jitter(position = position_jitter(seed = 1, width = .1) ) +
  geom_boxplot(width=0.4) +
  geom_text_repel(aes(label=outlier), size=2) + #position = position_jitter(seed = 1, width = .1),
  geom_hline(yintercept = 1, linetype=2, alpha=0.2) +
  scale_y_continuous(breaks=c(0.6,0.8,1.0,1.2,1.4,1.6,1.8)) +
  theme(axis.text.x = element_text(angle=0, vjust=.5),
        axis.title.x = element_blank(),
        legend.position="right") +
  ylab("Fold change in\nfraction charged") +
  #coord_equal() +
  #theme(aspect.ratio = 1) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))
fig2I
```


##Fig 2J: change in mod box-and-whisker
```{r}
#s4U
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Arg-ACG-1")
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Asn-GTT-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Asp-GCT-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Cys-GCA-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Gln-CTG-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Gln-TTG-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Gly-CCC-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Gly-GCC-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="His-GTG-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Ile-CAT-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Leu-CAA-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Leu-TAA-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Met-CAT-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Phe-GAA-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==9, Isodecoder=="Sec-TCA-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Ser-TGA-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Ser-GCT-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Ser-GGA-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Trp-CCA-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Tyr-GTA-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Tyr-GTA-2") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Val-GAC-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Val-GAC-2") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==8, Isodecoder=="Val-TAC-1") %>%
  rbind(temp)
#s4U summary
temp_s4U <- temp %>%
  # group_by(DM, stress, Isodecoder) %>%
  # mutate(mutation=mean(mutation)) %>%
  # filter(stress=="None") %>%
  # group_by(Isodecoder, DM) %>%
  # summarise(ns_mutation = mean(mutation) ) %>%
  # select(Isodecoder, DM, ns_mutation) %>%
  # full_join(., temp, by=c("Isodecoder", "DM")) %>%
  # mutate(change_mutation = mutation / ns_mutation) %>%
  mutate(modification = "s4U") %>%
  select(-position)



#m1G
temp <- Ecoli_base_data %>%
  filter(position==38, Isodecoder=="Arg-CCG-1")
temp <- Ecoli_base_data %>%
  filter(position==38, Isodecoder=="Leu-CAG-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==38, Isodecoder=="Leu-GAG-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==38, Isodecoder=="Pro-CGG-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==38, Isodecoder=="Leu-TAG-1") %>%
  rbind(temp)
#m1G summary
temp_m1G <- temp %>%
  # group_by(DM, stress, Isodecoder) %>%
  # mutate(mutation=mean(mutation)) %>%
  # filter(stress=="None") %>%
  # group_by(Isodecoder, DM) %>%
  # summarise(ns_mutation = mean(mutation) ) %>%
  # select(Isodecoder, DM, ns_mutation) %>%
  # full_join(., temp, by=c("Isodecoder", "DM")) %>%
  # mutate(change_mutation = mutation / ns_mutation) %>%
  mutate(modification = "m1G") %>%
  select(-position)



#Thiocytidine s2C
temp <- Ecoli_base_data %>%
  filter(position==33, Isodecoder=="Arg-ACG-1")
temp <- Ecoli_base_data %>%
  filter(position==33, Isodecoder=="Arg-TCT-1") %>%
  rbind(temp)
temp <- Ecoli_base_data %>%
  filter(position==33, Isodecoder=="Ser-GCT-1") %>%
  rbind(temp)
#s2C summary
temp_s2C <- temp %>%
  # group_by(DM, stress, Isodecoder) %>%
  # mutate(mutation=mean(mutation)) %>%
  # filter(stress=="None") %>%
  # group_by(Isodecoder, DM) %>%
  # summarise(ns_mutation = mean(mutation) ) %>%
  # select(Isodecoder, DM, ns_mutation) %>%
  # full_join(., temp, by=c("Isodecoder", "DM")) %>%
  # mutate(change_mutation = mutation / ns_mutation) %>%
  mutate(modification = "s2C") %>%
  select(-position)



#acp3U
temp <- Ecoli_base_data %>%
  filter(position==48, Isodecoder=="Arg-ACG-1") 
temp <- Ecoli_base_data %>%
  filter(position==48, Isodecoder=="Ile-GAT-1") %>%
  rbind(.,temp)
temp <- Ecoli_base_data %>%
  filter(position==47, Isodecoder=="Ile-CAT-1") %>%
  rbind(.,temp)
temp <- Ecoli_base_data %>%
  filter(position==47, Isodecoder=="Lys-TTT-1") %>%
  rbind(.,temp)
temp <- Ecoli_base_data %>%
  filter(position==48, Isodecoder=="Met-CAT-1") %>%
  rbind(.,temp)
temp <- Ecoli_base_data %>%
  filter(position==47, Isodecoder=="Phe-GAA-1") %>%
  rbind(.,temp)
temp <- Ecoli_base_data %>%
  filter(position==48, Isodecoder=="Val-GAC-1") %>%
  rbind(.,temp)
temp <- Ecoli_base_data %>%
  filter(position==48, Isodecoder=="Val-GAC-2") %>%
  rbind(.,temp)
#acp3U summary
temp_acp3U <- temp %>%
  # group_by(DM, stress, Isodecoder) %>%
  # mutate(mutation=mean(mutation)) %>%
  #filter(stress=="None") %>%
  #group_by(Isodecoder, DM) %>%
  #summarise(ns_mutation = mean(mutation) ) %>%
  #select(Isodecoder, DM, ns_mutation) %>%
  #full_join(., temp, by=c("Isodecoder", "DM")) %>%
  #mutate(change_mutation = mutation / ns_mutation) %>%
  mutate(modification = "acp3U") %>%
  select(-position)




#temp <- temp_acp3U
temp <- rbind(temp_s4U, temp_m1G, temp_s2C, temp_acp3U)

temp$stress <- recode(temp$stress, "None"="None","DIP"="DIP","aMG"="αMG","H2O2"="H2O2" )
temp$stress <- factor(temp$stress, levels=c("None","DIP","αMG","H2O2")) 

fig2J <- ggplot(filter(temp, DM=="-"), aes(x=modification, y=mutation, color=stress)) +
  geom_boxplot() +
  geom_point(position=position_jitterdodge(jitter.width = .1), alpha=0.2 ) +
  #geom_hline(yintercept = 1, linetype=2, alpha=.6) +
  #scale_y_continuous(trans="log2") +
  scale_color_manual(values = c("None"="#ffa900",
                                "DIP" ="#008f00",
                                "αMG" ="#021eaa",
                                "H2O2"="#b6192b")) +
  ylab("Mutation fraction") +
  #xlab("Stress") +
  labs(#title = "Change in modifications with stress",
       color="Stress") +
  #coord_equal() +
  #theme(aspect.ratio = 1) +
  theme(plot.title = element_text(hjust=0.5 ),
        axis.title.x = element_blank(),
        legend.position = "bottom",
        legend.box.background = element_rect(),
        legend.text = element_text(size=8),
        legend.title = element_text(size=10)) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))
  
fig2J
```

##old Save figure 2
```{r, save figure 2, echo=FALSE}

myplota <- arrangeGrob(fig2A, top = textGrob(expression(bold("a")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplotb <- arrangeGrob(fig2B, left = textGrob(expression(bold("b")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplotc <- arrangeGrob(fig2C, left = textGrob(expression(bold("c")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplotd <- arrangeGrob(fig2D, left = textGrob(expression(bold("d")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplote <- arrangeGrob(fig2E, left = textGrob(expression(bold("e")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplotf <- arrangeGrob(fig2F + theme(legend.position = "none"), left = textGrob(expression(bold("f")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplotf_legend <- arrangeGrob(get_legend(fig2F))

myplotg <- arrangeGrob(fig2G, left = textGrob(expression(bold("g")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplotfg <- grid.arrange(myplotf, myplotg, myplotf_legend, layout_matrix = rbind( c(1,2),c(1,2),c(1,2),c(1,2), c(3,3) ) )


myploth <- arrangeGrob(fig2H, left = textGrob(expression(bold("h")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myploti <- arrangeGrob(fig2I, left = textGrob(expression(bold("i")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplotj <- arrangeGrob(fig2J, left = textGrob(expression(bold("j")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

#Set the figure arrangement matrix
lay <- rbind(c(1,1,2,2,3,3),
             c(4,4,4,4,4,4),
             c(5,5,5,5,5,5),
             c(6,6,6,6,7,7),
             c(8,8,8,9,9,9))


#Arrange figures
figure2 <- grid.arrange(myplota, myplotb, myplotc, myplotd, myplote, myplotfg, myploth, myploti, myplotj, 
                        layout_matrix = lay)
figure2

ggsave("../figures/figure2.svg", 
       plot = figure2,
       scale = 1.3,
       width = 8, 
       height = 11)
ggsave("../figures/figure2.png", 
       plot = figure2,
       scale = 1.3,
       width = 8, 
       height = 11)

```

##New Save figure 2
```{r, save figure 2, echo=FALSE}

myplota <- arrangeGrob(fig2A, top = textGrob(expression(bold("a")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplotb <- arrangeGrob(figS2B, left = textGrob(expression(bold("b")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplotc <- arrangeGrob(fig2D, left = textGrob(expression(bold("c")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplotd <- arrangeGrob(figS2A, left = textGrob(expression(bold("d")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

# myplote <- arrangeGrob(fig2E, left = textGrob(expression(bold("e")), x = unit(0, "npc")
#         , y  = unit(1, "npc"), just=c("left","top"),
#         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplote <- arrangeGrob(fig2F + theme(legend.position = "none"), left = textGrob(expression(bold("e")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplote_legend <- arrangeGrob(get_legend(fig2F))

myplotf <- arrangeGrob(fig2G, left = textGrob(expression(bold("f")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplotef <- grid.arrange(myplote, myplotf, myplote_legend, empty_graph,
                         layout_matrix = rbind( c(1,1,1,2,2,2),
                                                c(1,1,1,2,2,2),
                                                c(1,1,1,2,2,2),
                                                c(1,1,1,2,2,2), 
                                                c(4,3,3,2,2,2) ) )

myplotg <- arrangeGrob(fig2H, left = textGrob(expression(bold("g")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myploth <- arrangeGrob(fig2I, left = textGrob(expression(bold("h")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myploti <- arrangeGrob(fig2J, left = textGrob(expression(bold("i")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

#Set the figure arrangement matrix
lay <- rbind(c(1,1,1,2,2,2),
             c(1,1,1,2,2,2),
             c(3,3,3,3,3,3),
             c(3,3,3,3,3,3),
             c(4,4,5,5,5,5),
             c(4,4,5,5,5,5),
             c(6,6,7,7,8,8),
             c(6,6,7,7,8,8))


#Arrange figures
figure2 <- grid.arrange(myplota, myplotb, myplotc, myplotd, myplotef, myplotg, myploth, myploti, 
                        layout_matrix = lay)
#figure2

ggsave("../figures/figure2.svg", 
       plot = figure2,
       scale = 1.3,
       width = 8.5, 
       height = 9)
ggsave("../figures/figure2.png", 
       plot = figure2,
       scale = 1.3,
       width = 8.5, 
       height = 9)

```

#_



#*Figure S2
##NA tRNA abundance chart
```{r}
temp <- Ecoli_counts_data %>%
  filter(gene_biotype=="tRNA") 

#Fix the stupid tRNA names
temp$gene_symbol <- recode(temp$gene_symbol, 
                           "alaT"="AlaUGC",
                           "alaU"="AlaUCG",
                           "alaV"="AlaUGC",
                           "alaW"="AlaGGC",
                           "alaX"="AlaGGC",
                           
                           "argQ"="ArgACG",
                           "argU"="ArgUCU",
                           "argV"="ArgACG",
                           "argW"="ArgCCU",
                           "argX"="ArgCCG",
                           "argY"="ArgACG",
                           "argZ"="ArgACG",
                           
                           "asnT"="AsnGUU",
                           "asnU"="AsnGUU",
                           "asnV"="AsnGUU",
                           "asnW"="AsnGUU",
                           
                           "aspT"="AspGUC",
                           "aspU"="AspGUC",
                           "aspV"="AspGUC",
                           
                           "cysT"="CysGCA",
                           
                           "glnU"="GlnUUG",
                           "glnV"="GlnCUG",
                           "glnW"="GlnUUG",
                           "glnX"="GlnCUG",
                           
                           "gltT"="GluUUC",
                           "gltU"="GluUUC",
                           "gltV"="GluUUC",
                           "gltW"="GluUUC",
                           
                           "glyT"="GlyUCC",
                           "glyU"="GlyUCC",
                           "glyV"="GlyGCC",
                           "glyW"="GlyGCC",
                           "glyX"="GlyGCC",
                           "glyY"="GlyGCC",
                           
                           "hisR"="HisGUG",
                           
                           "ileT"="IleGAU",
                           "ileU"="IleGAU",
                           "ileV"="IleGAU",
                           "ileX"="IleCAU",
                           "ileY"="IleCAU",
                           
                           "leuP"="LeuCAG",
                           "leuQ"="LeuCAG",
                           "leuT"="LeuCAG",
                           "leuU"="LeuGAG",
                           "leuV"="LeuCAG",
                           "leuW"="LeuUAG",
                           "leuX"="LeuCAA",
                           "leuZ"="LeuUAA",
                           
                           "lysQ"="LysUUU",
                           "lysT"="LysUUU",
                           "lysV"="LysUUU",
                           "lysW"="LysUUU",
                           "lysY"="LysUUU",
                           "lysZ"="LysUUU",
                           
                           "metT"="MetCAU",
                           "metU"="MetCAU",
                           "metV"="MetCAU",
                           "metW"="MetCAU",
                           "metY"="MetCAU",
                           "metZ"="MetCAU",
                           
                           "pheU"="PheGAA",
                           "pheV"="PheGAA",
                           
                           "proK"="ProCGG",
                           "proL"="ProGGG",
                           "proM"="ProUGG",
                           
                           "selC"="SelUCA",
                           
                           "serT"="SerUGA",
                           "serU"="SerUGA",
                           "serV"="SerGCU",
                           "serW"="SerGGA",
                           "serX"="SerGGA",
                           
                           "thrT"="ThrGGU",
                           "thrU"="ThrUGU",
                           "thrV"="ThrGGU",
                           "thrW"="ThrCGU",
                           
                           "trpT"="TrpCCA",
                           
                           "tyrT"="TyrGUA",
                           "tyrU"="TyrGUA",
                           "tyrV"="TyrGUA",
                           
                           "valT"="ValUAC",
                           "valU"="ValUAC",
                           "valV"="ValGAC",
                           "valW"="ValGAC",
                           "valX"="ValUAC",
                           "valY"="ValUAC",
                           "valZ"="ValUAC"
                           )

temp <- temp %>%
  group_by(DM, rep, stress, gene_symbol, gene_biotype) %>%
  summarise(counts = sum(counts)) %>% #combine genes that are isodecoders
  group_by(DM, rep, stress) %>%
  mutate(RPM = counts / sum(counts)*1000000) %>% #normalize counts to rpm
  group_by(gene_biotype, gene_symbol, DM, stress) %>%
  mutate(mean_RPM = mean(RPM)) %>%
  filter(stress=="None", DM=="+") %>%#filter for the plot
  group_by() 



UNUSED <- ggplot(filter(temp), aes(x=gene_symbol, y=mean_RPM)) +
  geom_point(shape=95, size=5) +
  geom_point(aes(y=RPM))+
  scale_y_continuous(trans="log2", breaks=c(16000,32000,64000))+
  theme(axis.text.x = element_text(angle = 90, hjust = 0, vjust=.5),
        axis.title.x = element_blank(),
        plot.title = element_text(hjust = 0.5) )+
  labs(title="tRNA abundance")+
  ylab("RPM") +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))

UNUSED

```

##Fig S2A Correlation between replicates
```{r}

#Don't average by replicate
temp <- Ecoli_counts_data %>%
  #filter(stress=="None") %>%
  select(-type) %>%
  group_by()

#rename "protein_coding" to "mRNA"
temp <- temp %>%
  filter(gene_biotype=="protein_coding") %>%
  mutate(gene_biotype="mRNA") %>%
  rbind(., filter(temp, gene_biotype != "protein_coding"))

#combine biotype of less abundant catagories into "ncRNA"
temp <- temp %>%
  filter(gene_biotype %nin% c("tRNA", "rRNA", "mRNA")) %>%
  mutate(gene_biotype = "ncRNA") %>%
  rbind(., filter(temp, gene_biotype %in% c("tRNA", "rRNA", "mRNA"))) %>%
  group_by(DM, stress, rep) %>%
  mutate(counts = (counts / sum(.$counts)))

#rename the reps and DM to make ggplot easier
temp$rep <- recode(temp$rep, "1"="rep1", "2"="rep2")
temp$DM <- recode(temp$DM, "-"=" - DM", "+"=" + DM")
temp$gene_biotype <- factor(temp$gene_biotype, levels = c("tRNA","rRNA","ncRNA","mRNA"))

#Make columns of counts for DM and rep
temp <- temp %>%
  spread(rep, counts)

temp <- temp %>%
  group_by() %>%
  group_by(gene_biotype, stress, DM) %>%
  summarise(correlation = cor(rep1, rep2, method=c("spearman"))**2) 


figS2A <- ggplot(temp, aes(x=gene_biotype, y=interaction(stress, DM), fill=correlation)) +
  geom_tile() +
  geom_text(aes(label = round(correlation, 2))) +
  scale_fill_gradient(low="grey70", high="red", limits = c(0,1), breaks=c(0, .5, 1))+
  theme_void()+
  theme(legend.position = "bottom",
        legend.text = element_text(angle=0, vjust=.5),
        plot.title = element_text(hjust = 0.5, size=12),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.text = element_text())+
  labs(title="Replicates",
       fill=bquote(~r^2) ) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))
  #theme(aspect.ratio = 2)

figS2A

```


##FigS2B correlation +/-DM scatter
```{r}
temp <- Ecoli_counts_data 

temp <- temp %>%
  group_by(DM, rep, stress, gene_symbol, gene_biotype) %>%
  summarise(counts = sum(counts)) %>% #combine genes that are isodecoders
  group_by(DM, rep, stress) %>%
  mutate(RPM = counts / sum(counts)*1000000) %>% #normalize counts to rpm
  group_by(gene_biotype, gene_symbol, DM, stress) %>%
  mutate(mean_RPM = mean(RPM)) %>%
  filter(stress=="None") %>%#filter for the plot
  group_by() 

temp$DM <- recode(temp$DM, "-"="minusDM", "+"="plusDM")

temp$gene_biotype <- recode(temp$gene_biotype, "ncRNA"="ncRNA", 
                                             "protein_coding"="protein coding",
                                            "rRNA" = "rRNA",
                                            "tmRNA" = "tmRNA",
                                             "tRNA" = "tRNA")

temp <- temp %>%
  select(-counts, -RPM) %>%
  pivot_wider(names_from = DM, values_from = mean_RPM)

figS2B <- ggplot(temp, aes(y=minusDM, x=plusDM, color = gene_biotype))+
  geom_point() +
  scale_y_continuous(trans="log2", breaks=c(1,100,10000), labels= trans_format('log10',math_format(10^.x)) )+
  scale_x_continuous(trans="log2", breaks=c(1,100,10000), labels= trans_format('log10',math_format(10^.x)) )+
  theme(legend.position = "right",
        legend.background = element_rect(fill="transparent"),
        legend.text = element_text()) +
  xlab("Demethylase\ntreated (RPM)")+
  ylab("Untreated (RPM)") +
  labs(color="Transcript family") +
  coord_equal() +
  #theme(aspect.ratio = 1) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))

figS2B
```

##FigS2C Trace Pro mutation +/-DM
```{r, echo=FALSE}
temp <- Ecoli_base_data %>%
  filter(Isodecoder=="Pro-GGG-1") %>%
  filter(stress=="None") %>%
  filter(position <73)

figS2C <- ggplot(temp, aes(x=position, y=mutation, color=DM, group=interaction(rep, DM) )) +
  geom_line() +
  #geom_point(size=1) +
  scale_y_continuous(breaks=c(0,.05,.1)) +
  scale_color_manual(values = c("-"="black","+"="red"))+
  theme(legend.position = c(.8, .8),
        plot.title = element_text(hjust=0.5)) +
  #labs(title = "Pro-GGG-1 mutation rate") +
  ylab("Mutation fraction")+
  xlab("Position (nt)")+
  annotate(geom="text", label=bquote(~m^1~'G'), x=37, y=.11) +
  annotate(geom="text", label=bquote(~S^4~'U'), x=8, y=.07) +
  annotate(geom="text", label="ProGGG", x=15, y=.11) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))
figS2C
```

##FigS2D Trace Pro stop +/-DM
```{r, echo=FALSE}
temp <- Ecoli_base_data %>%
  filter(Isodecoder=="Pro-GGG-1") %>%
  filter(stress=="None") %>%
  filter(position <73) %>%
  group_by(DM, Isodecoder, stress, rep) %>%
  mutate(pileup = pileup / max(pileup))

figS2D <- ggplot(temp, aes(x=position, y=pileup, color=DM, group=interaction(rep, DM) )) +
  geom_line() +
  #geom_point(size=1) +
  scale_color_manual(values = c("-"="black","+"="red"))+
  theme(legend.position = c(.8, .3),
        plot.title = element_text(hjust=0.5)) +
  #labs(title = "Pro-GGG-1 stop rate") +
  annotate(geom="text", label="ProGGG", x=15, y=max(temp$pileup)) +
  ylab("Normalized fraction\nof reads mapped") +
  xlab("Position (nt)") +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))
figS2D
```


##Fig S2E Change in small RNA abundance with stress
```{r, echo=FALSE}
temp <- Ecoli_counts_data %>%
  filter(counts >1) %>%
  group_by(DM, rep, stress) %>%
  mutate(counts = counts/sum(counts)*1000000 ) %>%
  filter(gene_biotype %nin% c("tRNA","rRNA","protein_coding")) %>%
  group_by()

temp <- temp %>%
  group_by() %>%
  filter(stress == "None") %>%
  mutate(ns_counts = counts) %>%
  select(Gene, gene_symbol, rep, DM, ns_counts) %>%
  full_join(., filter(temp, stress !="None"), by=c("Gene","gene_symbol", "DM", "rep")) %>%
  mutate(change_counts = counts / ns_counts) %>%
  group_by(Gene, stress, gene_symbol, DM) %>%
  mutate(mean_change_counts = mean(change_counts))


figS2E <- ggplot(filter(temp, DM=="-"), aes(x=gene_symbol, y=mean_change_counts))+
  geom_point(shape=95, size=5) +
  geom_point(aes(y=change_counts), size=1) +
  scale_y_continuous(trans="log2", breaks=c(0.06, .25, 1, 4, 16))+
  geom_hline(yintercept = 1, linetype=2)+
  theme(axis.text.x = element_text(angle=90, vjust=.5), 
        axis.title.x = element_blank(),
        plot.title = element_text(hjust=0.5),
        legend.position="right")+
  #labs(title="Change in abundance of ncRNA")+
  ylab("Fold change in abundance") +
  facet_grid(rows=vars(stress)) +
  theme(strip.background = element_blank()) +
  theme(strip.text.y = element_text(angle=0)) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))

figS2E
```

##Fig S2F: tRNA modificaiton change with stress
```{r, echo=FALSE}
temp <- Ecoli_base_data %>%
  mutate(mutation = ifelse(pileup<50, NA, mutation)) %>%
  group_by() %>%
  group_by(Isodecoder, DM, stress, position) %>%
  summarise(std=sd(mutation),
            mutation=mean(mutation)) %>%
  group_by()

temp <- temp %>%
  filter(stress != "None") %>%
  mutate(norm_mutation=mutation) %>%
  select(-mutation) %>%
  full_join(., select(filter(temp, stress=="None"), Isodecoder, position, mutation, DM), by=c("Isodecoder", "position","DM")) %>%
  mutate(norm_mutation = norm_mutation - mutation)


figS2F <- ggplot(filter(temp, DM=="-" ), aes(x=Isodecoder, y=position, fill=norm_mutation)) +
  geom_tile() +
  scale_y_reverse()+
  #zoom scale
  scale_fill_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="grey70", limits=c(-.1,.1))+
  #View things outside the scale
  geom_tile(data = filter(temp, DM=="-", norm_mutation > 0.1), fill="red4") +
  geom_tile(data = filter(temp, DM=="-", norm_mutation < -0.1), fill="blue4") +
  theme(legend.position = "right",
        legend.background = element_blank(),
        axis.text.x = element_text(angle = 90, hjust = 0, vjust=.5),
        axis.title.x = element_blank(),
        legend.text = element_text(angle=0, vjust=.5),
        plot.title = element_blank())+ #element_text(hjust = 0.5)) +
  labs(fill="Difference in\nmutation rate",
       title="Stress and mutation rate") +
  ylab("Position")+
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))+
  theme(strip.background = element_blank()) +
  theme(strip.text.y = element_text(angle=0)) +
  facet_grid(rows=vars(stress)) #, cols=vars(DM))



#Wonkey legend games
plt_legend <- get_legend(figS2F + theme(legend.title = element_blank() ) )
plt_legend <- plt_legend #+ theme(legend.title = element_blank())
#make a >0.25 legend
legend_data1 <-data.frame(Mutation=c("> 0.10"), x=c(1), y=c(1))
legend_legend1 <- ggplot(legend_data1, aes(x=y,y=y,fill=Mutation))+
  geom_tile() + scale_fill_manual(values= c("> 0.10"="red4")) +
  labs(fill="Difference in\nmutation rate")+
  theme(legend.position = "right", legend.background = element_blank(),
        legend.key.height = unit(c(0.2),"cm"))
legend_legend1 <- get_legend(legend_legend1)
#make a separea <0.25 legend
legend_data2 <-data.frame(Mutation=c("< -0.10"), x=c(1), y=c(1))
legend_legend2 <- ggplot(legend_data2, aes(x=y,y=y,fill=Mutation))+
  geom_tile() + scale_fill_manual(values= c("< -0.10"="blue4")) +
  theme(legend.position = "right", legend.background = element_blank(), legend.title = element_blank(),
        legend.key.height = unit(c(0.2),"cm"))
legend_legend2 <- get_legend(legend_legend2)

lay=rbind( c(1),c(2),c(3) )
legend_legend1 <- arrangeGrob(empty_graph,left=legend_legend1)
plt_legend <- arrangeGrob(empty_graph, left=plt_legend)
legend_legend2 <- arrangeGrob(empty_graph, left=legend_legend2)
plt_legend <- arrangeGrob(legend_legend1, plt_legend, legend_legend2, layout_matrix = lay  )
figS2F <- arrangeGrob(figS2F + theme(legend.position = "None") )

lay = rbind(c(1,1,1,1,2))
figS2F <- grid.arrange(figS2F, plt_legend, layout_matrix = lay)

figS2F



```




##NA: change in abundance box-and-whisker
```{r}
temp <- Ecoli_counts_data 
temp$gene_symbol <- recode(temp$gene_symbol, 
                           "alaT"="AlaUGC",
                           "alaU"="AlaUCG",
                           "alaV"="AlaUGC",
                           "alaW"="AlaGGC",
                           "alaX"="AlaGGC",
                           
                           "argQ"="ArgACG",
                           "argU"="ArgUCU",
                           "argV"="ArgACG",
                           "argW"="ArgCCU",
                           "argX"="ArgCCG",
                           "argY"="ArgACG",
                           "argZ"="ArgACG",
                           
                           "asnT"="AsnGUU",
                           "asnU"="AsnGUU",
                           "asnV"="AsnGUU",
                           "asnW"="AsnGUU",
                           
                           "aspT"="AspGUC",
                           "aspU"="AspGUC",
                           "aspV"="AspGUC",
                           
                           "cysT"="CysGCA",
                           
                           "glnU"="GlnUUG",
                           "glnV"="GlnCUG",
                           "glnW"="GlnUUG",
                           "glnX"="GlnCUG",
                           
                           "gltT"="GluUUC",
                           "gltU"="GluUUC",
                           "gltV"="GluUUC",
                           "gltW"="GluUUC",
                           
                           "glyT"="GlyUCC",
                           "glyU"="GlyUCC",
                           "glyV"="GlyGCC",
                           "glyW"="GlyGCC",
                           "glyX"="GlyGCC",
                           "glyY"="GlyGCC",
                           
                           "hisR"="HisGUG",
                           
                           "ileT"="IleGAU",
                           "ileU"="IleGAU",
                           "ileV"="IleGAU",
                           "ileX"="IleCAU",
                           "ileY"="IleCAU",
                           
                           "leuP"="LeuCAG",
                           "leuQ"="LeuCAG",
                           "leuT"="LeuCAG",
                           "leuU"="LeuGAG",
                           "leuV"="LeuCAG",
                           "leuW"="LeuUAG",
                           "leuX"="LeuCAA",
                           "leuZ"="LeuUAA",
                           
                           "lysQ"="LysUUU",
                           "lysT"="LysUUU",
                           "lysV"="LysUUU",
                           "lysW"="LysUUU",
                           "lysY"="LysUUU",
                           "lysZ"="LysUUU",
                           
                           "metT"="MetCAU",
                           "metU"="MetCAU",
                           "metV"="MetCAU",
                           "metW"="MetCAU",
                           "metY"="MetCAU",
                           "metZ"="MetCAU",
                           
                           "pheU"="PheGAA",
                           "pheV"="PheGAA",
                           
                           "proK"="ProCGG",
                           "proL"="ProGGG",
                           "proM"="ProUGG",
                           
                           "selC"="SelUCA",
                           
                           "serT"="SerUGA",
                           "serU"="SerUGA",
                           "serV"="SerGCU",
                           "serW"="SerGGA",
                           "serX"="SerGGA",
                           
                           "thrT"="ThrGGU",
                           "thrU"="ThrUGU",
                           "thrV"="ThrGGU",
                           "thrW"="ThrCGU",
                           
                           "trpT"="TrpCCA",
                           
                           "tyrT"="TyrGUA",
                           "tyrU"="TyrGUA",
                           "tyrV"="TyrGUA",
                           
                           "valT"="ValUAC",
                           "valU"="ValUAC",
                           "valV"="ValGAC",
                           "valW"="ValGAC",
                           "valX"="ValUAC",
                           "valY"="ValUAC",
                           "valZ"="ValUAC"
                           )

#Add codon and AA fields
temp <- temp %>%
  separate(gene_symbol, sep=3, c("AA","codon")) %>%
  mutate(gene_symbol = paste0(AA, codon))

#filter to just tRNA
temp1 <-temp %>%
  filter(gene_biotype =="tRNA") %>%
  select(-type)

#Normalize by total counts in library
temp <- temp1 %>%
  group_by(DM, stress, rep) %>%
  mutate(RPM = counts / sum(counts) *1000000) %>%
  group_by()

#Normalize to unstressed
temp <- temp %>%
  filter(stress=="None") %>%
  mutate(ns_RPM = RPM) %>%
  group_by(Gene, AA, codon, rep, DM, ns_RPM) %>%
  select(Gene, AA, codon, rep, DM, ns_RPM) %>%
  full_join(., filter(temp, stress!="None"), by=c("Gene","AA","codon","rep","DM")) %>%
  mutate(norm_RPM = RPM / ns_RPM)

#Filter for easy graphing
temp <- temp %>%
  group_by(DM, stress, AA) #%>%
  #filter(AA %in% c("Ser", "Arg", "Asn"))


ggplot(temp, aes(x=stress, y=norm_RPM, color=AA)) +
  geom_boxplot() +
  geom_jitter(width=0.1) +
  scale_y_continuous(trans="log2") +
  geom_hline(yintercept = 1, linetype=2, alpha=0.2) +
  theme(axis.text.x = element_text(angle=90, vjust=.5),
        axis.title.x = element_blank(),
        legend.position="None") +
  ylab("Fold change in abundance during stress") +
  facet_grid(cols=vars(AA))


```



#Save figure S2
```{r, save figure 2, echo=FALSE}

myplota <- arrangeGrob(fig2B, left = textGrob(expression(bold("a")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotb <- arrangeGrob(fig2C, top = textGrob(expression(bold("b")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotc <- arrangeGrob(figS2C, top = textGrob(expression(bold("c")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotd <- arrangeGrob(figS2D, top = textGrob(expression(bold("d")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplote <- arrangeGrob(fig2E, left = textGrob(expression(bold("e")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

#myplote <- grid.arrange(myplote, empty_graph, layout_matrix = rbind(c(1,1,1,1,1,1,1,1,1,1,2)))

myplotf <- arrangeGrob(figS2E, left = textGrob(expression(bold("f")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotg <- arrangeGrob(figS2F, left = textGrob(expression(bold("g")), x = unit(0, "npc")
        , y  = unit(1, "npc"), just=c("left","top"),
        gp=gpar(col="black", fontsize=18, fontfamily="Arial")))


#Set the figure arrangement matrix
lay <- rbind(c(1,2,3,4),
             c(1,2,3,4),
             c(1,2,3,4),
             c(5,5,5,5),
             c(5,5,5,5),
             c(6,6,6,6),
             c(6,6,6,6),
             c(6,6,6,6),
             c(7,7,7,7),
             c(7,7,7,7),
             c(7,7,7,7) )
#Arrange figures
figureS2 <- grid.arrange(myplota, myplotb, myplotc, myplotd, myplote, myplotf, myplotg, layout_matrix = lay)
figureS2

ggsave("../figures/figureS2.svg", 
       plot = figureS2,
       scale = 1.3,
       width = 8, 
       height = 9)
ggsave("../figures/figureS2.png", 
       plot = figureS2,
       scale = 1.3,
       width = 8, 
       height = 9)

```

#_


#Figure 3: human no stress
##Read in HEK catagory data
```{r, echo=FALSE}

#Rep1 minus DM
data1 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL7_S1_L004bc8.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep=1, stress="None")

# data2 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL7_S1_L004bc9.counts.tsv", header=T, sep="\t") %>%
#   mutate(DM="-", rep=1, stress="heat")
# 
# data3 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL7_S1_L004bc11.counts.tsv", header=T, sep="\t") %>%
#   mutate(DM="-", rep=1, stress="H2O2")
# 
# data4 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL7_S1_L004bc12.counts.tsv", header=T, sep="\t") %>%
#   mutate(DM="-", rep=1, stress="AsO2")


#Rep1 plus DM
data5 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL10_S2_L004bc8.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep=1, stress="None")

# data6 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL10_S2_L004bc9.counts.tsv", header=T, sep="\t") %>%
#   mutate(DM="+", rep=1, stress="heat")
# 
# data7 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL10_S2_L004bc11.counts.tsv", header=T, sep="\t") %>%
#   mutate(DM="+", rep=1, stress="H2O2")
# 
# data8 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL10_S2_L004bc12.counts.tsv", header=T, sep="\t") %>%
#   mutate(DM="+", rep=1, stress="AsO2")


#Rep2 minus DM
data9 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL11_S3_L004bc8.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep=2, stress="None")

# data10 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL11_S3_L004bc9.counts.tsv", header=T, sep="\t") %>%
#   mutate(DM="-", rep=2, stress="heat")
# 
# data11 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL11_S3_L004bc11.counts.tsv", header=T, sep="\t") %>%
#   mutate(DM="-", rep=2, stress="H2O2")
# 
# data12 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL11_S3_L004bc12.counts.tsv", header=T, sep="\t") %>%
#   mutate(DM="-", rep=2, stress="AsO2")

#Rep2 plus DM
data13 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL12_S4_L004bc8.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep=2, stress="None")

# data14 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL12_S4_L004bc9.counts.tsv", header=T, sep="\t") %>%
#   mutate(DM="+", rep=2, stress="heat")
# 
# data15 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL12_S4_L004bc11.counts.tsv", header=T, sep="\t") %>%
#   mutate(DM="+", rep=2, stress="H2O2")
# 
# data16 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL12_S4_L004bc12.counts.tsv", header=T, sep="\t") %>%
#   mutate(DM="+", rep=2, stress="AsO2")


#Rep3 minus DM
data17 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL13_S5_L004bc8.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="-", rep=3, stress="None")

# data18 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL13_S5_L004bc9.counts.tsv", header=T, sep="\t") %>%
#   mutate(DM="-", rep=3, stress="heat")
# 
# data19 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL13_S5_L004bc11.counts.tsv", header=T, sep="\t") %>%
#   mutate(DM="-", rep=3, stress="H2O2")
# 
# data20 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL13_S5_L004bc12.counts.tsv", header=T, sep="\t") %>%
#   mutate(DM="-", rep=3, stress="AsO2")


#Rep3 plus DM
data21 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL14_S6_L004bc8.counts.tsv", header=T, sep="\t") %>%
  mutate(DM="+", rep=3, stress="None")

# data22 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL14_S6_L004bc9.counts.tsv", header=T, sep="\t") %>%
#   mutate(DM="+", rep=3, stress="heat")
# 
# data23 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL14_S6_L004bc11.counts.tsv", header=T, sep="\t") %>%
#   mutate(DM="+", rep=3, stress="H2O2")
# 
# data24 = read.csv("../../2020 03 02 human figures/counts/TP-CK1-RPL14_S6_L004bc12.counts.tsv", header=T, sep="\t") %>%
#   mutate(DM="+", rep=3, stress="AsO2")

HEK_counts_data <- rbind(data1,
                      data5,
                      data9,
                      data13,
                      data17,
                      data21)

```

##Read in HEK base data
```{read in HEK basewise data}
#Rep1 minus DM
data1 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL7_S1_L004bc8.tsv", header=T, sep="\t") %>%
  filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="-", rep=1, stress="None")

# data2 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL7_S1_L004bc9.tsv", header=T, sep="\t") %>%
#   filter(gene_biotype !="protein_coding") %>%
#   #select(Gene, pileup, position, mutation, gene_biotype) %>%
#   mutate(DM="-", rep=1, stress="heat")
# 
# data3 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL7_S1_L004bc11.tsv", header=T, sep="\t") %>%
#   filter(gene_biotype!="protein_coding") %>%
#   #select(Gene, pileup, position, mutation, gene_biotype) %>%
#   mutate(DM="-", rep=1, stress="H2O2")
# 
# data4 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL7_S1_L004bc12.tsv", header=T, sep="\t") %>%
#   filter(gene_biotype!="protein_coding") %>%
#   #select(Gene, pileup, position, mutation, gene_biotype) %>%
#   mutate(DM="-", rep=1, stress="AsO2")
data_HEK <- rbind(data1)

#Rep1 plus DM
data1 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL10_S2_L004bc8.tsv", header=T, sep="\t") %>%
  filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="+", rep=1, stress="None")

# data2 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL10_S2_L004bc9.tsv", header=T, sep="\t") %>%
#   filter(gene_biotype!="protein_coding") %>%
#   #select(Gene, pileup, position, mutation, gene_biotype) %>%
#   mutate(DM="+", rep=1, stress="heat")
# 
# data3 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL10_S2_L004bc11.tsv", header=T, sep="\t") %>%
#   filter(gene_biotype!="protein_coding") %>%
#   #select(Gene, pileup, position, mutation, gene_biotype) %>%
#   mutate(DM="+", rep=1, stress="H2O2")
# 
# data4 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL10_S2_L004bc12.tsv", header=T, sep="\t") %>%
#   filter(gene_biotype!="protein_coding") %>%
#   #select(Gene, pileup, position, mutation, gene_biotype) %>%
#   mutate(DM="+", rep=1, stress="AsO2")
data_HEK <- rbind(data_HEK, data1)


#Rep2 minus DM
data1 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL11_S3_L004bc8.tsv", header=T, sep="\t") %>%
  filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="-", rep=2, stress="None")

# data2 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL11_S3_L004bc9.tsv", header=T, sep="\t") %>%
#   filter(gene_biotype!="protein_coding") %>%
#   #select(Gene, pileup, position, mutation, gene_biotype) %>%
#   mutate(DM="-", rep=2, stress="heat")
# 
# data3 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL11_S3_L004bc11.tsv", header=T, sep="\t") %>%
#   filter(gene_biotype!="protein_coding") %>%
#   #select(Gene, pileup, position, mutation, gene_biotype) %>%
#   mutate(DM="-", rep=2, stress="H2O2")
# 
# data4 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL11_S3_L004bc12.tsv", header=T, sep="\t") %>%
#   filter(gene_biotype!="protein_coding") %>%
#   #select(Gene, pileup, position, mutation, gene_biotype) %>%
#   mutate(DM="-", rep=2, stress="AsO2")
data_HEK <- rbind(data_HEK, data1)


#Rep2 plus DM
data1 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL12_S4_L004bc8.tsv", header=T, sep="\t") %>%
  filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="+", rep=2, stress="None")

# data2 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL12_S4_L004bc9.tsv", header=T, sep="\t") %>%
#   filter(gene_biotype!="protein_coding") %>%
#   #select(Gene, pileup, position, mutation, gene_biotype) %>%
#   mutate(DM="+", rep=2, stress="heat")
# 
# data3 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL12_S4_L004bc11.tsv", header=T, sep="\t") %>%
#   filter(gene_biotype!="protein_coding") %>%
#   #select(Gene, pileup, position, mutation, gene_biotype) %>%
#   mutate(DM="+", rep=2, stress="H2O2")
# 
# data4 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL12_S4_L004bc12.tsv", header=T, sep="\t") %>%
#   filter(gene_biotype!="protein_coding") %>%
#   #select(Gene, pileup, position, mutation, gene_biotype) %>%
#   mutate(DM="+", rep=2, stress="AsO2")
data_HEK <- rbind(data_HEK, data1)


#Rep3 minus DM
data1 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL13_S5_L004bc8.tsv", header=T, sep="\t") %>%
  filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="-", rep=3, stress="None")

# data2 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL13_S5_L004bc9.tsv", header=T, sep="\t") %>%
#   filter(gene_biotype!="protein_coding") %>%
#   #select(Gene, pileup, position, mutation, gene_biotype) %>%
#   mutate(DM="-", rep=3, stress="heat")
# 
# data3 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL13_S5_L004bc11.tsv", header=T, sep="\t") %>%
#   filter(gene_biotype!="protein_coding") %>%
#   #select(Gene, pileup, position, mutation, gene_biotype) %>%
#   mutate(DM="-", rep=3, stress="H2O2")
# 
# data4 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL13_S5_L004bc12.tsv", header=T, sep="\t") %>%
#   filter(gene_biotype!="protein_coding") %>%
#   #select(Gene, pileup, position, mutation, gene_biotype) %>%
#   mutate(DM="-", rep=3, stress="AsO2")
data_HEK <- rbind(data_HEK, data1)


#Rep3 plus DM
data1 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL14_S6_L004bc8.tsv", header=T, sep="\t") %>%
  filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="+", rep=3, stress="None")

# data2 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL14_S6_L004bc9.tsv", header=T, sep="\t") %>%
#   filter(gene_biotype!="protein_coding") %>%
#   #select(Gene, pileup, position, mutation, gene_biotype) %>%
#   mutate(DM="+", rep=3, stress="heat")
# 
# data3 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL14_S6_L004bc11.tsv", header=T, sep="\t") %>%
#   filter(gene_biotype!="protein_coding") %>%
#   #select(Gene, pileup, position, mutation, gene_biotype) %>%
#   mutate(DM="+", rep=3, stress="H2O2")
# 
# data4 = read.csv("../../2020 03 02 human figures/basewise/full_map/TP-CK1-RPL14_S6_L004bc12.tsv", header=T, sep="\t") %>%
#   filter(gene_biotype!="protein_coding") %>%
#   #select(Gene, pileup, position, mutation, gene_biotype) %>%
#   mutate(DM="+", rep=3, stress="AsO2")

data_HEK <- rbind(data_HEK, data1)



```

###Extra HEK base filter
```{r}


#Get rid of unused sequences
HEK_base_data <- data_HEK %>%
  filter(gene_biotype == "tRNA") %>%
  filter(!grepl("spike", Gene)) %>%
  filter(!grepl("Ecoli", Gene)) %>%
  filter(!grepl("Yeast", Gene)) %>%
  select(Gene, pileup, position, mutation, gene_biotype, DM, stress, rep, base) %>%
  select(-gene_biotype)

HEK_base_data$DM <- recode(HEK_base_data$DM, "-"="m", "+"="p")

#fix mitochondrial names
#Then rearrange all names
HEK_base_data <- HEK_base_data %>%
  filter(grepl("mt", Gene)) %>%
  mutate(Gene = paste0("Homo_sapiens_chrMT~trna1-z",Gene)) %>%
  rbind(filter(HEK_base_data, !grepl("mt",Gene))) %>%
  separate(Gene, sep="_", c("homo","sapiens","gene_symbol"))%>%
  select(-homo, -sapiens) %>%
  separate(gene_symbol, sep="-", c("chrom","codon")) %>%
  mutate(gene_symbol = paste0(codon, "_",chrom)) %>%
  select(-chrom, -codon)

# #filter for tRNAs that have high expression under
# #EVERY CONDITION at base ...50?
pileup_filter_value = 20
pileup_filter_position = c(40)
temp1 <- filter(HEK_base_data, position %in% pileup_filter_position) %>%
  pivot_wider(id_cols = c("gene_symbol","position"), names_from = c("stress","DM","rep"), values_from = pileup) %>%
  filter(None_m_1 > pileup_filter_value,
         #heat_m_1 > pileup_filter_value,
         #H2O2_m_1 > pileup_filter_value,
         #AsO2_m_1 > pileup_filter_value,
         None_p_1 > pileup_filter_value,
         #heat_p_1 > pileup_filter_value,
         #H2O2_p_1 > pileup_filter_value,
         #AsO2_p_1 > pileup_filter_value,
         None_m_2 > pileup_filter_value,
         #heat_m_2 > pileup_filter_value,
         #H2O2_m_2 > pileup_filter_value,
         #AsO2_m_2 > pileup_filter_value,
         None_p_2 > pileup_filter_value,
         #heat_p_2 > pileup_filter_value,
         #H2O2_p_2 > pileup_filter_value,
         #AsO2_p_2 > pileup_filter_value,
         None_m_3 > pileup_filter_value,
         #heat_m_3 > pileup_filter_value,
         #H2O2_m_3 > pileup_filter_value,
         #AsO2_m_3 > pileup_filter_value,
         None_p_3 > pileup_filter_value)
         #heat_p_3 > pileup_filter_value,
         #H2O2_p_3 > pileup_filter_value,
         #AsO2_p_3 > pileup_filter_value)


#fitler out eroniously long tRNA...
temp2 <- HEK_base_data %>%
  group_by(gene_symbol, DM, stress, rep) %>%
  summarize(position = max(position)) %>%
  filter(position <120) %>%
  select(gene_symbol)

#Apply filters to base data
HEK_base_data <- HEK_base_data %>%
  filter(gene_symbol %in% temp1$gene_symbol,
         gene_symbol %in% temp2$gene_symbol)

#FILTER FOR MOST ABUNDANT, and Make x labels nicer to read
temp <- HEK_base_data %>%
  mutate(gene_name=gene_symbol) %>%
  separate(gene_symbol, sep="_", c("gene_symbol", "number")) 
temp1 <- temp %>%
  filter(stress=="None", DM=="p") %>%
  group_by(stress, DM, gene_symbol) %>%
  mutate(max_pileup = max(pileup)) %>%
  filter(pileup == max_pileup) %>%
  group_by() %>%
  select(gene_name)
temp <- temp %>%
  filter(gene_name %in% temp1$gene_name)

HEK_base_data <- temp
```


##Fig 3A1: Piechart HEK all RNAs
```{r, echo=FALSE}

temp <- HEK_counts_data %>%
  filter(DM=="+" & stress=="None") %>%
  group_by() %>%
  group_by(Gene, type, gene_biotype, gene_symbol, DM, stress) %>%
  summarise(counts = mean(counts)) %>%
  group_by()

#Bin all ncRNAs under the name ncRNA
temp <- temp %>%
  filter(gene_biotype %nin% c("tRNA", "rRNA", "protein_coding", "Mt_rRNA")) %>%
  mutate(gene_biotype = "ncRNA") %>%
  rbind(., filter(temp, gene_biotype %in% c("tRNA", "rRNA", "protein_coding"))) %>%
  group_by(gene_biotype) 

#Bin all Mt_rRNA under the name rRNA
temp <- temp %>%
  filter(gene_biotype =="Mt_rRNA") %>%
  mutate(gene_biotype = "rRNA") %>%
  rbind(., filter(temp, gene_biotype!="Mt_rRNA" ))

#Calculate fraction of groups
temp <- temp %>%
  group_by(gene_biotype) %>%
  summarise(counts = sum(counts)) %>%
  mutate(counts = round((counts / sum(.$counts))*100,1))

#rename "protein_coding" to "mRNA"
temp <- temp %>%
  filter(gene_biotype=="protein_coding") %>%
  mutate(gene_biotype="mRNA") %>%
  rbind(., filter(temp, gene_biotype != "protein_coding"))

#transform value to be compatable with piechart
temp <- temp %>% 
  arrange(desc(gene_biotype)) %>%
  mutate(ypos = cumsum(counts)- 0.5*counts )

fig3A1 <-ggplot(temp, aes(x="", y=counts, fill=gene_biotype)) +
  geom_bar(stat="identity", width=1, color="white") +
  coord_polar("y", start=5) +
  #scale_y_continuous(breaks=cumsum(temp$counts) - temp$counts / 2, labels= temp$gene_biotype) +
  geom_text_repel(aes(y=ypos, label=paste0(gene_biotype, "\n", counts, "%")), box.padding = 1)+
  theme_void() +
  scale_fill_manual(values = c("tRNA"="#99CCFF", 
                                "rRNA"="#FFCC33", 
                               "mRNA"="#99CC99",
                                "ncRNA"="#CC99CC"))+
  labs(fill="Gene type") +
  guides(fill = guide_legend(reverse=TRUE)) + 
  theme(text = element_text(size=8),
        plot.title = element_text(size=14)) +
  theme(legend.position = "None",
        plot.title = element_text(hjust = 0.5)) +
  labs(title="All reads") +
  theme(plot.margin = unit(c(0.1, 0.05, 0.1, 0.3), "cm"))
fig3A1



```

##Fig 3A2: Piechart HEK ncRNA
```{r, echo=FALSE}
#Filter and simplify, minus DM, no stress
temp <- HEK_counts_data %>%
  filter(DM=="+" & stress=="None") %>%
  group_by() %>%
  group_by(Gene, type, gene_biotype, gene_symbol, DM, stress) %>%
  summarise(counts = mean(counts)) %>%
  group_by()

#filter for just ncRNAs
temp <- temp %>%
  filter(gene_biotype %nin% c("tRNA", "rRNA", "protein_coding", "Mt_rRNA"))

#Combine IG and TR genes
temp <- temp %>%
  filter(gene_biotype %in% c("IG_V_gene", "IG_D_gene", "IG_C_gene", "IG_J_gene", 
                             "TR_V_gene", "TR_D_gene", "TR_C_gene", "TR_J_gene")) %>%
  mutate(gene_biotype="Immune") %>%
  rbind(., filter(temp, gene_biotype %nin% c("IG_V_gene", "IG_D_gene", "IG_C_gene", "IG_J_gene", 
                             "TR_V_gene", "TR_D_gene", "TR_C_gene", "TR_J_gene")))

#Combine ribozyme, misc_RNA, and polymorphic_psuedogene into "OTHER"
temp <- temp %>%
  filter(gene_biotype %in% c("ribozyme","polymorphic_pseudogene","misc_RNA", "vaultRNA", "miRNA")) %>%
  mutate(gene_biotype="Other") %>%
  rbind(., filter(temp, gene_biotype %nin% c("ribozyme","polymorphic_pseudogene","misc_RNA", "vaultRNA", "miRNA")))

#Combine Immune, sRNA, scRNA, scaRNA with OTHER
temp <- temp %>%
  filter(gene_biotype %in% c("Other","Immune","sRNA", "scRNA", "scaRNA")) %>%
  mutate(gene_biotype="Other") %>%
  rbind(., filter(temp, gene_biotype %nin% c("Other","Immune","sRNA", "scRNA", "scaRNA")))

#summing catagory counts
temp <- temp %>%
  group_by(gene_biotype) %>%
  summarise(counts = sum(counts)) %>%
  mutate(counts = round((counts / sum(.$counts))*100,1))

#transform value to be compatable with piechart
temp <- temp %>% 
  arrange(desc(gene_biotype)) %>%
  mutate(ypos = cumsum(counts)- 0.5*counts ) 

fig3A2 <-ggplot(temp, aes(x="", y=counts, fill=gene_biotype)) +
  geom_bar(stat="identity", width=1, color="white") +
  coord_polar("y", start=4) +
  geom_text_repel(aes(y=ypos , label=paste0(gene_biotype, "\n", counts, "%")), box.padding = 1 )+
  theme_void() +
  labs(fill="Gene type") +
  guides(fill = guide_legend(reverse=TRUE)) + 
  theme(legend.position = "None",
        plot.title = element_text(hjust = 0.5))+
  theme(text = element_text(size=8),
        plot.title = element_text(size=14)) +
  labs(title="ncRNA reads") +
  theme(plot.margin = unit(c(0.1, 0.3, 0.05, 0.1), "cm"))
fig3A2

myplotb1 <- arrangeGrob(fig3A1)
myplotb2 <- arrangeGrob(fig3A2)
lay = rbind(c(1,2))
fig3A <- grid.arrange(myplotb1, myplotb2, layout_matrix = lay)
fig3A
```

##Fig 3B: comparison with microarray
```{r}

microarray_data <- read.csv("../microarray_data_human.txt", sep="\t", header=TRUE)
microarray_data$gene_symbol <- as.factor(microarray_data$gene_symbol)
microarray_data <- microarray_data %>%
  group_by(gene_symbol) %>%
  mutate(method="Microarray") %>%
  mutate("rep3"=NA) %>%
  filter(grepl("Arg", gene_symbol)) %>%
  group_by() 
microarray_data$rep1 <- as.numeric(microarray_data$rep1)
microarray_data$rep2 <- as.numeric(microarray_data$rep2)

temp <- HEK_counts_data %>%
  filter(gene_biotype=="tRNA") %>%
  group_by(DM, rep, stress) %>%
  mutate(RPM = counts / sum(counts)*1000000) %>% #normalize counts to rpm
  filter(grepl("Arg", gene_symbol)) %>%
  filter(!grepl(">", gene_symbol)) %>%
  separate(gene_symbol, sep="-", c("number", "gene_symbol"))

#COmbine ArgCCG and TCG into yCG
temp$gene_symbol <- recode(temp$gene_symbol, "ArgACG"="ArgACG", "ArgCCG"="ArgyCG",
                                            "ArgCCT"="ArgCCT", "ArgTCG"="ArgyCG",
                                            "ArgTCT"="ArgTCT")
temp <- temp %>%
  group_by(gene_symbol, stress, DM, rep) %>%
  summarise(RPM = sum(RPM)) %>%
  group_by(gene_symbol, DM, stress) %>%
  mutate(mean_RPM = mean(RPM)) %>%
  filter(stress=="None", DM=="+") %>%#filter for the plot
  group_by() 


#Fix the stupid tRNA names
temp$rep <- recode(temp$rep, "1"="rep1", "2"="rep2", "3"="rep3")

temp <- temp %>%
  filter(DM=="+", stress=="None") %>%
  select(gene_symbol, rep, RPM) %>%
  group_by(gene_symbol, rep) %>%
  summarise(RPM = sum(RPM)) %>%
  group_by(rep) %>%
  mutate(RPM = RPM/ sum(RPM)) %>%
  pivot_wider(names_from = rep, values_from = RPM) %>%
  mutate(method = "CHRIS-seq") %>%
  rbind(microarray_data) %>%
  pivot_longer(cols =c("rep1", "rep2", "rep3") ,names_to ="rep", values_to = "RPM") %>%
  group_by(method, gene_symbol) %>%
  mutate(mean_RPM = mean(RPM))

temp$gene_symbol <- as.character(temp$gene_symbol)
temp$gene_symbol <- factor(temp$gene_symbol, level=c("ArgyCG","ArgACG","ArgTCT","ArgCCT"))

fig3B <- ggplot(temp, aes(x=gene_symbol, y=mean_RPM, color=method)) +
  geom_point(shape=95, size=5, position= position_dodge(width=1) ) +
  geom_point(aes(y=RPM), position=position_dodge(width=1)) +
  theme(axis.text.x = element_text(angle = 45, hjust = 0.5, vjust=.5),
        axis.title.x = element_blank(),
        plot.title = element_text(hjust = 0.5),
        legend.position = "bottom",
        legend.text = element_text(size=8),
        legend.title = element_text(size=10),
        legend.background = element_blank())+
  scale_y_continuous(limits=c(0,1), breaks=c(0,0.5,1.0)) +
  ylab("Proportion of\nsignal") +
  labs(color="") +
  #annotate(geom="text", label="Arg", y=0.5, x="ArgACG") +
  scale_color_manual(values=c("CHRIS-seq"="red", "Microarray"="grey50")) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")) 
  #theme(aspect.ratio = 1.2)
  #theme(legend.key = element_rect(fill = "white", colour = "black") )
fig3B
```


##Fig 3C: titration of tRNA input
###Read in titration data
```{read in HEK basewise data}
#Rep1 minus DM
data1 = read.csv("../../2020 03 06 input titration/express/TP-CK-TP1-1_S1bc1.counts.tsv", header=T, sep="\t") %>%
  mutate(rep=1, input="ng1000", DM="-")

data2 = read.csv("../../2020 03 06 input titration/express/TP-CK-TP1-1_S1bc2.counts.tsv", header=T, sep="\t") %>%
  mutate(rep=1, input="ng100", DM="-")

data3 = read.csv("../../2020 03 06 input titration/express/TP-CK-TP1-1_S1bc3.counts.tsv", header=T, sep="\t") %>%
  mutate(rep=1, input="ng10", DM="-")

#rep2 minus DM
data4 = read.csv("../../2020 03 06 input titration/express/TP-CK-TP1-3_S3bc1.counts.tsv", header=T, sep="\t") %>%
  mutate(rep=2, input="ng10", DM="-")

data5 = read.csv("../../2020 03 06 input titration/express/TP-CK-TP1-3_S3bc2.counts.tsv", header=T, sep="\t") %>%
  mutate(rep=2, input="ng1000", DM="-")

data6 = read.csv("../../2020 03 06 input titration/express/TP-CK-TP1-3_S3bc3.counts.tsv", header=T, sep="\t") %>%
  mutate(rep=2, input="ng100", DM="-")

trna_titration <- rbind(data1, data2, data3, 
                         data4, data5, data6)
```

###correlation matrix
```{r}

temp <- trna_titration %>%
  filter(gene_biotype == "tRNA") %>%
  filter(rep=="2") %>%
  select(-type, -rep) %>%
  group_by(DM, Gene, gene_biotype) %>%
  pivot_wider(names_from = c("input"), values_from = counts) %>%
  group_by() #%>%
  # filter(ng1000_1>0, ng100_1>0, ng10_1>0,
  #        ng1000_2>0, ng100_2>0, ng10_2>0)

cormat <- round(cor(select(temp, -DM, -Gene, -gene_biotype, -gene_symbol),method=c("spearman"))**2 ,2) %>%
  melt()

#Change names for pretty graph
cormat$Var1 <- recode(cormat$Var1, "ng1000"="1000 ng", "ng100"="100 ng", "ng10"="10 ng")
cormat$Var2 <- recode(cormat$Var2, "ng1000"="1000 ng", "ng100"="100 ng", "ng10"="10 ng")

#Chage order for pretty graph
cormat$Var1 <- factor(cormat$Var1, levels=c("1000 ng","100 ng","10 ng"))
cormat$Var2 <- factor(cormat$Var2, levels=c("10 ng","100 ng","1000 ng"))

#Makr prety graph
fig3C <- ggplot(data = cormat, aes(x=Var1, y=Var2, fill=value)) + 
  geom_tile() +
  geom_text(aes(label=value)) +
  scale_fill_gradient(low="grey70", high="red", limits=c(0,1) )+
  theme_void() +
  theme(legend.position = "bottom",
        plot.title = element_text(hjust = 0.5, size=12),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.text = element_text(),
        axis.text.y = element_text(angle=90, hjust=0.5)) +
  labs(title="Input RNA",
       fill=bquote(~r^2)) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))

fig3C

```



##Fig 3D: Modification without DM
```{r, echo=FALSE}

#The good stuff
temp <- HEK_base_data %>%
  mutate(mutation = ifelse(pileup<10, NA, mutation)) %>%
  group_by(gene_symbol, position, DM, stress) %>%
  summarise(mutation = mean(mutation, na.rm=T),
            pileup= mean(pileup, na.rm=T)) %>%
  group_by()

#Split up mito etc
temp <- temp %>%
  mutate(source = "cytosolic")
temp <- temp %>%
  filter(grepl("z", gene_symbol)) %>%
  mutate(source = "mitochondrial") %>%
  rbind(filter(temp, !grepl("z", gene_symbol)))

#FILTER FOR MOST ABUNDANT, and Make x labels nicer to read
temp <- temp %>%
  separate(gene_symbol, sep="_", c("gene_symbol", "number")) %>%
  mutate(gene_name=paste0(gene_symbol,"_",number))
temp1 <- temp %>%
  filter(stress=="None", DM=="p") %>%
  group_by(stress, DM, gene_symbol) %>%
  mutate(max_pileup = max(pileup)) %>%
  filter(pileup == max_pileup) %>%
  group_by() %>%
  select(gene_name)
temp <- temp %>%
  filter(gene_name %in% temp1$gene_name)


fig3D1 <- ggplot(filter(temp, DM=="m", 
                       stress=="None",
                       source=="cytosolic"), 
                aes(x=gene_symbol, y=position, fill=mutation)) +
  geom_tile() +
  scale_y_reverse(breaks=c(9, 37, 58), limits=c(89, 1) )+
  scale_fill_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="grey70", limits=c(0,1), breaks=c(0,0.5,1) )+
  theme(legend.position = "None",
        axis.text.x = element_text(angle = 90, hjust = 0, vjust=.5, size=9),
        axis.title.x = element_blank(),
        legend.text = element_text(angle=0, vjust=.5),
        plot.title = element_text(hjust = 0.5, size=12, vjust=0), 
        axis.ticks.y = element_line(size=1),
        axis.ticks.length.y = unit(.2, "cm") )+ #element_text(hjust = 0.5)) +
  labs(#title="Mutation rate along tRNAs without demethylase treatment",
       fill="Mutation rate") +
  ylab("Position")  +
  labs(title="Nuclear-encoded") +
  theme(plot.title = element_text(hjust = 0) ) +
  theme(plot.margin = unit(c(0.05, 0.05, 0.3, 0.3), "cm"))
fig3D1


#Fix "z" gene
temp$gene_symbol <- str_replace_all(temp$gene_symbol, "z", "") 



fig3D2 <- ggplot(filter(temp, DM=="m", 
                       stress=="None",
                       source=="mitochondrial"), 
                aes(x=gene_symbol, y=position, fill=mutation)) +
  geom_tile() +
  scale_y_reverse(breaks=c(9, 37, 58), limits=c(89, 0), position="right" )+
  scale_fill_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="grey70", limits=c(0,1), breaks=c(0,0.5,1) )+
  theme(legend.position = "right",
        axis.text.x = element_text(angle = 90, hjust = 0, vjust=.5, size=9),
        axis.title.x = element_blank(),
        legend.text = element_text(angle=0, vjust=.5),
        plot.title = element_text(hjust = 0.5, size=12, vjust=0), 
        axis.ticks.y = element_line(size=1),
        axis.ticks.length.y = unit(.2, "cm"),
        axis.title.y = element_blank())+ #element_text(hjust = 0.5)) +
  labs(#title="Mutation rate along tRNAs without demethylase treatment",
       fill="Mutation rate") +
  ylab("Position")+
  labs(title="Mitochondrial-encoded") +
  theme(plot.title = element_text(hjust = 0) ) +
  theme(plot.margin = unit(c(0.05, 0.3, 0.3, 0.05), "cm"))
fig3D2


myplot1 <- arrangeGrob(fig3D1)
myplot2<- arrangeGrob(fig3D2)
lay = rbind(c(1,1,1,2,2))
fig3D <- grid.arrange(myplot1, myplot2, layout_matrix = lay)
fig3D
```


##Fig 3E: change in mod with DM
```{r, echo=FALSE}
temp <- HEK_base_data %>%
  mutate(mutation = ifelse(pileup<10, NA, mutation)) %>%
  #filter(stress == "None") %>%
  group_by(gene_symbol, position, DM, stress) %>%
  summarise(mutation = mean(mutation),
            pileup = mean(pileup, na.rm=T)) %>%
  group_by()

#Split up mito etc
temp <- temp %>%
  mutate(source = "cytosolic")
temp <- temp %>%
  filter(grepl("z", gene_symbol)) %>%
  mutate(source = "mitochondrial") %>%
  rbind(filter(temp, !grepl("z", gene_symbol)))

# FILTER FOR MOST ABUNDANT, and Make x labels nicer to read
temp <- temp %>%
  separate(gene_symbol, sep="_", c("gene_symbol", "number")) %>%
  mutate(gene_name=paste0(gene_symbol,"_",number))
temp1 <- temp %>%
  filter(stress=="None", DM=="p") %>%
  group_by(stress, DM, gene_symbol) %>%
  mutate(max_pileup = max(pileup)) %>%
  filter(pileup == max_pileup) %>%
  group_by() %>%
  select(gene_name)
temp <- temp %>%
  filter(gene_name %in% temp1$gene_name)

#Make data wide
temp <- temp %>%
  pivot_wider(id_cols = c("gene_symbol","position","stress","source"), names_from=DM, values_from=mutation)




fig3E1 <- ggplot(filter(temp, stress=="None",
                       source=="cytosolic"), 
                aes(x=gene_symbol, y=position, fill=p-m)) +
  geom_tile() +
  scale_y_reverse(breaks=c(9, 37, 58), limits=c(89, 1) )+
  scale_fill_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="grey70", 
                       limits=c(-1,.5), breaks=c(.5, 0,-.5,-1))+
  theme(legend.position = "None",
        axis.text.x = element_text(angle = 90, hjust = 0, vjust=.5, size=9),
        axis.title.x = element_blank(),
        legend.text = element_text(angle=0, vjust=.5),
        plot.title = element_text(hjust = 0.5, size=12, vjust=0), 
        axis.ticks.y = element_line(size=1),
        axis.ticks.length.y = unit(.2, "cm") )+ #element_text(hjust = 0.5)) +
  labs(#title="Mutation rate along tRNAs without demethylase treatment",
       fill="Mutation rate") +
  ylab("Position")  +
  labs(title="Nuclear-encoded") +
  theme(plot.title = element_text(hjust = 0) ) +
  theme(plot.margin = unit(c(0.05, 0.05, 0.3, 0.3), "cm"))
fig3E1

#Fix "z" gene
temp$gene_symbol <- str_replace_all(temp$gene_symbol, "z", "") 

fig3E2 <- ggplot(filter(temp, stress=="None",
                       source=="mitochondrial"), 
                aes(x=gene_symbol, y=position, fill=p-m)) +
  geom_tile() +
  scale_y_reverse(breaks=c(9, 37, 58), limits=c(89, 1), position="right" )+
  scale_fill_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="grey70", 
                       limits=c(-1,.5), breaks=c(.5, 0,-.5,-1))+
  theme(legend.position = "right",
        axis.text.x = element_text(angle = 90, hjust = 0, vjust=.5, size=9),
        axis.title.x = element_blank(),
        legend.text = element_text(angle=0, vjust=.5),
        plot.title = element_text(hjust = 0.5, size=12, vjust=0), 
        axis.ticks.y = element_line(size=1),
        axis.ticks.length.y = unit(.2, "cm"),
        axis.title.y = element_blank())+ #element_text(hjust = 0.5)) +
  labs(#title="Mutation rate along tRNAs without demethylase treatment",
       fill="Difference in \nmutation rate") +
  ylab("Position")  +
  labs(title="Mitochondrial-encoded") +
  theme(plot.title = element_text(hjust = 0) ) +
  theme(plot.margin = unit(c(0.05, 0.3, 0.3, 0.05), "cm"))
fig3E2


myplot1 <- arrangeGrob(fig3E1)
myplot2<- arrangeGrob(fig3E2)
lay = rbind(c(1,1,1,2,2))
fig3E <- grid.arrange(myplot1, myplot2, layout_matrix = lay)
fig3E
```



##Fig 3F: other RNA
```{r, echo=FALSE}

#Normalize to RPM and filter low stuff
temp <- HEK_counts_data %>%
  filter(stress=="None") %>%
  group_by(rep, DM, stress) %>% 
  mutate(RPM = counts/sum(counts)*1000000 ) %>%
  group_by(DM, stress) %>% 
  mutate(RPM = ifelse( counts>0, counts, 0) ) %>%
  filter(gene_biotype %in% c("snRNA", "snoRNA","miRNA")) 

#Collapse similar species
temp <- temp %>%
  separate(gene_symbol, sep="-", c("gene_symbol","number")) %>%
  select(-number)
#collapse similar species
temp <- temp %>%
  filter(grepl("\\.", gene_symbol)) %>%
  separate(gene_symbol, sep="\\.", c("gene_symbol","number")) %>%
  select(-number) %>%
  rbind(., filter(temp, !grepl("\\.", gene_symbol))) %>%
  group_by(gene_symbol, DM, stress, rep) %>%
  summarise(RPM = sum(RPM)) %>%
  group_by(DM, gene_symbol, stress) %>%
  mutate(mean_RPM = mean(RPM)) %>%
  filter(mean_RPM > 10) %>%
  group_by()

#Arrange by -DM value
temp <- temp %>%
  filter(DM=="-") %>%
  mutate(minus_mc = RPM) %>%
  select(gene_symbol, minus_mc) %>%
  full_join(.,temp, by=c("gene_symbol")) %>%
  mutate(gene_symbol = fct_reorder(gene_symbol, desc(minus_mc)))


#Make the pretty plot
fig3F <- ggplot(filter(temp, DM=="-"), aes(x=gene_symbol, y=mean_RPM))+
  geom_point(shape=95, size=5) +
  geom_point(aes(y=RPM), size=.4) +
  #scale_color_manual(values = c("-"="black", "+"="red")) +
  #geom_errorbar(aes(ymin=mean_count-std_count, ymax=mean_count+std_count)) +
  scale_y_log10(labels=trans_format('log10',math_format(10^.x)) ) +
  theme(axis.text.x = element_text(angle=90, vjust=0.5, hjust=0), 
        axis.title.x = element_blank(),
        legend.position= c(.8, .7) ) + 
  #facet_grid(cols=vars(stress)) + 
  #labs(title="Abundance of ncRNA")+
  ylab("Reads per million") +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.1), "cm")) 
fig3F

```




##old Save figure 3
```{r}
myplota <- arrangeGrob(fig3A, left = textGrob(expression(bold("a")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotb <- arrangeGrob(fig3B, left = textGrob(expression(bold("b")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotc <- arrangeGrob(fig3C, left = textGrob(expression(bold("c")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotd <- arrangeGrob(fig3D, left = textGrob(expression(bold("d")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplote <- arrangeGrob(fig3E, left = textGrob(expression(bold("e")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotf <- arrangeGrob(fig3F, left = textGrob(expression(bold("f")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotg <- arrangeGrob(fig3G, left = textGrob(expression(bold("g")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
# myploth <- arrangeGrob(fig4H, left = textGrob(expression(bold("h")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))


#Set the figure arrangement matrix
lay <- rbind(c(1, 2, 3),
             c(4, 4, 4),
             c(5, 5, 5),
             c(6, 6, 6))
#Arrange figures
figure3 <- grid.arrange(myplota, myplotb, myplotc, myplotd, myplote, myplotf, layout_matrix = lay)
figure3

setwd("~/4SR/data/20200317_Nature_methods_data/2020 02 24 ecoli figures/source/")
ggsave("../figures/figure3.svg", 
       plot = figure3,
       scale = 1.3,
       width = 9, 
       height = 7)
ggsave("../figures/figure3.png", 
       plot = figure3,
       scale = 1.3,
       width = 9, 
       height = 7)


```

##new Save figure 3
```{r}

myplota <- arrangeGrob(fig3A, left = textGrob(expression(bold("a")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotb <- arrangeGrob(figS3C, left = textGrob(expression(bold("b")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotc <- arrangeGrob(fig3D, top = textGrob(expression(bold("c")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotd <- arrangeGrob(fig3E, top = textGrob(expression(bold("d")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplote <- arrangeGrob(figS3G, left = textGrob(expression(bold("e")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
# myplotf <- arrangeGrob(figS3G, left = textGrob(expression(bold("f")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
# myplotg <- arrangeGrob(fig3G, left = textGrob(expression(bold("g")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
# myploth <- arrangeGrob(fig4H, left = textGrob(expression(bold("h")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))


#Set the figure arrangement matrix
lay <- rbind(c(1, 1, 2),
             c(1, 1, 2),
             c(1, 1, 2),
             c(3, 3, 3),
             c(3, 3, 3),
             c(4, 4, 4),
             c(4, 4, 4),
             c(5, 5, 6),
             c(5, 5, 6),
             c(5, 5, 6))
#Arrange figures
figure3 <- grid.arrange(myplota, myplotb, myplotc, myplotd, myplote, empty_graph, layout_matrix = lay)
figure3

setwd("~/4SR/data/20200317_Nature_methods_data/2020 02 24 ecoli figures/source/")
ggsave("../figures/figure3.svg", 
       plot = figure3,
       scale = 1.3,
       width = 7, 
       height = 8)

ggsave("../figures/figure3.png", 
       plot = figure3,
       scale = 1.3,
       width = 7, 
       height = 8)


```

#_


#Figure S3
##Fig S3A: HEK +/-DM correlation
```{r}
#Don't average by replicate
temp <- HEK_counts_data %>%
  filter(stress=="None") %>%
  select(-type) %>%
  group_by()

#rename "protein_coding" to "mRNA"
temp <- temp %>%
  filter(gene_biotype=="protein_coding") %>%
  mutate(gene_biotype="mRNA") %>%
  rbind(., filter(temp, gene_biotype != "protein_coding"))

#combine biotype of less abundant catagories into "ncRNA"
temp <- temp %>%
  filter(gene_biotype %nin% c("tRNA", "rRNA", "mRNA")) %>%
  mutate(gene_biotype = "ncRNA") %>%
  rbind(., filter(temp, gene_biotype %in% c("tRNA", "rRNA", "mRNA"))) %>%
  group_by(DM, stress, rep) %>%
  mutate(counts = (counts / sum(.$counts)))

#rename the reps and DM to make ggplot easier
temp$rep <- recode(temp$rep, "1"="rep 1", "2"="rep 2", "3"="rep 3")
temp$DM <- recode(temp$DM, "-"="minusDM", "+"="plusDM")
temp$gene_biotype <- factor(temp$gene_biotype, levels = c("tRNA","rRNA","ncRNA","mRNA"))

#Make columns of counts for DM and rep
temp <- temp %>%
  spread(DM, counts)

temp <- temp %>%
  group_by() %>%
  filter(minusDM >0 & plusDM>0) %>%
  group_by(gene_biotype, rep) %>%
  summarise(correlation = cor(minusDM, plusDM, method=c("spearman"))**2  )

figS3A <- ggplot(temp, aes(x=gene_biotype, y=rep, fill=correlation)) +
  geom_tile() +
  geom_text(aes(label=round(correlation,2) )) +
  scale_fill_gradient(low="grey70", high="red", limits=c(0,1))+
  theme_void() +
  theme(legend.position = "bottom",
        plot.title = element_text(hjust = 0.5, size=12),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.text = element_text() )+
  labs(title="±DM",
       fill=bquote(~r^2) ) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))

figS3A

```

##Fig S3B Correlation between replicates
```{r}

#Don't average by replicate
temp <- HEK_counts_data %>%
  filter(stress=="None") %>%
  select(-type) %>%
  group_by()

#rename "protein_coding" to "mRNA"
temp <- temp %>%
  filter(gene_biotype=="protein_coding") %>%
  mutate(gene_biotype="mRNA") %>%
  rbind(., filter(temp, gene_biotype != "protein_coding"))

#combine biotype of less abundant catagories into "ncRNA"
temp <- temp %>%
  filter(gene_biotype %nin% c("tRNA", "rRNA", "mRNA")) %>%
  mutate(gene_biotype = "ncRNA") %>%
  rbind(., filter(temp, gene_biotype %in% c("tRNA", "rRNA", "mRNA"))) %>%
  group_by(DM, stress, rep) %>%
  mutate(counts = (counts / sum(.$counts)))

#rename the reps and DM to make ggplot easier
temp$rep <- recode(temp$rep, "1"="rep1", "2"="rep2", "3"="rep3")
temp$DM <- recode(temp$DM, "-"=" - DM", "+"=" + DM")
temp$gene_biotype <- factor(temp$gene_biotype, levels = c("tRNA","rRNA","ncRNA","mRNA"))

#Make columns of counts for DM and rep
temp <- temp %>%
  spread(rep, counts)


temp <- temp %>%
  group_by() %>%
  group_by(gene_biotype, stress, DM) %>%
  summarise(correlation12 = cor(rep1, rep2, method=c("spearman"))**2,
            correlation13 = cor(rep1, rep3, method=c("spearman"))**2,
            correlation23 = cor(rep2, rep3, method=c("spearman"))**2) 

temp <- temp %>%
  pivot_longer(cols=c("correlation12","correlation13","correlation23"), names_to="variable", values_to = "value") #%>%
  #filter(DM ==" + DM")

temp$variable <- recode(temp$variable, "correlation12"="1 vs. 2", 
                                       "correlation13"="1 vs. 3",
                                       "correlation23"="2 vs. 3")


figS3B <- ggplot(filter(temp), aes(x=gene_biotype, y=interaction(variable, DM), fill=value)) +
  geom_tile() +
  geom_text(aes(label = round(value, 2))) +
  scale_fill_gradient(low="grey70", high="red", limits = c(0,1), breaks=c(0, .5, 1))+
  theme_void() +
  theme(legend.position = "bottom",
        legend.text = element_text(angle=0, vjust=.5),
        plot.title = element_text(hjust = 0.5, size=12),
        axis.title.x = element_blank(),
        axis.title.y = element_blank(),
        axis.text = element_text(),
        axis.text.x = element_text(angle=45, vjust=0.5) )+
  labs(title="Replicates",
    fill=bquote(~r^2) ) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")) 
figS3B

```



```{r}
# Former ##Fig S4C: scatter of + and - DM
# temp <- HEK_counts_data %>%
#   filter(stress == "None") %>%
#   group_by(rep, stress, DM, gene_biotype) %>%
#   mutate(counts = counts/sum(counts)*1000000  ) %>%
#   filter(gene_biotype %in% c("tRNA") )
# 
# temp <- temp %>%
#   group_by(DM, stress, gene_symbol, gene_biotype) %>%
#   filter(counts > 0)
# 
# temp <- temp %>%
#   summarise(counts = mean(counts) )
# 
# temp$DM <- recode(temp$DM, "-"="minus", "+"="plus")
# 
# temp <- temp %>%
#   filter(!grepl("Ecoli", gene_symbol)) %>%
#   filter(!grepl("yeast", gene_symbol)) %>%
#   filter(!grepl("spike", gene_symbol)) %>%
#   group_by()
# 
# temp <- temp %>%
#   pivot_wider(names_from = DM, values_from = counts) %>%
#   filter(plus > -1, minus > -1)
# 
# r_squared = as.character(round(cor(temp$minus, temp$plus, method="spearman") **2,2))
# temp$gene_symbol <- str_replace_all(temp$gene_symbol, ">mt", "MT-mt")
# 
# temp <- temp %>%
#   separate(gene_symbol, sep="-", c("number", "gene_symbol"))
# 
# 
# temp <- temp %>%
#   full_join(., DM_stops, by=c("gene_symbol", "stress"))
# 
# 
# 
# figS4C <- ggplot(temp, aes(x=minus, y=plus)) +
#   geom_point() +
#   geom_abline(linetype=2, slope=1, alpha=0.4) +
#   scale_y_continuous(trans="log2", breaks=c(1,100,10000))+
#   scale_x_continuous(trans="log2", breaks=c(1,100,10000))+
#   theme(legend.position = "right",
#         legend.background = element_rect(fill="transparent")) +
#   ylab("Demethylase treated (RPM)")+
#   xlab("Untreated (RPM)") +
#   #labs(color="Transcript type") +
#   coord_equal() +
#   annotate(geom="text", label=bquote(~r^2==.(r_squared) ), x=5, y=10000) +
#   theme(aspect.ratio = 1) +
#   theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")) 
#   #scale_color_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="grey50") 
#   #geom_point(data=filter(temp, gene_symbol %in% asdf$codon), color="blue")
#   #geom_text_repel(data = filter(temp, temp$plus / temp$minus >7, temp$minus >5), aes(label=gene_symbol))
# 
# figS4C

```

##Fig S3C: compare to 2015 DM-tRNA-seq
```{r}

data1 <- read.csv("../Zheng_2015/SRX900779.counts.tsv", header=T, sep="\t") %>%
  filter(gene_biotype=="tRNA") %>%
  mutate(DM="+", rep=1, source="tRNA")

data2 <- read.csv("../Zheng_2015/SRX900780.counts.tsv", header=T, sep="\t") %>%
  filter(gene_biotype=="tRNA") %>%
  mutate(DM="+", rep=2, source="tRNA")

data3 <- read.csv("../Zheng_2015/SRX900783.counts.tsv", header=T, sep="\t") %>%
  filter(gene_biotype=="tRNA") %>%
  mutate(DM="+", rep=1, source="total_RNA")

data4 <- read.csv("../Zheng_2015/SRX900784.counts.tsv", header=T, sep="\t") %>%
  filter(gene_biotype=="tRNA") %>%
  mutate(DM="+", rep=2, source="total_RNA")

Zheng_count_data <- rbind(data1, data2, data3, data3) %>%
  select(-type, -gene_biotype)
```

```{r}
temp <- HEK_counts_data %>%
  filter(gene_biotype=="tRNA", stress=="None", DM=="+") %>%
  select(-stress, -type, -gene_biotype) %>%
  mutate(source="CHRIS_seq")

temp <- Zheng_count_data %>%
  filter(source=="tRNA") %>%
  rbind(., temp) %>%
  group_by(DM, rep, source) %>%
  mutate(RPM = counts / sum(counts)*1000000) %>%
  select(-counts) %>%
  group_by(DM, source, Gene, gene_symbol) %>%
  summarise(mean_RPM = mean(RPM))

temp1 <- temp %>%
  pivot_wider(names_from = source, values_from = mean_RPM) %>%
  filter(!grepl("Ecoli", Gene)) %>%
  filter(!grepl("yeast", Gene)) %>%
  filter(!grepl("spike", Gene)) %>%
  group_by()

#The scatter plot :(
figS3C1 <- ggplot(temp1, aes(x = CHRIS_seq, y = tRNA ))+
  geom_point() +
  scale_y_continuous(trans="log2", breaks=c(1,100,10000), labels =trans_format('log10',math_format(10^.x))  ) +
  scale_x_continuous(trans="log2", breaks=c(1,100,10000), labels =trans_format('log10',math_format(10^.x)) ) +
  theme(plot.title = element_text(hjust = 0.5)) +
  xlab("CHRIS-seq (RPM)") +
  ylab("DM-tRNA-seq (RPM)") +
  #labs(title = "Abundance of tRNA\nby different techniques") +
  theme(aspect.ratio = 1) +
  coord_equal() +
  geom_abline(slope=1, linetype=2, alpha=0.4) +
  annotate(geom="text", label=bquote(~r^2~"=0.84")  , x=2000, y=1) 
figS3C1




#Correlation matrix
cormat <- round(cor(select(temp1, -Gene, -gene_symbol, -DM), method=c("spearman"))**2 ,2) %>%
  melt()

#Chage name for pretty graph
cormat$Var1 <- recode(cormat$Var1, "CHRIS_seq"="CHRIS-seq", 
                      "tRNA"="DM-tRNA-seq")
                      #"total_RNA"="DM-tRNA-seq\n(total RNA)")
cormat$Var2 <- recode(cormat$Var2, "CHRIS_seq"="CHRIS-seq", 
                      "tRNA"="DM-tRNA-seq")
                      #"total_RNA"="DM-tRNA-seq\n(total RNA)")
#reorder for pretty
cormat$Var1 <- factor (cormat$Var1, levels=c("CHRIS-seq",
                                             "DM-tRNA-seq"))
                                             #"DM-tRNA-seq\n(total RNA)"))
cormat$Var2 <- factor (cormat$Var2, levels=c(#"DM-tRNA-seq\n(total RNA)",
                                             "DM-tRNA-seq",
                                             "CHRIS-seq"))

# #Makr prety graph
# fig4C2 <- ggplot(data = cormat, aes(x=Var1, y=Var2, fill=value)) + 
#   geom_tile() +
#   geom_text(aes(label=value)) +
#   scale_fill_gradient(low="grey70", high="red", limits=c(0,1) )+
#   theme(legend.position = "None",
#         plot.title = element_text(hjust = 0.5),
#         axis.title.x = element_blank(),
#         axis.title.y = element_blank())+
#   labs(title="Correlation between\ntechniques",
#        fill="r^2")
# 
# fig4C2
# 
# 
# myplotb1 <- arrangeGrob(fig4C1)
# myplotb2 <- arrangeGrob(fig4C2)
# lay = rbind(c(1,2))
# fig4C <- grid.arrange(myplotb1, myplotb2, layout_matrix = lay)

figS3C <- figS3C1

figS3C
```

##Fig S3D: mutation trace
```{r}
temp <- HEK_base_data %>%
  #separate(gene_symbol, sep="_", c("Isodecoder", "chrom")) %>%
  filter(gene_symbol == "ArgACG", number=="chr14.trna7") %>%
  filter(stress == "None") %>%
  filter(position <73)
temp$DM <- recode(temp$DM, "m"="-", "p"="+")

figS3D <- ggplot(temp, aes(x=position, y=mutation, color=DM, group=interaction(rep, DM))) +
  geom_line() +
  scale_y_continuous(breaks=c(0,.5,1)) +
  scale_color_manual(values = c("-"="black","+"="red"))+
  theme(legend.position = c(.9, .4),
        plot.title = element_text(hjust=0.5)) +
  #labs(title = "Pro-GGG-1 mutation rate") +
  ylab("Mutation fraction")+
  xlab("Position (nt)")+
  annotate(geom="text", label=bquote(~m^1~'A'), x=58, y=.95) +
  annotate(geom="text", label=bquote('I'), x=34, y=1.1) +
  annotate(geom="text", label=bquote(~m[2]^2~'G'), x=26, y=.7) +
  annotate(geom="text", label=bquote(~m^1~'G'), x=8, y=.4) +
  annotate(geom="text", label="ArgACG", x=10, y=1.1) +
  theme(aspect.ratio = 0.7) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"),
        legend.background = element_blank())
figS3D
```

##Fig S3E: stop trace
```{r}
temp <- HEK_base_data %>%
  #mutate(gene_name= gene_symbol) %>%
  #eparate(gene_symbol, sep="_", c("Isodecoder", "chrom")) %>%
  filter(gene_symbol == "ArgACG", number=="chr14.trna7") %>%
  filter(stress=="None") %>%
  filter(position <73) %>%
  group_by(rep, DM, stress, gene_name) %>%
  mutate(pileup = pileup / max(pileup))
temp$DM <- recode(temp$DM, "m"="-", "p"="+")


figS3E <- ggplot(temp, aes(x=position, y=pileup, color=DM, group=interaction(rep, DM, gene_name))) +
  geom_line() +
  scale_y_continuous(breaks=c(0,.5,1)) +
  scale_color_manual(values = c("-"="black","+"="red"))+
  theme(legend.position = c(.9, .4),
        plot.title = element_text(hjust=0.5)) +
  #labs(title = "Pro-GGG-1 mutation rate") +
  ylab("Normalized fraction\nof reads mapped")+
  xlab("Position (nt)")+
  annotate(geom="text", label=bquote(~m^1~G), x=44, y=.6) +
  annotate(geom="text", label="ArgACG", x=10, y=max(temp$pileup)) +
  theme(aspect.ratio = 0.7) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"),
        legend.background = element_blank()) 
  #facet_wrap(~Isodecoder)
figS3E
```



##Fig S3F: miRNA
```{r, echo=FALSE}

temp <- HEK_counts_data %>%
  filter(stress=="None") %>%
  group_by(rep, DM, stress) %>% 
  mutate(counts = counts/sum(counts)*1000000 ) %>%
  group_by(DM, stress) %>% 
  mutate(counts = ifelse( counts>0, counts, 0) ) %>%
  filter(counts > 2) %>%
  filter(gene_biotype %in% c("miRNA")) %>%
  group_by(Gene, gene_symbol, DM, stress) %>% #sterss, DM
  mutate(std_count = sd(counts),
            mean_count = mean(counts)) %>%
  group_by() 


#Arrange by -DM value
temp <- temp %>%
  filter(DM=="-") %>%
  mutate(minus_mc = mean_count) %>%
  select(gene_symbol, minus_mc) %>%
  full_join(.,temp, by=c("gene_symbol")) %>%
  #group_by(stress) %>%
  mutate(gene_symbol = fct_reorder(gene_symbol, desc(minus_mc)))


#Make the pretty plot
figS3F <- ggplot(filter(temp), aes(x=gene_symbol, y=mean_count, color=DM ))+
  geom_point(shape=95, size=5) +
  geom_point(aes(y=counts), size=.4) +
  scale_color_manual(values = c("-"="black", "+"="red")) +
  #geom_errorbar(aes(ymin=mean_count-std_count, ymax=mean_count+std_count)) +
  scale_y_log10() +
  theme(axis.text.x = element_text(angle=90, vjust=.5, hjust=0), 
        axis.title.x = element_blank(),
        legend.position= c(.8, .6),
        legend.background = element_blank(),
        plot.title = element_text(size=12, hjust=0.5)) + 
  #facet_grid(cols=vars(stress)) + 
  labs(title="miRNA")+
  ylab("Reads per million") +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")) 
figS3F
```


##Fig S3G: mRNA input
###Read in poly(A) selected library
```{read in HEK basewise data}
#Rep1 minus DM
data1 = read.csv("../../2020 03 06 input titration/express/TP-CK-TP1-5_S5bc1.counts.tsv", header=T, sep="\t") %>%
  mutate(rep=1, input="ng1000")

data2 = read.csv("../../2020 03 06 input titration/express/TP-CK-TP1-5_S5bc2.counts.tsv", header=T, sep="\t") %>%
  mutate(rep=1, input="ng100")

data3 = read.csv("../../2020 03 06 input titration/express/TP-CK-TP1-5_S5bc3.counts.tsv", header=T, sep="\t") %>%
  mutate(rep=1, input="ng10")

#rep 2
data4 = read.csv("../../2020 03 06 input titration/express/TP-CK-TP1-6_S6bc1.counts.tsv", header=T, sep="\t") %>%
  mutate(rep=2, input="ng100")

data5 = read.csv("../../2020 03 06 input titration/express/TP-CK-TP1-6_S6bc2.counts.tsv", header=T, sep="\t") %>%
  mutate(rep=2, input="ng10")

data6 = read.csv("../../2020 03 06 input titration/express/TP-CK-TP1-6_S6bc3.counts.tsv", header=T, sep="\t") %>%
  mutate(rep=2, input="ng1000")

titration_express <- rbind(data1, data2, data3,
                           data4, data5, data6)
```

###Piechart
```{r, more different text, echo=FALSE}

temp <- titration_express %>%
  group_by(input, rep) %>%
  mutate(rpm = counts / sum(counts) *1000000) %>%
  group_by(Gene, type, gene_biotype, gene_symbol, input) %>%
  summarise(rpm = mean(rpm)) %>%
  group_by()

#combine biotype of less abundant catagories into "ncRNA"
temp <- temp %>%
  filter(gene_biotype %nin% c("tRNA", "rRNA", "protein_coding")) %>%
  mutate(gene_biotype = "ncRNA") %>%
  rbind(., filter(temp, gene_biotype %in% c("tRNA", "rRNA", "protein_coding"))) %>%
  group_by(gene_biotype) %>%
  summarise(rpm = sum(rpm)) %>%
  mutate(rpm = round((rpm / sum(.$rpm))*100,1))

#rename "protein_coding" to "mRNA"
temp <- temp %>%
  filter(gene_biotype=="protein_coding") %>%
  mutate(gene_biotype="mRNA") %>%
  rbind(., filter(temp, gene_biotype != "protein_coding"))


#transform value to be compatable with piechart
temp <- temp %>% 
  arrange(desc(gene_biotype)) %>%
  mutate(ypos = cumsum(rpm)- 0.5*rpm )


figS3G1 <- ggplot(temp, aes(x="", y=rpm, fill=gene_biotype)) +
  geom_bar(stat="identity", width=1, color="white") +
  coord_polar("y", start=-2) +
  geom_text_repel(aes(y=ypos, label=paste0(gene_biotype, "\n", rpm, "%") ), box.padding = 2)+
  theme_void() +
  scale_fill_manual(values = c("tRNA"="#99CCFF", 
                                "rRNA"="#FFCC33", 
                               "mRNA"="#99CC99",
                                "ncRNA"="#CC99CC"))+
  labs(fill="RNA family") +
  guides(fill = guide_legend(reverse=TRUE)) + 
  theme(legend.position = "None",
        plot.title = element_text(hjust = 0.5, size=14)) +
  labs(title="Poly(A)-selected reads")

figS3G1
```

###Agreement between 1ug replicates
```{r}
#COLLAPSE SPLICOFORMS TOGETHER
temp <- titration_express %>%
  filter(gene_biotype == "protein_coding") %>%
  select(-type) %>%
  group_by(gene_symbol, gene_biotype, rep, input) %>%
  summarise(counts = sum(counts)) %>%
  group_by(input, rep) %>%
  mutate(counts = counts / sum(counts)*1000000) %>%
  group_by(gene_symbol, gene_biotype) %>%
  filter(input=="ng1000")


temp$rep <- recode(temp$rep, "1"="rep1", "2"="rep2")

temp1 <- temp %>%
  pivot_wider(names_from = rep, values_from = counts) %>%
  filter(rep1> 0, #ng100_1>0, ng10_1>0,
         rep2 > 0 ) #ng100_2>0, ng10_2>0)


r_squared = as.character(round(cor(temp1$rep1, temp1$rep2, method="spearman") **2,2))

figS3G2 <- ggplot(data = temp1, aes(x=rep1, y=rep2)) + 
  geom_point() +
  annotate(geom="text", label=bquote(~r^2==.(r_squared) ), x=5, y=1000) +
  scale_y_continuous(trans="log2", breaks=c(1,100,10000),labels=trans_format('log10',math_format(10^.x)) )+
  scale_x_continuous(trans="log2", breaks=c(1,100,10000),labels=trans_format('log10',math_format(10^.x)) )+
  geom_abline(slope=1, linetype=2, color="grey60")+
  scale_fill_gradient(low="grey70", high="red")+
  theme(legend.position = "right",
        plot.background = element_blank(),
        plot.title = element_text(hjust = 0.5, size=12),
        axis.title = element_text(size=12),
        axis.text = element_text(size=8),
        axis.title.x = element_text(vjust = 0))+
  #labs(title="mRNA\nabundance" )+
  ylab("RPM rep 2") +
  xlab("RPM rep 1") +
  coord_equal() +
  theme(aspect.ratio = 1)
figS3G2

myplotb1 <- arrangeGrob(figS3G1)
myplotb2 <- arrangeGrob(figS3G2)
lay = rbind(c(1,2))

figS3G <- grid.arrange(myplotb1, myplotb2, layout_matrix = lay)

```


##old Save figure S3
```{r}
myplota <- arrangeGrob(figS3A, left = textGrob(expression(bold("a")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotb <- arrangeGrob(figS3B, left = textGrob(expression(bold("b")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotc <- arrangeGrob(figS3C, left = textGrob(expression(bold("c")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotd <- arrangeGrob(figS3D, left = textGrob(expression(bold("d")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplote <- arrangeGrob(figS3E, left = textGrob(expression(bold("e")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotf <- arrangeGrob(figS3F, left = textGrob(expression(bold("f")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotg <- arrangeGrob(figS3G, left = textGrob(expression(bold("g")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
# myploth <- arrangeGrob(figS3H, left = textGrob(expression(bold("h")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
# myploti <- arrangeGrob(figS3I, left = textGrob(expression(bold("i")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
# myplotj <- arrangeGrob(figS3J, left = textGrob(expression(bold("j")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
# myplotk <- arrangeGrob(figS3K, left = textGrob(expression(bold("k")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))


#Set the figure arrangement matrix
lay <- rbind(c(1, 2, 3),
             c(4, 5, 6),
             c(7, 7, 8))
#Arrange figures
figureS3 <- grid.arrange(myplota, myplotb, myplotc, myplotd, myplote, myplotf, myplotg, empty_graph,  layout_matrix = lay)
figureS3

setwd("~/4SR/data/20200317_Nature_methods_data/2020 02 24 ecoli figures/source/")

ggsave("../figures/figureS3.svg", 
       plot = figureS3,
       scale = 1.3,
       width = 8, 
       height = 7)
ggsave("../figures/figureS3.png", 
       plot = figureS3,
       scale = 1.3,
       width = 8, 
       height = 7)

```

##new Save figure S3
```{r}
myplota <- arrangeGrob(figS3A, top = textGrob(expression(bold("a")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotb <- arrangeGrob(figS3B, top = textGrob(expression(bold("b")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotc <- arrangeGrob(fig3B, top = textGrob(expression(bold("c")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotd <- arrangeGrob(fig3C, top = textGrob(expression(bold("d")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplote <- arrangeGrob(figS3D, top = textGrob(expression(bold("e")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotf <- arrangeGrob(figS3E, top = textGrob(expression(bold("f")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))


myplotg <- arrangeGrob(fig3F, left = textGrob(expression(bold("g")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myploth <- arrangeGrob(figS3F, top = textGrob(expression(bold("h")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplotef <- grid.arrange(myplote, myplotf, myploth, layout_matrix=rbind(c(1,1,2,2,3,3,3)))


# myploti <- arrangeGrob(figS3I, left = textGrob(expression(bold("i")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
# myplotj <- arrangeGrob(figS3J, left = textGrob(expression(bold("j")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
# myplotk <- arrangeGrob(figS3K, left = textGrob(expression(bold("k")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))


#Set the figure arrangement matrix
lay <- rbind(c(1, 2, 3, 4),
             c(5, 5, 5, 5),
             c(6, 6, 6, 6))
#Arrange figures
figureS3 <- grid.arrange(myplota, myplotb, myplotc, myplotd, myplotef, myplotg,  layout_matrix = lay)
figureS3

setwd("~/4SR/data/20200317_Nature_methods_data/2020 02 24 ecoli figures/source/")

ggsave("../figures/figureS3.svg", 
       plot = figureS3,
       scale = 1.3,
       width = 9, 
       height = 7)
ggsave("../figures/figureS3.png", 
       plot = figureS3,
       scale = 1.3,
       width = 9, 
       height = 7)

```

#_
#Figure S4
##*Fig S4A
```{r}

setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/EXTRA_FIGURES/")
img = readPNG("CMC_CHRIS-seq.png")
g <- rasterGrob(img, interpolate=TRUE)
figS4A <- qplot(0:3, 0:3, geom="blank") +
  annotation_custom(g, xmin=-Inf, xmax=Inf, ymin=-Inf, ymax=Inf) +
  theme_void() +
  theme(plot.margin = unit(c(0.3,0.3,0.3,0.3), "cm"))
figS4A


```

##Read in CMC data
```{r}
#Scatter the mutation rate and stop rate in 2 replicates for CMC rRNA. One dot per position. 
setwd("~/4SR/data/20200317_Nature_methods_data/rRNA_CMC/5_tsv/human_rRNA/")

data1 <- read.csv("./TP-WZ-1-RPI9_S1_L005L1_bc1_TTCC.tsv", sep = "\t") %>%
  mutate(rep="rep1", treatment="CMC_plus")

data2 <- read.csv("./TP-WZ-1-RPI10_S2_L005L1_bc1_TTCC.tsv", sep="\t") %>%
  mutate(rep="rep1", treatment="CMC_minus")

data3 <- read.csv("./TP-WZ-1-RPI11_S3_L005L1_bc1_TTCC.tsv", sep = "\t") %>%
  mutate(rep="rep2", treatment="CMC_plus")

data4 <- read.csv("./TP-WZ-1-RPI12_S4_L005L1_bc1_TTCC.tsv", sep = "\t") %>%
  mutate(rep="rep2", treatment="CMC_minus")

rRNA_CMC_data <- rbind(data1, data2, data3, data4)
```

##Fig S4B
```{r}

temp <- rRNA_CMC_data %>%
  filter(pileup >=1000) %>%
  filter(stop >=0, stop <=1) %>%
  filter(treatment=="CMC_plus") %>%
  group_by() %>%
  select(gene, mutation, stop, treatment, position, rep) %>%
  pivot_longer( -c(gene, treatment, rep, position), names_to = "metric", values_to = "rate") %>%
  pivot_wider(names_from = "rep", values_from = c("rate")) %>%
  filter(!is.na(rep1), 
         !is.na(rep2))

temp$metric <- recode(temp$metric, "stop"="Stop", "mutation"="Mutation")



r_squared1 = as.character(round(cor(filter(temp, metric=="Mutation")$rep1, filter(temp, metric=="Mutation")$rep2, method="pearson") **2,3))
r_squared2 = as.character(round(cor(filter(temp, metric=="Stop")$rep1, filter(temp, metric=="Stop")$rep2, method="pearson") **2,3))
rsquared_values = data.frame(rep1=c(.25, .25), rep2 = c(.01, .01), metric=c("Mutation","Stop"),text=c(r_squared1,r_squared2))
PLOT_titles = data.frame(rep1=c(.002, .002), rep2 = c(.4, .4), metric=c("Mutation","Stop"),text=c("Mutation", "Stop"))



r_squared=c("a","b")

figS4B <- ggplot(filter(temp, treatment=="CMC_plus"),
       aes(x=rep1, y=rep2)) +
  geom_point(size=1) +
  scale_y_continuous(trans="log2", breaks=c(0.01,0.1,1), 
                     labels =trans_format('log10',math_format(10^.x)),
                     limits = c(.001, .5)
                     ) +
  scale_x_continuous(trans="log2", breaks=c(0.01,0.1,1), 
                     labels =trans_format('log10',math_format(10^.x)),
                     limits = c(.001, .5)
                     ) +
  facet_grid(rows=vars(metric)) +
  xlab("Replicate 1") +
  ylab("Replicate 2") +
  #geom_text(data=test, aes(label=text)) +
  geom_text(data=PLOT_titles, aes(label=text)) +
  annotate(geom="text", label= paste0('~r^2'), x=.04, y=.01, parse=TRUE) +
  annotate(geom="text", label= paste0('='), x=.08, y=.01, parse=F) +
  geom_text(data=rsquared_values, aes(label=text) ) +
  theme(strip.placement = "outside",
        strip.background = element_blank(),
        panel.background = element_blank(),
        #plot.background = element_blank(),
        strip.text.y = element_blank(),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")) +
  coord_fixed(ratio=1)
figS4B 
```

##Fig S4C
```{r}

temp <- rRNA_CMC_data %>%
  mutate(stop = ifelse(stop <0, 0, stop))
temp$treatment <- recode(temp$treatment, "CMC_minus"="- CMC", "CMC_plus"="+ CMC")

pseudoU_18S <- c(34,36,93,105,109,119,210,218,406,572,609,649,651,681,686,801,814,815,822,863,866,918,966,1004,1056,1081,1174,1238,1244,1326,1347,1367,1445,1625,1643,1692) #https://people.biochem.umass.edu/fournierlab/3dmodmap/hum28sseq.php

temp <- temp %>%
  filter(rep=="rep1",position >=750, position <=1300,gene == "X03205.1")

#Stop and mutation sites from 800-1200 in 18S rRNA
A <- ggplot(temp,
       aes(x=position, y=mutation, color=treatment, group=interaction(rep, treatment))) +
  geom_line(size=.2) +
  geom_point(data=filter(temp, position %in% pseudoU_18S, treatment=="+ CMC"), color="grey40", shape="*", size=4) +
  coord_cartesian(ylim=c(0, .15)) +
  scale_y_continuous(breaks=c(0, 0.05, 0.10, 0.15)) +
  scale_x_continuous(breaks = c(800,1000,1200)) +
  scale_color_manual(values=c("- CMC"="black", "+ CMC"="red")) +
  ylab("Mutation Fraction") +
  labs(color="Treatment") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        legend.position = c(.35, .7),
        legend.background = element_blank(),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))

B <- ggplot(temp,
       aes(x=position, y=stop, color=treatment, group=interaction(rep, treatment))) +
  geom_line(size=.2) +
  geom_point(data=filter(temp, position %in% pseudoU_18S, treatment=="+ CMC"), color="grey40", shape="*", size=4) +
  coord_cartesian(ylim=c(0,.75)) +
  scale_y_continuous(breaks=c(0, 0.25, 0.50, 0.75)) +
  scale_x_continuous(breaks = c(800,1000,1200)) +
  ylab("Stop Fraction") +
  xlab("Position (nt)") +
  scale_color_manual(values=c("- CMC"="black", "+ CMC"="red"))+
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))


figS4C <- grid.arrange(A,B, layout_matrix = rbind(c(1), c(2)) )

```

##Fig S4D
```{r}
temp <- rRNA_CMC_data %>%
  mutate(stop = ifelse(stop <0, 0, stop))
temp$treatment <- recode(temp$treatment, "CMC_minus"="- CMC", "CMC_plus"="+ CMC")

pseudoU_28S <- c(1523,1569,1664,1670,1731,1766,1779,1789,1847,1849,2495,3616,3618,3674,3694,3709,3713,3737,3741,3743,3747,3749,3797,3801,3823,3830,3832,3863,3899,3938,4263,4266,4269,4282,4323,4331,4383,4390,4393,4401,4412,4426,4441,4470,4496,4501,4522,4546,4549,4598,4606,4643,4659,4937,4966,4975) +13 #Why is this off by 13? different reference sequence start position?

temp <- temp %>%
  filter(rep=="rep1",position >=1500, position <=1900, gene == "NR_003287.4")

#Stop and mutation sites from 800-1200 in 18S rRNA
A <- ggplot(temp,
       aes(x=position, y=mutation, color=treatment, group=interaction(rep, treatment))) +
  geom_line(size=.2) +
  geom_point(data=filter(temp, position %in% pseudoU_28S, treatment=="+ CMC"), color="grey40", shape="*", size=4) +
  coord_cartesian(ylim=c(0, .15)) +
  scale_y_continuous(breaks=c(0, 0.05, 0.10, 0.15)) +
  scale_x_continuous(breaks = c(1500,1700,1900)) +
  scale_color_manual(values=c("- CMC"="black", "+ CMC"="red")) +
  ylab("Mutation Fraction") +
  labs(color="Treatment") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        legend.position = c(.35, .7),
        legend.background = element_blank(),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))

B <- ggplot(temp,
       aes(x=position, y=stop, color=treatment, group=interaction(rep, treatment))) +
  geom_line(size=.2) +
  geom_point(data=filter(temp, position %in% pseudoU_28S, treatment=="+ CMC"), color="grey40", shape="*", size=4) +
  coord_cartesian(ylim=c(0,.75)) +
  scale_y_continuous(breaks=c(0, 0.25, 0.50, 0.75)) +
  scale_x_continuous(breaks = c(1500,1700,1900)) +
  ylab("Stop Fraction") +
  xlab("Position (nt)") +
  scale_color_manual(values=c("- CMC"="black", "+ CMC"="red")) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))

#Stop and mutaitonsites from 1500-2000 in 28S rRNA
figS4D <- grid.arrange(A,B, layout_matrix = rbind(c(1), c(2)) )

```

##Fig S4E
```{r}

temp <- rRNA_CMC_data %>%
  mutate(stop = ifelse(stop <0, 0, stop))
temp$treatment <- recode(temp$treatment, "CMC_minus"="- CMC", "CMC_plus"="+ CMC")

pseudoU_18S <- c(34,36,93,105,109,119,210,218,406,572,609,649,651,681,686,801,814,815,822,863,866,918,966,1004,1056,1081,1174,1238,1244,1326,1347,1367,1445,1625,1643,1692)

temp <- temp %>%
  filter(rep=="rep1",
         #position >=750, 
         #position <=1300,
         gene == "X03205.1")

#Stop and mutation sites from 800-1200 in 18S rRNA
A <- ggplot(temp,
       aes(x=position, y=mutation, color=treatment, group=interaction(rep, treatment))) +
  geom_line(size=.2) +
  geom_point(data=filter(temp, position %in% pseudoU_18S, treatment=="+ CMC"), color="grey40", shape="*", size=4) +
  coord_cartesian(ylim=c(0, .15)) +
  scale_y_continuous(breaks=c(0, 0.05, 0.10, 0.15)) +
  scale_color_manual(values=c("- CMC"="black", "+ CMC"="red")) +
  ylab("Mutation Fraction") +
  theme(axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        legend.background = element_blank(),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))

B <- ggplot(temp,
       aes(x=position, y=stop, color=treatment, group=interaction(rep, treatment))) +
  geom_line(size=.2) +
  geom_point(data=filter(temp, position %in% pseudoU_18S, treatment=="+ CMC"), color="grey40", shape="*", size=4) +
  coord_cartesian(ylim=c(0,.75)) +
  scale_y_continuous(breaks=c(0, 0.25, 0.50, 0.75)) +
  ylab("Stop Fraction") +
  xlab("Position (nt)") +
   labs(color="Treatment") +
  scale_color_manual(values=c("- CMC"="black", "+ CMC"="red"))+
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"),
        legend.position = c(.2, .7),
        legend.background = element_blank())


figS4E <- grid.arrange(A,B, layout_matrix = rbind(c(1), c(2)) )

```


##Save figure S4
```{r}

myplota <- arrangeGrob(figS4A, left = textGrob(expression(bold("a")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotb <- arrangeGrob(figS4B, left = textGrob(expression(bold("b")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotc <- arrangeGrob(figS4C, left = textGrob(expression(bold("c")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotd <- arrangeGrob(figS4D, left = textGrob(expression(bold("d")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplote <- arrangeGrob(figS4E, left = textGrob(expression(bold("e")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))


#Set the figure arrangement matrix
lay <- rbind(c(1, 1, 3, 4),
             c(1, 1, 3, 4),
             c(1, 1, 3, 4),
             c(2, 5, 5, 5),
             c(2, 5, 5, 5),
             c(2, 5, 5, 5))
#Arrange figures
figureS4 <- grid.arrange(myplota, myplotb, myplotc, myplotd, myplote,  layout_matrix = lay)
figureS4

setwd("~/4SR/data/20200317_Nature_methods_data/2020 02 24 ecoli figures/source/")

ggsave("../figures/figureS4.svg", 
       plot = figureS4,
       scale = 1.3,
       width = 8, 
       height = 7)
ggsave("../figures/figureS4.png", 
       plot = figureS4,
       scale = 1.3,
       width = 8, 
       height = 7)

```

#_
#Figure 4
##Fig 4A: Cartoon diagram
```{r}

setwd("~/4SR/data/20200317_Nature_methods_data/")
img = readPNG("./fig4B_cartoon.png")
fig4A <- rasterGrob(img, interpolate=TRUE)
# fig4A <- qplot(0:2, 0:2, geom="blank") +
#   annotation_custom(fig4B, xmin=-Inf, xmax=Inf, ymin=-Inf, ymax=Inf) +
#   theme_void()

fig4A
```

##Fig 4B: Piechart oral
```{r}
#How mnay reads
#How many MAPPED to bacterial tRNA
#How many mapped to SRP_combine
#How many map to 5S combine
#PLUS DM DATA
temp <- data.frame(RNA_type = c("tRNA", "SRP", "5S rRNA", "unmapped"),
                   average  = c(70.4, 1.1, 4.6, 23.8),
                   sdev     = c(6.1 , 0.2, 0.6, 5.8))

#transform value to be compatable with piechart - idk how this works, I copied it from stack exchange
temp <- temp %>% 
  arrange(desc(RNA_type)) %>%
  mutate(ypos = cumsum(average)- 0.5*average ) %>%
  mutate(label = paste0(RNA_type, "\n", average, " ±", sdev, "%") )

fig4B <-ggplot(temp, aes(x=1, y=average, fill=RNA_type)) +
  geom_bar(stat="identity", width=1, color="white") +
  coord_polar("y", start=1.1) +
  geom_text_repel(aes(y=ypos, x=1.5, label=paste0(RNA_type, "\n", average, " ±", sdev, "%") ), box.padding = 0.5, xlim = c(1.5, 3),size=3.2)+
  theme_void() +
  # scale_fill_manual(values = c("tRNA"="#99CCFF",
  #                               "rRNA"="#FFCC33",
  #                              "SRP"="#99CC99",
  #                               "unmapped"="#CC99CC"))+
  labs(fill="RNA type") +
  guides(fill = guide_legend(reverse=TRUE)) +
  theme(legend.position = "None",
        #plot.background = element_blank(),
        #panel.background = element_blank(),
        #panel.border = element_blank(),
        plot.title = element_text(hjust = 0.5),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")
        #axis.text.x = element_text()
        ) 
  #scale_y_continuous(breaks=cumsum(temp$average) - temp$average / 2, labels=temp$label )
  #labs(title="Read mapping from\noral micriobiome")

fig4B


# p <- ggplot(temp, aes(x=1,y=average,fill=RNA_type) ) + 
#   geom_bar(stat="identity", color = "black") + 
#   coord_polar(theta="y", start=70.3) + 
#   theme(axis.ticks=element_blank(),
#         axis.text.y=element_blank(),
#         axis.text.x= element_blank(),#element_text(colour='black'),
#         axis.title=element_blank() ) +
#   geom_text_repel(aes(y=ypos, x=1.4, label=paste0(RNA_type, "\n", average, " ±", sdev, "%") ), box.padding = 2) 
#   #scale_y_continuous(breaks=cumsum(temp$average) - temp$average / 2, labels=temp$label )
# p

```

##Fig 4CDE: scatter SRP 5S tRNA
###Read in SRP, 5S data
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_oral/6_sam_counter/")

#5s rRNA
data1 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc1_TTCC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="minus", type="rRNA_5S")
data2 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc2_TCTG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="minus", type="rRNA_5S")
data3 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc3_TGGT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="minus", type="rRNA_5S")
data4 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc4_CTGA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="minus", type="rRNA_5S")
data5 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc5_CCAT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="minus", type="rRNA_5S")
data6 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc6_CATC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="minus", type="rRNA_5S")
data7 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc7_GTAG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="minus", type="rRNA_5S")
data8 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc8_GGTA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="minus", type="rRNA_5S")

data9 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc1_TTCC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="plus", type="rRNA_5S")
data10 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc2_TCTG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="plus", type="rRNA_5S")
data11 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc3_TGGT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="plus", type="rRNA_5S")
data12 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc4_CTGA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="plus", type="rRNA_5S")
data13 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc5_CCAT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="plus", type="rRNA_5S")
data14 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc6_CATC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="plus", type="rRNA_5S")
data15 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc7_GTAG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="plus", type="rRNA_5S")
data16 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc8_GGTA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="plus", type="rRNA_5S")
#Re-read in rRNA 5S taxon info
rRNA_5S_taxon <- read.csv("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_SRP/6_sam_counter/rRNA_5S_combine/rRNA_5s_combine_lineage.tsv", sep="\t", na.strings = "NA") %>%
  mutate(type="rRNA_5S", sub_type="rRNA_5S") %>%
  separate(name, sep="_", c("name", "other")) %>%
  select(-other)
#combine data
rRNA_5S_mapping_data <- rbind(data1, data2, data3, data4,
                      data5, data6, data7, data8,
                      data9, data10,data11,data12,
                      data13,data14,data15,data16)



#rfam_SRP
data1 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc1_TTCC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="minus", type="rfam_SRP")
data2 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc2_TCTG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="minus", type="rfam_SRP")
data3 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc3_TGGT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="minus", type="rfam_SRP")
data4 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc4_CTGA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="minus", type="rfam_SRP")
data5 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc5_CCAT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="minus", type="rfam_SRP")
data6 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc6_CATC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="minus", type="rfam_SRP")
data7 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc7_GTAG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="minus", type="rfam_SRP")
data8 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc8_GGTA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="minus", type="rfam_SRP")

data9 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc1_TTCC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="plus", type="rfam_SRP")
data10 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc2_TCTG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="plus", type="rfam_SRP")
data11 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc3_TGGT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="plus", type="rfam_SRP")
data12 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc4_CTGA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="plus", type="rfam_SRP")
data13 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc5_CCAT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="plus", type="rfam_SRP")
data14 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc6_CATC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="plus", type="rfam_SRP")
data15 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc7_GTAG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="plus", type="rfam_SRP")
data16 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc8_GGTA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="plus", type="rfam_SRP")
#Re-read in SRP taxon info
SRP_short <- read.csv("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_SRP/6_sam_counter/SRP_combine/RF00169_lineage.tsv", sep="\t", na.strings = "NA") %>%
  mutate(sub_type="short_SRP")
SRP_long <- read.csv("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_SRP/6_sam_counter/SRP_combine/RF01854_lineage.tsv", sep="\t", na.strings = "NA") %>%
  mutate(sub_type="long_SRP")
rfam_SRP_taxon <- rbind(SRP_short, SRP_long) %>%
  mutate(type="rfam_SRP")
#combine data
rfam_SRP_mapping_data <- rbind(data1, data2, data3, data4,
                               data5, data6, data7, data8,
                               data9, data10,data11,data12,
                               data13,data14,data15,data16)



mapping_data <- rbind(rRNA_5S_mapping_data, 
                      rfam_SRP_mapping_data)
#Combine lineage data
taxon=rbind(rRNA_5S_taxon, rfam_SRP_taxon)
#Merge mapping and lineage
new_oral_abundance <- mapping_data %>%
  full_join(., taxon, by=c("name", "type")) %>%
  filter(!is.na(kingdom), !is.na(phylum), !is.na(class), !is.na(order)) #%>%
  #filter(count >= 0)

```

###Read in tRNA ORDER data
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_oral/ORAL_TAXONOMY/")

#tRNA mapped data
data1 <- read.csv("./tongue1_day1_untreated/tongue1_day1_untreated_order_aa_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="minus", type="tRNA")
data2 <- read.csv("./tongue1_day2_untreated/tongue1_day2_untreated_order_aa_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="minus", type="tRNA")
data3 <- read.csv("./tongue2_day1_untreated/tongue2_day1_untreated_order_aa_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="minus", type="tRNA")
data4 <- read.csv("./tongue2_day2_untreated/tongue2_day2_untreated_order_aa_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="minus", type="tRNA")
data5 <- read.csv("./tongue3_day1_untreated/tongue3_day1_untreated_order_aa_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="minus", type="tRNA")
data6 <- read.csv("./tongue3_day2_untreated/tongue3_day2_untreated_order_aa_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="minus", type="tRNA")
data7 <- read.csv("./tongue4_day1_untreated/tongue4_day1_untreated_order_aa_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="minus", type="tRNA")
data8 <- read.csv("./tongue4_day2_untreated/tongue4_day2_untreated_order_aa_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="minus", type="tRNA")

data9 <- read.csv("./tongue1_day1_demethylase/tongue1_day1_demethylase_order_aa_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="plus", type="tRNA")
data10 <- read.csv("./tongue1_day2_demethylase/tongue1_day2_demethylase_order_aa_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="plus", type="tRNA")
data11<- read.csv("./tongue2_day1_demethylase/tongue2_day1_demethylase_order_aa_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="plus", type="tRNA")
data12<- read.csv("./tongue2_day2_demethylase/tongue2_day2_demethylase_order_aa_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="plus", type="tRNA")
data13<- read.csv("./tongue3_day1_demethylase/tongue3_day1_demethylase_order_aa_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="plus", type="tRNA")
data14<- read.csv("./tongue3_day2_demethylase/tongue3_day2_demethylase_order_aa_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="plus", type="tRNA")
data15<- read.csv("./tongue4_day1_demethylase/tongue4_day1_demethylase_order_aa_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="plus", type="tRNA")
data16<- read.csv("./tongue4_day2_demethylase/tongue4_day2_demethylase_order_aa_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="plus", type="tRNA")

#combine data
new_oral_tRNA_mapping_data <- rbind(data1, data2, data3, data4,
                          data5, data6, data7, data8,
                          data9, data10,data11,data12,
                          data13,data14,data15,data16) %>%
  mutate(order=X) %>%
  select(-X, -X.1) %>%
  pivot_longer(c(-sample, -day, -order, -DM, -type) , names_to = "amino_acid", values_to = "counts") %>%
  filter(amino_acid != "STP", 
         order!="",
         counts >=0) 

```

###Process and combine data
```{r}

#Consider short or long SRP
temp <- new_oral_abundance %>%
  select(DM, sample, day, name, count, kingdom, phylum, class, order, family, genus, species, type) %>%
  group_by(DM, sample, day, name, kingdom, phylum, class, order, family, genus, species, type) %>% 
  summarise(est_counts = sum( count)) %>% #Combine species if they're for some reason separate
  group_by() %>%
  pivot_wider(names_from = type, values_from = est_counts) %>%
  group_by(DM, sample, day, kingdom, phylum, class, order, family, genus) %>% #Combine species
  summarise(rfam_SRP = sum(rfam_SRP, na.rm=T), 
            rRNA_5S = sum(rRNA_5S, na.rm=T)) %>%
  filter(rRNA_5S >0, rfam_SRP >0) %>% #REMOVE SPECIES FOR WHICH THERE IS NO DATA
  group_by(DM, sample, day, kingdom, phylum, class, order) %>% #ADD ORDER HERE IF YOU WANT ORDER LINEAGE
  summarise(rfam_SRP = sum(rfam_SRP, na.rm=T), 
            rRNA_5S = sum(rRNA_5S, na.rm=T)) 

#Add some Z-scores
temp <- temp %>%
  group_by(sample, day, DM) %>%
  mutate(Zscore_rRNA_5S  = (log10(rRNA_5S ) - mean(log10(rRNA_5S )))/sd(log10(rRNA_5S )),
         Zscore_rfam_SRP = (log10(rfam_SRP) - mean(log10(rfam_SRP)))/sd(log10(rfam_SRP)) )


#Sum up tRNAs by order and combine with SRP and 5S
temp <- new_oral_tRNA_mapping_data %>%
  group_by(sample, day, DM, order) %>%
  summarise(total_tRNA_counts=sum(counts)) %>%
  full_join(., temp, by=c("sample", "day", "DM", "order")) %>%
  filter(total_tRNA_counts >=1) %>%
  mutate(Zscore_total_tRNA_counts = (log10(total_tRNA_counts)-mean(log10(total_tRNA_counts))) /sd(log10(total_tRNA_counts)) )

Zscore_data <- temp %>%
  filter(!is.na(phylum),
         !is.na(Zscore_rRNA_5S),
         !is.na(Zscore_rfam_SRP),
         !is.na(Zscore_total_tRNA_counts))
```

###correlation play zone
```{r}

selected_phyla <- c("Actinobacteria","Bacteroidetes","Firmicutes","Proteobacteria","Spirochaetes","Tenericutes")

temp <- Zscore_data %>%
  filter(phylum %in% selected_phyla,
         #day=="b",
         sample !="Volunteer v04",
         DM=="plus"
         )

r_squared = as.character(round(cor(temp$Zscore_rRNA_5S, temp$Zscore_rfam_SRP, method="pearson") **2,2))
#SRP in actinobacteria might be affected by DM. Just saying

ggplot(temp, 
       #aes(x=Zscore_rRNA_5S, y=Zscore_rfam_SRP, color=phylum)) +
       aes(x=rRNA_5S, y=rfam_SRP, color=phylum, shape=day)) +
  geom_point() +
  geom_text(data=filter(temp, order=="Clostridiales", phylum =="Firmicutes"), aes(label=order)) +
  annotate(geom="text", label=bquote(~r^2==.(r_squared) ), x=1, y=-1) +
  xlab("5S rRNA") +
  ylab("SRP RNA") +
  labs(color="Phylum") +
  scale_color_manual(values=c("Actinobacteria"="grey20", "Bacteroidetes"="red", "Firmicutes"="dodgerblue2",
                              "Proteobacteria"="olivedrab3", "Spirochaetes"="mediumorchid1", "Tenericutes"="goldenrod1")) +
  theme(legend.position = c(.25, .75),
        legend.background = element_blank(),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))  +
  geom_abline(slope=0.15) +
  geom_abline(slope=0.05) +
  geom_abline(slope=0.25)


```



###Fig 4C: scatter SRP vs. 5s rRNA
```{r}

selected_phyla <- c("Actinobacteria","Bacteroidetes","Firmicutes","Proteobacteria","Spirochaetes","Tenericutes")

temp <- Zscore_data %>%
  filter(phylum %in% selected_phyla,
         #day=="a",
         #sample=="Volunteer v04",
         DM=="plus"
         )

r_squared = as.character(round(cor(temp$Zscore_rRNA_5S, temp$Zscore_rfam_SRP, method="pearson") **2,2))
#SRP in actinobacteria might be affected by DM. Just saying

fig4C <- ggplot(temp, 
       aes(x=Zscore_rRNA_5S, y=Zscore_rfam_SRP, color=phylum)) +
       #aes(x=rRNA_5S, y=rfam_SRP, color=phylum, shape=day)) +
  geom_point() +
  annotate(geom="text", label=bquote(~r^2==.(r_squared) ), x=-1, y=2) +
  xlab("5S rRNA") +
  ylab("SRP RNA") +
  labs(color="Phylum") +
  scale_color_manual(values=c("Actinobacteria"="grey20", "Bacteroidetes"="red", "Firmicutes"="dodgerblue2",
                              "Proteobacteria"="olivedrab3", "Spirochaetes"="mediumorchid1", "Tenericutes"="goldenrod1")) +
  theme(legend.position = "none",
        legend.background = element_rect(colour = "black", fill="white"),
        legend.key.height = unit(c(0.4),"cm"),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")) +
  coord_equal(xlim = c(-2.1,3.1), ylim = c(-2.1, 2.3))
fig4C

fig4C_legend <- get_legend(fig4C+theme(legend.position = "right", 
                                       legend.title = element_blank(),
                                       legend.background = element_blank()) )
  
```

###Fig 4D: scatter SRP vs. tRNA
```{r}

selected_phyla <- c("Actinobacteria","Bacteroidetes","Firmicutes","Proteobacteria","Spirochaetes","Tenericutes")

temp <- Zscore_data %>%
  filter(phylum %in% selected_phyla,
         #day=="b",
         DM=="plus"
         )

r_squared = as.character(round(cor(temp$Zscore_total_tRNA_counts, temp$Zscore_rfam_SRP, method="pearson") **2,2))

fig4D <- ggplot(temp, 
       aes(x=Zscore_total_tRNA_counts, y=Zscore_rfam_SRP, color=phylum)) +
  geom_point() +
  annotate(geom="text", label=bquote(~r^2==.(r_squared) ), x=-1, y=2) +
  xlab("tRNA") +
  ylab("SRP RNA") +
  labs(color="Phylum") +
  scale_color_manual(values=c("Actinobacteria"="grey20", "Bacteroidetes"="red", "Firmicutes"="dodgerblue2",
                              "Proteobacteria"="olivedrab3", "Spirochaetes"="mediumorchid1", "Tenericutes"="goldenrod1")) +
  theme(legend.position = "none",
        legend.background = element_blank(),
        legend.key.height = unit(c(0.4),"cm"),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")) +
  coord_equal(xlim = c(-2.1,3.1), ylim = c(-2.1, 2.3))
fig4D
```

###Fig 4E: scatter SRP vs. tRNA
```{r}

selected_phyla <- c("Actinobacteria","Bacteroidetes","Firmicutes","Proteobacteria","Spirochaetes","Tenericutes")

temp <- Zscore_data %>%
  filter(phylum %in% selected_phyla,
         #day=="b",
         DM=="plus"
         )

r_squared = as.character(round(cor(temp$Zscore_total_tRNA_counts, temp$Zscore_rRNA_5S, method="pearson") **2,2))

fig4E <- ggplot(filter(Zscore_data,
              phylum %in% selected_phyla,
              #sample=="Volunteer v03"
              day=="b"
              ), 
       aes(x=Zscore_total_tRNA_counts, y=Zscore_rRNA_5S, color=phylum)) +
  geom_point() +
  annotate(geom="text", label=bquote(~r^2==.(r_squared) ), x=-1, y=2) +
  xlab("tRNA") +
  ylab("5S rRNA") +
  labs(color="Phylum") +
  scale_color_manual(values=c("Actinobacteria"="grey20", "Bacteroidetes"="red", "Firmicutes"="dodgerblue2",
                              "Proteobacteria"="olivedrab3", "Spirochaetes"="mediumorchid1", "Tenericutes"="goldenrod1")) +
  theme(legend.position = "none",
        legend.background = element_blank(),
        legend.key.height = unit(c(0.4),"cm"),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")) +
  coord_equal(xlim = c(-2.1,3.1), ylim = c(-2.1, 2.3))
fig4E
```

##Fig4FG: SRP ends
###Read in SRP basewise new oral data 
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_oral/5_tsv/")

#Read in basewise mapping data for each rfa SRP gene
data1 <- read.csv("./SRP_combineX/TP-CK-2S-CW1_S1_L004L1_bc1_TTCC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="minus", type="rfam_SRP")
data2 <- read.csv("./SRP_combineX/TP-CK-2S-CW1_S1_L004L1_bc2_TCTG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="minus", type="rfam_SRP")
data3 <- read.csv("./SRP_combineX/TP-CK-2S-CW1_S1_L004L1_bc3_TGGT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="minus", type="rfam_SRP")
data4 <- read.csv("./SRP_combineX/TP-CK-2S-CW1_S1_L004L1_bc4_CTGA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="minus", type="rfam_SRP")
data5 <- read.csv("./SRP_combineX/TP-CK-2S-CW1_S1_L004L1_bc5_CCAT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="minus", type="rfam_SRP")
data6 <- read.csv("./SRP_combineX/TP-CK-2S-CW1_S1_L004L1_bc6_CATC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="minus", type="rfam_SRP")
data7 <- read.csv("./SRP_combineX/TP-CK-2S-CW1_S1_L004L1_bc7_GTAG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="minus", type="rfam_SRP")
data8 <- read.csv("./SRP_combineX/TP-CK-2S-CW1_S1_L004L1_bc8_GGTA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="minus", type="rfam_SRP")

data9 <- read.csv("./SRP_combineX/TP-CK-2S-CW2_S2_L004L1_bc1_TTCC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="plus", type="rfam_SRP")
data10 <- read.csv("./SRP_combineX/TP-CK-2S-CW2_S2_L004L1_bc2_TCTG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="plus", type="rfam_SRP")
data11 <- read.csv("./SRP_combineX/TP-CK-2S-CW2_S2_L004L1_bc3_TGGT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="plus", type="rfam_SRP")
data12 <- read.csv("./SRP_combineX/TP-CK-2S-CW2_S2_L004L1_bc4_CTGA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="plus", type="rfam_SRP")
data13 <- read.csv("./SRP_combineX/TP-CK-2S-CW2_S2_L004L1_bc5_CCAT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="plus", type="rfam_SRP")
data14 <- read.csv("./SRP_combineX/TP-CK-2S-CW2_S2_L004L1_bc6_CATC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="plus", type="rfam_SRP")
data15 <- read.csv("./SRP_combineX/TP-CK-2S-CW2_S2_L004L1_bc7_GTAG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="plus", type="rfam_SRP")
data16 <- read.csv("./SRP_combineX/TP-CK-2S-CW2_S2_L004L1_bc8_GGTA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="plus", type="rfam_SRP")

SRP_rfam_basewise <- rbind(data1, data2, data3, data4, 
                           data5, data6, data7, data8,
                           data9, data10,data11,data12,
                           data13,data14,data15,data16) %>%
  separate(gene, sep="\\.", c("name", "X")) %>%
  select(-X)




#Re-read in SRP taxon info
SRP_short <- read.csv("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_SRP/6_sam_counter/SRP_combine/RF00169_lineage.tsv", sep="\t", na.strings = "NA") %>%
  mutate(sub_type="short_SRP")
SRP_long <- read.csv("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_SRP/6_sam_counter/SRP_combine/RF01854_lineage.tsv", sep="\t", na.strings = "NA") %>%
  mutate(sub_type="long_SRP")
rfam_SRP_taxon <- rbind(SRP_short, SRP_long) %>%
  mutate(type="rfam_SRP")

#Re-read in rRNA 5S taxon info
rRNA_5S_taxon <- read.csv("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_SRP/6_sam_counter/rRNA_5S_combine/rRNA_5s_combine_lineage.tsv", sep="\t", na.strings = "NA") %>%
  mutate(type="rRNA_5S", sub_type="rRNA_5S") %>%
  separate(name, sep="_", c("name", "other")) %>%
  select(-other)

taxon=rbind(rRNA_5S_taxon, rfam_SRP_taxon)

```

###Fig4F: CP002122
```{r}

#5' end of CP002122
temp <- SRP_rfam_basewise %>%
  select(sample, day, DM, position, stop, mutation, pileup, name, base) %>%
  group_by(sample, day, DM, name)
temp <- temp %>%
  group_by(sample, day, DM, name) %>%
  filter(position == 8) %>%
  mutate(scale_position = pileup) %>%
  select(sample, day, DM, name, scale_position) %>%
  full_join(temp,., by=c("sample", "day", "DM", "name")) %>%
  mutate(scale_pileup = pileup / scale_position)
fig4F1 <- ggplot(filter(temp, name =="CP002122X", position <=8, DM=="plus"),
       aes(x=position, y=scale_pileup, group=interaction(name, day, sample, DM)))+
  geom_line() +
  geom_text(aes(label=base, y= -0.1, x=position)) +
  scale_y_continuous(breaks=c(1, 0.5, 0), limits=c(-.15, 1)) +
  scale_x_continuous(breaks=c(8,6,4,2)) +
  ylab("Normalized\npileup") +
  xlab("Position (nt)") +
  annotate(geom="text", label= "5'" , x=1, y=.1, parse=F) +
  annotate(geom="text", label="//", x=8, y=1) +
  theme(plot.title = element_text(size=14),
        axis.title.x = element_blank(),
        axis.title.y = element_blank() ) +
  theme(panel.background = element_blank(),
        panel.border = element_blank(),
        axis.line = element_line(),
        plot.margin = unit(c(top=0.3, right=0.3, bottom=0.3, left=0.3), "cm")) 
  #labs(title="Prevotella SRP")
#fig4F1

#3' end of CP002122
temp <- SRP_rfam_basewise %>%
  select(sample, day, DM, position, stop, mutation, pileup, name, base) %>%
  group_by(sample, day, DM, name)
temp <- temp %>%
  group_by(sample, day, DM, name) %>%
  filter(position == 100) %>%
  mutate(scale_position = pileup) %>%
  select(sample, day, DM, name, scale_position) %>%
  full_join(temp,., by=c("sample", "day", "DM", "name")) %>%
  mutate(scale_pileup = pileup / scale_position)

fig4F2 <- ggplot(filter(temp, name =="CP002122X", position >=100, DM=="plus"),
       aes(x=position, y=scale_pileup, group=interaction(name, day, sample, DM)))+
  geom_line() +
  geom_text(aes(label=base, y= -0.1, x=position)) +
  scale_y_continuous(breaks=c(1, 0.5, 0), limits=c(-.15, 1)) +
  scale_x_continuous(breaks=c(100,104,108)) +
  ylab("Normalized 3'\npileup") +
  xlab("Position (nt)") +
  annotate(geom="text", label= "3'" , x=111, y=.1, parse=F) +
  annotate(geom="text", label="//", x=100, y=1) +
  theme(plot.title = element_blank(),
        axis.title.y = element_blank(),
        axis.text.y = element_blank(),
        axis.line.y = element_blank(),
        axis.ticks.y = element_blank(),
        axis.title.x = element_blank()) +
  theme(panel.background = element_blank(),
        panel.border = element_blank(),
        axis.line = element_line(),
        plot.margin = unit(c(top=0.3, right=0.3, bottom=0.3, left=0.3), "cm") ) 
  #labs(title="Prevotella 3' end")
#fig4F2

myplotF1 <- arrangeGrob(fig4F1)
myplotF2 <- arrangeGrob(fig4F2)


setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/EXTRA_FIGURES/")
img = readJPEG("CP002122.jpeg")
g <- rasterGrob(img, interpolate=TRUE)
myplotE3 <- qplot(0:3, 0:3, geom="blank") +
  annotation_custom(g, xmin=-Inf, xmax=Inf, ymin=-Inf, ymax=Inf) +
  theme_void()
myplotE3

lay = rbind(c(1, 2))

fig4F <- grid.arrange(myplotF1, myplotF2, layout_matrix = lay)
fig4F <- arrangeGrob(fig4F, 
          top = textGrob(expression("Prevotella SRP"), x = unit(0.5, "npc")
         , y   = unit(0.5, "npc"), just=c( "top"),
         gp=gpar(col="black", fontsize=12, fontfamily="Arial")),
         bottom = textGrob(expression("Position (nt)"), x = unit(0.5, "npc")
         , y   = unit(0.5, "npc"), just=c("bottom"),
         gp=gpar(col="black", fontsize=12, fontfamily="Arial")),
         left = textGrob(expression("Normalized pileup"), rot=90, x = unit(0.5, "npc")
         , y   = unit(0.5, "npc"), just=c( "center"),
         gp=gpar(col="black", fontsize=12, fontfamily="Arial") )
         )

fig4F <- grid.arrange(fig4F)

# myplotb <- arrangeGrob(fig4B, top = textGrob(expression(bold("b")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")) )

```

###Fig4G: AP011540
```{r}

#5' end of CP002122
temp <- SRP_rfam_basewise %>%
  select(sample, day, DM, position, stop, mutation, pileup, name, base) %>%
  group_by(sample, day, DM, name)
temp <- temp %>%
  group_by(sample, day, DM, name) %>%
  filter(position == 7) %>%
  mutate(scale_position = pileup) %>%
  select(sample, day, DM, name, scale_position) %>%
  full_join(temp,., by=c("sample", "day", "DM", "name")) %>%
  mutate(scale_pileup = pileup / scale_position)

fig4G1 <- ggplot(filter(temp, name =="AP011540", position <=7, DM=="plus"),
       aes(x=position, y=scale_pileup, group=interaction(name, day, sample, DM)))+
  geom_line() +
  geom_text(aes(label=base, y= -0.1, x=position)) +
  scale_y_continuous(breaks=c(1, 0.5, 0), limits=c(-.15, 1)) +
  scale_x_continuous(breaks=c(8,6,4,2)) +
  ylab("Normalized\npileup") +
  xlab("Position (nt)") +
  annotate(geom="text", label= "5'" , x=1, y=.1, parse=F) +
  annotate(geom="text", label="//", x=7, y=1) +
  theme(plot.title = element_text(size=14),
        axis.title.x = element_blank(),
        axis.title.y = element_blank() ) +
  theme(panel.background = element_blank(),
        panel.border = element_blank(),
        axis.line = element_line(),
        plot.margin = unit(c(top=0.3, right=0.3, bottom=0.3, left=0.3), "cm")) 
#fig4G1

#3' end of CP002122
temp <- SRP_rfam_basewise %>%
  select(sample, day, DM, position, stop, mutation, pileup, name, base) %>%
  group_by(sample, day, DM, name)
temp <- temp %>%
  group_by(sample, day, DM, name) %>%
  filter(position == 88) %>%
  mutate(scale_position = pileup) %>%
  select(sample, day, DM, name, scale_position) %>%
  full_join(temp,., by=c("sample", "day", "DM", "name")) %>%
  mutate(scale_pileup = pileup / scale_position)

fig4G2 <- ggplot(filter(temp, name =="AP011540", position >=88, DM=="plus"),
       aes(x=position, y=scale_pileup, group=interaction(name, day, sample, DM)))+
  geom_line() +
  geom_text(aes(label=base, y= -0.1, x=position)) +
  scale_y_continuous(breaks=c(1, 0.5, 0), limits=c(-.15, 1) ) +
  scale_x_continuous(breaks=c(88,92,96)) +
  ylab("Normalized pileup") +
  xlab("Position (nt)") +
  annotate(geom="text", label= "3'" , x=96, y=.1, parse=F) +
  annotate(geom="text", label="//", x=88, y=1) +
  theme(plot.title = element_blank(),
        axis.title.y = element_blank(),
        axis.text.y = element_blank(),
        axis.line.y = element_blank(),
        axis.ticks.y = element_blank(),
        axis.title.x = element_blank()) +
  theme(panel.background = element_blank(),
        panel.border = element_blank(),
        axis.line = element_line(),
        plot.margin = unit(c(top=0.3, right=0.3, bottom=0.3, left=0.3), "cm") ) 
#fig4G2

myplotG1 <- arrangeGrob(fig4G1)
myplotG2 <- arrangeGrob(fig4G2)


setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/EXTRA_FIGURES/")
img = readJPEG("AP011540.jpeg")
g <- rasterGrob(img, interpolate=TRUE)
myplotG3 <- qplot(0:2, 0:2, geom="blank") +
  annotation_custom(g, xmin=-Inf, xmax=Inf, ymin=-Inf, ymax=Inf) +
  theme_void()
#myplotG3

lay = rbind(c(1, 2))

fig4G <- grid.arrange(myplotG1, myplotG2, layout_matrix = lay)
fig4G <- arrangeGrob(fig4G, 
          top = textGrob(expression("Rothia SRP"), x = unit(0.5, "npc")
         , y   = unit(0.5, "npc"), just=c( "top"),
         gp=gpar(col="black", fontsize=12, fontfamily="Arial")),
         bottom = textGrob(expression("Position (nt)"), x = unit(0.5, "npc")
         , y   = unit(0.5, "npc"), just=c("bottom"),
         gp=gpar(col="black", fontsize=12, fontfamily="Arial")),
         left = textGrob(expression("Normalized pileup"), rot=90, x = unit(0.5, "npc")
         , y   = unit(0.5, "npc"), just=c( "center"),
         gp=gpar(col="black", fontsize=12, fontfamily="Arial") )
         )

fig4G <- grid.arrange(fig4G)

```

##Save figure 4
```{r}

myplota <- arrangeGrob(fig4A, top = textGrob(expression(bold("b")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplotb <- arrangeGrob(fig4B, top = textGrob(expression(bold("a")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
#myplotb <- grid.arrange(empty_graph, myplotb, empty_graph, layout_matrix= rbind(c(1,2,2,3)) )


myplotc <- arrangeGrob(fig4C_legend, top = textGrob(expression(bold("")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplotc <- grid.arrange(empty_graph, myplotc, layout_matrix= rbind(c(1), c(2)) )

myplotc2<- arrangeGrob(fig4C, top = textGrob(expression(bold("c")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotd <- arrangeGrob(fig4D, top = textGrob(expression(bold("d")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplote <- arrangeGrob(fig4E, top = textGrob(expression(bold("e")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotf <- arrangeGrob(fig4F, top = textGrob(expression(bold("f")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotg <- arrangeGrob(fig4G, top = textGrob(expression(bold("g")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplotfg <- grid.arrange(myplotf, myplotg, layout_matrix=rbind(c(1,2)))

#Set the figure arrangement matrix
lay <- rbind(c(1,2,4),
             c(3,5,6),
             c(7,7,7) )
#Arrange figures
figure4 <- grid.arrange(myplotb, myplota, myplotc2, myplotc, myplotd, myplote, myplotfg,  layout_matrix = lay)
figure4

setwd("~/4SR/data/20200317_Nature_methods_data/2020 02 24 ecoli figures/source/")
ggsave("../figures/figure4.svg", 
       plot = figure4,
       scale = 1.3,
       width = 8, 
       height = 6)
ggsave("../figures/figure4.png", 
       plot = figure4,
       scale = 1.3,
       width = 8, 
       height = 6)

```
#_



#Figure 5
##Fig 5AB: compositon SRP 5S tRNA
###Read in tRNA composoiton data
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_oral/ORAL_TAXONOMY/")

#tRNA mapped data
data1 <- read.csv("./tongue1_day1_untreated/tongue1_day1_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="minus", type="tRNA")
data2 <- read.csv("./tongue1_day2_untreated/tongue1_day2_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="minus", type="tRNA")
data3 <- read.csv("./tongue2_day1_untreated/tongue2_day1_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="minus", type="tRNA")
data4 <- read.csv("./tongue2_day2_untreated/tongue2_day2_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="minus", type="tRNA")
data5 <- read.csv("./tongue3_day1_untreated/tongue3_day1_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="minus", type="tRNA")
data6 <- read.csv("./tongue3_day2_untreated/tongue3_day2_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="minus", type="tRNA")
data7 <- read.csv("./tongue4_day1_untreated/tongue4_day1_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="minus", type="tRNA")
data8 <- read.csv("./tongue4_day2_untreated/tongue4_day2_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="minus", type="tRNA")

data9 <- read.csv("./tongue1_day1_demethylase/tongue1_day1_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="plus", type="tRNA")
data10 <- read.csv("./tongue1_day2_demethylase/tongue1_day2_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="plus", type="tRNA")
data11<- read.csv("./tongue2_day1_demethylase/tongue2_day1_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="plus", type="tRNA")
data12<- read.csv("./tongue2_day2_demethylase/tongue2_day2_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="plus", type="tRNA")
data13<- read.csv("./tongue3_day1_demethylase/tongue3_day1_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="plus", type="tRNA")
data14<- read.csv("./tongue3_day2_demethylase/tongue3_day2_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="plus", type="tRNA")
data15<- read.csv("./tongue4_day1_demethylase/tongue4_day1_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="plus", type="tRNA")
data16<- read.csv("./tongue4_day2_demethylase/tongue4_day2_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="plus", type="tRNA")

#combine data
tRNA_mapping_data <- bind_rows(data1, data2, data3, data4,
                          data5, data6, data7, data8,
                          data9, data10,data11,data12,
                          data13,data14,data15,data16) %>%
  mutate(class=X) %>%
  select(-X, -X.1) %>%
  pivot_longer(c(-sample, -day, -class, -DM, -type) , names_to = "anticodon", values_to = "counts") %>%
  filter(anticodon != "STP", 
         class!="",
         counts >=0) %>%
  group_by(sample, day, DM, class, type) %>%
  summarise(count = sum(counts))

```

###Read in composition SRP data
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_oral/6_sam_counter/")
#rfam_SRP
data1 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc1_TTCC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="minus", type="rfam_SRP")
data2 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc2_TCTG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="minus", type="rfam_SRP")
data3 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc3_TGGT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="minus", type="rfam_SRP")
data4 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc4_CTGA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="minus", type="rfam_SRP")
data5 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc5_CCAT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="minus", type="rfam_SRP")
data6 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc6_CATC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="minus", type="rfam_SRP")
data7 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc7_GTAG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="minus", type="rfam_SRP")
data8 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc8_GGTA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="minus", type="rfam_SRP")

data9 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc1_TTCC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="plus", type="rfam_SRP")
data10 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc2_TCTG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="plus", type="rfam_SRP")
data11 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc3_TGGT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="plus", type="rfam_SRP")
data12 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc4_CTGA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="plus", type="rfam_SRP")
data13 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc5_CCAT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="plus", type="rfam_SRP")
data14 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc6_CATC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="plus", type="rfam_SRP")
data15 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc7_GTAG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="plus", type="rfam_SRP")
data16 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc8_GGTA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="plus", type="rfam_SRP")

#Re-read in SRP taxon info
SRP_short <- read.csv("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_SRP/6_sam_counter/SRP_combine/RF00169_lineage.tsv", sep="\t", na.strings = "NA") %>%
  mutate(sub_type="short_SRP")
# SRP_long <- read.csv("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_SRP/6_sam_counter/SRP_combine/RF01854_lineage.tsv", sep="\t", na.strings = "NA") %>%
#   mutate(sub_type="long_SRP")
rfam_SRP_taxon <- rbind(SRP_short )%>%#SRP_long) %>%
  mutate(type="rfam_SRP")
#combine data
rfam_SRP_mapping_data <- rbind(data1, data2, data3, data4,
                               data5, data6, data7, data8,
                               data9, data10,data11,data12,
                               data13,data14,data15,data16)

```

###Read in composition by RNA_5S
```{r}

setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_oral/6_sam_counter/")
#5s rRNA
data1 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc1_TTCC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="minus", type="rRNA_5S")
data2 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc2_TCTG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="minus", type="rRNA_5S")
data3 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc3_TGGT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="minus", type="rRNA_5S")
data4 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc4_CTGA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="minus", type="rRNA_5S")
data5 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc5_CCAT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="minus", type="rRNA_5S")
data6 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc6_CATC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="minus", type="rRNA_5S")
data7 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc7_GTAG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="minus", type="rRNA_5S")
data8 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc8_GGTA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="minus", type="rRNA_5S")

data9 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc1_TTCC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="plus", type="rRNA_5S")
data10 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc2_TCTG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="plus", type="rRNA_5S")
data11 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc3_TGGT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="plus", type="rRNA_5S")
data12 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc4_CTGA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="plus", type="rRNA_5S")
data13 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc5_CCAT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="plus", type="rRNA_5S")
data14 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc6_CATC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="plus", type="rRNA_5S")
data15 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc7_GTAG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="plus", type="rRNA_5S")
data16 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc8_GGTA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="plus", type="rRNA_5S")
#Re-read in rRNA 5S taxon info
rRNA_5S_taxon <- read.csv("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_SRP/6_sam_counter/rRNA_5S_combine/rRNA_5s_combine_lineage.tsv", sep="\t", na.strings = "NA") %>%
  mutate(type="rRNA_5S", sub_type="rRNA_5S") %>%
  separate(name, sep="_", c("name", "other")) %>%
  select(-other)
#combine data
rRNA_5S_mapping_data <- rbind(data1, data2, data3, data4,
                      data5, data6, data7, data8,
                      data9, data10,data11,data12,
                      data13,data14,data15,data16)

```

###Read in composition by 16S DNA
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_oral/oral_16S/")

data1 <- read.csv("./ORAL-TAXONOMY-CLASS.txt", header=T, sep="\t") %>%
  pivot_longer(c(-"CLASS"), names_to = "sample", values_to = "sum_count") %>%
  separate(sample, sep="_", c("junk","sample", "day")) %>%
  select(-junk) %>%
  mutate(type="rRNA_16S",
         DM="plus", 
         class=CLASS) %>%
  select(-CLASS) %>%
  group_by(day, sample) %>%
  mutate(sum_count = sum_count / sum(sum_count)) %>%
  group_by()

data1$day <- recode(data1$day, "A"="a", "B"="b")
data1$sample <- recode(data1$sample, "S3"="Volunteer v03", "S4"="Volunteer v04", "S5"="Volunteer v05", "S6"="Volunteer v06")
data1$class <- factor(data1$class)

oral_16S <- data1
```


###Fig 5A with Combine composition
```{r}

mapping_data <- rbind(rRNA_5S_mapping_data, rfam_SRP_mapping_data)
lineage_data <- rRNA_5S_taxon %>%
  full_join(.,rfam_SRP_taxon, by=c("name","type", "kingdom","phylum","class","order","family","genus")) %>%
  select(c("name","type", "kingdom","phylum","class","order","family","genus"))

mapping_data <- mapping_data %>%
  full_join(., lineage_data, by=c("name","type")) %>%
  filter(!is.na(kingdom), !is.na(phylum), !is.na(class), !is.na(order),
         count >0) %>% 
  select(sample, day, DM, class, type, count) %>%
  group_by(sample, day, DM, class, type) %>%
  summarise(count = sum(count)) %>%
  rbind( tRNA_mapping_data) %>%
  filter( !is.na(class)) %>%
  filter(class %nin% c("Mammalia", "Aves")) %>% #SRP only represents prokaryotes
  group_by()


composition_data_common <- mapping_data %>%
  #asdfadsfasdf
  group_by(DM, sample, day, type) %>%
  mutate(sum_count = count / sum(count)) %>%
  group_by() %>%
  bind_rows(oral_16S) %>% #FOLD IN THE 16S data HERE
  mutate(class = as.character(class)) %>%
  mutate(class = ifelse(class=="", "Unassigned", class)) %>% #Get the unnamed class into the other chatagory
  mutate(class = ifelse(sum_count >=0.01, class, "< 1%" )) %>%
  mutate(class = as.factor(class)) %>%
  group_by(class, sample, day, DM, type) %>%
  summarise(sum_count = sum(sum_count) )  %>%#combine all the things newly named "other"
  group_by()  %>%
  filter(sample=="Volunteer v03",
         day=="a",
         DM=="plus") %>%
  mutate(class = fct_reorder(class, sum_count ))


order_temp <- levels(composition_data_common$class)
order_temp <- order_temp[order_temp !="< 1%"]
order_temp <- append("< 1%", order_temp)
composition_data_common$class <- factor(composition_data_common$class, levels = order_temp )

composition_data_common$type <- recode(composition_data_common$type, "tRNA"="tRNA", "rRNA_5S"="5S rRNA", "rfam_SRP"="SRP",
                                       "rRNA_16S" = "16S rRNA")
composition_data_common$type <- factor(composition_data_common$type, levels=c("16S rRNA", "SRP", "5S rRNA", "tRNA"))

fig5A <- ggplot(filter(composition_data_common
                       #type !="16S rRNA\nv4"
                       ), 
                  aes(y=sum_count, x=type, fill=class)) +
  geom_bar(stat="identity", position="stack") +
  coord_flip() +
  theme(legend.position = "right", axis.title.y = element_blank(),
        legend.title = element_blank()) +
  #scale_fill_manual(values=c("#ffa900", "#008f00","#fffc79","#7b1979","#021eaa","#b6192b"), guide=guide_legend(reverse=TRUE) ) +
  scale_fill_manual(values=c("Bacteroidia"="#b6192b", "Actinobacteria"="#021eaa", "Negativicutes"="#7b1979",
                             "Clostridia"="#fffc79", "Betaproteobacteria"="#008f00", "Bacilli"="#ffa900",
                             "Epsilonproteobacteria"="#abc837", "Gammaproteobacteria"="#3c97ac", "Cytophagia"="#ff4c00",
                             "< 1%"="#00acff"),
                    guide=guide_legend(reverse=TRUE)) +
  #scale_fill_discrete(guide=guide_legend(reverse=TRUE)) +
  ylab("Fraction of reads") +
  #scale_y_discrete(position = "right") +
  labs(fill="Class") +
  # facet_grid(rows=vars(day),
  #            cols=vars(sample)) +
  theme(panel.border = element_blank(),
        panel.background = element_blank(),
        panel.spacing = unit(c(0, 0, 0, 0), "cm"),
        legend.background = element_blank(),
        axis.ticks = element_blank(),
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0, size=12),
        strip.placement = "left",
        axis.text.y = element_text(),
        axis.text.x = element_blank(),
        axis.title.x = element_blank(),
        #legend.text = element_text(size=8),
        legend.key.height = unit(c(0.4),"cm"),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm") )
fig5A
```

###Fig 5B: 16S rRNA
```{r}

# fig5B <- ggplot(filter(composition_data_common,
#                        type =="16S rRNA\nv4"
#                        ), 
#                   aes(y=sum_count, x=type, fill=class)) +
#   geom_bar(stat="identity", position="stack") +
#   coord_flip() +
#   theme(legend.position = "right", axis.title.y = element_blank(),
#         legend.title = element_blank()) +
#   #scale_fill_manual(values=c("#ffa900", "#008f00","#fffc79","#7b1979","#021eaa","#b6192b"), guide=guide_legend(reverse=TRUE) ) +
#   scale_fill_manual(values=c("Bacteroidia"="#b6192b", "Actinobacteria"="#021eaa", "Negativicutes"="#7b1979",
#                              "Clostridia"="#fffc79", "Betaproteobacteria"="#008f00", "Bacilli"="#ffa900",
#                              "Epsilonproteobacteria"="#abc837", "Gammaproteobacteria"="#3c97ac", "Cytophagia"="#ff4c00",
#                              "< 1%"="#00acff"),
#                     guide=guide_legend(reverse=TRUE)) +
#   #scale_fill_discrete(guide=guide_legend(reverse=TRUE)) +
#   ylab("Fraction of reads") +
#   #scale_y_discrete(position = "right") +
#   labs(fill="Class") +
#   # facet_grid(rows=vars(day),
#   #            cols=vars(sample)) +
#   theme(panel.border = element_blank(),
#         panel.background = element_blank(),
#         panel.spacing = unit(c(0, 0, 0, 0), "cm"),
#         legend.background = element_blank(),
#         axis.ticks = element_blank(),
#         strip.background = element_blank(),
#         strip.text.y = element_text(angle=0, size=12),
#         strip.placement = "left",
#         axis.text.y = element_text(),
#         axis.text.x = element_blank(),
#         axis.title.x = element_blank(),
#         #legend.text = element_text(size=8),
#         legend.key.height = unit(c(0.4),"cm"),
#         plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm") )
# fig5B

```

##Fig 5B: Daily variation
```{r}


mapping_data <- rbind(rRNA_5S_mapping_data, rfam_SRP_mapping_data)
lineage_data <- rRNA_5S_taxon %>%
  full_join(.,rfam_SRP_taxon, by=c("name","type", "kingdom","phylum","class","order","family","genus")) %>%
  select(c("name","type", "kingdom","phylum","class","order","family","genus"))

mapping_data <- mapping_data %>%
  full_join(., lineage_data, by=c("name","type")) %>%
  filter(!is.na(kingdom), !is.na(phylum), !is.na(class), !is.na(order),
         count >0) %>% 
  select(sample, day, DM, class, type, count) %>%
  group_by(sample, day, DM, class, type) %>%
  summarise(count = sum(count)) %>%
  rbind( tRNA_mapping_data) %>%
  filter( !is.na(class)) %>%
  filter(class %nin% c("Mammalia", "Aves")) %>% #SRP only represents prokaryotes
  group_by()


composition_data_common <- mapping_data %>%
  #asdfadsfasdf
  group_by(DM, sample, day, type) %>%
  mutate(sum_count = count / sum(count)) %>%
  group_by() %>%
  bind_rows(oral_16S) %>% #FOLD IN THE 16S data HERE
  mutate(class = as.character(class)) %>%
  mutate(class = ifelse(class=="", "Unassigned", class)) %>% #Get the unnamed class into the other chatagory
  #mutate(class = ifelse(sum_count >=0.01, class, "< 1%" )) %>%
  filter(class %in% c("Bacteroidia","Actinobacteria","Negativicutes","Clostridia","Betaproteobacteria",
                      "Bacilli","Gammaproteobacteria","Epsilonproteobacteria","Fusobacteriia")) %>%
  mutate(class = as.factor(class)) %>%
  group_by(class, sample, day, DM, type) %>%
  summarise(sum_count = sum(sum_count) )  %>%#combine all the things newly named "other"
  group_by()  %>%
  filter(DM=="plus") %>%
  mutate(class = fct_reorder(class, sum_count ))

composition_data_common$type <- recode(composition_data_common$type, "tRNA"="tRNA", 
                                       "rRNA_5S"="5S rRNA", "rfam_SRP"="SRP", "rRNA_16S" = "16S rRNA")

composition_data_common$type <- factor(composition_data_common$type, levels=c("tRNA","5S rRNA","SRP","16S rRNA"))

composition_data_common$sample <- recode(composition_data_common$sample, "Volunteer v03"="Individual 1", "Volunteer v04"="Individual 2", "Volunteer v05"="Individual 3", "Volunteer v06"="Individual 4")

#write.table(composition_data_common,file='composition_heat_map_data.csv', quote=FALSE, sep=',', col.names = NA)


temp <- composition_data_common %>%
  #filter(sample=="Individual 1") %>%
  mutate(class = fct_reorder(class, sum_count )) %>%
  group_by(sample, DM, class, type)  %>%
  pivot_wider(names_from = day, values_from = sum_count) %>%
  filter(a>0, b>0) %>%
  mutate(difference=b - a) %>%
  select(class, sample, DM, type, difference) #===============//================
# order_temp <- levels( filter(temp, type=="tRNA")$class)
# order_temp <- order_temp[order_temp !="< 1%"]
# order_temp <- append("< 1%", order_temp)
# temp$class <- factor(temp$class, levels = order_temp )

#==================//=========================//==================//=========================
#Tao's choice data from 16S
# temp2 <- data.frame(
#   class = c("Bacteroidia","Actinobacteria","Negativicutes","Clostridia", "Betaproteobacteria",
#             "Bacilli", "Gammaproteobacteria","Epsilonproteobacteria","Fusobacteriia"),
#   v03 = c(0.202252, -0.01258, 0.11271, 0.003999, -0.08938,
#           -0.00272, -0.19767, 0.008131, -0.02011),
#   v04 = c(0.023837, 0.013707, 0.044257, 0.021175, -0.00101, 
#           -0.00788, -0.15358, 0.001525, 0.057162),
#   v05 = c(0.159634, 0.00398, -0.06151, -0.0045, 0.002772, 
#           0.003193, 0.007798, -0.00358, -0.10879),
#   v06 = c(0.135797, -0.00973,  -0.10577, -0.02567, 0.00378,
#           0.011074, 0.161376, -0.00511, -0.16596)) %>%
#   pivot_longer(c(-class),names_to = "sample", values_to = "difference") %>%
#   mutate(type="16S DNA", 
#          DM = "plus")
# temp2$sample <- recode(temp2$sample, "v03"="Individual 1", "v04"="Individual 2", "v05"="Individual 3", "v06"="Individual 4")
#==================//=========================//==================//=========================

#temp <- bind_rows(temp, temp2)

temp$class <- factor(temp$class, levels=rev(c("Bacteroidia","Actinobacteria","Negativicutes","Clostridia", "Betaproteobacteria",
            "Bacilli", "Gammaproteobacteria","Epsilonproteobacteria","Fusobacteriia")) )
#temp$sample <- recode(temp$sample, "Individual 1"="1", "Individual 2"="2", "Individual 3"="3", "Individual 4"="4")
temp$type <- factor(temp$type, levels=c("tRNA","5S rRNA","SRP", "16S rRNA"))


fig5B <- ggplot(filter(temp
              #type != "16S DNA"
              ),
       aes(x=interaction(type), y=class, fill=difference)) +
  geom_tile() +
  scale_fill_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="white", limits=c(-.25,.25), breaks=c(-.25,0,.25))+
  labs(fill="Fraction change") +
  theme(axis.text.x = element_text(angle=90, hjust=1, vjust=0.5, face="bold"),
        axis.text.y = element_text(),
        axis.title = element_blank(),
        axis.ticks = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        plot.background = element_blank()) +
  theme(legend.position = "top",
        strip.background = element_blank(),
        plot.title = element_text(face="bold", size=12),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")) +
  #coord_fixed(ratio=0.5) +
  #coord_equal()+
  #labs(title="Individual 1") +
  facet_grid(cols=vars(sample))
fig5B

```

##Fig 5C: piechart stool
```{r}
#How mnay reads
#How many MAPPED to bacterial tRNA
#How many mapped to SRP_combine
#How many map to 5S combine
temp <- data.frame(RNA_type = c("tRNA", "SRP", "5S rRNA", "unmapped"),
                   average  = c(62.9, 0.4, 0.8, 35.9),
                   sdev     = c(14.7, 0.01, 0.2, 14.5))

#transform value to be compatable with piechart - idk how this works, I copied it from stack exchange
temp <- temp %>% 
  arrange(desc(RNA_type)) %>%
  mutate(ypos = cumsum(average)- 0.5*average )

fig5C <- ggplot(temp, aes(x="", y=average, fill=RNA_type)) +
  geom_bar(stat="identity", width=1, color="white") +
  coord_polar("y", start=-2) +
  geom_text_repel(aes(y=ypos, label=paste0(RNA_type, "\n", average, " ±", sdev, "%") ), box.padding = 1)+
  theme_void() +
  # scale_fill_manual(values = c("tRNA"="#99CCFF",
  #                               "rRNA"="#FFCC33",
  #                              "SRP"="#99CC99",
  #                               "unmapped"="#CC99CC"))+
  labs(fill="RNA type") +
  guides(fill = guide_legend(reverse=TRUE)) +
  theme(legend.position = "None",
        plot.title = element_text(hjust = 0.5),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")) +
  labs(title="Stool reads")

fig5C

```

##Fig 5D: Stool composition
###Composititoin by CLASS - tRNA
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/stool_biome/tRNA/stool/")

#Read in data by anticodon
data1 <- read.csv("./s9_untreated_taxonomy/s9_untreated_class_anticodon_counts.tsv", sep="\t", header=T) %>%
  mutate(sample="Stool 1", DM="minus", type="tRNA", day="c") 
data2 <- read.csv("./s10_untreated_taxonomy/s10_untreated_class_anticodon_counts.tsv", sep="\t", header=T) %>%
  mutate(sample="Stool 2", DM="minus", type="tRNA", day="c") 
data3 <- read.csv("./s9_demethylase_taxonomy/s9_demethylase_class_anticodon_counts.tsv", sep="\t", header=T) %>%
  mutate(sample="Stool 1", DM="plus", type="tRNA", day="c") 
data4 <- read.csv("./s10_demethylase_taxonomy/s10_demethylase_class_anticodon_counts.tsv", sep="\t", header=T) %>%
  mutate(sample="Stool 2", DM="plus", type="tRNA", day="c")

#combine data
tRNA_mapping_data <- bind_rows(data1, data2, data3, data4) %>%
  mutate(class=X) %>%
  select(-X, -X.1) %>%
  pivot_longer(c(-sample, -day, -class, -DM, -type) , names_to = "anticodon", values_to = "counts") %>%
  filter(anticodon != "STP", 
         class!="",
         counts >=0) %>%
  group_by(sample, day, DM, class, type) %>%
  summarise(count = sum(counts))

```

###Composition by CLASS - SRP
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/stool_biome/6_sam_counter/")

data1 <- read.csv("./SRP_combine/TP-CK-3S-CW2_S2_L001L1_bc9_GACT.tsv", header=T, sep="\t") %>%
  mutate(sample="Stool 1", DM="minus", type="rfam_SRP", day="c")
data2 <- read.csv("./SRP_combine/TP-CK-3S-CW2_S2_L001L1_bc10_ACCA.tsv", header=T, sep="\t") %>%
  mutate(sample="Stool 2", DM="minus", type="rfam_SRP", day="c")
data3 <- read.csv("./SRP_combine/TP-CK-3S-CW3_S3_L001L1_bc9_GACT.tsv", header=T, sep="\t") %>%
  mutate(sample="Stool 1", DM="plus", type="rfam_SRP", day="c")
data4 <- read.csv("./SRP_combine/TP-CK-3S-CW3_S3_L001L1_bc10_ACCA.tsv", header=T, sep="\t") %>%
  mutate(sample="Stool 2", DM="plus", type="rfam_SRP", day="c")

rfam_SRP_mapping_data <- rbind(data1, data2, data3, data4)


```

###Composition strip by CLASS - rRNA
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/stool_biome/6_sam_counter/")

data1 <- read.csv("./rRNA_5S_combine/TP-CK-3S-CW2_S2_L001L1_bc9_GACT.tsv", header=T, sep="\t") %>%
  mutate(sample="Stool 1", DM="minus", type="rRNA_5S", day="c")
data2 <- read.csv("./rRNA_5S_combine/TP-CK-3S-CW2_S2_L001L1_bc10_ACCA.tsv", header=T, sep="\t") %>%
  mutate(sample="Stool 2", DM="minus", type="rRNA_5S", day="c")
data3 <- read.csv("./rRNA_5S_combine/TP-CK-3S-CW3_S3_L001L1_bc9_GACT.tsv", header=T, sep="\t") %>%
  mutate(sample="Stool 1", DM="plus", type="rRNA_5S", day="c")
data4 <- read.csv("./rRNA_5S_combine/TP-CK-3S-CW3_S3_L001L1_bc10_ACCA.tsv", header=T, sep="\t") %>%
  mutate(sample="Stool 2", DM="plus", type="rRNA_5S", day="c")

rRNA_5S_mapping_data <- rbind(data1, data2, data3, data4) 

```

###Composition 16S
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/stool_biome/STOOL_TRNA_16S_TAXONOMY/")
#This is the stupidest data. All the data is in each of the 4 sheets. So dumb
#somebody better give me real data later. This makes me angry. 
#Dm field doesn't really apply here
data1 <- read_excel("./s9_untreated_trna_rrna_comparison.xlsx", sheet="class") %>%
  mutate(count = Stool_S1_B1_16S_reads) %>%
         # count2 = Stool_S1_B2_16S_reads,
         # count3 = Stool_S1_B3_16S_reads) %>%
  select(class, count) %>%
  mutate(sample="Stool 1", DM="plus", type="rRNA_16S", day="c")


data2 <- read_excel("./s10_untreated_trna_rrna_comparison.xlsx", sheet="class") %>%
  mutate(count = Stool_S2_B1_16S_reads) %>%
  select(class, count) %>%
  mutate(sample="Stool 2", DM="plus", type="rRNA_16S", day="c")

#I don't think these are actually stool samples
# data3 <- read_excel("./s9_demethylase_trna_rrna_comparison.xlsx", sheet="class") %>%
#   mutate(count = Stool_S2_B1_16S_reads) %>%
#   select(class, count) %>%
#   mutate(sample="Stool 1", DM="plus", type="rRNA_16S", day="c")
# data4 <- read_excel("./s10_demethylase_trna_rrna_comparison.xlsx", sheet="class") %>%
#   mutate(count = Stool_S2_B2_16S_reads) %>%
#   select(class, count) %>%
#   mutate(sample="Stool 2", DM="plus", type="rRNA_16S", day="c")

rRNA_16S_mapping_data <- rbind(data1, data2) %>%
  filter(!is.na(class),
         count >=0)

```

###Combine composition
```{r}

mapping_data <- rbind(rRNA_5S_mapping_data, rfam_SRP_mapping_data)
lineage_data <- rRNA_5S_taxon %>%
  full_join(.,rfam_SRP_taxon, by=c("name","type", "kingdom","phylum","class","order","family","genus")) %>%
  select(c("name","type", "kingdom","phylum","class","order","family","genus"))

mapping_data <- mapping_data %>%
  full_join(., lineage_data, by=c("name","type")) %>%
  filter(!is.na(kingdom), !is.na(phylum), !is.na(class), !is.na(order),
         count >0) %>% 
  select(sample, day, DM, class, type, count) %>%
  group_by(sample, day, DM, class, type) %>%
  summarise(count = sum(count)) %>%
  rbind(tRNA_mapping_data) %>%
  bind_rows(rRNA_16S_mapping_data) %>%
  filter(!is.na(class)) %>%
  filter(class %nin% c("Mammalia", "Aves")) %>% #SRP only represents prokaryotes
  group_by()


composition_data_common <- mapping_data %>%
  #asdfadsfasdf
  group_by(DM, sample, day, type) %>%
  mutate(sum_count = count / sum(count)) %>%
  group_by() %>%
  mutate(class = as.character(class)) %>%
  mutate(class = ifelse(class=="", "Unassigned", class)) %>% #Get the unnamed class into the other chatagory
  mutate(class = ifelse(sum_count >=0.01, class, "< 1%" )) %>%
  mutate(class =  ifelse(class != "Verrucomicrobiae", class, "< 1%" )) %>% #GROSS GROSS GROSS GROSS GROSS
  mutate(class = as.factor(class)) %>%
  group_by(class, sample, day, DM, type) %>%
  summarise(sum_count = sum(sum_count) )  %>%#combine all the things newly named "other"
  group_by()  %>%
  filter(#sample=="Individual 5",
         day=="c",
         DM=="plus"
         ) %>%
  mutate(class = fct_reorder(class, sum_count ))

order_temp <- levels(composition_data_common$class)
order_temp <- order_temp[order_temp !="< 1%"]
order_temp <- append("< 1%", order_temp)
composition_data_common$class <- factor(composition_data_common$class, levels = order_temp )

composition_data_common$type <- recode(composition_data_common$type, "tRNA"="tRNA", "rRNA_5S"="5S rRNA", 
                                       "rfam_SRP"="SRP", "rRNA_16S"="16S rRNA")
composition_data_common$type <- factor(composition_data_common$type, levels=c("16S rRNA", "SRP", "5S rRNA", "tRNA"))



fig5D <- ggplot(filter(composition_data_common
                         #sample=="Stool 1"
                         #type =="16S rRNA"
                         ), 
                  aes(y=sum_count, x=type, fill=class)) +
  geom_bar(stat="identity", position="stack") +
  coord_flip() +
  theme(legend.position = "right", axis.title.y = element_blank(),
        legend.title = element_blank()) +
  # scale_fill_manual(values=c("Clostridia"="#fffc79", #Sensible colors
  #                            "Bacteroidia"="#b6192b",
  #                            "Nitrospira"="#0055d4",
  #                            "Bacilli"="#ffa900",
  #                            "Gammaproteobacteria"="#3c97ac",
  #                            "Erysipelotrichia"="#e9afaf",
  #                            "Magnoliopsida"="#aaaaff",
  #                            "Betaproteobacteria"="#008f00",
  #                            "Deltaproteobacteria"="#aaaaad",
  #                            "Negativicutes"="#7b1979",
  #                            "Alphaproteobacteria"="#dd55ff",
  #                            "Coriobacteriia"="#ccffaa",
  #                            "Actinobacteria"="#916f7c",
  #                            "Mollicutes"="#b6192b",
  #                            "< 1%"="#00acff"),       guide=guide_legend(reverse=TRUE)) +
    scale_fill_manual(values=c("Clostridia"="#b6192b",
                             "Bacteroidia"="#021eaa",
                             "Nitrospira"="#7b1979",
                             "Bacilli"="#fffc79",
                             "Erysipelotrichia"="#916f7c",
                             "Gammaproteobacteria"="#008f00",
                             "Betaproteobacteria"="#ffa900",
                             "Negativicutes"="#abc837",
                             "Deltaproteobacteria"="#3c97ac",
                             "Erysipelotrichia"="#e9afaf",
                             "Magnoliopsida"="#aaaaff",
                             "Alphaproteobacteria"="#dd55ff",
                             "Coriobacteriia"="#ccffaa",
                             "Actinobacteria"="#ff4c00",
                             "Mollicutes"="#b6192b",
                             "Verrucomicrobiae"="grey70",
                             "< 1%"="#00acff"),       guide=guide_legend(reverse=TRUE)) +
  ylab("Fraction of reads") +
  #scale_y_discrete(position = "right") +
  labs(fill="Class") +
  facet_grid(#rows=vars(DM),
             rows=vars(sample)) +
  theme(panel.border = element_blank(),
        panel.background = element_blank(),
        #panel.spacing = unit(c(0, 0, 0, 0), "cm"), #FUCKs things up when there is a facet grid
        legend.background = element_blank(),
        axis.ticks = element_blank(),
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0, size=12),
        strip.placement = "left",
        axis.text.y = element_text(),
        axis.text.x = element_blank(),
        axis.title.x = element_blank(),
        legend.text = element_text(size=8),
        legend.key.height = unit(c(0.4),"cm"),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm") )
fig5D


```

##Save figure 5
```{r}
myplota <- arrangeGrob(fig5A, left = textGrob(expression(bold("a")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotb <- arrangeGrob(fig5B, left = textGrob(expression(bold("b")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotc <- arrangeGrob(fig5C, left = textGrob(expression(bold("c")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotd <- arrangeGrob(fig5D, left = textGrob(expression(bold("d")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))


#Set the figure arrangement matrix
lay <- rbind(c(1,3),
             c(1,3),
             c(2,4),
             c(2,4),
             c(2,4))
#Arrange figures
figure5 <- grid.arrange(myplota, myplotb, myplotc, myplotd,  layout_matrix = lay)
figure5

setwd("~/4SR/data/20200317_Nature_methods_data/2020 02 24 ecoli figures/source/")
ggsave("../figures/figure5.svg", 
       plot = figure5,
       scale = 1.3,
       width = 8.5, 
       height = 6)
ggsave("../figures/figure5.png", 
       plot = figure5,
       scale = 1.3,
       width = 8.5, 
       height = 6)

```

#_
#Figure S5
##Read in data
###Read in tRNA composoiton data
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_oral/ORAL_TAXONOMY/")

#tRNA mapped data
data1 <- read.csv("./tongue1_day1_untreated/tongue1_day1_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="minus", type="tRNA")
data2 <- read.csv("./tongue1_day2_untreated/tongue1_day2_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="minus", type="tRNA")
data3 <- read.csv("./tongue2_day1_untreated/tongue2_day1_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="minus", type="tRNA")
data4 <- read.csv("./tongue2_day2_untreated/tongue2_day2_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="minus", type="tRNA")
data5 <- read.csv("./tongue3_day1_untreated/tongue3_day1_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="minus", type="tRNA")
data6 <- read.csv("./tongue3_day2_untreated/tongue3_day2_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="minus", type="tRNA")
data7 <- read.csv("./tongue4_day1_untreated/tongue4_day1_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="minus", type="tRNA")
data8 <- read.csv("./tongue4_day2_untreated/tongue4_day2_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="minus", type="tRNA")

data9 <- read.csv("./tongue1_day1_demethylase/tongue1_day1_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="plus", type="tRNA")
data10 <- read.csv("./tongue1_day2_demethylase/tongue1_day2_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="plus", type="tRNA")
data11<- read.csv("./tongue2_day1_demethylase/tongue2_day1_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="plus", type="tRNA")
data12<- read.csv("./tongue2_day2_demethylase/tongue2_day2_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="plus", type="tRNA")
data13<- read.csv("./tongue3_day1_demethylase/tongue3_day1_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="plus", type="tRNA")
data14<- read.csv("./tongue3_day2_demethylase/tongue3_day2_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="plus", type="tRNA")
data15<- read.csv("./tongue4_day1_demethylase/tongue4_day1_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="plus", type="tRNA")
data16<- read.csv("./tongue4_day2_demethylase/tongue4_day2_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="plus", type="tRNA")

#combine data
tRNA_mapping_data <- bind_rows(data1, data2, data3, data4,
                          data5, data6, data7, data8,
                          data9, data10,data11,data12,
                          data13,data14,data15,data16) %>%
  mutate(class=X) %>%
  select(-X, -X.1) %>%
  pivot_longer(c(-sample, -day, -class, -DM, -type) , names_to = "anticodon", values_to = "counts") %>%
  filter(anticodon != "STP", 
         class!="",
         counts >=0) %>%
  group_by(sample, day, DM, class, type) %>%
  summarise(count = sum(counts))

```

###Read in composition SRP data
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_oral/6_sam_counter/")
#rfam_SRP
data1 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc1_TTCC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="minus", type="rfam_SRP")
data2 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc2_TCTG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="minus", type="rfam_SRP")
data3 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc3_TGGT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="minus", type="rfam_SRP")
data4 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc4_CTGA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="minus", type="rfam_SRP")
data5 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc5_CCAT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="minus", type="rfam_SRP")
data6 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc6_CATC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="minus", type="rfam_SRP")
data7 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc7_GTAG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="minus", type="rfam_SRP")
data8 <- read.csv("./SRP_combine/TP-CK-2S-CW1_S1_L004L1_bc8_GGTA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="minus", type="rfam_SRP")

data9 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc1_TTCC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="plus", type="rfam_SRP")
data10 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc2_TCTG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="plus", type="rfam_SRP")
data11 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc3_TGGT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="plus", type="rfam_SRP")
data12 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc4_CTGA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="plus", type="rfam_SRP")
data13 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc5_CCAT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="plus", type="rfam_SRP")
data14 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc6_CATC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="plus", type="rfam_SRP")
data15 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc7_GTAG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="plus", type="rfam_SRP")
data16 <- read.csv("./SRP_combine/TP-CK-2S-CW2_S2_L004L1_bc8_GGTA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="plus", type="rfam_SRP")

#Re-read in SRP taxon info
SRP_short <- read.csv("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_SRP/6_sam_counter/SRP_combine/RF00169_lineage.tsv", sep="\t", na.strings = "NA") %>%
  mutate(sub_type="short_SRP")
# SRP_long <- read.csv("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_SRP/6_sam_counter/SRP_combine/RF01854_lineage.tsv", sep="\t", na.strings = "NA") %>%
#   mutate(sub_type="long_SRP")
rfam_SRP_taxon <- rbind(SRP_short )%>%#SRP_long) %>%
  mutate(type="rfam_SRP")
#combine data
rfam_SRP_mapping_data <- rbind(data1, data2, data3, data4,
                               data5, data6, data7, data8,
                               data9, data10,data11,data12,
                               data13,data14,data15,data16)

```

###Read in composition by RNA_5S
```{r}

setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_oral/6_sam_counter/")
#5s rRNA
data1 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc1_TTCC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="minus", type="rRNA_5S")
data2 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc2_TCTG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="minus", type="rRNA_5S")
data3 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc3_TGGT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="minus", type="rRNA_5S")
data4 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc4_CTGA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="minus", type="rRNA_5S")
data5 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc5_CCAT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="minus", type="rRNA_5S")
data6 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc6_CATC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="minus", type="rRNA_5S")
data7 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc7_GTAG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="minus", type="rRNA_5S")
data8 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW1_S1_L004L1_bc8_GGTA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="minus", type="rRNA_5S")

data9 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc1_TTCC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="plus", type="rRNA_5S")
data10 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc2_TCTG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="plus", type="rRNA_5S")
data11 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc3_TGGT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="plus", type="rRNA_5S")
data12 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc4_CTGA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="plus", type="rRNA_5S")
data13 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc5_CCAT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="plus", type="rRNA_5S")
data14 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc6_CATC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="plus", type="rRNA_5S")
data15 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc7_GTAG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="plus", type="rRNA_5S")
data16 <- read.csv("./rRNA_5S_combine/TP-CK-2S-CW2_S2_L004L1_bc8_GGTA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="plus", type="rRNA_5S")
#Re-read in rRNA 5S taxon info
rRNA_5S_taxon <- read.csv("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_SRP/6_sam_counter/rRNA_5S_combine/rRNA_5s_combine_lineage.tsv", sep="\t", na.strings = "NA") %>%
  mutate(type="rRNA_5S", sub_type="rRNA_5S") %>%
  separate(name, sep="_", c("name", "other")) %>%
  select(-other)
#combine data
rRNA_5S_mapping_data <- rbind(data1, data2, data3, data4,
                      data5, data6, data7, data8,
                      data9, data10,data11,data12,
                      data13,data14,data15,data16)

```

###Read in composition by 16S DNA
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_oral/oral_16S/")

data1 <- read.csv("./ORAL-TAXONOMY-CLASS.txt", header=T, sep="\t") %>%
  pivot_longer(c(-"CLASS"), names_to = "sample", values_to = "sum_count") %>%
  separate(sample, sep="_", c("junk","sample", "day")) %>%
  select(-junk) %>%
  mutate(type="rRNA_16S",
         DM="plus", 
         class=CLASS) %>%
  select(-CLASS) %>%
  group_by(day, sample) %>%
  mutate(sum_count = sum_count / sum(sum_count)) %>%
  group_by()

data1$day <- recode(data1$day, "A"="a", "B"="b")
data1$sample <- recode(data1$sample, "S3"="Volunteer v03", "S4"="Volunteer v04", "S5"="Volunteer v05", "S6"="Volunteer v06")
data1$class <- factor(data1$class)

oral_16S <- data1
```
##Fig S5A: taxonomy for other individuals
```{r}

mapping_data <- rbind(rRNA_5S_mapping_data, rfam_SRP_mapping_data)
lineage_data <- rRNA_5S_taxon %>%
  full_join(.,rfam_SRP_taxon, by=c("name","type", "kingdom","phylum","class","order","family","genus")) %>%
  select(c("name","type", "kingdom","phylum","class","order","family","genus"))

mapping_data <- mapping_data %>%
  full_join(., lineage_data, by=c("name","type")) %>%
  filter(!is.na(kingdom), !is.na(phylum), !is.na(class), !is.na(order),
         count >0) %>% 
  select(sample, day, DM, class, type, count) %>%
  group_by(sample, day, DM, class, type) %>%
  summarise(count = sum(count)) %>%
  rbind( tRNA_mapping_data) %>%
  filter( !is.na(class)) %>%
  filter(class %nin% c("Mammalia", "Aves")) %>% #SRP only represents prokaryotes
  group_by()


composition_data_common <- mapping_data %>%
  #asdfadsfasdf
  group_by(DM, sample, day, type) %>%
  mutate(sum_count = count / sum(count)) %>%
  group_by() %>%
  bind_rows(oral_16S) %>% #FOLD IN THE 16S data HERE
  mutate(class = as.character(class)) %>%
  mutate(class = ifelse(class=="", "Unassigned", class)) %>% #Get the unnamed class into the other chatagory
  mutate(class = ifelse(sum_count >=0.01, class, "< 1%" )) %>%
  mutate(class = as.factor(class)) %>%
  group_by(class, sample, day, DM, type) %>%
  summarise(sum_count = sum(sum_count) )  %>%#combine all the things newly named "other"
  group_by()  %>%
  filter(#sample!="Volunteer v03",
         day=="a",
         DM=="plus") %>%
  mutate(class = fct_reorder(class, sum_count ))


order_temp <- levels(composition_data_common$class)
order_temp <- order_temp[order_temp !="< 1%"]
order_temp <- append("< 1%", order_temp)
composition_data_common$class <- factor(composition_data_common$class, levels = order_temp )

composition_data_common$type <- recode(composition_data_common$type, "tRNA"="tRNA", "rRNA_5S"="5S rRNA", "rfam_SRP"="SRP",
                                       "rRNA_16S" = "16S rRNA")
composition_data_common$type <- factor(composition_data_common$type, levels=c("16S rRNA", "SRP", "5S rRNA", "tRNA"))
composition_data_common$sample <- recode(composition_data_common$sample,"Volunteer v03"="1", "Volunteer v04"="2", "Volunteer v05"="3", "Volunteer v06"="4")

figS5A <- ggplot(filter(composition_data_common
                       #type !="16S rRNA\nv4"
                       ), 
                  aes(y=sum_count, x=type, fill=class)) +
  geom_bar(stat="identity", position="stack") +
  coord_flip() +
  theme(legend.position = "right", axis.title.y = element_blank(),
        legend.title = element_blank()) +
  scale_fill_manual(values=c("Bacteroidia"="#b6192b", "Actinobacteria"="#021eaa", "Negativicutes"="#7b1979",
                             "Clostridia"="#fffc79", "Betaproteobacteria"="#008f00", "Bacilli"="#ffa900",
                             "Epsilonproteobacteria"="#abc837", "Gammaproteobacteria"="#3c97ac", "Cytophagia"="#ff4c00",
                             "< 1%"="#00acff", 
                             "Fusobacteriia"="skyblue", "Acidimicrobiia"="pink", 
                             "Erysipelotrichia"="goldenrod", "Flavobacteriia"="grey60"),
                    guide=guide_legend(reverse=TRUE)) +
  #scale_fill_discrete(guide=guide_legend(reverse=TRUE)) +
  ylab("Fraction of reads") +
  #scale_y_discrete(position = "right") +
  labs(fill="Class") +
  facet_grid(cols=vars(sample)) +
  #            cols=vars(sample)) +
  theme(panel.border = element_blank(),
        panel.background = element_blank(),
        #panel.spacing = unit(c(0, 0, 0, 0), "cm"),
        legend.background = element_blank(),
        axis.ticks = element_blank(),
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0, size=12),
        strip.placement = "left",
        axis.text.y = element_text(),
        axis.text.x = element_blank(),
        axis.title.x = element_blank(),
        #legend.text = element_text(size=8),
        legend.key.height = unit(c(0.4),"cm"),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm") )
figS5A

```

##Fig S5B: anticodon taxonomy
###Read in tRNA composoiton data ORAL
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_oral/ORAL_TAXONOMY/")

#tRNA mapped data
data1 <- read.csv("./tongue1_day1_untreated/tongue1_day1_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="minus", type="tRNA")
data2 <- read.csv("./tongue1_day2_untreated/tongue1_day2_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="minus", type="tRNA")
data3 <- read.csv("./tongue2_day1_untreated/tongue2_day1_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="minus", type="tRNA")
data4 <- read.csv("./tongue2_day2_untreated/tongue2_day2_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="minus", type="tRNA")
data5 <- read.csv("./tongue3_day1_untreated/tongue3_day1_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="minus", type="tRNA")
data6 <- read.csv("./tongue3_day2_untreated/tongue3_day2_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="minus", type="tRNA")
data7 <- read.csv("./tongue4_day1_untreated/tongue4_day1_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="minus", type="tRNA")
data8 <- read.csv("./tongue4_day2_untreated/tongue4_day2_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="minus", type="tRNA")

data9 <- read.csv("./tongue1_day1_demethylase/tongue1_day1_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="plus", type="tRNA")
data10 <- read.csv("./tongue1_day2_demethylase/tongue1_day2_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="plus", type="tRNA")
data11<- read.csv("./tongue2_day1_demethylase/tongue2_day1_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="plus", type="tRNA")
data12<- read.csv("./tongue2_day2_demethylase/tongue2_day2_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="plus", type="tRNA")
data13<- read.csv("./tongue3_day1_demethylase/tongue3_day1_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="plus", type="tRNA")
data14<- read.csv("./tongue3_day2_demethylase/tongue3_day2_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="plus", type="tRNA")
data15<- read.csv("./tongue4_day1_demethylase/tongue4_day1_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="plus", type="tRNA")
data16<- read.csv("./tongue4_day2_demethylase/tongue4_day2_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="plus", type="tRNA")

#combine data
tRNA_mapping_data <- bind_rows(data1, data2, data3, data4,
                          data5, data6, data7, data8,
                          data9, data10,data11,data12,
                          data13,data14,data15,data16) %>%
  mutate(class=X) %>%
  select(-X, -X.1) %>%
  pivot_longer(c(-sample, -day, -class, -DM, -type) , names_to = "anticodon", values_to = "counts") %>%
  filter(anticodon != "STP", 
         class!="",
         counts >=0) 
```

###Read in tRNA composition STOOL
```{r}
# setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/stool_biome/tRNA/stool/")
# 
# #Read in data by anticodon
# data1 <- read.csv("./s9_untreated_taxonomy/s9_untreated_class_anticodon_counts.tsv", sep="\t", header=T) %>%
#   mutate(sample="Stool 1", DM="minus", type="tRNA", day="c") 
# data2 <- read.csv("./s10_untreated_taxonomy/s10_untreated_class_anticodon_counts.tsv", sep="\t", header=T) %>%
#   mutate(sample="Stool 2", DM="minus", type="tRNA", day="c") 
# data3 <- read.csv("./s9_demethylase_taxonomy/s9_demethylase_class_anticodon_counts.tsv", sep="\t", header=T) %>%
#   mutate(sample="Stool 1", DM="plus", type="tRNA", day="c") 
# data4 <- read.csv("./s10_demethylase_taxonomy/s10_demethylase_class_anticodon_counts.tsv", sep="\t", header=T) %>%
#   mutate(sample="Stool 2", DM="plus", type="tRNA", day="c")
# 
# #combine data
# tRNA_mapping_data <- bind_rows(data1, data2, data3, data4) %>%
#   mutate(class=X) %>%
#   select(-X, -X.1) %>%
#   pivot_longer(c(-sample, -day, -class, -DM, -type) , names_to = "anticodon", values_to = "counts") %>%
#   filter(anticodon != "STP", 
#          class!="",
#          counts >=0) 
```

###Make a graph
```{r}

temp <- tRNA_mapping_data %>%
  filter( !is.na(class)) %>%
  filter(class %nin% c("Mammalia", "Aves")) %>% #SRP only represents prokaryotes
  filter(anticodon %in% c("CTT","TTT")) %>%
  group_by(sample, day, DM, anticodon) %>%
  mutate(sum_count = counts / sum(counts)) %>%
  group_by() %>%
  mutate(class = as.character(class)) %>%
  mutate(class = ifelse(class=="", "Unassigned", class)) %>% #Get the unnamed class into the other chatagory
  mutate(class = ifelse(sum_count >=0.02, class, "< 2%" )) %>%
  mutate(class = as.factor(class)) %>%
  group_by(class, sample, day, DM, type, anticodon) %>%
  summarise(sum_count = sum(sum_count) )  %>%#combine all the things newly named "other"
  group_by()  %>%
  mutate(class = fct_reorder(class, sum_count )) %>%
  filter(DM=="plus", 
         day=="a"
         )
  


order_temp <- levels(temp$class)
order_temp <- order_temp[order_temp !="< 2%"]
order_temp <- append("< 2%", order_temp)
#temp$class <- factor(temp$class, levels = order_temp )
#DO it manually to match S5A
temp$class <- factor(temp$class, levels=rev(c("Bacteroidia","Actinobacteria","Negativicutes","Gammaproteobacteria","Clostridia",
                                          "Betaproteobacteria","Bacilli","Fusobacteriia","Epsilonproteobacteria","< 2%")))

temp$type <- recode(temp$type, "tRNA"="tRNA", "rRNA_5S"="5S rRNA", "rfam_SRP"="SRP",
                                       "rRNA_16S" = "16S rRNA")
temp$type <- factor(temp$type, levels=c("16S rRNA", "SRP", "5S rRNA", "tRNA"))
temp$sample <- recode(temp$sample,"Volunteer v03"="1", "Volunteer v04"="2", "Volunteer v05"="3", "Volunteer v06"="4")



figS5B <- ggplot(filter(temp
                       #type !="16S rRNA\nv4"
                       ), 
                  aes(y=sum_count, x=anticodon, fill=class)) +
  geom_bar(stat="identity", position="stack") +
  coord_flip() +
  theme(legend.position = "right", axis.title.y = element_blank(),
        legend.title = element_blank()) +
  scale_fill_manual(values=c("Bacteroidia"="#b6192b", "Actinobacteria"="#021eaa", "Negativicutes"="#7b1979",
                             "Clostridia"="#fffc79", "Betaproteobacteria"="#008f00", "Bacilli"="#ffa900",
                             "Epsilonproteobacteria"="#abc837", "Gammaproteobacteria"="#3c97ac", "Cytophagia"="#ff4c00",
                             "< 2%"="#00acff",
                             "Fusobacteriia"="skyblue", "Acidimicrobiia"="pink", "Acidobacteriia"="black",
                             "Alphaproteobacteria"="black", "Betaproteobacteria"="black", "Cytophagia"="black", 
                             "Chitinophagia" = "black", "Methanobacteria"="orangered", "Mollicutes"="lightblue1",
                             "Erysipelotrichia"="goldenrod", "Flavobacteriia"="grey60", "Sphingobacteriia"="black"),
                    guide=guide_legend(reverse=TRUE)) +
  ylab("Fraction of reads") +
  #scale_y_discrete(position = "right") +
  labs(fill="Class") +
  facet_grid(cols=vars(sample)
             #rows=vars(day)
             ) +
  theme(panel.border = element_blank(),
        panel.background = element_blank(),
        #panel.spacing = unit(c(0, 0, 0, 0), "cm"),
        legend.background = element_blank(),
        axis.ticks = element_blank(),
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0, size=12),
        strip.placement = "left",
        axis.text.y = element_text(),
        axis.text.x = element_blank(),
        axis.title.x = element_blank(),
        #legend.text = element_text(size=8),
        legend.key.height = unit(c(0.4),"cm"),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm") )
figS5B

```


##Save figure S5
```{r}

myplota <- arrangeGrob(figS5A, top = textGrob(expression(bold("a")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplotb <- arrangeGrob(figS5B, top = textGrob(expression(bold("b")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))


#Set the figure arrangement matrix
lay <- rbind(c(1),
             c(2))
#Arrange figures
figureS5 <- grid.arrange(myplota, myplotb,  layout_matrix = lay)
figureS5

setwd("~/4SR/data/20200317_Nature_methods_data/2020 02 24 ecoli figures/source/")
ggsave("../figures/figureS5.svg", 
       plot = figureS5,
       scale = 1.3,
       width = 8, 
       height = 5)
ggsave("../figures/figureS5.png", 
       plot = figureS5,
       scale = 1.3,
       width = 8, 
       height = 5)

```


#_
#Figure 6
##Read in Rothia basewise data
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_oral/5_tsv/")


#Read in basewise mapping data for each rfa SRP gene
data1 <- read.csv("./microbe_trna/TP-CK-2S-CW1_S1_L004L1_bc1_TTCC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="minus", type="trna")
data2 <- read.csv("./microbe_trna/TP-CK-2S-CW1_S1_L004L1_bc2_TCTG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="minus", type="trna")
data3 <- read.csv("./microbe_trna/TP-CK-2S-CW1_S1_L004L1_bc3_TGGT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="minus", type="trna")
data4 <- read.csv("./microbe_trna/TP-CK-2S-CW1_S1_L004L1_bc4_CTGA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="minus", type="trna")
data5 <- read.csv("./microbe_trna/TP-CK-2S-CW1_S1_L004L1_bc5_CCAT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="minus", type="trna")
data6 <- read.csv("./microbe_trna/TP-CK-2S-CW1_S1_L004L1_bc6_CATC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="minus", type="trna")
data7 <- read.csv("./microbe_trna/TP-CK-2S-CW1_S1_L004L1_bc7_GTAG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="minus", type="trna")
data8 <- read.csv("./microbe_trna/TP-CK-2S-CW1_S1_L004L1_bc8_GGTA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="minus", type="trna")

data9 <- read.csv("./microbe_trna/TP-CK-2S-CW2_S2_L004L1_bc1_TTCC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="a", DM="plus", type="trna")
data10 <- read.csv("./microbe_trna/TP-CK-2S-CW2_S2_L004L1_bc2_TCTG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v03", day="b", DM="plus", type="trna")
data11 <- read.csv("./microbe_trna/TP-CK-2S-CW2_S2_L004L1_bc3_TGGT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="a", DM="plus", type="trna")
data12 <- read.csv("./microbe_trna/TP-CK-2S-CW2_S2_L004L1_bc4_CTGA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v04", day="b", DM="plus", type="trna")
data13 <- read.csv("./microbe_trna/TP-CK-2S-CW2_S2_L004L1_bc5_CCAT.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="a", DM="plus", type="trna")
data14 <- read.csv("./microbe_trna/TP-CK-2S-CW2_S2_L004L1_bc6_CATC.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v05", day="b", DM="plus", type="trna")
data15 <- read.csv("./microbe_trna/TP-CK-2S-CW2_S2_L004L1_bc7_GTAG.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="a", DM="plus", type="trna")
data16 <- read.csv("./microbe_trna/TP-CK-2S-CW2_S2_L004L1_bc8_GGTA.tsv", header=T, sep="\t") %>%
  mutate(sample="Volunteer v06", day="b", DM="plus", type="trna")

oral_trna_basewise <- rbind(data1, data2, data3, data4, 
                           data5, data6, data7, data8,
                           data9, data10,data11,data12,
                           data13,data14,data15,data16) %>%
  separate(gene, sep="_", c("genus", "species", "X","Y","Z")) %>%
  mutate(Z = ifelse(grepl("Rothia", genus), Y, Z)) %>%
  select(-X, -Y) %>%
  separate(Z, sep="-", c("X", "AA", "isodecoder", "Y", "Z")) %>%
  #mutate(name = paste0(AA, isodecoder,"-", Y,"-", Z)) %>% #the right way
  filter(Y =="1", Z=="1") %>%
  mutate(name = paste0(AA, isodecoder)) %>% #the wrong way
  select(-X, -Y, -Z) %>%
  filter( AA %nin% c("Und"))

# write.csv(filter(oral_trna_basewise,
#                  genus=="Rothia"), "oral_tRNA_mutaiton_data.csv")
```

##Fig 6A: tRNA modifications in oral species
```{r}
temp <- oral_trna_basewise %>%
  mutate(mutation = ifelse(pileup<50, NA, mutation)) %>%
  filter(#DM=="plus",
         day =="b",
         sample == "Volunteer v03") #%>%
  # group_by(ge, position, stress, DM) %>%
  # #mutate(mutation = mean(mutation) ) %>%
  # group_by()
  # # mutate(mutation = round(mutation/5, 2)*5 )


fig6A <- ggplot(filter(temp,
              genus=="Rothia",
              DM=="minus"
              ), aes(name, y=position, fill=mutation))+
  geom_tile()+
  scale_y_reverse(breaks = c(8, 37, 47, 76)
                  #sec.axis = dup_axis()
                  ) +
  scale_fill_gradient(low="grey70", high="red", limits=c(0,.25), breaks=c(0, .12, .25), na.value = "grey70") +
  geom_tile(data=filter(temp, mutation>=0.25, genus=="Rothia", DM=="minus"), fill="red4") +
  theme(legend.position = "right",
        legend.text = element_text(angle=0, vjust=.5),
        legend.background = element_blank(),
        axis.text.x = element_text(angle = 90, hjust = 0, vjust=.5, size=10),
        axis.title.x = element_blank(),
        axis.ticks.y = element_line(size=1),
        axis.ticks.length.y = unit(.2, "cm"),
        plot.title = element_blank()) +
  labs(fill="Mutation\nrate") + #element_text(hjust = 0.5)) +
  #labs(title="E. coli mutation rate\nNo stress, -DM, average of replicates") +
  ylab("Position") +
  # facet_grid(#cols=vars(genus),
  #            #rows=vars(DM)
  #            ) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))


#Wonkey legend games
plt_legend <- get_legend(fig6A + theme(legend.title = element_blank() ) )
plt_legend <- plt_legend #+ theme(legend.title = element_blank())
#make a >0.25 legend
legend_data <-data.frame(Mutation=c(">0.25",">0.25"), x=c(0,1), y=c(0,1))
legend_legend <- ggplot(legend_data, aes(x=y,y=y,fill=Mutation))+
  geom_tile() +
  scale_fill_manual(values= c(">0.25"="red4")) +
  labs(fill="Mutation\nrate") +
  theme(legend.position = "right", legend.background = element_blank(),
        legend.key.height = unit(c(0.2),"cm") )
legend_legend <- get_legend(legend_legend)

lay=rbind(c(3),
          c(1),
          c(2),
          c(2),
          c(3))

legend_legend <- arrangeGrob(empty_graph,left=legend_legend)
plt_legend <- arrangeGrob(empty_graph, left=plt_legend)
plt_legend <- arrangeGrob(legend_legend, plt_legend, empty_graph, layout_matrix = lay  )


fig6A <- arrangeGrob(fig6A + theme(legend.position = "None") )
lay = rbind(c(1,1,1,1,2))
fig6A <- grid.arrange(fig6A, plt_legend, layout_matrix = lay)

fig6A
```

##Fig6B: difference with DM
```{r}

temp2 <- oral_trna_basewise %>%
  group_by(sample, day, position, species, genus, AA, isodecoder, DM, name) %>%
  select(mutation) %>%
  filter(#DM=="plus",
         day =="b",
         sample == "Volunteer v03") %>%
  group_by() %>%
  pivot_wider(names_from = DM, values_from = mutation)

fig6B <- ggplot(filter(temp2,
              genus=="Rothia"
              #isodecoder=="TGA"
              ), aes(name, y=position, fill=plus-minus))+
  geom_tile() +
  scale_y_reverse(breaks = c(8, 37, 47, 76)
                  #sec.axis = dup_axis()
                  ) +
  scale_fill_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="grey70", limits=c(-.25, .25), breaks=c(-0.25,0,0.25))+
  geom_tile(data=filter(temp2, plus-minus>=0.25, genus=="Rothia"), fill="red4") +
  geom_tile(data=filter(temp2, plus-minus<=-0.25, genus=="Rothia"), fill="blue4") +
  theme(legend.position = "right",
        legend.text = element_text(angle=0, vjust=.5),
        legend.background = element_blank(),
        axis.text.x = element_text(angle = 90, hjust = 0, vjust=.5, size=10),
        axis.title.x = element_blank(),
        axis.ticks.y = element_line(size=1),
        axis.ticks.length.y = unit(.2, "cm"),
        plot.title = element_blank()) +
  labs(fill="Mutation\ndifference\nwith DM") + #element_text(hjust = 0.5)) +
  #labs(title="E. coli mutation rate\nNo stress, -DM, average of replicates") +
  ylab("Position") +
  # facet_grid(cols=vars(genus)
  #            ) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))


#Wonkey legend games
plt_legend <- get_legend(fig6B + theme(legend.title = element_blank() ) )
plt_legend <- plt_legend #+ theme(legend.title = element_blank())
#make a >0.25 legend
legend_data1 <-data.frame(Mutation=c("> 0.25"), x=c(1), y=c(1))
legend_legend1 <- ggplot(legend_data1, aes(x=y,y=y,fill=Mutation))+
  geom_tile() + scale_fill_manual(values= c("> 0.25"="red4")) +
  labs(fill="Difference in\nmutation rate")+
  theme(legend.position = "right", legend.background = element_blank(),
        legend.key.height = unit(c(0.2),"cm"))
legend_legend1 <- get_legend(legend_legend1)
#make a separea <0.25 legend
legend_data2 <-data.frame(Mutation=c("< -0.25"), x=c(1), y=c(1))
legend_legend2 <- ggplot(legend_data2, aes(x=y,y=y,fill=Mutation))+
  geom_tile() + scale_fill_manual(values= c("< -0.25"="blue4")) +
  theme(legend.position = "right", legend.background = element_blank(), legend.title = element_blank(),
        legend.key.height = unit(c(0.2),"cm"))
legend_legend2 <- get_legend(legend_legend2)

lay=rbind(c(4), c(1),c(2),c(2),c(3), c(4) )
legend_legend1 <- arrangeGrob(empty_graph,left=legend_legend1)
plt_legend <- arrangeGrob(empty_graph, left=plt_legend)
legend_legend2 <- arrangeGrob(empty_graph, left=legend_legend2)
plt_legend <- arrangeGrob(legend_legend1, plt_legend, legend_legend2,empty_graph, layout_matrix = lay  )
fig6B <- arrangeGrob(fig6B + theme(legend.position = "None") )

lay = rbind(c(1,1,1,1,2))
fig6B <- grid.arrange(fig6B, plt_legend, layout_matrix = lay)

fig6B
```

##Fig 6C
```{r}

temp <- oral_trna_basewise %>%
  mutate(mutation = ifelse(pileup<50, NA, mutation)) %>%
  filter(genus=="Rothia",
         day =="b",
         sample=="Volunteer v03") %>%
  separate(name, sep="-", c("name", "num_1", "num_2"))

temp$DM <- recode(temp$DM, "minus"="minus DM", "plus"="plus DM")

fig6C <- ggplot(filter(temp, 
              position >=32, 
              position <=42,
              isodecoder %in% c("ACG","CCG","CTG","TTG","GTG","CAG","GAG","TAG","CGG","GGG")
              ),
       aes(x=position, y=mutation, color=DM)) +
  geom_line() +
  coord_cartesian(ylim=c(0,.25)) +
  scale_x_continuous(breaks=c(33, 37, 42)) +
  scale_y_continuous(breaks=c(0,.12,.25)) +
  scale_color_manual(values = c("minus DM"="black","plus DM"="red")) +
  facet_grid(cols=vars(name), switch="both") +
  labs(color="") +
  ylab("Mutation rate") +
  theme(panel.background = element_blank(),
        panel.border = element_blank(),
        axis.title.x = element_blank(),
        axis.ticks.x = element_blank(),
        axis.text.x = element_blank(),
        axis.line.y = element_line(),
        axis.line.x = element_line(),
        strip.background = element_blank(),
        strip.placement = "outside",
        strip.text = element_text(size=10),
        legend.position = c(.9,.8),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")
        )
fig6C
```

##Fig6D: oral pos22
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_oral/tRNA_modifications/")

data1 <- read.csv("./tongue1_day1/tongue1_day1_position_22_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v03", day="a")
data2 <- read.csv("./tongue1_day2/tongue1_day2_position_22_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v03", day="b")
data3 <- read.csv("./tongue2_day1/tongue2_day1_position_22_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v04", day="a")
data4 <- read.csv("./tongue2_day2/tongue2_day2_position_22_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v04", day="b")
data5 <- read.csv("./tongue3_day1/tongue3_day1_position_22_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v05", day="a")
data6 <- read.csv("./tongue3_day2/tongue3_day2_position_22_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v05", day="b")
data7 <- read.csv("./tongue4_day1/tongue4_day1_position_22_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v06", day="a")
data8 <- read.csv("./tongue4_day2/tongue4_day2_position_22_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v06", day="b")


ORAL_MOD_DATA_pos22 <- rbind(data1, data2, data3, data4, 
                       data5, data6, data7, data8) %>%
  # mutate(anticodon_sequence = ifelse(anticodon_sequence=="", NA, anticodon_sequence ),
  #        amino_acid = ifelse(amino_acid=="", NA, amino_acid )) %>%
  #group_by(sample, day, anticodon_sequence, amino_acid, class) %>%
  mutate(mutation_A_un = A_un / (A_un + C_un + G_un +T_un ),
         mutation_A_dm = A_dm / (A_dm + C_dm + G_dm +T_dm ),
         position = 22)


position_22_codons <- c("GCA","CTG","TTG","CTC","TTC","GTG","CAA","CAG","GAG","TAG","TAA","CGA","GGA","TGA","GCT","GTA")
SELECTED_CLASSES <- c(#"Bacteroidia",
                      #"Actinobacteria",
                      "Negativicutes",
                      "Clostridia",
                      #"Betaproteobacteria",
                      "Bacilli"
                      #"Gammaproteobacteria",
                      #"Epsilonproteobacteria",
                      #"Fusobacteriia"
                      )


temp <- ORAL_MOD_DATA_pos22 %>%
  filter(day=="a", sample=="Volunteer v03") %>%
  mutate(diff = mutation_A_dm - mutation_A_un) %>%
  filter(#diff > 0.15,
         anticodon_sequence %in% position_22_codons,
         class %in% SELECTED_CLASSES) %>%
  #full_join(., select(taxon, phylum, class), by=c("class")) %>%
  select(class, mutation_A_un, mutation_A_dm, anticodon_sequence, amino_acid, sample, day) %>%
  pivot_longer(c(-class, -anticodon_sequence, -amino_acid, -sample, -day), 
               names_to = "DM", values_to = "A_mut")
temp$DM <- recode(temp$DM, "mutation_A_un"="-", "mutation_A_dm"="+")
temp$DM <- factor(temp$DM, levels=c("-", "+"))
temp$amino_acid <- factor(temp$amino_acid, levels = rev(levels(temp$amino_acid)))


fig6D <- ggplot(filter(temp),
       aes(y = interaction(anticodon_sequence, amino_acid), x=DM, fill= 1-A_mut )) +
  geom_tile() +
  scale_y_discrete() +
  scale_fill_gradient(low="grey70",  high="red", na.value="grey70")+
  labs(fill="Mutation\nrate") +
  ylab("tRNA (anticodon)") +
  xlab("Demethylase treatment") +
  #coord_fixed(ratio=0.25) +
  theme(axis.text.x = element_text(size=12, vjust=0.5),
        axis.title.x = element_text(size=10),
        axis.title.y = element_blank(),
        axis.ticks = element_blank(),
        panel.background = element_blank(),
        panel.border = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(size=10),
        plot.margin = unit(c(0.05, 0.3, 0.05, 0.3), "cm"),
        legend.position = "right",
        legend.title = element_text(size=10),
        legend.text = element_text(size=8) ) +
  facet_grid(cols=vars(class))
fig6D

```

##Fig6E: oral pos58/59

##Read in pos 58 and 59 ORDER modifications
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/reference/new_oral/tRNA_modifications/")

data1 <- read.csv("./tongue1_day1/tongue1_day1_position_58_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v03", day="a")
data2 <- read.csv("./tongue1_day2/tongue1_day2_position_58_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v03", day="b")
data3 <- read.csv("./tongue2_day1/tongue2_day1_position_58_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v04", day="a")
data4 <- read.csv("./tongue2_day2/tongue2_day2_position_58_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v04", day="b")
data5 <- read.csv("./tongue3_day1/tongue3_day1_position_58_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v05", day="a")
data6 <- read.csv("./tongue3_day2/tongue3_day2_position_58_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v05", day="b")
data7 <- read.csv("./tongue4_day1/tongue4_day1_position_58_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v06", day="a")
data8 <- read.csv("./tongue4_day2/tongue4_day2_position_58_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v06", day="b")

ORAL_MOD_DATA_pos58 <- rbind(data1, data2, data3, data4, 
                       data5, data6, data7, data8) %>%
  # mutate(anticodon_sequence = ifelse(anticodon_sequence=="", NA, anticodon_sequence ),
  #        amino_acid = ifelse(amino_acid=="", NA, amino_acid )) %>%
  #group_by(sample, day, anticodon_sequence, amino_acid, class) %>%
  mutate(mutation_A_un = A_un / (A_un + C_un + G_un +T_un ),
         mutation_A_dm = A_dm / (A_dm + C_dm + G_dm +T_dm ), 
         position = "pos58")

#===============//======================//=====================


data1 <- read.csv("./tongue1_day1/tongue1_day1_position_59_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v03", day="a")
data2 <- read.csv("./tongue1_day2/tongue1_day2_position_59_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v03", day="b")
data3 <- read.csv("./tongue2_day1/tongue2_day1_position_59_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v04", day="a")
data4 <- read.csv("./tongue2_day2/tongue2_day2_position_59_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v04", day="b")
data5 <- read.csv("./tongue3_day1/tongue3_day1_position_59_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v05", day="a")
data6 <- read.csv("./tongue3_day2/tongue3_day2_position_59_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v05", day="b")
data7 <- read.csv("./tongue4_day1/tongue4_day1_position_59_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v06", day="a")
data8 <- read.csv("./tongue4_day2/tongue4_day2_position_59_class_profile.tsv", sep="\t") %>%
  mutate(sample="Volunteer v06", day="b")

ORAL_MOD_DATA_pos59 <- rbind(data1, data2, data3, data4, 
                       data5, data6, data7, data8) %>%
  # mutate(anticodon_sequence = ifelse(anticodon_sequence=="", NA, anticodon_sequence ),
  #        amino_acid = ifelse(amino_acid=="", NA, amino_acid )) %>%
  #group_by(sample, day, anticodon_sequence, amino_acid, class) %>%
  mutate(mutation_A_un = A_un / (A_un + C_un + G_un +T_un ),
         mutation_A_dm = A_dm / (A_dm + C_dm + G_dm +T_dm ), 
         position = "pos59")
  
#"ACG"

ORAL_MOD_DATA_pos5859 <- rbind(ORAL_MOD_DATA_pos58, ORAL_MOD_DATA_pos59) %>%
  filter( anticodon_sequence %nin% c("AGC","ATT","ATC","ACA","ACC","AAT","AAG","AAA",
                                     "AGA","AGT","ATA","AGG","ACT","ATG","AAC","TAT") ) %>%
  filter(!is.na(anticodon_sequence),
         amino_acid !="STP",
         amino_acid !="")



SELECTED_CLASSES <- c(#"Bacteroidia",
                      "Actinobacteria"
                      #"Negativicutes",
                      #"Clostridia",
                      #"Betaproteobacteria",
                      #"Bacilli",
                      #"Gammaproteobacteria",
                      #"Epsilonproteobacteria",
                      #"Fusobacteriia"
                      )


temp <- ORAL_MOD_DATA_pos5859 %>%
  mutate(diff = mutation_A_dm - mutation_A_un) %>%
  select(sample, day, class, position, diff, anticodon_sequence) %>%
  pivot_wider(names_from = position, values_from = diff) %>%
  select(sample, day, class, "pos58", "pos59", anticodon_sequence) %>%
  mutate(position2 = ifelse(is.na(pos58), "pos59", ifelse(is.na(pos59), "pos58", ifelse(pos58 > pos59, "pos58", "pos59" ))) ) %>%
  select(sample, day, class, position2,anticodon_sequence) %>%
  full_join(ORAL_MOD_DATA_pos5859,. , by=c("sample", "day", "class", "anticodon_sequence")) %>%
  filter(position2 == position)


temp <- temp %>%
  filter(day=="a", sample=="Volunteer v03") %>%
  mutate(diff = mutation_A_dm - mutation_A_un) %>%
  filter(#diff > 0.15,
         #anticodon_sequence %in% position_22_codons,
         class %in% SELECTED_CLASSES) %>%
  #full_join(., select(taxon, phylum, class), by=c("class")) %>%
  select(class, mutation_A_un, mutation_A_dm, anticodon_sequence, amino_acid, sample, day, position) %>%
  pivot_longer(c(-class, -anticodon_sequence, -amino_acid, -sample, -day, -position), 
               names_to = "DM", values_to = "A_mut") %>%
  filter( anticodon_sequence != "TCG") #GET RID OF AN ARG AT THE LAST MINUTE
temp$DM <- recode(temp$DM, "mutation_A_un"="-", "mutation_A_dm"="+")
temp$DM <- factor(temp$DM, levels=c("-", "+"))
temp$amino_acid <- factor(temp$amino_acid, levels = rev(levels(temp$amino_acid)))

fig6E <- ggplot(filter(temp),
       aes(y = interaction(anticodon_sequence, amino_acid), x=DM, fill= 1-A_mut )) +
  geom_tile() +
  scale_y_discrete() +
  scale_fill_gradient(low="grey70",  high="red", na.value="grey40", breaks=c(0,0.50,1.0), limits=c(0,1))+
  labs(fill="Mutation rate") +
  ylab("tRNA (anticodon)") +
  xlab("Demethylase\ntreatment") +
  #coord_fixed(ratio=0.7) +
  theme(axis.text.x = element_text(size=16, vjust=0.5),
        axis.ticks = element_blank(),
        axis.title.y = element_blank(),
        legend.position = "bottom",
        legend.text = element_text(angle=45),
        panel.background = element_blank(),
        panel.border = element_blank(),
        strip.background = element_blank(),
        strip.text = element_text(),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")) +
  facet_grid(cols=vars(class)
             #rows=vars(position)
             )
fig6E

```


##Fig6F: daily changes, pos 22 modificaiton
```{r}

SELECTED_CLASSES <- c(#"Bacteroidia",
                      #"Actinobacteria",
                      "Negativicutes",
                      "Clostridia",
                      #"Betaproteobacteria",
                      "Bacilli"
                      #"Gammaproteobacteria",
                      #"Epsilonproteobacteria",
                      #"Fusobacteriia"
                      )

#========================//=========================//===================
position_22_codons <- c("GCA","CTG","TTG","CTC","TTC","GTG","CAG","TAA","CGA","GGA","TGA","GCT","GTA"
                       # "TAG","GAG","CAA"
                        )


temp <- ORAL_MOD_DATA_pos22 %>%
  #filter(sample=="Volunteer v03") %>%
  mutate(diff = mutation_A_dm - mutation_A_un) %>%
  filter(anticodon_sequence %in% position_22_codons,
         class %in% SELECTED_CLASSES) %>%
  #full_join(., select(taxon, phylum, class), by=c("class")) %>%
  select(class, mutation_A_un, anticodon_sequence, amino_acid, sample, day) #%>%
  #pivot_wider(names_from = day, values_from = mutation_A_un)
temp$day <- recode(temp$day, "a"="Day 1", "b"="Day 2")
temp$sample <- recode(temp$sample, "Volunteer v03"="Individual 1", "Volunteer v04"="Individual 2", "Volunteer v05"="Individual 3", "Volunteer v06"="Individual 4")

fig6F <- ggplot(filter(temp),
       aes(x=class,
           y=1- mutation_A_un,
           group=interaction(day, class),
           color=day)) +
  geom_boxplot() +
  geom_point(position=position_jitterdodge(jitter.width = .1), alpha=0.2 ) +
  labs(color="") +
  # scale_fill_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="grey50"#, limits=c(-.25,.25)
  #                      ) +
  facet_grid(cols=vars(sample)) +
  scale_color_manual(values= c("Day 1"="grey20", "Day 2"="red")) +
  ylab("Mutation rate\nat position 22") +
  theme(axis.text.x = element_text(angle=90, hjust=1, vjust=0.5),
        strip.background = element_blank(),
        axis.title.x = element_blank(),
        legend.position = "right") 
fig6F

```

##Fig6G: daily changes, pos58/59 modification
```{r}

temp <- ORAL_MOD_DATA_pos5859 %>%
  mutate(diff = mutation_A_dm - mutation_A_un) %>%
  pivot_wider(names_from = position, values_from = diff) %>%
  select(sample, day, class, "pos58", "pos59", anticodon_sequence) %>%
  mutate(position2 = ifelse(is.na(pos58), "pos59", ifelse(is.na(pos59), "pos58", ifelse(pos58 > pos59, "pos58", "pos59" ))) ) %>%
  select(sample, day, class, position2,anticodon_sequence) %>%
  full_join(ORAL_MOD_DATA_pos5859,. , by=c("sample", "day", "class", "anticodon_sequence")) %>%
  filter(position2 == position)


temp <- temp %>%
  mutate(diff = mutation_A_dm - mutation_A_un) %>%
  filter(class == "Actinobacteria") #%>%
  #select(class, mutation_A_un, anticodon_sequence, amino_acid, sample, day) #%>
temp$day <- recode(temp$day, "a"="Day 1", "b"="Day 2")
temp$sample <- recode(temp$sample, "Volunteer v03"="Individual 1", "Volunteer v04"="Individual 2", "Volunteer v05"="Individual 3", "Volunteer v06"="Individual 4")


fig6G <- ggplot(filter(temp),
       aes(x=class,
           y=1- mutation_A_un,
           group=interaction(day, sample),
           color=day)) +
  geom_boxplot() +
  geom_point(position=position_jitterdodge(jitter.width = .1), alpha=0.2 ) +
  labs(color="") +
  xlab("Actinobacteria") +
  #facet_grid(cols=vars(class), switch = "x") +
  scale_color_manual(values= c("Day 1"="grey20", "Day 2"="red")) +
  scale_x_discrete(position="bottom") +
  ylab("Mutation rate\nat position 58/59") +
  theme(axis.text.x = element_blank(),# element_text(angle=0, hjust=0.5, vjust=0.5),
        strip.background = element_blank(),
        axis.title.x = element_text(),
        axis.ticks.x = element_blank()
        ) +
  facet_grid(cols=vars(sample))
fig6G

```

##Fig 6H: stool pos22
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/stool_biome/modification_results/")

data1 <- read.csv("./s9_position_22_class_profile.tsv", sep="\t", header=T) %>%
  mutate(sample="Stool 1", day="c") 
data2 <- read.csv("./s10_position_22_class_profile.tsv", sep="\t", header=T) %>%
  mutate(sample="Stool 2", day="c") 



Stool_MOD_DATA_pos22 <- rbind(data1, data2) %>%
  # mutate(anticodon_sequence = ifelse(anticodon_sequence=="", NA, anticodon_sequence ),
  #        amino_acid = ifelse(amino_acid=="", NA, amino_acid )) %>%
  #group_by(sample, day, anticodon_sequence, amino_acid, class) %>%
  mutate(mutation_A_un = A_un / (A_un + C_un + G_un +T_un ),
         mutation_A_dm = A_dm / (A_dm + C_dm + G_dm +T_dm ),
         position = 22)


position_22_codons <- c("GCA","CTG","TTG","CTC","TTC","GTG","CAA","CAG","GAG","TAG","TAA","CGA","GGA","TGA","GCT","GTA")
SELECTED_CLASSES <- c(#"Bacteroidia",
                      #"Actinobacteria",
                      #"Negativicutes",
                      "Clostridia",
                      #"Betaproteobacteria",
                      "Bacilli"
                      #"Gammaproteobacteria"
                      #"Epsilonproteobacteria",
                      #"Fusobacteriia"
                      )


temp <- Stool_MOD_DATA_pos22 %>%
  filter(sample=="Stool 1") %>%
  mutate(diff = mutation_A_dm - mutation_A_un) %>%
  filter(#diff > 0.15,
         anticodon_sequence %in% position_22_codons,
         class %in% SELECTED_CLASSES
         ) %>%
  #full_join(., select(taxon, phylum, class), by=c("class")) %>%
  select(class, mutation_A_un, mutation_A_dm, anticodon_sequence, amino_acid, sample, day) %>%
  pivot_longer(c(-class, -anticodon_sequence, -amino_acid, -sample, -day), 
               names_to = "DM", values_to = "A_mut")
temp$DM <- recode(temp$DM, "mutation_A_un"="-", "mutation_A_dm"="+")
temp$DM <- factor(temp$DM, levels=c("-", "+"))
temp$class <- recode(temp$class, "Clostridia"="Clostridia", "Bacilli"="Bacilli")
temp$amino_acid <- factor(temp$amino_acid, levels = rev(levels(temp$amino_acid)))


fig6H <- ggplot(filter(temp),
       aes(y = interaction(anticodon_sequence, amino_acid), x=DM, fill= 1-A_mut )) +
  geom_tile() +
  scale_y_discrete() +
  scale_fill_gradient(low="grey70",  high="red", na.value="grey40")+
  labs(fill="Mutation rate") +
  ylab("tRNA (anticodon)") +
  xlab("Demethylase treatment") +
  coord_fixed(ratio=0.4) +
  theme(axis.text.x = element_text(size=16, vjust=0.5),
        axis.ticks = element_blank(),
        axis.title.y = element_blank(),
        panel.background = element_blank(),
        panel.border = element_blank(),
        strip.background = element_blank(),
        plot.margin = unit(c(0.05, 0.3, 0.3, 0.3), "cm")) +
  facet_grid(cols=vars(class))
fig6H

```


##Fig 6I: stool pos58/59
```{r}
setwd("~/4SR/data/20200317_Nature_methods_data/2020 06 15 SRP analysis/stool_biome/modification_results/")

data1 <- read.csv("./s9_position_58_class_profile.tsv", sep="\t", header=T) %>%
  mutate(sample="Stool 1", day="c", position="pos58") 
data2 <- read.csv("./s10_position_58_class_profile.tsv", sep="\t", header=T) %>%
  mutate(sample="Stool 2", day="c", position="pos58")
STOOL_MOD_DATA_pos58 <- rbind(data1, data2) %>%
  mutate(frac_A_un = A_un / (A_un + C_un + G_un +T_un ),
         frac_A_dm = A_dm / (A_dm + C_dm + G_dm +T_dm ), 
         position = "pos58")

data3 <- read.csv("./s9_position_59_class_profile.tsv", sep="\t", header=T) %>%
  mutate(sample="Stool 1", day="c", position="pos59") 
data4 <- read.csv("./s10_position_59_class_profile.tsv", sep="\t", header=T) %>%
  mutate(sample="Stool 2", day="c", position="pos59")
STOOL_MOD_DATA_pos59 <- rbind(data3, data4) %>%
  mutate(frac_A_un = A_un / (A_un + C_un + G_un +T_un ),
         frac_A_dm = A_dm / (A_dm + C_dm + G_dm +T_dm ), 
         position = "pos59")


STOOL_MOD_DATA_pos5859 <- rbind(STOOL_MOD_DATA_pos58, STOOL_MOD_DATA_pos59) %>%
  filter( anticodon_sequence %nin% c("AGC","ATT","ATC","ACA","ACC","AAT","AAG","AAA",
                                     "AGA","AGT","ATA","AGG","ACT","ATG","AAC","TAT") ) %>%
  filter(!is.na(anticodon_sequence),
         amino_acid !="STP",
         amino_acid !="")



SELECTED_CLASSES <- c(#"Bacteroidia",
                      "Actinobacteria"
                      #"Negativicutes",
                      #"Clostridia",
                      #"Betaproteobacteria",
                      #"Bacilli",
                      #"Gammaproteobacteria",
                      #"Epsilonproteobacteria",
                      #"Fusobacteriia"
                      )



temp <- STOOL_MOD_DATA_pos5859 %>%
  group_by() %>%
  mutate(diff = frac_A_dm - frac_A_un) %>%
  select(sample, day, class, position, diff, anticodon_sequence) %>%
  pivot_wider(names_from = position, values_from = diff) %>%
  select(sample, day, class, pos58, pos59, anticodon_sequence) %>%
  mutate(position2 = ifelse(is.na(pos58), "pos59", ifelse(is.na(pos59), "pos58", ifelse(pos58 > pos59, "pos58", "pos59" ))) ) %>%
  select(sample, day, class, position2,anticodon_sequence) %>%
  full_join(STOOL_MOD_DATA_pos5859,. , by=c("sample", "day", "class", "anticodon_sequence")) %>%
  filter(position2 == position)



A <- STOOL_MOD_DATA_pos5859 %>%
  filter(position=="pos58",
         anticodon_sequence %in% c("GTT","GTG","CGT","TGT"))
B <- STOOL_MOD_DATA_pos5859 %>%
  filter(position=="pos59",
         anticodon_sequence %in% c("GGA","TGA","GCT","ACG","CGA","CGG"))
# C <- data.frame(
#   anticodon_sequence = c("GTT", "GTG", "GAA", "TGA", "CGT", "TGT"),
#   amino_acid = c("Asn", "His", "Phe", "Ser", "Thr", "Thr"),
#   frac_A_dm = c(.365, .326, .857, .133, .269, .155),
#   frac_A_un = c(.004, .003, .004, 0, .002, .001)
# ) %>% 
#   mutate(class="Actinobacteria", sample="Stool 1", day="c", position="X")


temp <- rbind(A,B) %>%
  filter(day=="c", sample=="Stool 1") %>%
  mutate(diff = frac_A_dm - frac_A_un) %>%
  filter(#diff > 0.15,
         #anticodon_sequence %in% position_22_codons,
         class %in% SELECTED_CLASSES) %>%
  #full_join(., select(taxon, phylum, class), by=c("class")) %>%
  select(class, frac_A_un, frac_A_dm, anticodon_sequence, amino_acid, sample, day, position) %>%
  pivot_longer(c(-class, -anticodon_sequence, -amino_acid, -sample, -day, -position), 
               names_to = "DM", values_to = "frac_A")
temp$DM <- recode(temp$DM, "frac_A_un"="-", "frac_A_dm"="+")
temp$DM <- factor(temp$DM, levels=c("-", "+"))
temp$amino_acid <- factor(temp$amino_acid, levels = rev(levels(temp$amino_acid)))


fig6I <- ggplot(filter(temp),
       aes(y = interaction(anticodon_sequence, amino_acid), x=DM, fill= 1-frac_A )) +
  geom_tile() +
  scale_y_discrete() +
  scale_fill_gradient(low="grey70",  high="red", na.value="grey40")+
  labs(fill="Mutation rate") +
  ylab("tRNA (anticodon)") +
  xlab("Demethylase\ntreatment") +
  coord_fixed(ratio=0.3) +
  theme(axis.text.x = element_text(size=16, vjust=0.5),
        axis.ticks = element_blank(),
        axis.title.y = element_blank(),
        panel.background = element_blank(),
        panel.border = element_blank(),
        strip.background = element_blank(),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")) +
  facet_grid(cols=vars(class)
             #rows=vars(position)
             )
fig6I

```

##Save figure 6
```{r}
myplota <- arrangeGrob(fig6A, left = textGrob(expression(bold("a")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotb <- arrangeGrob(fig6B, left = textGrob(expression(bold("b")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotc <- arrangeGrob(fig6C, left = textGrob(expression(bold("c")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotd <- arrangeGrob(fig6D, top = textGrob(expression(bold("d")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

myplotcd <- grid.arrange(myplotc, myplotd, layout_matrix=rbind(c(1,1,1,2,2)))

myplote <- arrangeGrob(fig6F, top = textGrob(expression(bold("e")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotf <- arrangeGrob(fig6G, top = textGrob(expression(bold("f")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
# myplotg <- arrangeGrob(fig6H, top = textGrob(expression(bold("g")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
# myploth <- arrangeGrob(fig6H, left = textGrob(expression(bold("h")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
# myploti <- arrangeGrob(fig6I, left = textGrob(expression(bold("i")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))


#Set the figure arrangement matrix
lay <- rbind(c(1,1,1,1),
             c(1,1,1,1),
             c(2,2,2,2),
             c(2,2,2,2),
             c(3,3,3,3),
             c(3,3,3,3),
             c(3,3,3,3),
             c(4,4,5,5),
             c(4,4,5,5),
             c(4,4,5,5))
#Arrange figures
figure6 <- grid.arrange(myplota, myplotb, myplotcd,
                        myplote, myplotf, 
                        layout_matrix = lay)
figure6

setwd("~/4SR/data/20200317_Nature_methods_data/2020 02 24 ecoli figures/source/")
ggsave("../figures/figure6.svg", 
       plot = figure6,
       scale = 1.3,
       width = 9, 
       height = 8)
ggsave("../figures/figure6.png", 
       plot = figure6,
       scale = 1.3,
       width = 9, 
       height = 8)

```
#_
##Save figure S6
```{r}
# myplota <- arrangeGrob(fig6A, left = textGrob(expression(bold("a")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
# myplotb <- arrangeGrob(fig6B, left = textGrob(expression(bold("b")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
# myplotc <- arrangeGrob(fig6C, left = textGrob(expression(bold("c")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
# myplotd <- arrangeGrob(fig6D, top = textGrob(expression(bold("d")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
# 
# myplotcd <- grid.arrange(myplotc, myplotd, layout_matrix=rbind(c(1,1,1,2,2)))
# 
# myplote <- arrangeGrob(fig6F, top = textGrob(expression(bold("e")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
# myplotf <- arrangeGrob(fig6G, top = textGrob(expression(bold("f")), x = unit(0, "npc")
#          , y   = unit(1, "npc"), just=c("left","top"),
#          gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotg <- arrangeGrob(fig6H, top = textGrob(expression(bold("b")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myploth <- arrangeGrob(fig6I, left = textGrob(expression(bold("c")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))

#myploth <- grid.arrange(myploth, empty_graph, layout_matrix=rbind(c(1,1,2)))

myploti <- arrangeGrob(fig6E, left = textGrob(expression(bold("a")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))


#Set the figure arrangement matrix
lay <- rbind(c(1,2),
             c(1,3))
#Arrange figures
figureS6 <- grid.arrange(myploti, myplotg, myploth,
                        layout_matrix = lay)
figureS6

setwd("~/4SR/data/20200317_Nature_methods_data/2020 02 24 ecoli figures/source/")
ggsave("../figures/figureS6.svg", 
       plot = figureS6,
       scale = 1.3,
       width = 5, 
       height = 7)
ggsave("../figures/figureS6.png", 
       plot = figureS6,
       scale = 1.3,
       width = 5, 
       height = 7)

```


#_
#_









#OLD

#Figure 6
##Fig 6A: stool taxonomy
```{r}

stool_counts_9_plus <- read.csv("../../2020 04 30 microbiome/taxonomy/stool/s9_demethylase_taxonomy/s9_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Stool 1", DM="+DM")
stool_counts_9_minus <- read.csv("../../2020 04 30 microbiome/taxonomy/stool/s9_untreated_taxonomy/s9_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Stool 1", DM="-DM")
stool_counts_10_plus <- read.csv("../../2020 04 30 microbiome/taxonomy/stool/s10_demethylase_taxonomy/s10_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Stool 2", DM="+DM")
stool_counts_10_minus <- read.csv("../../2020 04 30 microbiome/taxonomy/stool/s10_untreated_taxonomy/s10_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Stool 2", DM="-DM")


stool_taxa <- rbind(stool_counts_9_plus, stool_counts_9_minus, stool_counts_10_plus, stool_counts_10_minus) %>%
  mutate(class=X, count=X.1) %>%
  select(-X, -X.1) %>%
  #mutate(class = as.factor(class), sample= as.factor(sample), DM=as.factor(DM)) %>%
  group_by(class, DM, sample) %>%
  pivot_longer( c(-class, -DM, -sample), names_to="anticodon", values_to = "count") %>%
  summarise(sum_count = sum(count, na.rm=T)) %>%
  group_by(DM, sample) %>%
  mutate(sum_count = sum_count / sum(sum_count)) %>%
  group_by() %>%
  mutate(class = as.character(class)) %>%
  mutate(class = ifelse(class=="", "Unassigned", class)) %>% #Get the unnamed class into the other chatagory
  mutate(class = ifelse(sum_count >=0.02, class, "< 2%" )) %>%
  mutate(class = as.factor(class)) %>%
  group_by(class, sample, DM) %>%
  summarise(sum_count = sum(sum_count) )  %>%#combine all the things newly named "other"
  group_by()  %>%
  mutate(sample = fct_reorder(sample, desc(sample) ))
stool_taxa$DM <- factor(stool_taxa$DM, levels= c("-DM", "+DM"))


stool_taxa_rare <- rbind(stool_counts_9_plus, stool_counts_9_minus, stool_counts_10_plus, stool_counts_10_minus) %>%
  mutate(class=X, count=X.1) %>%
  select(-X, -X.1) %>%
  #mutate(class = as.factor(class), sample= as.factor(sample), DM=as.factor(DM)) %>%
  group_by(class, DM, sample) %>%
  pivot_longer( c(-class, -DM, -sample), names_to="anticodon", values_to = "count") %>%
  summarise(sum_count = sum(count, na.rm=T)) %>%
  group_by(DM, sample) %>%
  mutate(sum_count = sum_count / sum(sum_count)) %>%
  group_by() %>%
  mutate(class = as.character(class)) %>%
  mutate(class = ifelse(class=="", "Unassigned", class)) %>% #Get the unnamed class into the other chatagory
  mutate(class = ifelse(sum_count <0.02, ifelse(sum_count > 0.007, class, "< 0.7%"), "too_big" )) %>%
  mutate(class = as.factor(class)) %>%
  group_by(class, sample, DM) %>%
  summarise(sum_count = sum(sum_count) )  %>%#combine all the things newly named "other"
  group_by() %>%
  filter(class != "too_big") %>%
  mutate(sample = fct_reorder(sample, desc(sample) ))
stool_taxa_rare$DM <- factor(stool_taxa_rare$DM, levels= c("-DM", "+DM"))

stool_taxa <- stool_taxa %>%
  filter(DM=="+DM")
stool_taxa_rare <- stool_taxa_rare %>%
  filter(DM=="+DM")

stool_taxa$class <- recode(stool_taxa$class, "< 2%"="< 2%", "Bacilli"="Bacilli", "Gammaproteobacteria"="γ-proteobacteria",
                           "Unassigned"="Unassigned", "Bacteroidia"="Bacteroidia","Clostridia"="Clostridia")
stool_taxa$class <- factor(stool_taxa$class, levels = c("< 2%", "Bacilli","γ-proteobacteria","Unassigned","Bacteroidia","Clostridia") )

fig6A_1 <- ggplot(stool_taxa, aes(y=sum_count, x=sample, fill=class)) +
  geom_bar(stat="identity", position="stack") +
  coord_flip() +
  theme(legend.position = "right", axis.title.y = element_blank(),
        legend.title = element_blank()) +
  scale_fill_manual(values=c("#ffa900", "#008f00","#fffc79","#7b1979","#021eaa","#b6192b"), guide=guide_legend(reverse=TRUE) ) +
  ylab("Fraction of reads") +
  #scale_y_discrete(position = "right") +
  labs(fill="Class") +
  #facet_grid(rows=vars(DM)) +
  theme(panel.border = element_blank(),
        panel.background = element_blank(),
        panel.spacing = unit(c(0, 0, 0, 0), "cm"),
        legend.background = element_blank(),
        axis.ticks = element_blank(),
        strip.background = element_blank(), 
        strip.text.y = element_text(angle=0, size=12),
        strip.placement = "left",
        axis.text.y = element_text(),
        axis.text.x = element_blank(),
        axis.title.x = element_blank(),
        #legend.text = element_text(size=8),
        legend.key.height = unit(c(0.4),"cm"),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0), "cm") ) 
fig6A_1


stool_taxa_rare$class <- recode(stool_taxa_rare$class,  "< 0.7%"="< 0.7%","Negativicutes"="Negativicutes","Mollicutes"="Mollicutes",
    "Actinobacteria"="Actinobacteria", "Tissierellia"="Tissierellia", "Alphaproteobacteria"="α-proteobacteria","Coriobacteriia"="Coriobacteriia",
    "Deltaproteobacteria"="δ-proteobacteria","Betaproteobacteria"="β-proteobacteria")

stool_taxa_rare$class <- factor(stool_taxa_rare$class, levels = c("< 0.7%","Negativicutes",
    "Mollicutes","Actinobacteria","Tissierellia", "α-proteobacteria","Coriobacteriia", "δ-proteobacteria",
    "β-proteobacteria", "γ-proteobacteria") )

fig6A_2 <- ggplot(stool_taxa_rare, aes(y=sum_count, x=sample, fill=class)) +
  geom_bar(stat="identity", position="stack" ) +
  coord_flip() +
  theme(legend.position = "right", axis.title.y = element_blank(),
        legend.title = element_blank()) +
  scale_fill_manual(values=c("#abc837","#3c97ac","#ff4c00","#00acff","#8a6f19","#a6fb79","#ab82ff","#ffaaaa","#dd55ff","#fffc79"), guide=guide_legend(reverse=TRUE) ) +
  ylab("Fraction of reads") +
  #facet_grid(rows=vars(DM)) +
  theme(panel.border = element_blank(),
        panel.background = element_blank(),
        panel.spacing = unit(c(0, 0, 0, 0), "cm"),
        axis.ticks = element_blank(),
        strip.background = element_blank(), 
        strip.text.y = element_text(angle=0, size=12),
        strip.placement = "left",
        axis.text.y = element_text(),
        axis.text.x=element_blank(),
        axis.title.x = element_blank(),
        legend.title = element_blank(),
        legend.background = element_blank(),
        #legend.text = element_text(size=8),
        legend.key.height = unit(c(0.4),"cm"),
        plot.margin = unit(c(0, 0.3, 0.3, 0.3), "cm"))
fig6A_2

legend1 <- arrangeGrob(left= get_legend(fig6A_1), empty_graph)
legend2 <- arrangeGrob(left= get_legend(fig6A_2), empty_graph)
legend <- arrangeGrob(legend1, legend2)
myplota <- arrangeGrob(fig6A_1+theme(legend.position="None") )
myplotb <- arrangeGrob(fig6A_2+theme(legend.position="None") )

quick_data <-data.frame(x=c(.3,.55,.65,.63), y=c(0,1,0,1), line=c(1,1,2,2))
annotation_grob <- ggplot(quick_data,aes(x=x,y=y,group=line))+theme_void()+
  geom_line(color="grey80",size=1) +scale_x_continuous(limits=c(0,1))+
  theme(plot.margin = unit(c(0, 0, 0, 0), "cm"))
#Set the figure arrangement matrix
lay <- rbind(c(1,1,1,1,3,3),
             c(1,1,1,1,3,3),
             c(1,1,1,1,3,3),
             c(5,5,5,5,5,5),
             c(4,2,2,2,3,3),
             c(4,2,2,2,6,6))
#Arrange figures
fig6A <- grid.arrange(myplota, myplotb, legend, empty_graph, annotation_grob,empty_graph, layout_matrix = lay)

```


##Fig6B: tongue taxonomy
```{r}
tongue_counts_1_1_plus <- read.csv("../../2020 04 30 microbiome/taxonomy/tongue/tongue1_day1_demethylase_taxonomy/tongue1_day1_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Day 1", DM="+DM")
tongue_counts_1_1_minus <- read.csv("../../2020 04 30 microbiome/taxonomy/tongue/tongue1_day1_untreated_taxonomy/tongue1_day1_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Day 1", DM="-DM")
tongue_counts_1_2_plus <- read.csv("../../2020 04 30 microbiome/taxonomy/tongue/tongue1_day2_demethylase_taxonomy/tongue1_day2_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Day 2", DM="+DM")
tongue_counts_1_2_minus <- read.csv("../../2020 04 30 microbiome/taxonomy/tongue/tongue1_day2_untreated_taxonomy/tongue1_day2_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Day 2", DM="-DM")


tongue_taxa <- rbind(tongue_counts_1_1_plus, tongue_counts_1_1_minus, tongue_counts_1_2_plus, tongue_counts_1_2_minus) %>%
  mutate(class=X, count=X.1) %>%
  select(-X, -X.1) %>%
  #mutate(class = as.factor(class), sample= as.factor(sample), DM=as.factor(DM)) %>%
  group_by(class, DM, sample) %>%
  pivot_longer( c(-class, -DM, -sample), names_to="anticodon", values_to = "count") %>%
  summarise(sum_count = sum(count, na.rm=T)) %>%
  group_by(DM, sample) %>%
  mutate(sum_count = sum_count / sum(sum_count, na.rm=T)) %>%
  group_by() %>%
  mutate(class = as.character(class)) %>%
  mutate(class = ifelse(class=="", "Unassigned", class)) %>% #Get the unnamed class into the other chatagory
  mutate(class = ifelse(sum_count >=0.11, class, "< 11%" )) %>%
  mutate(class = as.factor(class)) %>%
  group_by(class, sample, DM) %>%
  summarise(sum_count = sum(sum_count) )  %>%#combine all the things newly named "other"
  group_by()  %>%
  mutate(sample = fct_reorder(sample, desc(sample) ))
tongue_taxa$DM <- factor(tongue_taxa$DM, levels= c("-DM", "+DM"))


tongue_taxa_rare <- rbind(tongue_counts_1_1_plus, tongue_counts_1_1_minus, tongue_counts_1_2_plus, tongue_counts_1_2_minus) %>%
  mutate(class=X, count=X.1) %>%
  select(-X, -X.1) %>%
  mutate(class = as.factor(class), sample= as.factor(sample), DM=as.factor(DM)) %>%
  group_by(class, DM, sample) %>%
  pivot_longer( c(-class, -DM, -sample), names_to="anticodon", values_to = "count") %>%
  summarise(sum_count = sum(count, na.rm=T)) %>%
  group_by(DM, sample) %>%
  mutate(sum_count = sum_count / sum(sum_count, na.rm=T)) %>%
  group_by() %>%
  mutate(class = as.character(class)) %>%
  mutate(class = ifelse(class=="", "Unassigned", class)) %>% #Get the unnamed class into the other chatagory
  mutate(class = ifelse(sum_count <0.11, ifelse(sum_count > 0.007, class, "< 0.7%"), "too_big" )) %>%
  mutate(class = as.factor(class)) %>%
  group_by(class, sample, DM) %>%
  summarise(sum_count = sum(sum_count) )  %>%#combine all the things newly named "other"
  group_by() %>%
  filter(class != "too_big") %>%
  mutate(sample = fct_reorder(sample, desc(sample) ))
tongue_taxa_rare$DM <- factor(tongue_taxa_rare$DM, levels= c("-DM", "+DM"))


tongue_taxa <- tongue_taxa %>%
  filter(DM=="+DM")
tongue_taxa_rare <- tongue_taxa_rare %>%
  filter(DM=="+DM")

tongue_taxa$class <- recode(tongue_taxa$class,  "< 11%"="< 11%","Negativicutes"="Negativicutes","Bacilli"="Bacilli")
tongue_taxa$class <- factor(tongue_taxa$class, levels = c("< 11%","Negativicutes","Bacilli") )

fig6B_1 <- ggplot(tongue_taxa, aes(y=sum_count, x=sample, fill=class)) +
  geom_bar(stat="identity", position="stack") +
  coord_flip() +
  theme(legend.position = "right", axis.title.y = element_blank(),
        legend.title = element_blank()) +
  scale_fill_manual(values=c("#ffa900","#3c97ac","#008f00"),guide=guide_legend(reverse=TRUE) ) +
  ylab("Fraction of reads") +
  #facet_grid(rows=vars(DM)) +
  theme(panel.border = element_blank(),
        panel.background = element_blank(),
        panel.spacing = unit(c(0, 0, 0, 0), "cm"),
        axis.ticks = element_blank(),
        strip.background = element_blank(), 
        strip.text.y = element_text(angle=0, size=12),
        strip.placement = "left",
        axis.text.y = element_text(),
        axis.text.x = element_blank(),
        axis.title.x = element_blank(),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0), "cm"),
        #legend.text = element_text(size=8),
        legend.background = element_blank(),
        legend.key.height = unit(c(0.4),"cm") ) 
fig6B_1


#"Fusobacteriia", "Betaproteobacteria","Gammaproteobacteria","Clostridia",
# "#aaaaaa", "#dd55ff" ,"#fffc79" ,"#b6192b","#00acff","#021eaa"



tongue_taxa_rare$class <- recode(tongue_taxa_rare$class,  "< 0.7%"="< 0.7%","Unassigned"="Unassigned","Fusobacteriia"="Fusobacteriia",
                                  "Betaproteobacteria"="β-proteobacteria","Gammaproteobacteria"="γ-proteobacteria",
                                 "Clostridia"="Clostridia", "Actinobacteria"="Actinobacteria","Bacteroidia"="Bacteroidia",
                                 "Flavobacteriia"="Flavobacteriia", "Mammalian"="Mammalia")
tongue_taxa_rare$class <- factor(tongue_taxa_rare$class, levels = c("< 0.7%","Unassigned", "Mammalia","Flavobacteriia","Fusobacteriia",
                                  "β-proteobacteria","γ-proteobacteria","Clostridia","Actinobacteria","Bacteroidia") )

fig6B_2 <- ggplot(tongue_taxa_rare, aes(y=sum_count, x=sample, fill=class)) +
  geom_bar(stat="identity", position="stack" ) +
  coord_flip() +
  theme(legend.position = "right", axis.title.y = element_blank(),
        legend.title = element_blank()) +
  scale_fill_manual(values=c("#abc837", "#7b1979","#e9afaf", "#aaaaff", "#aaaaad","#dd55ff","#fffc79","#b6192b","#00acff","#021eaa"), guide=guide_legend(reverse=TRUE) ) +
  ylab("Fraction of reads") +
  #facet_grid(rows=vars(DM)) +
  theme(panel.border = element_blank(),
        panel.background = element_blank(),
        panel.spacing = unit(c(0, 0, 0, 0), "cm"),
        axis.ticks = element_blank(),
        strip.background = element_blank(), 
        strip.text.y = element_text(angle=0, size=12),
        strip.placement = "left",
        axis.text.y = element_text(),
        axis.text.x=element_blank(),
        axis.title.x = element_blank(),
        legend.title = element_blank(),
        legend.background = element_blank(),
       #legend.text = element_text(size=8),
        legend.key.height = unit(c(0.4),"cm"),
        plot.margin = unit(c(0, 0.3, 0.3, 0.3), "cm"))
fig6B_2

legend1 <- arrangeGrob(left= get_legend(fig6B_1), empty_graph)
legend2 <- arrangeGrob(left= get_legend(fig6B_2), empty_graph)
legend <- arrangeGrob(legend1, legend2)
myplota <- arrangeGrob(fig6B_1+theme(legend.position="None") )
myplotb <- arrangeGrob(fig6B_2+theme(legend.position="None") )

quick_data <-data.frame(x=c(.3,.55,.65,.63), y=c(0,1,0,1), line=c(1,1,2,2))
annotation_grob <- ggplot(quick_data,aes(x=x,y=y,group=line))+theme_void()+
  geom_line(color="grey80",size=1) +scale_x_continuous(limits=c(0,1))+
  theme(plot.margin = unit(c(0, 0, 0, 0), "cm"))
#Set the figure arrangement matrix
lay <- rbind(c(1,1,1,1,3,3),
             c(1,1,1,1,3,3),
             c(1,1,1,1,3,3),
             c(5,5,5,5,5,5),
             c(4,2,2,2,3,3),
             c(4,2,2,2,6,6))
#Arrange figures
fig6B <- grid.arrange(myplota, myplotb, legend, empty_graph, annotation_grob, layout_matrix = lay)

```

##Save figure 6
```{r}
myplota <- arrangeGrob(fig6A, left = textGrob(expression(bold("a")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotb <- arrangeGrob(fig6B, left = textGrob(expression(bold("b")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotc <- arrangeGrob(empty_graph, left = textGrob(expression(bold("c")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotd <- arrangeGrob(empty_graph, left = textGrob(expression(bold("d")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplote <- arrangeGrob(empty_graph, left = textGrob(expression(bold("e")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotf <- arrangeGrob(empty_graph, left = textGrob(expression(bold("f")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))


#Set the figure arrangement matrix
lay <- rbind(c(1, 2),
             c(1, 2),
             c(3, 5),
             c(3, 5),
             c(4, 6),
             c(4, 6))
#Arrange figures
figure6 <- grid.arrange(myplota, myplotb, myplotc, myplotd, myplote, myplotf,  layout_matrix = lay)
figure6

ggsave("../figures/figure6.svg", 
       plot = figure6,
       scale = 1.3,
       width = 8, 
       height = 7)
ggsave("../figures/figure6.png", 
       plot = figure6,
       scale = 1.3,
       width = 8, 
       height = 7)

```


#_
#Figures S6
##FigS6A: stool taxonomy CTC TTC
```{r}

stool_counts_9_plus <- read.csv("../../2020 04 30 microbiome/taxonomy/stool/s9_demethylase_taxonomy/s9_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Stool 1", DM="+DM")
stool_counts_9_minus <- read.csv("../../2020 04 30 microbiome/taxonomy/stool/s9_untreated_taxonomy/s9_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Stool 1", DM="-DM")
stool_counts_10_plus <- read.csv("../../2020 04 30 microbiome/taxonomy/stool/s10_demethylase_taxonomy/s10_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Stool 2", DM="+DM")
stool_counts_10_minus <- read.csv("../../2020 04 30 microbiome/taxonomy/stool/s10_untreated_taxonomy/s10_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Stool 2", DM="-DM")



stool_taxa <- rbind(stool_counts_9_plus, stool_counts_9_minus, stool_counts_10_plus, stool_counts_10_minus) %>%
  mutate(class=X, count=X.1) %>%
  select(-X, -X.1) %>%
  #mutate(class = as.factor(class), sample= as.factor(sample), DM=as.factor(DM)) %>%
  group_by(class, DM, sample) %>%
  pivot_longer( c(-class, -DM, -sample), names_to="anticodon", values_to = "count") %>%
  #summarise(sum_count = sum(count)) %>%
  group_by(DM, sample, anticodon) %>%
  mutate(sum_count = count / sum(count, na.rm=T) ) %>%
  group_by() %>%
  mutate(class = as.character(class)) %>%
  mutate(class = ifelse(class=="", "Unassigned", class)) %>% #Get the unnamed class into the other chatagory
  mutate(class = ifelse(sum_count >=0.02, class, "< 2%" )) %>%
  mutate(class = as.factor(class)) %>%
  group_by(class, sample, DM, anticodon) %>%
  summarise(sum_count = sum(sum_count) )  %>%#combine all the things newly named "other"
  group_by()  %>%
  mutate(sample = fct_reorder(sample, desc(sum_count) ))
stool_taxa$DM <- factor(stool_taxa$DM, levels= c("-DM", "+DM"))




stool_taxa <- stool_taxa %>%
  filter(DM=="+DM")


stool_taxa$class <- recode(stool_taxa$class,  "< 2%"="< 2%","Negativicutes"="Negativicutes", "Clostridia"="Clostridia",
    "Actinobacteria"="Actinobacteria", "Alphaproteobacteria"="α-proteobacteria",
    "Coriobacteriia"="Coriobacteriia","Bacilli"="Bacilli", "Bacteroidia"="Bacteroidia")

stool_taxa$class <- factor(stool_taxa$class, levels=c("< 2%", "Unassigned" ,"Actinobacteria","Negativicutes","α-proteobacteria",
                                                      "Coriobacteriia","Bacilli" ,"Bacteroidia","Clostridia"))

figS6A <- ggplot(filter(stool_taxa, anticodon %in% c("CTC","TTC")), aes(y=sum_count, x=sample, fill=class)) +
  geom_bar(stat="identity", position="stack") +
  coord_flip() +
  theme(legend.position = "right", axis.title.y = element_blank(),
        legend.title = element_blank()) +
  scale_fill_manual(values=c("#ffa900","#7b1979","#00acff","#3c97ac",
                             "#a6fb79","#ab82ff","#008f00","#021eaa","#b6192b"),guide=guide_legend(reverse=TRUE) ) +
  ylab("Fraction of reads") +
  #scale_y_discrete(position = "right") +
  theme(panel.border = element_blank(),
        panel.background = element_blank(),
        #panel.spacing = unit(c(0, 0, 0, 0), "cm"),
        legend.background = element_blank(),
        axis.ticks = element_blank(),
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0, size=12),
        strip.placement = "left",
        axis.text.y = element_text(),
        axis.text.x = element_blank(),
        axis.title.x = element_blank(),
        legend.text = element_text(size=8),
        legend.key.height = unit(c(0.4),"cm"),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0), "cm") ) +
  facet_grid(rows=vars(anticodon))
  
figS6A
```

##FigS6B
```{r}
tongue_counts_1_1_plus <- read.csv("../../2020 04 30 microbiome/taxonomy/tongue/tongue1_day1_demethylase_taxonomy/tongue1_day1_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Day 1", DM="+DM")
tongue_counts_1_1_minus <- read.csv("../../2020 04 30 microbiome/taxonomy/tongue/tongue1_day1_untreated_taxonomy/tongue1_day1_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Day 1", DM="-DM")
tongue_counts_1_2_plus <- read.csv("../../2020 04 30 microbiome/taxonomy/tongue/tongue1_day2_demethylase_taxonomy/tongue1_day2_demethylase_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Day 2", DM="+DM")
tongue_counts_1_2_minus <- read.csv("../../2020 04 30 microbiome/taxonomy/tongue/tongue1_day2_untreated_taxonomy/tongue1_day2_untreated_class_anticodon_counts.tsv", header=T, sep="\t") %>%
  mutate(sample="Day 2", DM="-DM")


tongue_taxa <- rbind(tongue_counts_1_1_plus, tongue_counts_1_1_minus, tongue_counts_1_2_plus, tongue_counts_1_2_minus) %>%
  mutate(class=X, count=X.1) %>%
  select(-X, -X.1) %>%
  #mutate(class = as.factor(class), sample= as.factor(sample), DM=as.factor(DM)) %>%
  group_by(class, DM, sample) %>%
  pivot_longer( c(-class, -DM, -sample), names_to="anticodon", values_to = "count") %>%
  #summarise(sum_count = sum(count, na.rm=T)) %>%
  group_by(DM, sample, anticodon) %>%
  mutate(sum_count = count / sum(count, na.rm=T)) %>%
  group_by() %>%
  mutate(class = as.character(class)) %>%
  mutate(class = ifelse(class=="", "Unassigned", class)) %>% #Get the unnamed class into the other chatagory
  mutate(class = ifelse(sum_count >=0.11, class, "< 11%" )) %>%
  mutate(class = as.factor(class)) %>%
  group_by(class, sample, DM, anticodon) %>%
  summarise(sum_count = sum(sum_count) )  %>%#combine all the things newly named "other"
  group_by()  %>%
  mutate(sample = fct_reorder(sample, desc(sample) ))
tongue_taxa$DM <- factor(tongue_taxa$DM, levels= c("-DM", "+DM"))

tongue_taxa <- tongue_taxa %>%
  filter(DM=="+DM")

tongue_taxa$class <- recode(tongue_taxa$class,  "< 11%"="< 11%","Bacilli"="Bacilli","Actinobacteria"="Actinobacteria",
                            "Clostridia"="Clostridia","Unassigned"="Unassigned")
tongue_taxa$class <- factor(tongue_taxa$class, levels = c("< 11%", "Unassigned","Clostridia","Actinobacteria","Negativicutes","Bacilli") )

figS6B <- ggplot(filter(tongue_taxa, anticodon %in% c("CTC","TTC")), aes(y=sum_count, x=sample, fill=class)) +
  geom_bar(stat="identity", position="stack") +
  coord_flip() +
  theme(legend.position = "right", axis.title.y = element_blank(),
        legend.title = element_blank()) +
  scale_fill_manual(values=c("#ffa900","#7b1979", "#b6192b", "#00acff", "#008f00"),guide=guide_legend(reverse=TRUE) ) +
  ylab("Fraction of reads") +
  #facet_grid(rows=vars(DM)) +
  theme(panel.border = element_blank(),
        panel.background = element_blank(),
       # panel.spacing = unit(c(0, 0, 0, 0), "cm"),
        axis.ticks = element_blank(),
        strip.background = element_blank(), 
        strip.text.y = element_text(angle=0, size=12),
        strip.placement = "left",
        axis.text.y = element_text(),
        axis.text.x = element_blank(),
        axis.title.x = element_blank(),
        plot.margin = unit(c(0.3, 0.3, 0.3, 0), "cm"),
        #legend.text = element_text(size=8),
        legend.background = element_blank(),
        legend.key.height = unit(c(0.4),"cm") ) +
  facet_grid((rows=vars(anticodon)))
figS6B

```


##FigS6C
```{r}
data1 <- read.csv("../../2020 04 30 microbiome/mapped reads/stool/TP-CK-3S-CW2_S2_L001L1_bc9.tsv", header = TRUE, sep="\t") %>%
  mutate(DM="-", sample="Stool 1")
data2 <- read.csv("../../2020 04 30 microbiome/mapped reads/stool/TP-CK-3S-CW2_S2_L001L1_bc10.tsv", header = TRUE, sep="\t") %>%
  mutate(DM="-", sample="Stool 2")
data3 <- read.csv("../../2020 04 30 microbiome/mapped reads/stool/TP-CK-3S-CW3_S3_L001L1_bc9.tsv", header = TRUE, sep="\t") %>%
  mutate(DM="+", sample="Stool 1")
data4 <- read.csv("../../2020 04 30 microbiome/mapped reads/stool/TP-CK-3S-CW3_S3_L001L1_bc10.tsv", header = TRUE, sep="\t") %>%
  mutate(DM="+", sample="Stool 2")

stool_basewise <- rbind(data1, data2, data3, data4) %>%
  group_by(gene, DM, sample) %>%
  mutate(norm_pileup = pileup / max(pileup)) %>%
  filter(gene == "Biffido_SerGGA") %>%
  group_by() %>%
  mutate(gene = "Bifidobacteriaceae\nSerGGA")



figS6C1 <- ggplot(filter(stool_basewise, position <77, position >61 ), 
                  aes(x=position, y=mutation, color=DM))+
  geom_line()+
  scale_color_manual(values=c("-"="black", "+"="red")) +
  facet_grid(cols=vars(sample), rows=vars(gene)) + 
  ylab("Mutation fraction\n") +
  xlab("Position (nt)") +
  scale_y_continuous(breaks=c(0,.5,1)) +
  theme(legend.position = c(.9, .6), plot.title = element_text(hjust=0.5)) +
  annotate(geom="text", label= "~m^1~A", x=68, y=.7, parse=TRUE) +
  annotate(geom="text", label="SerGGA", x=64, y=1.1) +
  theme(strip.background = element_blank(),
        strip.text.y = element_blank(),
        legend.background = element_blank())
figS6C1

figS6C2 <- ggplot(filter(stool_basewise, position <86 ), 
                  aes(x=position, y=norm_pileup, color=DM))+
  geom_line()+
  scale_color_manual(values=c("-"="black", "+"="red")) +
  facet_grid(cols=vars(sample), rows=vars(gene)) + 
  ylab("Mutation fraction\n") +
  xlab("Position (nt)") +
  scale_y_continuous(breaks=c(0,.5,1)) +
  theme(legend.position = c(.9, .3), plot.title = element_text(hjust=0.5)) +
  #annotate(geom="text", label= "~m^1~A", x=68, y=.7, parse=TRUE) +
  annotate(geom="text", label="SerGGA", x=12, y=1.1) +
  theme(strip.background = element_blank(),
        strip.text.y = element_blank(),
        legend.background = element_blank())
figS6C2

lay=rbind(c(1),c(2))
figS6C <- grid.arrange(figS6C2, figS6C1, layout_matrix=lay)
figS6C
```


##FigS6D
```{r}
data1 <- read.csv("../../2020 04 30 microbiome/mapped reads/tongue/tongue_TP-CK-TP1-10_S12bc1.tsv", header = TRUE, sep="\t") %>%
  mutate(DM="-", sample="Day 1")
data2 <- read.csv("../../2020 04 30 microbiome/mapped reads/tongue/tongue_TP-CK-TP1-10_S13bc1.tsv", header = TRUE, sep="\t") %>%
  mutate(DM="+", sample="Day 1")
data3 <- read.csv("../../2020 04 30 microbiome/mapped reads/tongue/tongue_TP-CK-TP1-10_S14bc1.tsv", header = TRUE, sep="\t") %>%
  mutate(DM="-", sample="Day 2")
data4 <- read.csv("../../2020 04 30 microbiome/mapped reads/tongue/tongue_TP-CK-TP1-10_S15bc1.tsv", header = TRUE, sep="\t") %>%
  mutate(DM="+", sample="Day 2")

tongue_basewise <- rbind(data1, data2, data3, data4) %>%
  group_by(gene, DM, sample) %>%
  mutate(norm_pileup = pileup / max(pileup)) %>%
  separate(gene, sep="-", c("species", "amino_acid", "anticodon", "n1","n2"))
tongue_basewise$species <- recode(tongue_basewise$species, "StreptococcTs_pyogenes_ATCC_19615_tRNA"="Streptococcus_pyogenes_ATCC_19615_tRNA","Veillonella_parvTla_DSM_2008_tRNA"="Veillonella_parvula_DSM_2008_tRNA")
tongue_basewise$amino_acid <- recode(tongue_basewise$amino_acid, "Ala"="Ala","Asp"="Asp","Asn"="Asn","Cys"="Cys","Gln"="Gln","GlT"="Glu",
                                                                  "Arg"="Arg","Gly"="Gly","His"="His","Ile"="Ile","Ile2"="Ile2","LeT"="Leu",
                                                                  "Lys"="Lys","Met"="Met","Phe"="Phe","Pro"="Pro","Ser"="Ser","Thr"="Thr",
                                                                  "Trp"="Trp","Tyr"="Tyr","Val"="Val","fMet"="fMer")


tongue_basewise <- tongue_basewise %>%
  filter(amino_acid != "Tnd") %>%
  separate(species, sep="_", c("genus", "species", "a1", "a2", "a3") ) %>%
  mutate(gene=paste0(genus, "\n",amino_acid, anticodon)) %>%
  filter(n1==1, n2==1)


figS6D1 <- ggplot(filter(tongue_basewise, gene %in% c("Veillonella\nSerGGA"), 
                         position <35, position >15 ), 
                  aes(x=position, y=mutation, color=DM))+
  geom_line()+
  scale_color_manual(values=c("-"="black", "+"="red")) +
  facet_grid(cols=vars(sample), rows=vars(gene)) + 
  ylab("Mutation fraction\n") +
  xlab("Position (nt)") +
  scale_y_continuous(breaks=c(0,.5,1)) +
  theme(legend.position = c(.9, .6), plot.title = element_text(hjust=0.5)) +
  annotate(geom="text", label= "~m^1~A", x=20, y=.8, parse=TRUE) +
  annotate(geom="text", label="SerGGA", x=18, y=1.1) +
  theme(strip.background = element_blank(),
        strip.text.y = element_blank(),
        legend.background = element_blank() )
figS6D1

figS6D2 <- ggplot(filter(tongue_basewise, gene %in% c("Veillonella\nSerGGA" ), position<90 ),
                  aes(x=position, y=norm_pileup, color=DM))+
  geom_line()+
  scale_color_manual(values=c("-"="black", "+"="red")) +
  facet_grid(cols=vars(sample), rows=vars(gene)) + 
  ylab("Normalized fraction\nof mapped reads ") +
  xlab("Position (nt)") +
  scale_y_continuous(breaks=c(0,.5,1), limits=c(0,1.1)) +
  theme(legend.position = c(.9, .5), plot.title = element_text(hjust=0.5)) +
  #annotate(geom="text", label= "~m^1~A", x=69, y=.7, parse=TRUE) +
  annotate(geom="text", label="SerGGA", x=12, y=1.1) +
  theme(strip.background = element_blank(),
        strip.text.y = element_blank(),
        legend.background = element_blank() )
figS6D2

lay=rbind(c(1),c(2))
figS6D <- grid.arrange(figS6D2, figS6D1, layout_matrix=lay)
figS6D
```

##Save figure S6
```{r}
myplota <- arrangeGrob(figS6A, left = textGrob(expression(bold("a")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotb <- arrangeGrob(figS6B, left = textGrob(expression(bold("b")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotc <- arrangeGrob(figS6C, left = textGrob(expression(bold("c")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotd <- arrangeGrob(figS6D, left = textGrob(expression(bold("d")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))


#Set the figure arrangement matrix
lay <- rbind(c(1, 2),
             c(3, 4))
#Arrange figures
figureS6 <- grid.arrange(myplota, myplotb, myplotc, myplotd,  layout_matrix = lay)
figureS6

ggsave("../figures/figureS6.svg", 
       plot = figureS6,
       scale = 1.3,
       width = 8, 
       height = 7)
ggsave("../figures/figureS6.png", 
       plot = figureS6,
       scale = 1.3,
       width = 8, 
       height = 7)

```

#_
#Figure Y
```{r}



```




#_
#_
#_
#_
#_
#_
#_
#_
#_













##HEK charging playground
```{r}


pileup_filter_value = 20
pileup_filter_position = c(30)
temp1 <- filter(HEK_base_data, position %in% pileup_filter_position) %>%
  pivot_wider(id_cols = c("gene_symbol","position"), names_from = c("stress","DM","rep"), values_from = pileup) %>%
  filter(None_m_1 > pileup_filter_value,
         heat_m_1 > pileup_filter_value,
         H2O2_m_1 > pileup_filter_value,
         AsO2_m_1 > pileup_filter_value,
         None_p_1 > pileup_filter_value,
         heat_p_1 > pileup_filter_value,
         H2O2_p_1 > pileup_filter_value,
         AsO2_p_1 > pileup_filter_value,
         None_m_2 > pileup_filter_value,
         heat_m_2 > pileup_filter_value,
         H2O2_m_2 > pileup_filter_value,
         AsO2_m_2 > pileup_filter_value,
         None_p_2 > pileup_filter_value,
         heat_p_2 > pileup_filter_value,
         H2O2_p_2 > pileup_filter_value,
         AsO2_p_2 > pileup_filter_value,
         None_m_3 > pileup_filter_value,
         heat_m_3 > pileup_filter_value,
         H2O2_m_3 > pileup_filter_value,
         AsO2_m_3 > pileup_filter_value,
         None_p_3 > pileup_filter_value,
         heat_p_3 > pileup_filter_value,
         H2O2_p_3 > pileup_filter_value,
         AsO2_p_3 > pileup_filter_value)





#Get the last base pileup
temp <- HEK_base_data %>%
  filter(gene_symbol %in% temp1$gene_symbol) %>%
  group_by(gene_symbol) %>%
  filter(position %in% c(max(position)) ) %>%
  group_by(gene_symbol, DM, stress, rep) %>%
  mutate(last_pileup = pileup) %>%
  select(last_pileup, gene_symbol, rep, DM, stress)

#calculate charging
temp <- HEK_base_data %>%
  filter(gene_symbol %in% temp1$gene_symbol) %>%
  group_by(gene_symbol) %>%
  filter(position %in% c(max(position)-1) ) %>%
  group_by(gene_symbol, DM, stress, rep) %>%
  full_join(., temp, by=c("gene_symbol","DM","rep","stress")) %>%
  mutate(charging = last_pileup / (pileup) )

#average replicates
temp <- temp %>%
  group_by(gene_symbol, DM, stress)  %>%
  summarise(mean_charging = mean(charging) )

#normalize to none stress
temp <- temp %>%
  group_by(gene_symbol, DM) %>%
  filter(stress=="None") %>%
  mutate(ns_charging = mean_charging) %>%
  select(DM, gene_symbol, ns_charging) %>%
  full_join(., temp, by=c("DM", "gene_symbol")) %>%
  mutate(norm_charging = mean_charging / ns_charging)

#Get AA and codon
temp <- temp %>%
  separate(gene_symbol, sep="-", c("AA","codon","number")) %>%
  mutate(gene_symbol = paste0(AA, codon, number),
         gene_symbol = paste0(AA,codon)) %>%
  separate(gene_symbol, sep="_", c("gene_symbol", "number"))

# #Collapse by AA
# temp <- temp %>%
#   group_by(stress, AA, DM, gene_symbol) %>%
#   summarise(norm_charging = mean(norm_charging))
# 
# 
# #A function to identify boxplot outliers
# is_outlier <- function(x) {
#   return(x < quantile(x, 0.25) - 1.5 * IQR(x)  | x > quantile(x, 0.75) + 1.5 * IQR(x) )
# }
# 
# #label outliar points
# temp <- temp %>%
#   group_by(stress) %>%
#   mutate(outlier = ifelse(is_outlier(norm_charging), gene_symbol, as.character(NA) ) )
# 
# 
# ggplot(filter(temp, DM=="-", stress!="None"), aes(x=stress, y=norm_charging)) +
#   #geom_jitter(position = position_jitter(seed = 1, width = .1) ) +
#   geom_boxplot(width=0.4) +
#   geom_text_repel(aes(label=outlier), size=2) + #position = position_jitter(seed = 1, width = .1),
#   #geom_hline(yintercept = 1, linetype=2, alpha=0.2) +
#   theme(axis.text.x = element_text(angle=0, vjust=.5),
#         axis.title.x = element_blank(),
#         legend.position="right") +
#   ylab("Fold change in fraction charged") +
#   coord_equal() +
#   theme(aspect.ratio = 1) +
#   theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))

ggplot(filter(temp), aes(x=gene_symbol, y=norm_charging) ) +
  geom_point() +
  theme(axis.text.x = element_text(angle=90, size=10)) +
  facet_grid(rows=vars(interaction(stress, DM)))



```

#_
#_
#A very HEK stress figure
##Bespoke data filtering
```{r}
#Get rid of unused sequences
HEK_base_data <- data_HEK %>%
  filter(gene_biotype == "tRNA") %>%
  filter(!grepl("spike", Gene)) %>%
  filter(!grepl("Ecoli", Gene)) %>%
  filter(!grepl("Yeast", Gene)) %>%
  filter(!grepl("SeCe",  Gene)) %>%
  select(Gene, pileup, position, mutation, gene_biotype, DM, stress, rep, base, stop) %>%
  select(-gene_biotype)

HEK_base_data$DM <- recode(HEK_base_data$DM, "-"="m", "+"="p")

#fix mitochondrial names
#Then rearrange all names
HEK_base_data <- HEK_base_data %>%
  filter(grepl("mt", Gene)) %>%
  mutate(Gene = paste0("Homo_sapiens_chrMT~trna1-z",Gene)) %>%
  rbind(., filter(HEK_base_data, !grepl("mt",Gene))) %>%
  separate(Gene, sep="_", c("homo","sapiens","gene_symbol"))%>%
  select(-homo, -sapiens) %>%
  separate(gene_symbol, sep="-", c("chrom","codon")) %>%
  mutate(gene_symbol = paste0(codon, "_",chrom)) %>%
  select(-chrom, -codon)

# #filter for tRNAs that have high expression under
# #EVERY CONDITION at base ...50?
pileup_filter_value = 2
pileup_filter_position = c(40)
temp1 <- filter(HEK_base_data, position %in% pileup_filter_position) %>%
  select(-stop) %>%
  pivot_wider(id_cols = c("gene_symbol","position"), names_from = c("stress","DM","rep"), values_from = pileup) %>%
  filter(None_m_1 > pileup_filter_value,
         #heat_m_1 > pileup_filter_value,
         #H2O2_m_1 > pileup_filter_value,
         #AsO2_m_1 > pileup_filter_value,
         None_p_1 > pileup_filter_value,
         #heat_p_1 > pileup_filter_value,
         #H2O2_p_1 > pileup_filter_value,
         #AsO2_p_1 > pileup_filter_value,
         None_m_2 > pileup_filter_value,
         #heat_m_2 > pileup_filter_value,
         #H2O2_m_2 > pileup_filter_value,
         #AsO2_m_2 > pileup_filter_value,
         None_p_2 > pileup_filter_value,
         #heat_p_2 > pileup_filter_value,
         #H2O2_p_2 > pileup_filter_value,
         #AsO2_p_2 > pileup_filter_value,
         None_m_3 > pileup_filter_value,
         #heat_m_3 > pileup_filter_value,
         #H2O2_m_3 > pileup_filter_value,
         #AsO2_m_3 > pileup_filter_value,
         None_p_3 > pileup_filter_value)
         #heat_p_3 > pileup_filter_value,
         #H2O2_p_3 > pileup_filter_value,
         #AsO2_p_3 > pileup_filter_value)


#fitler out eroniously long tRNA...
temp2 <- HEK_base_data %>%
  group_by(gene_symbol, DM, stress, rep) %>%
  summarize(position = max(position)) %>%
  filter(position <120) %>%
  select(gene_symbol)

#Apply filters to base data
HEK_base_data <- HEK_base_data %>%
  filter(gene_symbol %in% temp1$gene_symbol,
         gene_symbol %in% temp2$gene_symbol)

#Rename Met genes as elongator or initiator based on 5' A/C or G respectively
temp <- HEK_base_data %>%
  filter(grepl("Met", gene_symbol)) %>%
  filter(position==1) %>%
  mutate(gene_symbol2 = ifelse(base=="G", paste0("i",gene_symbol), gene_symbol) ) %>%
  filter(rep==1, stress=="None", DM=="m") %>%
  select(gene_symbol, gene_symbol2) 
#interlude to get the fMet genes to HEK_counts data later
temp_iMet <- temp
#Resume fMet filtering
temp <- temp %>%
  full_join(HEK_base_data, ., by=c("gene_symbol")) %>%
  mutate(gene_symbol = ifelse(is.na(gene_symbol2), gene_symbol, gene_symbol2) ) %>%
  select(-gene_symbol2) 
HEK_base_data <- temp

#FILTER FOR MOST ABUNDANT, and Make x labels nicer to read
temp <- HEK_base_data %>%
  mutate(gene_name=gene_symbol) %>%
  separate(gene_symbol, sep="_", c("gene_symbol", "number"))
temp1 <- temp %>%
  filter(stress=="None", DM=="p") %>%
  group_by(stress, DM, gene_symbol) %>%
  mutate(max_pileup = max(pileup)) %>%
  filter(pileup == max_pileup) %>%
  group_by() %>%
  select(gene_name)
temp <- temp %>%
  filter(gene_name %in% temp1$gene_name)

HEK_base_data <- temp
```

##HEK Chnge in abundance
```{r}
#Filter for tRNAs and average over rep
temp <- HEK_counts_data %>%
  group_by(DM, stress, rep) %>%
  mutate(rpm = counts / sum(counts) *1000000) %>%
  filter(gene_biotype=="tRNA") %>%
  group_by(Gene, gene_symbol, stress, DM) %>%
  mutate(mean_rpm = mean(rpm)) %>%
  group_by()

#temp$gene_symbol <- str_replace_all(temp$gene_symbol, ">mt", "trnaMT-mt") 
#Add iMet names
temp <- temp %>%
  group_by() %>%
  filter(grepl("mt", Gene)) %>%
  mutate(Gene = paste0("Homo_sapiens_chrMT.trna1-",Gene)) %>%
  rbind(filter(temp, !grepl("mt",Gene))) %>%
  separate(Gene, sep="_", c("homo","sapiens","gene_symbol")) %>%
  select(-homo, -sapiens) %>%
  separate(gene_symbol, sep="-", c("chrom","codon")) %>%
  mutate(gene_symbol = paste0(codon, "_",chrom)) %>%
  select(-chrom, -codon) %>%
  full_join(., temp_iMet, by=c("gene_symbol")) %>% #temp_fMet was defined in the above chunk
  mutate(gene_symbol = ifelse(is.na(gene_symbol2), gene_symbol, gene_symbol2)) %>%
  select(-gene_symbol2)


#Pool isodecoders
temp <- temp %>%
  mutate(gene_symbol2=gene_symbol) %>%
  separate(gene_symbol, sep="_", c("anticodon", "number")) %>%
  group_by(anticodon, DM, stress, rep) %>%
  summarise(group_mean_rpm = sum(mean_rpm),
            group_rpm = sum(rpm)) %>%
  filter(group_rpm >10)

#remove weird Isodecoders
temp <- temp %>%
  filter(anticodon %nin% c("Undet???","SubCTA","SupTTA", "SupCTA", "NA")) %>%
  filter(!is.na(anticodon))

#Normalize to no stress
temp <- temp %>%
  filter(stress=="None") %>%
  group_by() %>%
  mutate(ns_rpm = group_mean_rpm) %>%
  filter(rep==1) %>%
  select(ns_rpm, DM, anticodon) %>%
  full_join(temp, ., by=c("DM", "anticodon")) %>%
  mutate(norm_group_mean_rpm = group_mean_rpm / ns_rpm,
         norm_group_rpm = group_rpm / ns_rpm)

figXA <- ggplot(filter(temp, DM=="+"), aes(x=anticodon, y=norm_group_rpm))+
  geom_point() +
  geom_point(aes(y=norm_group_mean_rpm), shape=95, size=5) +
  geom_hline(yintercept = 1, linetype=2)+
  scale_y_continuous(trans="log2", breaks=c(0.66, 0.8, 1, 1.25, 1.5)) +
  theme(axis.text.x = element_text(angle=90, size=10, vjust = 0.5, hjust=1)) +
  theme(strip.background = element_blank()) +
  theme(strip.text.y = element_text(angle=0)) +
  facet_grid(rows=vars(stress)) +
  ylab("Fold change in RPM") +
  xlab("Anticodon")
figXA




#Split up mito etc
temp <- temp %>%
  mutate(source = "cytosolic")
temp <- temp %>%
  filter(grepl("mt", anticodon)) %>%
  mutate(source = "mitochondrial") %>%
  rbind(filter(temp, !grepl("mt", anticodon)))

#A function to identify boxplot outliers
is_outlier <- function(x) {
  return(x < quantile(x, 0.25) - 1.5 * IQR(x)  | x > quantile(x, 0.75) + 1.5 * IQR(x) )
  #return(quantile(x, 0.25) - 1.5 * IQR(x)   )
}

#label outliar points
temp1 <- temp %>%
  filter(rep== 2) %>%
  group_by(stress, DM, source) %>%
  mutate(outlier = ifelse(is_outlier(norm_group_mean_rpm), anticodon, as.character(NA) ) )


#Figure time
figXB <- ggplot(filter(temp1, DM=="+",  stress!="None"), aes(x=stress, y=norm_group_mean_rpm, fill=source )) +
  geom_boxplot(width=0.4, notch=F) +
  geom_text_repel(aes(label=outlier), size=2) + #position = position_jitter(seed = 1, width = .1),
  scale_y_continuous(trans="log2", breaks=c(0.66, 0.8, 1, 1.25, 1.5)) +
  scale_fill_manual(values=c("cytosolic"="white", "mitochondrial"="grey60")) +
  geom_hline(yintercept = 1, linetype=2, alpha=0.2) +
  theme(axis.text.x = element_text(angle=0, vjust=.5),
        axis.title.x = element_blank(),
        legend.position="right") +
  ylab("Fold change in abundance") +
  #coord_equal() +
  #theme(aspect.ratio = 1) +
  theme(strip.background = element_blank() ) +
  theme(strip.text.y = element_text(angle=0)) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))
figXB

figXBmito <- ggplot(filter(temp1, DM=="+",  stress!="None", source=="mitochondrial"), aes(x=stress, y=norm_group_mean_rpm )) +
  geom_boxplot(width=0.4, notch=F, fill="grey60") +
  geom_text_repel(aes(label=outlier), size=2) + #position = position_jitter(seed = 1, width = .1),
  scale_y_continuous(trans="log2", breaks=c(0.66, 0.8, 1, 1.25, 1.5)) +
  #scale_fill_manual(values=c("cytosolic"="white", "mitochondrial"="grey60")) +
  geom_hline(yintercept = 1, linetype=2, alpha=0.2) +
  theme(axis.text.x = element_text(angle=0, vjust=.5),
        axis.title.x = element_blank(),
        legend.position="right") +
  ylab("Fold change in abundance") +
  #coord_equal() +
  #theme(aspect.ratio = 1) +
  theme(strip.background = element_blank() ) +
  theme(strip.text.y = element_text(angle=0)) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))
figXBmito

figXBcyto <- ggplot(filter(temp1, DM=="+",  stress!="None", source=="cytosolic"), aes(x=stress, y=norm_group_mean_rpm )) +
  geom_boxplot(width=0.4, notch=F, fill="grey60") +
  geom_text_repel(aes(label=outlier), size=2) + #position = position_jitter(seed = 1, width = .1),
  scale_y_continuous(trans="log2", breaks=c(0.66, 0.8, 1, 1.25, 1.5)) +
  #scale_fill_manual(values=c("cytosolic"="white", "mitochondrial"="grey60")) +
  geom_hline(yintercept = 1, linetype=2, alpha=0.2) +
  theme(axis.text.x = element_text(angle=0, vjust=.5),
        axis.title.x = element_blank(),
        legend.position="right") +
  ylab("Fold change in abundance") +
  #coord_equal() +
  #theme(aspect.ratio = 1) +
  theme(strip.background = element_blank() ) +
  theme(strip.text.y = element_text(angle=0)) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))
figXBcyto


```


##HEK changes in charging
```{r}

#Doesn't really do anything
temp <- HEK_base_data %>%
  mutate(mutation = ifelse(pileup<1, NA, mutation)) %>%
  group_by(gene_symbol, position, DM, stress) %>%
  group_by()

#Split up mito etc
temp <- temp %>%
  mutate(source = "cytosolic")
temp <- temp %>%
  filter(grepl("z", gene_symbol)) %>%
  mutate(source = "mitochondrial") %>%
  rbind(filter(temp, !grepl("z", gene_symbol)))

# FILTER FOR MOST ABUNDANT, and Make x labels nicer to read
temp <- temp %>%
  separate(gene_symbol, sep="_", c("gene_symbol", "number")) %>%
  mutate(gene_name=paste0(gene_symbol,"_",number))
temp1 <- temp %>%
  filter(stress=="None", DM=="p") %>%
  group_by(stress, DM, gene_symbol) %>%
  mutate(max_pileup = max(pileup)) %>%
  filter(pileup == max_pileup) %>%
  group_by() %>%
  select(gene_name)
temp <- temp %>%
  filter(gene_name %in% temp1$gene_name) 




#calculate charging for each rep
temp <- temp %>%
  group_by(gene_symbol, DM, stress, rep) %>%
  filter(position %in% c(max(position)) ) %>%
  mutate(last_pileup = pileup) %>%
  group_by() %>%
  select(gene_symbol, DM, stress, rep, last_pileup) %>%
  full_join(temp, ., by=c("gene_symbol","DM","rep","stress")) %>%
  group_by(gene_symbol, DM, stress, rep) %>%
  filter(position == max(position)-1 ) %>%
  mutate(charging = last_pileup / pileup )

#Fix mt tRNA names, remove "z"
temp$gene_symbol <- str_replace_all(temp$gene_symbol, "z", "") 

#average replicates
temp <- temp %>%
  group_by(gene_symbol, DM, stress)  %>%
  mutate(mean_charging = mean(charging) )

#normalize to none stress
temp <- temp %>%
  group_by(gene_symbol, DM) %>%
  filter(stress=="None") %>%
  mutate(ns_charging = mean_charging) %>%
  select(DM, gene_symbol, ns_charging) %>%
  full_join(., temp, by=c("DM", "gene_symbol")) %>%
  mutate(norm_charging = mean_charging - ns_charging,
         norm_charging_rep = charging - ns_charging)

#Get AA and codon
temp <- temp %>%
  separate(gene_symbol, sep="_", c("codon","number")) %>%
  mutate(gene_symbol = codon) %>%
  separate(codon, sep=3, c("AA", "codon"))

figXC <- ggplot(filter(temp, DM=="p"), aes(x=gene_symbol, y=norm_charging_rep))+
  geom_point()+
  geom_point(aes(y=norm_charging), shape=95, size=5) +
  facet_grid(rows=vars(stress)) +
  ylab("Difference in fraction charged") +
  geom_hline(yintercept = 0, linetype=2, alpha=0.2) +
  theme(axis.text.x = element_text(angle=90, vjust=.5),
        axis.title.x = element_blank(),
        legend.position="right") +
  theme(strip.background = element_blank()) +
  theme(strip.text.y = element_text(angle=0)) 
figXC






#Split up mito etc
temp <- temp %>%
  mutate(source = "cytosolic")
temp <- temp %>%
  filter(grepl("mt", gene_symbol)) %>%
  mutate(source = "mitochondrial") %>%
  rbind(filter(temp, !grepl("mt", gene_symbol)))

#A function to identify boxplot outliers
is_outlier <- function(x) {
  return(x < quantile(x, 0.25) - 1.5 * IQR(x)  | x > quantile(x, 0.75) + 1.5 * IQR(x) )
}

#Combaine reps, to avoid odd problems
temp <- temp %>%
  group_by(gene_symbol, DM, stress, source) %>%
  summarise(norm_charging = mean(norm_charging, na.rm=T))

#label outliar points
temp1 <- temp %>%
  group_by(stress, DM, source) %>%
  mutate(outlier = ifelse(is_outlier(norm_charging), gene_symbol, as.character(NA) ) )

figXD <- ggplot(filter(temp1, DM=="p", stress!="None"), aes(x=stress, y=norm_charging, fill=source)) +
  geom_boxplot(width=0.4) +
  geom_text_repel(aes(label=outlier), size=2) + #position = position_jitter(seed = 1, width = .1),
  scale_fill_manual(values=c("cytosolic"="white", "mitochondrial"="grey60")) +
  geom_hline(yintercept = 0, linetype=2, alpha=0.2) +
  theme(axis.text.x = element_text(angle=0, vjust=.5),
        axis.title.x = element_blank(),
        legend.position="right") +
  ylab("Difference in fraction charged") +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))
figXD


figXDmito <- ggplot(filter(temp1, DM=="p", stress!="None", source=="mitochondrial"), aes(x=stress, y=norm_charging)) +
  geom_boxplot(width=0.4, fill="grey60") +
  geom_text_repel(aes(label=outlier), size=2) + #position = position_jitter(seed = 1, width = .1),
  #scale_fill_manual(values=c("cytosolic"="white", "mitochondrial"="grey60")) +
  geom_hline(yintercept = 0, linetype=2, alpha=0.2) +
  theme(axis.text.x = element_text(angle=0, vjust=.5),
        axis.title.x = element_blank(),
        legend.position="right") +
  ylab("Difference in fraction charged") +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))
figXDmito

figXDcyto <- ggplot(filter(temp1, DM=="p", stress!="None", source=="cytosolic"), aes(x=stress, y=norm_charging)) +
  geom_boxplot(width=0.4, fill="grey60") +
  geom_text_repel(aes(label=outlier), size=2) + #position = position_jitter(seed = 1, width = .1),
  #scale_fill_manual(values=c("cytosolic"="white", "mitochondrial"="grey60")) +
  geom_hline(yintercept = 0, linetype=2, alpha=0.2) +
  theme(axis.text.x = element_text(angle=0, vjust=.5),
        axis.title.x = element_blank(),
        legend.position="right") +
  ylab("Difference in fraction charged") +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))
figXDcyto


```

##Combine figure of abundance and charging
```{r}

myplot1 <- arrangeGrob(figXB+theme(legend.position="None"), left = textGrob(expression(bold("A")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top")) ) 
myplot2 <- arrangeGrob(figXD+theme(legend.position="None"), left = textGrob(expression(bold("B")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top")) )
plt_legend <- get_legend(figXD + theme(legend.title = element_blank(), legend.background = element_blank() ) )
lay = rbind(c(1,1,2,2,3))
figXBD <- grid.arrange(myplot1, myplot2, plt_legend, layout_matrix=lay) #, myplot2, layout_matrix = lay)
figXBD





myplot1 <- arrangeGrob(figXBmito+theme(legend.position="None"), left = textGrob(expression(bold("A")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top")) ) 
myplot2 <- arrangeGrob(figXDmito+theme(legend.position="None"), left = textGrob(expression(bold("B")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top")) )
lay = rbind(c(1,1,2,2))
figXBDmito <- grid.arrange(myplot1, myplot2, layout_matrix=lay, top=textGrob(expression("Mitochondrial tRNAs"))) #, myplot2, layout_matrix = lay)
figXBDmito





myplot1 <- arrangeGrob(figXBcyto+theme(legend.position="None"), left = textGrob(expression(bold("A")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top")) ) 
myplot2 <- arrangeGrob(figXDcyto+theme(legend.position="None"), left = textGrob(expression(bold("B")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top")) )
lay = rbind(c(1,1,2,2))
figXBDcyto <- grid.arrange(myplot1, myplot2, layout_matrix=lay, top=textGrob(expression("Cytosolic tRNAs"))) #, myplot2, layout_matrix = lay)
figXBDcyto



```


##HEK modification changes
```{r}

temp <- HEK_base_data %>%
  mutate(mutation = ifelse(pileup<1, NA, mutation)) %>%
  #filter(stress == "None") %>%
  group_by(gene_symbol, position, DM, stress) %>%
  summarise(mutation = mean(mutation),
            pileup=mean(pileup, na.rm=T)) %>%
  group_by()

#Split up mito etc
temp <- temp %>%
  mutate(source = "cytosolic")
temp <- temp %>%
  filter(grepl("z", gene_symbol)) %>%
  mutate(source = "mitochondrial") %>%
  rbind(filter(temp, !grepl("z", gene_symbol)))

# FILTER FOR MOST ABUNDANT, and Make x labels nicer to read
temp <- temp %>%
  separate(gene_symbol, sep="_", c("gene_symbol", "number")) %>%
  mutate(gene_name=paste0(gene_symbol,"_",number))
temp1 <- temp %>%
  filter(stress=="None", DM=="p") %>%
  group_by(stress, DM, gene_symbol) %>%
  mutate(max_pileup = max(pileup)) %>%
  filter(pileup == max_pileup) %>%
  group_by() %>%
  select(gene_name)
temp <- temp %>%
  filter(gene_name %in% temp1$gene_name)

#Normalize to no stress
temp <- temp %>%
  filter(stress=="None") %>%
  group_by() %>%
  mutate(ns_mutation = mutation) %>%
  select(position, gene_symbol, source, ns_mutation, DM) %>%
  full_join(., temp, by=c("position","gene_symbol","source", "DM")) %>%
  mutate(norm_mutation = mutation - ns_mutation) #%>%
  #mutate(norm_mutation = ifelse( norm_mutation <0.05 & norm_mutation >-0.05, 0, norm_mutation) ) 


temp1 <- filter(temp, stress !="None", source=="cytosolic", DM=="m")
temp1$position <- as.factor(temp1$position)


#The meh figure
figXE1 <- ggplot(temp1,
                aes(x=gene_symbol, y=position, fill=norm_mutation)) +
  geom_tile() +
  geom_tile(data=filter(temp1, norm_mutation <0.2 & norm_mutation >-0.2)) +
  scale_y_discrete(breaks=c(9,26,37,58), limits=rev(levels(temp1$position)))+
  scale_fill_gradient2(low="blue4", mid="grey70", midpoint=0, high="red4", na.value="grey70",
                       limits=c(-.2,.2), breaks=c(.2, .05,-.05,-.2)) +
  geom_tile(data=filter(temp1, norm_mutation>=  0.2), fill="red") +
  geom_tile(data=filter(temp1, norm_mutation<= -0.2), fill="blue") +
  theme(legend.position = "None",
        axis.text.x = element_text(angle = 90, hjust = 1, vjust=.5, size=10),
        axis.title.x = element_blank(),
        legend.text = element_text(angle=0, vjust=.5),
        plot.title = element_text(hjust = 0.5),
        axis.ticks.y = element_line(size=1),
        axis.ticks.length.y = unit(.2, "cm") )+ #element_text(hjust = 0.5)) +
  labs(#title="Mutation rate along tRNAs without demethylase treatment",
       fill="Mutation rate") +
  ylab("Position")  +
  labs(title="Nuclear-encoded") +
  theme(plot.title = element_text(hjust = 0) ) +
  theme(plot.margin = unit(c(0.3, 0.05, 0.3, 0.3), "cm"))+
  theme(strip.background = element_blank()) +
  theme(strip.text.y = element_blank()) +
  facet_grid(rows=vars(stress))
figXE1




#Fix "z" gene
temp$gene_symbol <- str_replace_all(temp$gene_symbol, "z", "")
temp1 <- filter(temp, stress !="None", source=="mitochondrial", DM=="m")
temp1$position <- as.factor(temp1$position)


figXE2 <- ggplot(temp1,
                aes(x=gene_symbol, y=position, fill= norm_mutation )) +
  geom_tile() +
  geom_tile(data=filter(temp1, norm_mutation <0.2 & norm_mutation >-0.2)) +
  #scale_y_reverse(breaks=c(9, 26, 37, 58), limits=c(89, 1) )+
  scale_y_discrete(breaks=c(9,26,37,58), limits=rev(levels(temp1$position))) +
  scale_fill_gradient2(low="blue4", mid="grey70", midpoint=0, high="red4", na.value="grey70",
                       limits=c(-.2,.2), breaks=c(.2, .05,-.05,-.2))+
  geom_tile(data=filter(temp1, norm_mutation >=  0.2), fill="red") +
  geom_tile(data=filter(temp1, norm_mutation <=  -0.2), fill="blue") +
  theme(legend.position = "None",
        axis.text.x = element_text(angle = 90, hjust = 1, vjust=.5, size=10),
        axis.title.x = element_blank(),
        legend.text = element_text(angle=0, vjust=.5),
        plot.title = element_text(hjust = 0.5),
        axis.ticks.y = element_line(size=1),
        axis.ticks.length.y = unit(.2, "cm") )+ #element_text(hjust = 0.5)) +
  labs(#title="Mutation rate along tRNAs without demethylase treatment",
       fill="Mutation rate") +
  ylab("Position")  +
  labs(title="Mitochondiral-encoded") +
  theme(plot.title = element_text(hjust = 0) ) +
  theme(#axis.title.y  = element_blank(),
        #axis.ticks.y = element_blank(),
        #axis.text.y = element_blank()
        ) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.05), "cm"))+
  theme(legend.position = "right") +
  theme(strip.background = element_blank()) +
  theme(strip.text.y = element_text(angle=0)) +
  facet_grid(rows=vars(stress))



#Wonkey legend games
plt_legend <- get_legend(figXE2 + theme(legend.title = element_blank(), legend.background = element_blank() ) )
plt_legend <- plt_legend #+ theme(legend.title = element_blank())
#make a >0.25 legend
legend_data1 <-data.frame(Mutation=c("> 0.20"), x=c(1), y=c(1))
legend_legend1 <- ggplot(legend_data1, aes(x=y,y=y,fill=Mutation))+
  geom_tile() + scale_fill_manual(values= c("> 0.20"="red")) +
  labs(fill="")+
  theme(legend.position = "right", legend.background = element_blank(),
        legend.key.height = unit(c(0.2),"cm"))
legend_legend1 <- get_legend(legend_legend1)
#make a separea <0.25 legend
legend_data2 <-data.frame(Mutation=c("< -0.20"), x=c(1), y=c(1))
legend_legend2 <- ggplot(legend_data2, aes(x=y,y=y,fill=Mutation))+
  geom_tile() + scale_fill_manual(values= c("< -0.20"="blue")) +
  theme(legend.position = "right", legend.background = element_blank(), legend.title = element_blank(),
        legend.key.height = unit(c(0.2),"cm"))
legend_legend2 <- get_legend(legend_legend2)

lay=rbind(c(5),c(1),c(2),c(3),c(3),c(4),c(5),c(5))
legend_legend1 <- arrangeGrob(empty_graph,left=legend_legend1)
plt_legend <- arrangeGrob(empty_graph, left=plt_legend)
legend_legend2 <- arrangeGrob(empty_graph, left=legend_legend2)
plt_legend <- arrangeGrob(text_grob("Difference in\nmutation rate"), legend_legend1, plt_legend, legend_legend2, empty_graph, layout_matrix = lay  )


myplot1 <- arrangeGrob(figXE1+theme(strip.text.y = element_text(angle=0),
                                    plot.title = element_blank()) +
                         ylab("Position along tRNA") )
myplot2<- arrangeGrob(figXE2+theme(legend.position = "None"))
lay = rbind(c(1,1,1,2,2))
figXE <- grid.arrange(myplot1) #, myplot2, layout_matrix = lay)
figXE


lay = rbind(c(1,1,1,1,1,2))
figXE <- grid.arrange(figXE, plt_legend, layout_matrix = lay)
figXE

#Make individual nuclear or mito heat maps
ggsave("../../../../grants/2020 05 18 tRNA stress R01/figures/cyto_stress_mod.png", 
       plot = figXE,
       scale = 1.0,
       width = 8, 
       height = 5)

```


##Make a trace for Thr with inset for stress mutation
```{r}
#Make a fun trace along ThrAGT
temp1 <- filter(temp, source=="cytosolic", DM=="m")
temp1$stress <- factor(temp1$stress, levels=c("None","heat","H2O2","AsO2"))

figXE_trace <- ggplot(filter(temp1, gene_symbol=="ThrAGT", stress %in% c("None", "AsO2","H2O2","heat")), aes(x=position, y=mutation, color=stress))+
  #geom_point() +
  geom_line(size=.4) +
  labs(color="Stress") +
  xlab("Position (nt)") +
  ylab("Mutation fraction") +
  annotate(geom="text", label="ThrAGT", x=4, y=.95) +
  scale_color_manual(values = c("None"="#ffa900",
                                "heat" ="#008f00",
                                "AsO2" ="#021eaa",
                                "H2O2"="#b6192b")) +
  theme(legend.position = "right") 
figXE_trace

#inset positions c(m1A59, I35, 3mC33, m1G27)



#Don't average reps
tempA <- HEK_base_data %>%
  mutate(mutation = ifelse(pileup<1, NA, mutation)) %>%
  #filter(stress == "None") %>%
  group_by(gene_symbol, position, DM, stress) %>%
  mutate(mean_mutation = mean(mutation),
            pileup=mean(pileup, na.rm=T)) %>%
  group_by()

#Split up mito etc
tempA <- tempA %>%
  mutate(source = "cytosolic")
tempA <- tempA %>%
  filter(grepl("z", gene_symbol)) %>%
  mutate(source = "mitochondrial") %>%
  rbind(filter(tempA, !grepl("z", gene_symbol)))

# FILTER FOR MOST ABUNDANT, and Make x labels nicer to read
tempA <- tempA %>%
  separate(gene_symbol, sep="_", c("gene_symbol", "number")) %>%
  mutate(gene_name=paste0(gene_symbol,"_",number))
temp2 <- tempA %>%
  filter(stress=="None", DM=="p") %>%
  group_by(stress, DM, gene_symbol) %>%
  mutate(max_pileup = max(pileup)) %>%
  filter(pileup == max_pileup) %>%
  group_by() %>%
  select(gene_name)
tempA <- tempA %>%
  filter(gene_name %in% temp2$gene_name)
tempA$stress <- factor(tempA$stress, levels=c("None","heat","H2O2","AsO2"))

figXE_trace_59<- ggplot(filter(tempA, position==59, gene_symbol=="ThrAGT", source=="cytosolic", DM=="m"), aes(x=stress, y=mutation)) +
  geom_point() +
  labs(title=bquote(~m^1~'A58')) +
  geom_point(aes(y=mean_mutation), shape=95, size=5) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(), plot.title=element_text(size=10),
        axis.text.x = element_text(angle=90))
  
figXE_trace_35<- ggplot(filter(tempA, position==35, gene_symbol=="ThrAGT", source=="cytosolic", DM=="m"), aes(x=stress, y=mutation)) +
  geom_point() +
  labs(title=bquote('I34')) +
  geom_point(aes(y=mean_mutation), shape=95, size=5) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(), plot.title=element_text(size=10),
        axis.text.x = element_text(angle=90))

figXE_trace_33<- ggplot(filter(tempA, position==33, gene_symbol=="ThrAGT", source=="cytosolic", DM=="m"), aes(x=stress, y=mutation)) +
  geom_point() +
  labs(title=bquote(~m^3~'C32')) +
  geom_point(aes(y=mean_mutation), shape=95, size=5) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(), plot.title=element_text(size=10),
        axis.text.x = element_text(angle=90))

figXE_trace_27<- ggplot(filter(tempA, position==27, gene_symbol=="ThrAGT", source=="cytosolic", DM=="m"), aes(x=stress, y=mutation)) +
  geom_point() +
  labs(title=bquote(~m[2]^2~'G26')) +
  geom_point(aes(y=mean_mutation), shape=95, size=5) +
  theme(axis.title.x = element_blank(), axis.title.y = element_blank(), plot.title=element_text(size=10),
        axis.text.x = element_text(angle=90))

lay=rbind(c(1,1,1,1),
          c(2,3,4,5))

figXE_trace_inset <- grid.arrange(figXE_trace+theme(legend.position=c(.85, 0.55), legend.background = element_blank()), 
             figXE_trace_27, figXE_trace_33, figXE_trace_35, figXE_trace_59, layout_matrix=lay)

#Make individual nuclear or mito heat maps
ggsave("../../../../grants/2020 05 18 tRNA stress R01/figures/cyto_stress_trace.png", 
       plot = figXE_trace_inset,
       scale = 1.3,
       width = 4, 
       height = 3)

```


##Make a trace for Cys and Trp cytosolic
```{r}

#Make a fun trace along ThrAGT
temp1 <- filter(temp, source=="cytosolic", DM=="m")
temp1$stress <- factor(temp1$stress, levels=c("None","heat","H2O2","AsO2"))

#Cys trace
figXE_trace <- ggplot(filter(temp1, grepl("SerCGA", gene_symbol), stress %in% c("None", "AsO2","H2O2","heat")), aes(x=position, y=mutation, color=stress))+
  geom_point() +
  geom_line(size=.4) +
  labs(color="Stress") +
  xlab("Position (nt)") +
  ylab("Mutation fraction") +
  #annotate(geom="text", label="CysGCA", x=4, y=.95) +
  scale_color_manual(values = c("None" ="grey60",
                                "heat" ="#008f00",
                                "AsO2" ="#021eaa",
                                "H2O2" ="#b6192b")) +
  theme(legend.position = "right") 
figXE_trace

#57 for trp and cys, 67 for Ser
figXE_trace_59_Cys<- ggplot(filter(tempA, position==57, gene_symbol=="CysGCA", source=="cytosolic", DM=="m"), aes(x=stress, y=mutation)) +
  geom_point() +
  labs(title= bquote(CysGCA ~m^1~'A58') ) +
  ylab("Mutation rate") +
  #annotate(geom="text", label=bquote(~m^1~'A58'), x=1, y=.15) +
  geom_point(aes(y=mean_mutation), shape=95, size=5) +
  theme(axis.title.x = element_blank(), 
        #axis.title.y = element_blank(), 
        plot.title=element_text(size=12),
        axis.text.x = element_text(angle=90)) +
  theme(plot.title =element_text(hjust=0.5) ) 
figXE_trace_59_Cys

#57 for trp and cys, 67 for Ser
figXE_trace_59_Trp <- ggplot(filter(tempA, position==57, gene_symbol=="TrpCCA", source=="cytosolic", DM=="m"), aes(x=stress, y=mutation)) +
  geom_point() +
  labs(title= bquote(TrpCCA ~m^1~'A58') ) +
  #annotate(geom="text", label=bquote(~m^1~'A58'), x=1, y=.55) +
  geom_point(aes(y=mean_mutation), shape=95, size=5) +
  theme(axis.title.x = element_blank(), 
        axis.title.y = element_blank(), 
        plot.title=element_text(size=12),
        axis.text.x = element_text(angle=90)) +
  theme(plot.title =element_text(hjust=0.5) )
figXE_trace_59_Trp

Cys_trp_trace <-grid.arrange(figXE_trace_59_Cys, figXE_trace_59_Trp, layout_matrix=rbind(c(1,2)))
Cys_trp_trace
# #Make individual nuclear or mito heat maps
# ggsave("../../../../grants/2020 05 18 tRNA stress R01/figures/Cys_trp_trace.png", 
#        plot = Cys_trp_trace,
#        scale = 1.4,
#        width = 3, 
#        height = 1.5)

```


##Make trace for mito
```{r}

#57 for trp and cys, 67 for Ser
figXE_trace_9_mtPhe<- ggplot(filter(tempA, position==9, grepl("PheGAA", gene_symbol), DM=="m", source=="mitochondrial"), aes(x=stress, y=mutation)) +
  geom_point() +
  labs(title= bquote(mtPheGAA ~m^1~'A9') ) +
  ylab("Mutation rate") +
  #annotate(geom="text", label=bquote(~m^1~'A58'), x=1, y=.15) +
  geom_point(aes(y=mean_mutation), shape=95, size=5) +
  theme(axis.title.x = element_blank(), 
        #axis.title.y = element_blank(), 
        plot.title=element_text(size=12),
        axis.text.x = element_text(angle=90)) +
  theme(plot.title =element_text(hjust=0.5) ) 
figXE_trace_9_mtPhe

#57 for trp and cys, 67 for Ser
figXE_trace_9_mtPro <- ggplot(filter(tempA, position==9, grepl("mtProTGG", gene_symbol), DM=="m"), aes(x=stress, y=mutation)) +
  geom_point() +
  labs(title= bquote(mtProTGG ~m^1~'A9') ) +
  ylab("Mutation rate") +
  #annotate(geom="text", label=bquote(~m^1~'A58'), x=1, y=.55) +
  geom_point(aes(y=mean_mutation), shape=95, size=5) +
  theme(axis.title.x = element_blank(), 
        #axis.title.y = element_blank(), 
        plot.title=element_text(size=12),
        axis.text.x = element_text(angle=90)) +
  theme(plot.title =element_text(hjust=0.5) )
figXE_trace_9_mtPro




#myplot1 <- arrangeGrob(figXBcyto+theme(legend.position="None"), left = textGrob(expression(bold("A")), x = unit(0, "npc")
#         , y = unit(1, "npc"), just=c("left","top")) ) 


A <- arrangeGrob(figXBmito, left = textGrob(expression(bold("C")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top")))
B <- arrangeGrob(figXDmito,left = textGrob(expression(bold("D")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top")))
C <- arrangeGrob(figXE_trace_9_mtPhe,left = textGrob(expression(bold("A")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top")))
D <- arrangeGrob(figXE_trace_9_mtPro,left = textGrob(expression(bold("B")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top")))

mt_Phe_Pro_trace <-grid.arrange(C,D,A,B, layout_matrix=rbind(c(1,2),c(3,4)))

#Make individual nuclear or mito heat maps
ggsave("../../../../grants/2020 05 18 tRNA stress R01/figures/mt_Phe_Pro_trace.png", 
       plot = mt_Phe_Pro_trace,
       scale = 1.4,
       width = 4, 
       height = 4)

```


##HEK stop changes
```{r}

temp <- HEK_base_data %>%
  mutate(mutation = ifelse(pileup<1, NA, mutation)) %>%
  #filter(stress == "None") %>%
  group_by(gene_symbol, position, DM, stress) %>%
  summarise(stop = mean(stop),
            pileup=mean(pileup, na.rm=T)) %>%
  group_by(gene_symbol, DM, stress) %>%
  mutate(pileup = pileup / max(pileup)) %>%
  group_by()

#Split up mito etc
temp <- temp %>%
  mutate(source = "cytosolic")
temp <- temp %>%
  filter(grepl("z", gene_symbol)) %>%
  mutate(source = "mitochondrial") %>%
  rbind(filter(temp, !grepl("z", gene_symbol)))

# FILTER FOR MOST ABUNDANT, and Make x labels nicer to read
temp <- temp %>%
  separate(gene_symbol, sep="_", c("gene_symbol", "number")) %>%
  mutate(gene_name=paste0(gene_symbol,"_",number))
temp1 <- temp %>%
  filter(stress=="None", DM=="m") %>%
  group_by(stress, DM, gene_symbol) %>%
  mutate(max_pileup = max(pileup)) %>%
  filter(pileup == max_pileup) %>%
  group_by() %>%
  select(gene_name)
temp <- temp %>%
  filter(gene_name %in% temp1$gene_name)

#Normalize to no stress
temp <- temp %>%
  filter(stress=="None") %>%
  group_by() %>%
  mutate(ns_stop = stop) %>%
  select(position, gene_symbol, source, ns_stop, DM) %>%
  full_join(., temp, by=c("position","gene_symbol","source", "DM")) %>%
  mutate(norm_stop = stop - ns_stop) #%>%
  #mutate(norm_mutation = ifelse( norm_mutation <0.05 & norm_mutation >-0.05, 0, norm_mutation) ) 


temp1 <- filter(temp, stress !="None", source=="cytosolic", DM=="m")
temp1$position <- as.factor(temp1$position)


#The meh figure
figXE1 <- ggplot(temp1,
                aes(x=gene_symbol, y=position, fill=norm_stop)) +
  geom_tile() +
  #geom_tile(data=filter(temp1, norm_stop <0.2 & norm_stop >-0.2)) +
  scale_y_discrete(breaks=c(9,26,37,58), limits=rev(levels(temp1$position)))+
  scale_fill_gradient2(low="blue4", mid="grey70", midpoint=0, high="red4", na.value="grey50",
                       limits=c(-1,1), breaks=c(1, .5,-.5,-1)) +
  geom_tile(data=filter(temp1, norm_stop>=  1), fill="red") +
  geom_tile(data=filter(temp1, norm_stop<= -1), fill="blue") +
  theme(legend.position = "None",
        axis.text.x = element_text(angle = 90, hjust = 1, vjust=.5, size=10),
        axis.title.x = element_blank(),
        legend.text = element_text(angle=0, vjust=.5),
        plot.title = element_text(hjust = 0.5),
        axis.ticks.y = element_line(size=1),
        axis.ticks.length.y = unit(.2, "cm") )+ #element_text(hjust = 0.5)) +
  labs(#title="Mutation rate along tRNAs without demethylase treatment",
       fill="Difference in stop rate") +
  ylab("Position")  +
  labs(title="Nuclear-encoded") +
  theme(plot.title = element_text(hjust = 0) ) +
  theme(plot.margin = unit(c(0.3, 0.05, 0.3, 0.3), "cm"))+
  theme(strip.background = element_blank()) +
  theme(strip.text.y = element_blank()) +
  facet_grid(rows=vars(stress))
figXE1



#Fix "z" gene
temp$gene_symbol <- str_replace_all(temp$gene_symbol, "z", "")
temp1 <- filter(temp, stress !="None", source=="mitochondrial", DM=="m")
temp1$position <- as.factor(temp1$position)


figXE2 <- ggplot(temp1,
                aes(x=gene_symbol, y=position, fill= norm_stop )) +
  geom_tile() +
  #geom_tile(data=filter(temp1, norm_stop <0.2 & norm_stop >-0.2)) +
  #scale_y_reverse(breaks=c(9, 26, 37, 58), limits=c(89, 1) )+
  scale_y_discrete(breaks=c(9,26,37,58), limits=rev(levels(temp1$position)))+
  scale_fill_gradient2(low="blue4", mid="grey70", midpoint=0, high="red4", na.value="grey50",
                       limits=c(-1,1), breaks=c(1, .5,-.5,-1))+
  geom_tile(data=filter(temp1, norm_stop >=  1), fill="red") +
  geom_tile(data=filter(temp1, norm_stop <=  -1), fill="blue") +
  theme(legend.position = "None",
        axis.text.x = element_text(angle = 90, hjust = 1, vjust=.5, size=10),
        axis.title.x = element_blank(),
        legend.text = element_text(angle=0, vjust=.5),
        plot.title = element_text(hjust = 0.5),
        axis.ticks.y = element_line(size=1),
        axis.ticks.length.y = unit(.2, "cm") )+ #element_text(hjust = 0.5)) +
  labs(#title="Mutation rate along tRNAs without demethylase treatment",
       fill="Difference in stop rate") +
  ylab("Position")  +
  labs(title="Mitochondiral-encoded") +
  theme(plot.title = element_text(hjust = 0) ) +
  theme(axis.title.y  = element_blank(),
        axis.ticks.y = element_blank(),
        axis.text.y = element_blank()) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.05), "cm"))+
  theme(legend.position = "right") +
  theme(strip.background = element_blank()) +
  theme(strip.text.y = element_text(angle=0)) +
  facet_grid(rows=vars(stress))



#Wonkey legend games
plt_legend <- get_legend(figXE1 + theme(legend.position = "right", legend.title = element_blank() ) )
plt_legend <- plt_legend #+ theme(legend.title = element_blank())
#make a >0.25 legend
legend_data1 <-data.frame(Mutation=c("> 1"), x=c(1), y=c(1))
legend_legend1 <- ggplot(legend_data1, aes(x=y,y=y,fill=Mutation))+
  geom_tile() + scale_fill_manual(values= c("> 1"="red")) +
  labs(fill="Difference in\nstop rate")+
  theme(legend.position = "right", legend.background = element_blank(),
        legend.key.height = unit(c(0.2),"cm"))
legend_legend1 <- get_legend(legend_legend1)
#make a separea <0.25 legend
legend_data2 <-data.frame(Mutation=c("< 1"), x=c(1), y=c(1))
legend_legend2 <- ggplot(legend_data2, aes(x=y,y=y,fill=Mutation))+
  geom_tile() + scale_fill_manual(values= c("< 1"="blue")) +
  theme(legend.position = "right", legend.background = element_blank(), legend.title = element_blank(),
        legend.key.height = unit(c(0.2),"cm"))
legend_legend2 <- get_legend(legend_legend2)

lay=rbind(c(4),c(4),c(1),c(2),c(2),c(3),c(4),c(4))
legend_legend1 <- arrangeGrob(empty_graph,left=legend_legend1)
plt_legend <- arrangeGrob(empty_graph, left=plt_legend)
legend_legend2 <- arrangeGrob(empty_graph, left=legend_legend2)
plt_legend <- arrangeGrob(legend_legend1, plt_legend, legend_legend2, empty_graph, layout_matrix = lay  )


myplot1 <- arrangeGrob(figXE1)
myplot2<- arrangeGrob(figXE2+theme(legend.position = "None"))
lay = rbind(c(1,1,1,2,2))
figXE <- grid.arrange(myplot1, myplot2, layout_matrix = lay)
figXE


lay = rbind(c(1,1,1,1,1,2))
figXE <- grid.arrange(figXE, plt_legend, layout_matrix = lay)
figXE



```


##Quantify individual modifcations
```{r}
temp <- HEK_base_data %>%
  mutate(gene_symbol = gene_name) %>%
  mutate(mutation = ifelse(pileup<1, NA, mutation)) %>%
  #filter(stress == "None") %>%
  group_by(position, DM, stress, gene_symbol, base) %>%
  summarise(mutation = mean(mutation),
            pileup=mean(pileup, na.rm=T)) %>%
  group_by()

#Split up mito etc
temp <- temp %>%
  mutate(source = "cytosolic")
temp <- temp %>%
  filter(grepl("z", gene_symbol)) %>%
  mutate(source = "mitochondrial") %>%
  rbind(filter(temp, !grepl("z", gene_symbol)))

#Fix "z" gene
temp$gene_symbol <- str_replace_all(temp$gene_symbol, "z", "") 
temp$stress <- factor(temp$stress, levels = c("None", "heat", "H2O2", "AsO2"))

# FILTER FOR MOST ABUNDANT, and Make x labels nicer to read
temp <- temp %>%
  separate(gene_symbol, sep="_", c("gene_symbol", "number")) %>%
  mutate(gene_name=paste0(gene_symbol,"_",number))
temp1 <- temp %>%
  filter(stress=="None", DM=="p") %>%
  group_by(stress, DM, gene_symbol) %>%
  mutate(max_pileup = max(pileup)) %>%
  filter(pileup == max_pileup) %>%
  group_by(DM, stress, gene_name) %>%
  filter(position == max(position)) %>%
  group_by() %>%
  select(gene_name)
temp <- temp %>%
  filter(gene_name %in% temp1$gene_name)


#temporary
#temp <- filter(temp, stress=="None")

#Aggregate m1A
#Ala
temp_m1A <- temp %>%
  filter(position %in% c(58), gene_name=="AlaAGC_chr2.trna3")
temp_m1A <- temp %>%
  filter(position %in% c(57), gene_name=="AlaCGC_chr6.trna119") %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(57), gene_name=="AlaTGC_chr12.trna13") %>% rbind(temp_m1A)
#Arg
temp_m1A <- temp %>%
  filter(position %in% c(58), gene_name=="ArgACG_chr14.trna7")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(58), gene_name=="ArgCCG_chr17.trna23")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(58), gene_name=="ArgCCT_chr7.trna3")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(58), gene_name=="ArgTCG_chr17.trna19")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(58), gene_name=="ArgTCT_chr1.trna9")  %>% rbind(temp_m1A)
#Asn
temp_m1A <- temp %>%
  filter(position %in% c(59), gene_name=="AsnGTT_chr1.trna107")  %>% rbind(temp_m1A)
#Asp
# temp_m1A <- temp %>%
#   filter(position %in% c(57), gene_name=="AspGTC_chr1.trna69")  %>% rbind(temp_m1A)
#Cys
temp_m1A <- temp %>%
  filter(position %in% c(57), gene_name=="CysGCA_chr17.trna15")  %>% rbind(temp_m1A)
#Gln
temp_m1A <- temp %>%
  filter(position %in% c(57), gene_name=="GlnCTG_chr15.trna7")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(57), gene_name=="GlnTTG_chr17.trna16")  %>% rbind(temp_m1A)
#Glu
temp_m1A <- temp %>%
  filter(position %in% c(57), gene_name=="GluCTC_chr1.trna116")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(57), gene_name=="GluTTC_chr1.trna134")  %>% rbind(temp_m1A)
#Gly
temp_m1A <- temp %>%
  filter(position %in% c(56), gene_name=="GlyCCC_chr16.trna34")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(56), gene_name=="GlyGCC_chr1.trna68")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(57), gene_name=="GlyTCC_chr1.trna117")  %>% rbind(temp_m1A)
#His
temp_m1A <- temp %>%
  filter(position %in% c(57), gene_name=="HisGTG_chr1.trna111")  %>% rbind(temp_m1A)
#Ile
temp_m1A <- temp %>%
  filter(position %in% c(59), gene_name=="IleAAT_chr6.trna57")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(59), gene_name=="IleTAT_chr19.trna10")  %>% rbind(temp_m1A)
#Leu
temp_m1A <- temp %>%
  filter(position %in% c(67), gene_name=="LeuAAG_chr14.trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(69), gene_name=="LeuCAA_chr1.trna58")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(68), gene_name=="LeuCAG_chr16.trna17")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(68), gene_name=="LeuTAA_chr6.trna83")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(67), gene_name=="LeuTAG_chr16.trna27")  %>% rbind(temp_m1A)
#Lys
temp_m1A <- temp %>%
  filter(position %in% c(58), gene_name=="LysCTT_chr1.trna119")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(58), gene_name=="LysTTT_chr1.trna54")  %>% rbind(temp_m1A)
#Met
temp_m1A <- temp %>%
  filter(position %in% c(58), gene_name=="MetCAT_chr8.trna10")  %>% rbind(temp_m1A)
#Phe
temp_m1A <- temp %>%
  filter(position %in% c(58), gene_name=="PheGAA_chr11.trna15")  %>% rbind(temp_m1A)
#Pro
temp_m1A <- temp %>%
  filter(position %in% c(57), gene_name=="ProAGG_chr1.trna65")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(57), gene_name=="ProCGG_chr1.trna52")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(57), gene_name=="ProTGG_chr14.trna6")  %>% rbind(temp_m1A)
#Sece
temp_m1A <- temp %>%
  filter(position %in% c(71), gene_name=="SeCeTCA_chr19.trna8")  %>% rbind(temp_m1A)
#Ser
temp_m1A <- temp %>%
  filter(position %in% c(67), gene_name=="SerCGA_chr12.trna2")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(67), gene_name=="SerGCT_chr15.trna10")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(67), gene_name=="SerTGA_chr10.trna2")  %>% rbind(temp_m1A)
#Thr
temp_m1A <- temp %>%
  filter(position %in% c(58), gene_name=="ThrAGT_chr17.trna36")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(57), gene_name=="ThrCGT_chr17.trna14")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(59), gene_name=="ThrTGT_chr6.trna127")  %>% rbind(temp_m1A)
#Trp
temp_m1A <- temp %>%
  filter(position %in% c(57), gene_name=="TrpCCA_chr12.trna6")  %>% rbind(temp_m1A)
#Tyr
temp_m1A <- temp %>%
  filter(position %in% c(58), gene_name=="TyrGTA_chr6.trna14")  %>% rbind(temp_m1A)
#Val
temp_m1A <- temp %>%
  filter(position %in% c(58), gene_name=="ValAAC_chr3.trna2")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(58), gene_name=="ValCAC_chr1.trna85")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(58), gene_name=="ValTAC_chr11.trna17")  %>% rbind(temp_m1A)

#Mitochondrial
#m1A9
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtAlaTGC_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtArgTCG_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtAsnGTT_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtAspGTC_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtCysGCA_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtGlnTTG_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtGluTTC_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtGlyTCC_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtHisGTG_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtIleGAT_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtLeuTAA_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtLeuTAG_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtLysTTT_chrMT~trna1")  %>% rbind(temp_m1A)
# temp_m1A <- temp %>%
#   filter(position %in% c(9), gene_name=="mtMetCAT_chrMT~trna1")
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtPheGAA_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtProTGG_chrMT~trna1")  %>% rbind(temp_m1A)
# temp_m1A <- temp %>%
#   filter(position %in% c(7,8,9,10), gene_name=="mtSerGCT_chrMT~trna1")
# temp_m1A <- temp %>%
#   filter(position %in% c(7,8,9,10), gene_name=="mtSerTGA_chrMT~trna1")
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtThrTGT_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtTrpTCA_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtTyrGTA_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(9), gene_name=="mtValTAC_chrMT~trna1")  %>% rbind(temp_m1A)

#mito m1A 58
temp_m1A <- temp %>%
  filter(position %in% c(60), gene_name=="mtLeuTAA_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(54), gene_name=="mtLysTTT_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(54), gene_name=="mtSerTGA_chrMT~trna1")  %>% rbind(temp_m1A)
temp_m1A <- temp %>%
  filter(position %in% c(53), gene_name=="mtThrTGT_chrMT~trna1")  %>% rbind(temp_m1A)



figXF <- ggplot(filter(temp_m1A, DM=="m"), aes(x=stress, y=mutation, color=source)) +
  geom_boxplot(width=0.4) +
  labs(title="m1A during stress") +
  ylab("Mutation rate at m1A sites") +
  theme(plot.title = element_text(hjust = 0) ) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))+
  theme(legend.position = "right") 
figXF 

```

##Save figure X
```{r}
myplota <- arrangeGrob(figXA, left = textGrob(expression(bold("a")), x = unit(0, "npc")
         , y   = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotb <- arrangeGrob(figXB, left = textGrob(expression(bold("b")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotc <- arrangeGrob(figXC, left = textGrob(expression(bold("c")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotd <- arrangeGrob(figXD, left = textGrob(expression(bold("d")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplote <- arrangeGrob(figXE, left = textGrob(expression(bold("e")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))
myplotf <- arrangeGrob(figXF, left = textGrob(expression(bold("f")), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top"),
         gp=gpar(col="black", fontsize=18, fontfamily="Arial")))


#Set the figure arrangement matrix
lay <- rbind(c(1,1,2),
             c(3,3,4),
             c(5,5,6))
#Arrange figures
figureX <- grid.arrange(myplote)#, myplotb, myplotc, myplotd, myplote, myplotf,  layout_matrix = lay)
figureX

# ggsave("../figures/figureX.svg", 
#        plot = figureX,
#        scale = 1.3,
#        width = 8, 
#        height = 7)
ggsave("../figures/figureXE.png", 
       plot = figureX,
       scale = 1.3,
       width = 12, 
       height = 5)


```


##More modifications
```{r}
#m1G
temp_m1G <- temp %>%
  filter(position %in% c(35,36,37,38,39), gene_name=="CysGCA_chr17.trna15") #Nothing here
temp_m1G <- temp %>%
  filter(position %in% c(37), gene_name=="HisGTG_chr1.trna111")
temp_m1G <- temp %>%
  filter(position %in% c(35), gene_name=="LeuAAG_chr14.trna1")
temp_m1G <- temp %>%
  filter(position %in% c(38), gene_name=="LeuCAG_chr16.trna17")
temp_m1G <- temp %>%
  filter(position %in% c(38), gene_name=="LeuTAG_chr16.trna27")
temp_m1G <- temp %>%
  filter(position %in% c(38), gene_name=="LeuCAA_chr1.trna58")
temp_m1G <- temp %>%
  filter(position %in% c(37), gene_name=="LeuTAA_chr6.trna83")
temp_m1G <- temp %>%
  filter(position %in% c(36), gene_name=="ProAGG_chr1.trna65")
temp_m1G <- temp %>%
  filter(position %in% c(36), gene_name=="ProCGG_chr1.trna52")
temp_m1G <- temp %>%
  filter(position %in% c(36), gene_name=="ProTGG_chr14.trna6")
temp_m1G <- temp %>%
  filter(position %in% c(37), gene_name=="ArgACG_chr14.trna7")
temp_m1G <- temp %>%
  filter(position %in% c(37), gene_name=="ArgCCG_chr17.trna23")
temp_m1G <- temp %>%
  filter(position %in% c(37), gene_name=="ArgTCG_chr17.trna19")
temp_m1G <- temp %>%
  filter(position %in% c(36), gene_name=="TrpCCA_chr12.trna6")
temp_m1G <- temp %>%
  filter(position %in% c(37), gene_name=="TyrGTA_chr6.trna14")



#MODOMICS BACKED

#reset variables
temp_m2G   = NA
temp_m22G  = NA
temp_I     = NA
temp_m1I   = NA
temp_om2G  = NA
temp_m7G   = NA
temp_dhU   = NA
temp_m3C   = NA
temp_m1G   = NA
temp_acp3U = NA
temp_q     = NA
temp_m6A   = NA
temp_m1A   = NA
temp_m5C   = NA
temp_om2C  = NA
temp_om25U = NA
temp_o2mP  = NA
temp_om2U  = NA
temp_W     = NA
temp_i6A   = NA
temp_m6t6A = NA
temp_galQ  = NA
temp_m1P   = NA


#By gene
#======================AlaAGC_chr2.trna3
temp_m2G <-  temp %>% filter(position %in% c(10), gene_name=="AlaAGC_chr2.trna3")
temp_m22G <- temp %>% filter(position %in% c(26), gene_name=="AlaAGC_chr2.trna3") #better with plus DM
temp_I <-    temp %>% filter(position %in% c(34), gene_name=="AlaAGC_chr2.trna3")
temp_m1I <-  temp %>% filter(position %in% c(37), gene_name=="AlaAGC_chr2.trna3")
temp_om2G <- temp %>% filter(position %in% c(39), gene_name=="AlaAGC_chr2.trna3")
temp_m7G <-  temp %>% filter(position %in% c(46), gene_name=="AlaAGC_chr2.trna3") #better with plus DM
temp_dhU <-  temp %>% filter(position %in% c(17), gene_name=="AlaAGC_chr2.trna3")
temp_dhU <-  temp %>% filter(position %in% c(20), gene_name=="AlaAGC_chr2.trna3") %>% rbind(temp_dhU)
#temp_m1A <-  temp %>% filter(position %in% c(58), gene_name=="AlaAGC_chr2.trna3")
#======================AlaTGC_chr12.trna13
temp_m2G <-  temp %>% filter(position %in% c(10), gene_name=="AlaTGC_chr12.trna13")  %>% rbind(temp_m2G)
temp_m22G <- temp %>% filter(position %in% c(24), gene_name=="AlaTGC_chr12.trna13")  %>% rbind(temp_m22G) #better with plus DM
#temp_I <-    temp %>% filter(position %in% c(33), gene_name=="AlaTGC_chr12.trna13") # %>% rbind(temp_I)
temp_m1I <-  temp %>% filter(position %in% c(36), gene_name=="AlaTGC_chr12.trna13")  %>% rbind(temp_m1I)
temp_om2G <- temp %>% filter(position %in% c(38), gene_name=="AlaTGC_chr12.trna13")  %>% rbind(temp_om2G)
temp_m7G <-  temp %>% filter(position %in% c(45), gene_name=="AlaTGC_chr12.trna13")  %>% rbind(temp_m7G) #better with plus DM
temp_dhU <-  temp %>% filter(position %in% c(16), gene_name=="AlaTGC_chr12.trna13")  %>% rbind(temp_dhU)
temp_dhU <-  temp %>% filter(position %in% c(19), gene_name=="AlaTGC_chr12.trna13")  %>% rbind(temp_dhU)
#temp_m1A <-  temp %>% filter(position %in% c(57), gene_name=="AlaCGC_chr6.trna119") # %>% rbind(temp_m1A)
#======================AlaTGC_chr12.trna13
temp_m2G <-  temp %>% filter(position %in% c(10), gene_name=="AlaTGC_chr12.trna13")  %>% rbind(temp_m2G)
temp_m22G <- temp %>% filter(position %in% c(24), gene_name=="AlaTGC_chr12.trna13")  %>% rbind(temp_m22G) #better with plus DM
#temp_I <-    temp %>% filter(position %in% c(33), gene_name=="AlaTGC_chr12.trna13")  %>% rbind(temp_I)
temp_m1I <-  temp %>% filter(position %in% c(36), gene_name=="AlaTGC_chr12.trna13")  %>% rbind(temp_m1I)
temp_om2G <- temp %>% filter(position %in% c(38), gene_name=="AlaTGC_chr12.trna13")  %>% rbind(temp_om2G)
temp_m7G <-  temp %>% filter(position %in% c(45), gene_name=="AlaTGC_chr12.trna13")  %>% rbind(temp_m7G) #better with plus DM
temp_dhU <-  temp %>% filter(position %in% c(16), gene_name=="AlaTGC_chr12.trna13")  %>% rbind(temp_dhU)
temp_dhU <-  temp %>% filter(position %in% c(19), gene_name=="AlaTGC_chr12.trna13")  %>% rbind(temp_dhU)
#temp_m1A <-  temp %>% filter(position %in% c(57), gene_name=="AlaTGC_chr12.trna13")  %>% rbind(temp_m1A)
#======================ArgACG_chr14.trna7
#======================ArgCCG_chr17
#======================ArgCCT_chr7.trna3
temp_m3C  <- temp %>% filter(position %in% c(32), gene_name=="ArgCCT_chr7.trna3") %>% rbind(temp_m3C)
#======================ArgTCG_chr17.trna19
#======================ArgTCT_chr1.trna9
temp_m3C  <- temp %>% filter(position %in% c(32), gene_name=="ArgTCT_chr1.trna9") %>% rbind(temp_m3C)
#======================AsnGTT_chr1.trna107
temp_m1G <-  temp %>% filter(position %in% c(9),  gene_name=="AsnGTT_chr1.trna107") %>% rbind(temp_m1G)
temp_m2G <-  temp %>% filter(position %in% c(10), gene_name=="AsnGTT_chr1.trna107") %>% rbind(temp_m2G)
temp_dhU <-  temp %>% filter(position %in% c(16), gene_name=="AsnGTT_chr1.trna107") %>% rbind(temp_dhU)
temp_dhU <-  temp %>% filter(position %in% c(20), gene_name=="AsnGTT_chr1.trna107") %>% rbind(temp_dhU)
temp_acp3U <-temp %>% filter(position %in% c(21), gene_name=="AsnGTT_chr1.trna107") %>% rbind(temp_acp3U)
temp_m22G <- temp %>% filter(position %in% c(27), gene_name=="AsnGTT_chr1.trna107") %>% rbind(temp_m22G)
temp_q <-    temp %>% filter(position %in% c(33), gene_name=="AsnGTT_chr1.trna107") %>% rbind(temp_q)
temp_m6A <-  temp %>% filter(position %in% c(38), gene_name=="AsnGTT_chr1.trna107") %>% rbind(temp_m6A)
temp_m7G <-  temp %>% filter(position %in% c(47), gene_name=="AsnGTT_chr1.trna107") %>% rbind(temp_m7G)
temp_dhU <-  temp %>% filter(position %in% c(48), gene_name=="AsnGTT_chr1.trna107") %>% rbind(temp_dhU)
temp_m1A <-  temp %>% filter(position %in% c(58), gene_name=="AsnGTT_chr1.trna107") %>% rbind(temp_m1A)
#======================AspGTC_chr1.trna69
temp_m1A <-  temp %>% filter(position %in% c(9),  gene_name=="AspGTC_chr1.trna69") %>% rbind(temp_m1A)
temp_m2G <-  temp %>% filter(position %in% c(10), gene_name=="AspGTC_chr1.trna69") %>% rbind(temp_m2G)
temp_q <-    temp %>% filter(position %in% c(34), gene_name=="AspGTC_chr1.trna69") %>% rbind(temp_q)
#======================CysGCA_chr17.trna15
#======================GlnCTG_chr15.trna7
temp_m1G <-  temp %>% filter(position %in% c(9),  gene_name=="GlnCTG_chr15.trna7") %>% rbind(temp_m1G)
temp_om2G <- temp %>% filter(position %in% c(17), gene_name=="GlnCTG_chr15.trna7") %>% rbind(temp_om2G)
temp_dhU <-  temp %>% filter(position %in% c(19), gene_name=="GlnCTG_chr15.trna7") %>% rbind(temp_dhU)
temp_dhU <-  temp %>% filter(position %in% c(20), gene_name=="GlnCTG_chr15.trna7") %>% rbind(temp_dhU)
temp_om2C <- temp %>% filter(position %in% c(32), gene_name=="GlnCTG_chr15.trna7") %>% rbind(temp_om2C)
temp_m5C <-  temp %>% filter(position %in% c(48), gene_name=="GlnCTG_chr15.trna7") %>% rbind(temp_m5C)
temp_m5C <-  temp %>% filter(position %in% c(48), gene_name=="GlnCTG_chr15.trna7") %>% rbind(temp_m5C)
temp_m1A <-  temp %>% filter(position %in% c(57), gene_name=="GlnCTG_chr15.trna7") %>% rbind(temp_m1A)
#======================GlnTTG_chr17.trna16
temp_m1G <-  temp %>% filter(position %in% c(9),  gene_name=="GlnTTG_chr17.trna16") %>% rbind(temp_m1G)
temp_om2G <- temp %>% filter(position %in% c(17), gene_name=="GlnTTG_chr17.trna16") %>% rbind(temp_om2G)
temp_dhU <-  temp %>% filter(position %in% c(19), gene_name=="GlnTTG_chr17.trna16") %>% rbind(temp_dhU)
temp_dhU <-  temp %>% filter(position %in% c(20), gene_name=="GlnTTG_chr17.trna16") %>% rbind(temp_dhU)
temp_om2C <- temp %>% filter(position %in% c(32), gene_name=="GlnTTG_chr17.trna16") %>% rbind(temp_om2C)
temp_m5C <-  temp %>% filter(position %in% c(48), gene_name=="GlnTTG_chr17.trna16") %>% rbind(temp_m5C)
temp_m5C <-  temp %>% filter(position %in% c(48), gene_name=="GlnTTG_chr17.trna16") %>% rbind(temp_m5C)
temp_m1A <-  temp %>% filter(position %in% c(57), gene_name=="GlnTTG_chr17.trna16") %>% rbind(temp_m1A)
#======================GluCTC_chr1.trna116
temp_m2G <-  temp %>% filter(position %in% c(9),  gene_name=="GluCTC_chr1.trna116") %>% rbind(temp_m2G)
temp_dhU <-  temp %>% filter(position %in% c(19), gene_name=="GluCTC_chr1.trna116") %>% rbind(temp_dhU)
temp_m5C <-  temp %>% filter(position %in% c(48), gene_name=="GluCTC_chr1.trna116") %>% rbind(temp_m5C)
temp_m5C <-  temp %>% filter(position %in% c(49), gene_name=="GluCTC_chr1.trna116") %>% rbind(temp_m5C)
temp_om25U<- temp %>% filter(position %in% c(53), gene_name=="GluCTC_chr1.trna116") %>% rbind(temp_om25U)
#======================GluTTC_chr1.trna134
#======================GlyCCC_chr16.trna34
temp_om2C <- temp %>% filter(position %in% c(4),  gene_name=="GlyCCC_chr16.trna34") %>% rbind(temp_om2C)
temp_m2G <-  temp %>% filter(position %in% c(6),  gene_name=="GlyCCC_chr16.trna34") %>% rbind(temp_m2G)
temp_dhU <-  temp %>% filter(position %in% c(19), gene_name=="GlyCCC_chr16.trna34") %>% rbind(temp_dhU)
temp_om2U <- temp %>% filter(position %in% c(32), gene_name=="GlyCCC_chr16.trna34") %>% rbind(temp_om2U)
temp_o2mP <- temp %>% filter(position %in% c(39), gene_name=="GlyCCC_chr16.trna34") %>% rbind(temp_o2mP)
temp_m5C <-  temp %>% filter(position %in% c(47), gene_name=="GlyCCC_chr16.trna34") %>% rbind(temp_m5C)
temp_m5C <-  temp %>% filter(position %in% c(48), gene_name=="GlyCCC_chr16.trna34") %>% rbind(temp_m5C)
temp_m1A <-  temp %>% filter(position %in% c(56), gene_name=="GlyCCC_chr16.trna34") %>% rbind(temp_m1A)
#======================GlyGCC_chr1.trna68
temp_om2U <- temp %>% filter(position %in% c(4),  gene_name=="GlyGCC_chr1.trna68") %>% rbind(temp_om2U)
temp_m2G <-  temp %>% filter(position %in% c(6),  gene_name=="GlyGCC_chr1.trna68") %>% rbind(temp_m2G)
temp_dhU <-  temp %>% filter(position %in% c(19), gene_name=="GlyGCC_chr1.trna68") %>% rbind(temp_dhU)
temp_m5C <-  temp %>% filter(position %in% c(38), gene_name=="GlyGCC_chr1.trna68") %>% rbind(temp_m5C)
temp_m5C <-  temp %>% filter(position %in% c(46), gene_name=="GlyGCC_chr1.trna68") %>% rbind(temp_m5C)
temp_m5C <-  temp %>% filter(position %in% c(47), gene_name=="GlyGCC_chr1.trna68") %>% rbind(temp_m5C)
temp_m5C <-  temp %>% filter(position %in% c(48), gene_name=="GlyGCC_chr1.trna68") %>% rbind(temp_m5C)
temp_m1A <-  temp %>% filter(position %in% c(56), gene_name=="GlyGCC_chr1.trna68") %>% rbind(temp_m1A)
#======================GlyTCC_chr1.trna117
#======================HisGTG_chr1.trna111
temp_dhU <-  temp %>% filter(position %in% c(16), gene_name=="HisGTG_chr1.trna111") %>% rbind(temp_dhU)
temp_dhU <-  temp %>% filter(position %in% c(19,20), gene_name=="HisGTG_chr1.trna111") %>% rbind(temp_dhU)
temp_m5C <-  temp %>% filter(position %in% c(47,48), gene_name=="HisGTG_chr1.trna111") %>% rbind(temp_m5C)
temp_m1A <-  temp %>% filter(position %in% c(57), gene_name=="HisGTG_chr1.trna111") %>% rbind(temp_m1A)
#======================IleAAT_chr6.trna57
#======================IleTAT_chr19.trna10
#======================LeuAAG_chr14.trna1
#======================LeuCAA_chr1.trna58
temp_m5C <-  temp %>% filter(position %in% c(33), gene_name=="LeuCAA_chr1.trna58") %>% rbind(temp_m5C)
#======================LeuCAG_chr16.trna17
temp_m3C <-  temp %>% filter(position %in% c(46), gene_name=="LeuCAG_chr16.trna17") %>% rbind(temp_m3C)
#======================LeuTAA_chr6.trna83
# sequence is all wrong. ignore
# temp_m2G <-  temp %>% filter(position %in% c(6), gene_name=="LeuTAA_chr6.trna83")
# temp_m2G <-  temp %>% filter(position %in% c(10), gene_name=="LeuTAA_chr6.trna83")
# temp_ac4C<-  temp %>% filter(position %in% c(12), gene_name=="LeuTAA_chr6.trna83")
# temp_dhU <-  temp %>% filter(position %in% c(19), gene_name=="LeuTAA_chr6.trna83")
# temp_m22G <- temp %>% filter(position %in% c(26), gene_name=="LeuTAA_chr6.trna83")
# temp_m1G  <- temp %>% filter(position %in% c(37), gene_name=="LeuTAA_chr6.trna83")
# #temp_m1G  <- temp %>% filter(position %in% c(38), gene_name=="LeuTAA_chr6.trna83")
# temp_om2U <- temp %>% filter(position %in% c(44), gene_name=="LeuTAA_chr6.trna83")
# temp_om2U <- temp %>% filter(position %in% c(45), gene_name=="LeuTAA_chr6.trna83")
# temp_m5C  <- temp %>% filter(position %in% c(48), gene_name=="LeuTAA_chr6.trna83")
# temp_m1A  <- temp %>% filter(position %in% c(58), gene_name=="LeuTAA_chr6.trna83")
#======================LeuTAG_chr16.trna27
#======================LysCTT_chr1.trna119
#======================LysTTT_chr1.trna54
#======================MetCAT_chr8.trna10
temp_m3C  <- temp %>% filter(position %in% c(20), gene_name=="MetCAT_chr8.trna10") %>% rbind(temp_m3C)
#======================PheGAA_chr11.trna15
temp_m2G <-  temp %>% filter(position %in% c(10), gene_name=="PheGAA_chr11.trna15") %>% rbind(temp_m2G)
temp_m1A <-  temp %>% filter(position %in% c(14), gene_name=="PheGAA_chr11.trna15") %>% rbind(temp_m1A)
temp_dhU <-  temp %>% filter(position %in% c(16), gene_name=="PheGAA_chr11.trna15") %>% rbind(temp_dhU)
temp_dhU <-  temp %>% filter(position %in% c(17), gene_name=="PheGAA_chr11.trna15") %>% rbind(temp_dhU)
temp_m22G<-  temp %>% filter(position %in% c(26), gene_name=="PheGAA_chr11.trna15") %>% rbind(temp_m22G)
temp_om2C<-  temp %>% filter(position %in% c(32), gene_name=="PheGAA_chr11.trna15") %>% rbind(temp_om2C)
temp_om2G<-  temp %>% filter(position %in% c(34), gene_name=="PheGAA_chr11.trna15") %>% rbind(temp_om2G)
temp_W   <-  temp %>% filter(position %in% c(37), gene_name=="PheGAA_chr11.trna15") %>% rbind(temp_W)
temp_m7G <-  temp %>% filter(position %in% c(46), gene_name=="PheGAA_chr11.trna15") %>% rbind(temp_m7G)
temp_dhU <-  temp %>% filter(position %in% c(47), gene_name=="PheGAA_chr11.trna15") %>% rbind(temp_dhU)
temp_m5C <-  temp %>% filter(position %in% c(49), gene_name=="PheGAA_chr11.trna15") %>% rbind(temp_m5C)
temp_m1A <-  temp %>% filter(position %in% c(58), gene_name=="PheGAA_chr11.trna15") %>% rbind(temp_m1A)
#======================ProAGG_chr1.trna65
#======================ProCGG_chr1.trna52
#======================ProTGG_chr14.trna6
#======================SeCeTCA_chr19.trna8
temp_i6A <-  temp %>% filter(position %in% c(39), gene_name=="SeCeTCA_chr19.trna8") %>% rbind(temp_i6A)
temp_m1A <-  temp %>% filter(position %in% c(58), gene_name=="PheGAA_chr11.trna15") %>% rbind(temp_m1A)
#======================SerAGA_chr17.trna35
temp_m3C  <- temp %>% filter(position %in% c(32), gene_name=="SerAGA_chr17.trna35") %>% rbind(temp_m3C)
temp_m3C  <- temp %>% filter(position %in% c(47), gene_name=="SerAGA_chr17.trna35") %>% rbind(temp_m3C)
#======================SerCGA_chr17.trna41
temp_m3C  <- temp %>% filter(position %in% c(32), gene_name=="SerCGA_chr17.trna41") %>% rbind(temp_m3C)
temp_m3C  <- temp %>% filter(position %in% c(47), gene_name=="SerCGA_chr17.trna41") %>% rbind(temp_m3C)
#======================SerGCT_chr6.trna175
temp_m6t6A<- temp %>% filter(position %in% c(39), gene_name=="SerGCT_chr6.trna175") %>% rbind(temp_m6t6A)
temp_m3C  <- temp %>% filter(position %in% c(49), gene_name=="SerGCT_chr6.trna175") %>% rbind(temp_m3C)
#======================SerTGA_chr6.trna172
temp_m3C  <- temp %>% filter(position %in% c(32), gene_name=="SerTGA_chr6.trna172") %>% rbind(temp_m3C)
temp_m3C  <- temp %>% filter(position %in% c(47), gene_name=="SerTGA_chr6.trna172") %>% rbind(temp_m3C)
#======================ThrAGT_chr17.trna40
temp_m3C  <- temp %>% filter(position %in% c(33), gene_name=="ThrAGT_chr17.trna40") %>% rbind(temp_m3C)
#======================ThrCGT_chr17.trna14
##Literally none
#======================ThrTGT_chr6.trna127
temp_m3C  <- temp %>% filter(position %in% c(33), gene_name=="ThrTGT_chr6.trna127") %>% rbind(temp_m3C)
#======================TrpCCA_chr12.trna6
#======================TyrGTA_chr6.trna14
temp_m2G <-  temp %>% filter(position %in% c(6),  gene_name=="TyrGTA_chr6.trna14") %>% rbind(temp_m2G)
temp_m2G <-  temp %>% filter(position %in% c(10), gene_name=="TyrGTA_chr6.trna14") %>% rbind(temp_m2G)
temp_dhU <-  temp %>% filter(position %in% c(16), gene_name=="TyrGTA_chr6.trna14") %>% rbind(temp_dhU)
temp_dhU <-  temp %>% filter(position %in% c(17), gene_name=="TyrGTA_chr6.trna14") %>% rbind(temp_dhU)
temp_acp3U <-temp %>% filter(position %in% c(20), gene_name=="TyrGTA_chr6.trna14") %>% rbind(temp_acp3U)
temp_m22G<-  temp %>% filter(position %in% c(26), gene_name=="TyrGTA_chr6.trna14") %>% rbind(temp_m22G)
temp_m22G<-  temp %>% filter(position %in% c(27), gene_name=="TyrGTA_chr6.trna14") %>% rbind(temp_m22G)
temp_galQ<-  temp %>% filter(position %in% c(34), gene_name=="TyrGTA_chr6.trna14") %>% rbind(temp_galQ)
temp_m1G <-  temp %>% filter(position %in% c(37), gene_name=="TyrGTA_chr6.trna14") %>% rbind(temp_m1G)
temp_m1P <-  temp %>% filter(position %in% c(39), gene_name=="TyrGTA_chr6.trna14") %>% rbind(temp_m1P)
temp_m7G <-  temp %>% filter(position %in% c(46), gene_name=="TyrGTA_chr6.trna14") %>% rbind(temp_m7G)
temp_dhU <-  temp %>% filter(position %in% c(47), gene_name=="TyrGTA_chr6.trna14") %>% rbind(temp_dhU)
temp_m5C <-  temp %>% filter(position %in% c(48), gene_name=="TyrGTA_chr6.trna14") %>% rbind(temp_m5C)
temp_m1A <-  temp %>% filter(position %in% c(58), gene_name=="TyrGTA_chr6.trna14") %>% rbind(temp_m1A)
#======================ValAAC_chr3.trna2
temp_dhU <-  temp %>% filter(position %in% c(16), gene_name=="ValAAC_chr3.trna2") %>% rbind(temp_dhU)
temp_dhU <-  temp %>% filter(position %in% c(19), gene_name=="ValAAC_chr3.trna2") %>% rbind(temp_dhU)
temp_dhU <-  temp %>% filter(position %in% c(20), gene_name=="ValAAC_chr3.trna2") %>% rbind(temp_dhU)
temp_m2G <-  temp %>% filter(position %in% c(26), gene_name=="ValAAC_chr3.trna2") %>% rbind(temp_m2G)
temp_m7G <-  temp %>% filter(position %in% c(46), gene_name=="ValAAC_chr3.trna2") %>% rbind(temp_m7G)
temp_dhU <-  temp %>% filter(position %in% c(47), gene_name=="ValAAC_chr3.trna2") %>% rbind(temp_dhU)
temp_m5C <-  temp %>% filter(position %in% c(48), gene_name=="ValAAC_chr3.trna2") %>% rbind(temp_m5C)
temp_m5C <-  temp %>% filter(position %in% c(49), gene_name=="ValAAC_chr3.trna2") %>% rbind(temp_m5C)
temp_m1A <-  temp %>% filter(position %in% c(58), gene_name=="ValAAC_chr3.trna2") %>% rbind(temp_m1A)
#======================ValCAC_chr1.trna85
temp_dhU <-  temp %>% filter(position %in% c(16), gene_name=="ValCAC_chr1.trna85") %>% rbind(temp_dhU)
temp_dhU <-  temp %>% filter(position %in% c(19), gene_name=="ValCAC_chr1.trna85") %>% rbind(temp_dhU)
temp_dhU <-  temp %>% filter(position %in% c(20), gene_name=="ValCAC_chr1.trna85") %>% rbind(temp_dhU)
temp_m2G <-  temp %>% filter(position %in% c(26), gene_name=="ValCAC_chr1.trna85") %>% rbind(temp_m2G)
temp_m7G <-  temp %>% filter(position %in% c(46), gene_name=="ValCAC_chr1.trna85") %>% rbind(temp_m7G)
temp_dhU <-  temp %>% filter(position %in% c(47), gene_name=="ValCAC_chr1.trna85") %>% rbind(temp_dhU)
temp_m5C <-  temp %>% filter(position %in% c(48), gene_name=="ValCAC_chr1.trna85") %>% rbind(temp_m5C)
temp_m5C <-  temp %>% filter(position %in% c(49), gene_name=="ValCAC_chr1.trna85") %>% rbind(temp_m5C)
temp_m1A <-  temp %>% filter(position %in% c(58), gene_name=="ValCAC_chr1.trna85") %>% rbind(temp_m1A)
#======================ValTAC_chr11.trna17
temp_dhU <-  temp %>% filter(position %in% c(16), gene_name=="ValTAC_chr11.trna17") %>% rbind(temp_dhU)
temp_dhU <-  temp %>% filter(position %in% c(19), gene_name=="ValTAC_chr11.trna17") %>% rbind(temp_dhU)
temp_dhU <-  temp %>% filter(position %in% c(20), gene_name=="ValTAC_chr11.trna17") %>% rbind(temp_dhU)
temp_m2G <-  temp %>% filter(position %in% c(26), gene_name=="ValTAC_chr11.trna17") %>% rbind(temp_m2G)
temp_m7G <-  temp %>% filter(position %in% c(46), gene_name=="ValTAC_chr11.trna17") %>% rbind(temp_m7G)
temp_dhU <-  temp %>% filter(position %in% c(47), gene_name=="ValTAC_chr11.trna17") %>% rbind(temp_dhU)
temp_m5C <-  temp %>% filter(position %in% c(48), gene_name=="ValTAC_chr11.trna17") %>% rbind(temp_m5C)
temp_m5C <-  temp %>% filter(position %in% c(49), gene_name=="ValTAC_chr11.trna17") %>% rbind(temp_m5C)
temp_m1A <-  temp %>% filter(position %in% c(58), gene_name=="ValTAC_chr11.trna17") %>% rbind(temp_m1A)
#======================


temp_m2G   <- temp_m2G   %>% mutate(modification = "m2G")
temp_m22G  <- temp_m22G  %>% mutate(modification = "m22G")
temp_I     <- temp_I     %>% mutate(modification = "I")
temp_m1I   <- temp_m1I   %>% mutate(modification = "m1I")
temp_om2G  <- temp_om2G  %>% mutate(modification = "om2G")
temp_m7G   <- temp_m7G   %>% mutate(modification = "m7G")
temp_dhU   <- temp_dhU   %>% mutate(modification = "dhU")
temp_m3C   <- temp_m3C   %>% mutate(modification = "m3C")
temp_m1G   <- temp_m1G   %>% mutate(modification = "m1G")
temp_acp3U <- temp_acp3U %>% mutate(modification = "acp3U")
temp_q     <- temp_q     %>% mutate(modification = "Q")
temp_m6A   <- temp_m6A   %>% mutate(modification = "m6A")
temp_m1A   <- temp_m1A   %>% mutate(modification = "m1A")
temp_m5C   <- temp_m5C   %>% mutate(modification = "m5C")
temp_om2C  <- temp_om2C  %>% mutate(modification = "om2C")
temp_om25U <- temp_om25U %>% mutate(modification = "om25C")
temp_o2mP  <- temp_o2mP  %>% mutate(modification = "om2P")
temp_om2U  <- temp_om2U  %>% mutate(modification = "om2U")
temp_W     <- temp_W     %>% mutate(modification = "W")
temp_i6A   <- temp_i6A   %>% mutate(modification = "i6A")
temp_m6t6A <- temp_m6t6A %>% mutate(modification = "m6t6A")
temp_galQ  <- temp_galQ  %>% mutate(modification = "galQ")
temp_m1P   <- temp_m1P   %>% mutate(modification = "m1P")

temp_mods <- rbind(temp_m2G,   temp_m22G,  temp_I,    temp_m1I,
                   temp_om2G,  temp_m7G,   temp_dhU,  temp_m3C, 
                   temp_m1G,   temp_acp3U, temp_q,    temp_m6A,
                   temp_m1A,   temp_m5C,   temp_om2C, temp_om25U,
                   temp_o2mP,  temp_om2U,  temp_W,    temp_i6A, 
                   temp_m6t6A, temp_galQ,  temp_m1P)

temp_mods <- temp_mods %>%
  filter(gene_name %nin% c(NA)) %>%
  filter(pileup > 50)
  #filter(stress !="AsO2")

ggplot(data=temp_mods, aes(x=stress, y=mutation, color=DM)) +
  #geom_point() +
  #geom_line() +
  geom_boxplot(width=0.4) +
  theme(legend.position = "right") +
  scale_y_log10() +
  facet_wrap(~modification, scales="free_y") 

#M2G
ggplot(data=filter(temp_mods, DM=="m", modification=="m2G", stress!="AsO2"), aes(x=stress, y=mutation)) +
  geom_boxplot(color="blue") +
  geom_point() +
  geom_line(aes(group=interaction(gene_name, position)))

#acp3U
ggplot(data=filter(temp_mods, DM=="p", modification=="acp3U", stress!="AsO2"), aes(x=stress, y=mutation)) +
  #geom_boxplot(color="blue") +
  geom_point() +
  geom_line(aes(group=interaction(gene_name, position)))

#om2G
ggplot(data=filter(temp_mods, DM=="p", modification=="om2G", stress!="AsO2"), aes(x=stress, y=mutation)) +
  #geom_boxplot(color="blue") +
  geom_point() +
  geom_line(aes(group=interaction(gene_name, position)))


```

test graph
```{r}

ggplot(data= temp_I , aes(x=stress, y=mutation)) +
  geom_boxplot()


```

#_
#_
#Brief look for tRFs during stress
```{r}

#Get rid of unused sequences
temp <- data_HEK %>%
  filter(gene_biotype == "tRNA") %>%
  filter(!grepl("spike", Gene)) %>%
  filter(!grepl("Ecoli", Gene)) %>%
  filter(!grepl("Yeast", Gene)) %>%
  filter(!grepl("SeCe",  Gene)) %>%
  select(Gene, pileup, position, mutation, gene_biotype, DM, stress, rep, base) %>%
  select(-gene_biotype)

temp$DM <- recode(temp$DM, "-"="m", "+"="p")

#fix mitochondrial names
#Then rearrange all names
temp <- temp %>%
  filter(grepl("mt", Gene)) %>%
  mutate(Gene = paste0("Homo_sapiens_chrMT~trna1-z",Gene)) %>%
  rbind(., filter(temp, !grepl("mt",Gene))) %>%
  separate(Gene, sep="_", c("homo","sapiens","gene_symbol"))%>%
  select(-homo, -sapiens) %>%
  separate(gene_symbol, sep="-", c("chrom","codon")) %>%
  mutate(gene_symbol = paste0(codon, "_",chrom)) %>%
  select(-chrom, -codon)

temp$DM <- recode(temp$DM, "m"="-", "p"="+")
temp$stress <- factor(temp$stress, levels = c("None","AsO2","H2O2","heat"))
temp$gene_symbol <- str_replace_all(temp$gene_symbol, "z", "") #Fix Z mitochondria names


temp1 <- temp %>% 
  filter(grepl("Trp", gene_symbol)) %>%
  filter(position < 73) %>%
  group_by(rep, DM, stress, gene_symbol) %>%
  mutate(pileup = pileup / max(pileup))

ggplot(filter(temp1, DM=="-"), aes(x=position, y=pileup, color=stress, group=interaction(rep, DM, stress)) ) +
  geom_line() +
  scale_y_continuous(breaks=c(0,.5,1)) +
  geom_vline(xintercept = 31, linetype=2, alpha=0.2) +
  #scale_color_manual(values = c("-"="black","+"="red"))+
  theme(legend.position = "right",
        plot.title = element_text(hjust=0.5)) +
  ylab("Normalized fraction\nof reads mapped")+
  xlab("Position (nt)")+
  theme(aspect.ratio = 0.7) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"),
        legend.background = element_blank(),
        strip.background = element_blank(),
        strip.text.y = element_text(angle=0)) +
  facet_wrap(~gene_symbol) 


#ArgACG_3.11
#Glu is crazy

```

#_
#_
#A vry detailed look at tRNA fragments
##Read in HEK tRF base data
```{read in HEK basewise data}
setwd("~/4SR/data/2020 05 11 tRF analysis/")

#Rep1 minus DM
data1 = read.csv("./TP-CK1-RPL7_S1_L004bc8.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="-", rep=1, stress="None")

data2 = read.csv("./TP-CK1-RPL7_S1_L004bc9.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype !="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="-", rep=1, stress="heat")

data3 = read.csv("./TP-CK1-RPL7_S1_L004bc11.tsv", header=T, sep="\t") %>%
  #xfilter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="-", rep=1, stress="H2O2")

data4 = read.csv("./TP-CK1-RPL7_S1_L004bc12.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="-", rep=1, stress="AsO2")
data_HEK_tRF <- rbind(data1, data2, data3, data4)

#Rep1 plus DM
data1 = read.csv("./TP-CK1-RPL10_S2_L004bc8.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="+", rep=1, stress="None")

data2 = read.csv("./TP-CK1-RPL10_S2_L004bc9.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="+", rep=1, stress="heat")

data3 = read.csv("./TP-CK1-RPL10_S2_L004bc11.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="+", rep=1, stress="H2O2")

data4 = read.csv("./TP-CK1-RPL10_S2_L004bc12.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="+", rep=1, stress="AsO2")
data_HEK_tRF <- rbind(data_HEK_tRF, data1, data2, data3, data4)


#Rep2 minus DM
data1 = read.csv("./TP-CK1-RPL11_S3_L004bc8.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="-", rep=2, stress="None")

data2 = read.csv("./TP-CK1-RPL11_S3_L004bc9.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="-", rep=2, stress="heat")

data3 = read.csv("./TP-CK1-RPL11_S3_L004bc11.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="-", rep=2, stress="H2O2")

data4 = read.csv("./TP-CK1-RPL11_S3_L004bc12.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="-", rep=2, stress="AsO2")
data_HEK_tRF <- rbind(data_HEK_tRF, data1, data2, data3, data4)


#Rep2 plus DM
data1 = read.csv("./TP-CK1-RPL12_S4_L004bc8.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="+", rep=2, stress="None")

data2 = read.csv("./TP-CK1-RPL12_S4_L004bc9.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="+", rep=2, stress="heat")

data3 = read.csv("./TP-CK1-RPL12_S4_L004bc11.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="+", rep=2, stress="H2O2")

data4 = read.csv("./TP-CK1-RPL12_S4_L004bc12.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="+", rep=2, stress="AsO2")
data_HEK_tRF <- rbind(data_HEK_tRF, data1, data2, data3, data4)


#Rep3 minus DM
data1 = read.csv("./TP-CK1-RPL13_S5_L004bc8.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="-", rep=3, stress="None")

data2 = read.csv("./TP-CK1-RPL13_S5_L004bc9.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="-", rep=3, stress="heat")

data3 = read.csv("./TP-CK1-RPL13_S5_L004bc11.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="-", rep=3, stress="H2O2")

data4 = read.csv("./TP-CK1-RPL13_S5_L004bc12.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="-", rep=3, stress="AsO2")
data_HEK_tRF <- rbind(data_HEK_tRF, data1, data2, data3, data4)


#Rep3 plus DM
data1 = read.csv("./TP-CK1-RPL14_S6_L004bc8.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="+", rep=3, stress="None")

data2 = read.csv("./TP-CK1-RPL14_S6_L004bc9.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="+", rep=3, stress="heat")

data3 = read.csv("./TP-CK1-RPL14_S6_L004bc11.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="+", rep=3, stress="H2O2")

data4 = read.csv("./TP-CK1-RPL14_S6_L004bc12.tsv", header=T, sep="\t") %>%
  #filter(gene_biotype!="protein_coding") %>%
  #select(Gene, pileup, position, mutation, gene_biotype) %>%
  mutate(DM="+", rep=3, stress="AsO2")

data_HEK_tRF <- rbind(data_HEK_tRF, data1, data2, data3, data4)



```

##Extra HEK base filter
```{r}


#Get rid of unused sequences
HEK_base_data_tRF <- data_HEK_tRF %>%
  filter(!grepl("spike", gene_name)) %>%
  filter(!grepl("Ecoli", gene_name)) %>%
  filter(!grepl("random", gene_name)) %>%
  filter(!grepl("Yeast", gene_name)) 

HEK_base_data_tRF$DM <- recode(HEK_base_data_tRF$DM, "-"="m", "+"="p")

#fix mitochondrial names
#Then rearrange all names
HEK_base_data_tRF$bin_start <- as.factor(HEK_base_data_tRF$bin_start)
HEK_base_data_tRF$bin_stop <- as.factor(HEK_base_data_tRF$bin_stop)

HEK_base_data_tRF <- HEK_base_data_tRF %>%
  filter(grepl("mt", gene_name)) %>%
  mutate(gene_name = paste0("Homo_sapiens_chrMT~trna1-z",gene_name)) %>%
  rbind(filter(HEK_base_data_tRF, !grepl("mt",gene_name))) %>%
  separate(gene_name, sep="_", c("homo","sapiens","gene_name"))%>%
  select(-homo, -sapiens) %>%
  separate(gene_name, sep="-", c("chrom","codon")) %>%
  mutate(gene_symbol = paste0(codon, "_",chrom)) %>%
  select(-chrom, -codon)

```


##His tRNAs
```{r}

# 
temp_trf <- HEK_base_data_tRF %>%
  mutate(gene_name=gene_symbol) %>%
  separate(gene_name, sep="_", c("amino_acid", "number")) %>%
  group_by(DM, stress, rep) %>%
  mutate(norm_pileup = pileup / sum(starts),
         norm_starts = starts / sum(starts)) %>%
  group_by(gene_symbol, DM, stress, position, bin_start, rep) %>%
  mutate(mean_norm_pileup = mean(norm_pileup),
         mean_norm_starts = mean(norm_starts)) %>%
  group_by()
#temp$bin_start <- recode(temp$bin_start, "0"="0-20", "20"="20-40", "40"="40-60", "60"="60+" )
#temp$bin_start <- as.factor(temp$bin_start)



Arg_fragment <- ggplot(filter(temp_trf, stress %in% c("None","AsO2", "heat", "H2O2"), 
              DM=="p", 
              grepl("ArgACG_chr14", gene_symbol), 
              !grepl("mt", gene_symbol, bin_start),
              #grepl("GlnCTG_chr6.trna131", gene_symbol),
              #bin_start=="20-40" ,
              pileup >=10
              ), 
       aes(x=position, y= mean_norm_starts*1000000, 
           color=stress, 
           group=interaction(rep, DM, stress, gene_symbol, bin_start)))+
  geom_line(alpha=0.8) +
  #geom_line(aes(y=mean_norm_starts), size=1) +
  #geom_vline(xintercept = c(31), alpha=0.2, linetype=2) +
  #geom_vline(xintercept = c(36), alpha=0.2, linetype=2) +
  theme(legend.position = "right") +
  labs(color="Stress", y="Read starts\nper million") +
  #scale_x_continuous(breaks=c(20,25,30,35,40)) +
  # facet_grid(rows=vars(bin_start), cols=vars(gene_symbol)) +
  # geom_hline(yintercept = c(0.00019,0.000145)) +
  # facet_grid(rows=vars(gene_symbol), cols=vars(DM)) +
  #facet_wrap(~gene_symbol) +
  #facet_grid(cols=vars(amino_acid), rows=vars(bin_start)) +
  coord_cartesian(xlim=c(25,38), ylim=c(0,200))
  #coord_cartesian(xlim=c(0,75), ylim=c(0,1))
  #labs(strip="Read starting\nposition", color="Stress", y="Read starts\nper million")
Arg_fragment

Arg_fig <- arrangeGrob(Arg_fragment, top=textGrob(expression(bold("ArgACG"))) )


setwd("~/4SR/data/20200317_Nature_methods_data/2020 02 24 ecoli figures/source/")
#Make individual nuclear or mito heat maps
ggsave("../../../../grants/2020 05 18 tRNA stress R01/figures/Arg_fig.png",
       plot = Arg_fig,
       scale = 1.4,
       width = 3,
       height = 2)

#Write data for Tao
#write.table(temp1,file='HisGTG_AsO2.txt', quote=FALSE, sep='\t', col.names = NA)

#LeuCAG

```

##His mutation rates
```{r}

#Get rid of unused sequences
HEK_base_data <- data_HEK %>%
  filter(gene_biotype == "tRNA") %>%
  filter(!grepl("spike", Gene)) %>%
  filter(!grepl("Ecoli", Gene)) %>%
  filter(!grepl("Yeast", Gene)) %>%
  select(Gene, pileup, position, mutation, gene_biotype, DM, stress, rep, base) %>%
  select(-gene_biotype)

HEK_base_data$DM <- recode(HEK_base_data$DM, "-"="m", "+"="p")

#fix mitochondrial names
#Then rearrange all names
HEK_base_data <- HEK_base_data %>%
  filter(grepl("mt", Gene)) %>%
  mutate(Gene = paste0("Homo_sapiens_chrMT~trna1-z",Gene)) %>%
  rbind(filter(HEK_base_data, !grepl("mt",Gene))) %>%
  separate(Gene, sep="_", c("homo","sapiens","gene_symbol"))%>%
  select(-homo, -sapiens) %>%
  separate(gene_symbol, sep="-", c("chrom","codon")) %>%
  mutate(gene_symbol = paste0(codon, "_",chrom)) %>%
  select(-chrom, -codon)

#fitler out eroniously long tRNA...
temp2 <- HEK_base_data %>%
  group_by(gene_symbol, DM, stress, rep) %>%
  summarize(position = max(position)) %>%
  filter(position <120) %>%
  select(gene_symbol)

#Apply filters to base data
HEK_base_data <- HEK_base_data %>%
  filter(#gene_symbol %in% temp1$gene_symbol,
         gene_symbol %in% temp2$gene_symbol)




```

```{r}

temp <- HEK_base_data %>%
  filter(pileup >=10)

ggplot(filter(temp, stress %in% c("None","AsO2", "heat", "H2O2"), DM=="m", 
              grepl("Trp", gene_symbol), 
              !grepl("mt", gene_symbol), 
              pileup>=10 ), 
       aes(x=position, y= mutation, color=stress, group=interaction(rep, DM, stress, gene_symbol)))+
  geom_line(alpha=0.8) +
  #geom_line(aes(y=mean_norm_starts), size=1) +
  geom_vline(xintercept = c(32), alpha=0.2, linetype=2) +
  geom_vline(xintercept = c(36), alpha=0.2, linetype=2) +
  theme(legend.position = "right") +
  #scale_x_continuous(breaks=c(20,25,30,35,40)) +
  # facet_grid(rows=vars(bin_start), cols=vars(gene_symbol)) +
  # geom_hline(yintercept = c(0.00019,0.000145)) +
  #facet_wrap(~amino_acid) +
  #facet_grid(cols=vars(amino_acid), rows=vars(bin_start)) +
  facet_wrap(~gene_symbol) +
  coord_cartesian( xlim=c(0,70), ylim=c(0,0.9)) +
  ylab("Mutation Rate") +
  labs(strip="Read starting\nposition", color="Stress")

# myplot1 <- arrangeGrob(tRF_His_starts+theme(legend.position="None"), top = textGrob(expression(bold("HisGTG"))  )) 
# myplot2 <- arrangeGrob(tRF_His_mutation+theme(legend.position="None"), left = textGrob(expression(bold()), x = unit(0, "npc")
#          , y = unit(1, "npc"), just=c("left","top")) )
# plt_legend <- get_legend(tRF_Glu_starts + theme( legend.background = element_blank() ) )
# lay = rbind(c(1,1,3),
#             c(2,2,3))
# tRF_Glu_fig <- grid.arrange(myplot1, myplot2, plt_legend, layout_matrix=lay) #, myplot2, layout_matrix = lay)
# tRF_Glu_fig


```

#_
##fragment mutation rate
```{r}

# 
temp <- HEK_base_data_tRF %>%
  mutate(gene_name=gene_symbol) %>%
  separate(gene_name, sep="_", c("amino_acid", "number")) %>%
  group_by(DM, stress, rep) %>%
  mutate(norm_pileup = pileup / sum(starts),
         norm_starts = starts / sum(starts)) %>%
  group_by(gene_symbol, DM, stress, position, bin_start) %>%
  mutate(mean_norm_pileup = mean(norm_pileup),
         mean_norm_starts = mean(norm_starts)) %>%
  group_by()
temp$bin_start <- recode(temp$bin_start, "0"="0-20", "20"="20-40", "40"="40-60", "60"="60+" )
temp$bin_start <- as.factor(temp$bin_start)


temp1 <- temp %>%
  #filter(grepl("Trp", gene_symbol)) %>%
  filter(bin_start %in% c("0-20","20-40", "0-20", "40-60", "60+"))

ggplot(filter(temp1, stress %in% c("None","AsO2"), grepl("SerCGA", gene_symbol), pileup>=5 ), aes(x=position, y= (mutation) / pileup, color=stress, group=interaction(rep, DM, stress, gene_symbol)))+
  geom_line(alpha=0.4) +
  #geom_line(aes(y=mean_norm_starts), size=1) +
  geom_vline(xintercept = c(24), alpha=0.2, linetype=2) +
  theme(legend.position = "right") +
  #scale_x_continuous(breaks=c(20,25,30,35,40)) +
  # facet_grid(rows=vars(bin_start), cols=vars(gene_symbol)) +
  # geom_hline(yintercept = c(0.00019,0.000145)) +
  # facet_wrap(~amino_acid) 
  facet_grid(rows=vars(amino_acid, bin_start), cols=vars(DM))


##Candidate sites by mutation+deletion
#AspGTC @24nt
#ArgACG, ArgCCG @25
#ArgTCT @35
#Lys CTT @35 and 18
#TyrGTA @20
#ValAAC @18
#ValCAC45


#GlnCTG @8
#mtGlnTTG during AsO2
#ThrCGT @27
#mtThrTGT during AsO2 stress
#mtSerGCT @27
#Proline fragments in AsO2 stress
#LeuAAG @30
#LeuCAG @30


##Candidates sites by stops
#Asp @37
#ArgACG, ArgCCG @37
#Arg TCT @10
#LysCCT @35
#LysTTT @18
#HisGTG @8
#HisGTG @37
#GluCTC @37
#GlnCTG @37





```

##fragment Glu start rate analysis
```{r}

temp <- HEK_base_data_tRF %>%
  mutate(gene_name=gene_symbol) %>%
  separate(gene_name, sep="_", c("amino_acid", "number")) %>%
  group_by(DM, stress, rep) %>%
  mutate(norm_pileup = pileup / sum(starts),
         norm_starts = starts / sum(starts)) %>%
  group_by(gene_symbol, DM, stress, position, bin_start) %>%
  mutate(mean_norm_pileup = mean(norm_pileup),
         mean_norm_starts = mean(norm_starts)) %>%
  group_by()
temp$bin_start <- recode(temp$bin_start, "0"="0-20", "20"="20-40", "40"="40-60", "60"="60+" )
temp$bin_start <- as.factor(temp$bin_start)


temp1 <- temp %>%
  #filter(grepl("Trp", gene_symbol)) %>%
  filter(bin_start %in% c("0-20","20-40", "0-20", "40-60", "60+"))

tRF_Glu_starts <- ggplot(filter(temp1, stress %in% c("None","AsO2", "heat", "H2O2"), DM=="p", 
                                grepl("GluCTC", gene_symbol), !grepl("mt", gene_symbol), pileup>=1,
                                bin_start=="20-40"), 
                         aes(x=position, y= mean_norm_starts*1000000, color=stress, group=interaction(rep, DM, stress, gene_symbol, bin_start)))+
  geom_line(alpha=0.7) +
  #geom_line(aes(y=mean_norm_starts), size=1) +
  geom_vline(xintercept = c(32), alpha=0.2, linetype=2) +
  geom_vline(xintercept = c(36), alpha=0.2, linetype=2) +
  theme(legend.position = "right") +
  #scale_x_continuous(breaks=c(20,25,30,35,40)) +
  # facet_grid(rows=vars(bin_start), cols=vars(gene_symbol)) +
  # geom_hline(yintercept = c(0.00019,0.000145)) +
  #facet_wrap(~amino_acid) +
  #facet_grid(cols=vars(amino_acid), rows=vars(bin_start)) +
  coord_cartesian( ylim=c(0,50), xlim=c(25,40)) +
  labs(strip="Read starting\nposition", color="Stress", y="Read starts\nper million")
tRF_Glu_starts

```

##Glu mutation rates
```{r}
# temp <- HEK_base_data %>%
#   mutate(gene_name=gene_symbol) %>%
#   separate(gene_name, sep="_", c("amino_acid", "number")) %>%
#   group_by(DM, stress, rep) %>%
#   group_by(gene_symbol, DM, stress, position) %>%
#   #group_by(gene_symbol, DM, stress, position, rep) %>%
#   #mutate(mutation = mutation / pileup) %>%
#   group_by()


tRF_Glu_mutation <- ggplot(filter(temp, stress %in% c("None","AsO2", "heat", "H2O2"), DM=="m", grepl("GluCTC", gene_symbol), !grepl("mt", gene_symbol), pileup>=10 ), 
       aes(x=position, y= mutation, color=stress, group=interaction(rep, DM, stress, gene_symbol)))+
  geom_line(alpha=0.7) +
  #geom_line(aes(y=mean_norm_starts), size=1) +
  geom_vline(xintercept = c(32), alpha=0.2, linetype=2) +
  geom_vline(xintercept = c(36), alpha=0.2, linetype=2) +
  theme(legend.position = "right") +
  #scale_x_continuous(breaks=c(20,25,30,35,40)) +
  # facet_grid(rows=vars(bin_start), cols=vars(gene_symbol)) +
  # geom_hline(yintercept = c(0.00019,0.000145)) +
  #facet_wrap(~amino_acid) +
  #facet_grid(cols=vars(amino_acid), rows=vars(bin_start)) +
  coord_cartesian( xlim=c(15,20), ylim=c(0,0.25)) +
  ylab("Mutation Rate") +
  labs(strip="Read starting\nposition", color="Stress")
tRF_Glu_mutation

myplot1 <- arrangeGrob(tRF_Glu_starts+theme(legend.position="right", axis.title.x = element_blank()), top=textGrob(expression(bold("GluCTC")))) 
myplot2 <- arrangeGrob(tRF_Glu_mutation+theme(legend.position="None"), left = textGrob(expression(bold()), x = unit(0, "npc")
         , y = unit(1, "npc"), just=c("left","top")) )
plt_legend <- get_legend(tRF_Glu_starts + theme( legend.background = element_blank() ) )
lay = rbind(c(1))
tRF_Glu_fig <- grid.arrange(myplot1, layout_matrix=lay) #, myplot2, layout_matrix = lay)
tRF_Glu_fig

#Make individual nuclear or mito heat maps
ggsave("../../../../grants/2020 05 18 tRNA stress R01/figures/tRF_Glu_fig.png",
       plot = tRF_Glu_fig,
       scale = 1.4,
       width = 3,
       height = 2)

```

#_
#_
#_




#Send Tao HEK count data to plot against microarray
```{r}
temp <- HEK_counts_data %>%
  filter(gene_biotype=="tRNA") %>%
  filter(stress=="None") %>%
  select(-type, -gene_biotype, -stress)

temp$gene_symbol <- str_replace_all(temp$gene_symbol, ">mt", "trnaMT-mt") 

temp <- temp %>%
  filter(!grepl(">", gene_symbol)) %>%
  separate(gene_symbol, sep="-", c("pos", "anticodon")) %>%
  group_by(anticodon, DM, rep) %>%
  summarise(counts = sum(counts)) %>%
  pivot_wider(names_from = c("rep", "DM"), values_from = counts)

wtemp <- temp 
#write.table(wtemp,file='HEK_abundance_no_stress.tsv', quote=FALSE, sep='\t', col.names = NA)

```


#_
#_
#_
#Stop rates
```{r, echo=FALSE}

#The good stuff
temp <- HEK_base_data %>%
  mutate(mutation = ifelse(pileup<10, NA, mutation)) %>%
  group_by(gene_symbol, position, DM, stress) %>%
  summarise(mutation = mean(mutation, na.rm=T),
            pileup= mean(pileup, na.rm=T)) %>%
  group_by()

#Split up mito etc
temp <- temp %>%
  mutate(source = "cytosolic")
temp <- temp %>%
  filter(grepl("z", gene_symbol)) %>%
  mutate(source = "mitochondrial") %>%
  rbind(filter(temp, !grepl("z", gene_symbol)))

#FILTER FOR MOST ABUNDANT, and Make x labels nicer to read
temp <- temp %>%
  separate(gene_symbol, sep="_", c("gene_symbol", "number")) %>%
  mutate(gene_name=paste0(gene_symbol,"_",number))
temp1 <- temp %>%
  filter(stress=="None", DM=="p") %>%
  group_by(stress, DM, gene_symbol) %>%
  mutate(max_pileup = max(pileup)) %>%
  filter(pileup == max_pileup) %>%
  group_by() %>%
  select(gene_name)
temp <- temp %>%
  filter(gene_name %in% temp1$gene_name)

#Scale pileup
temp <- temp %>%
  group_by(gene_name, source, stress, DM) %>%
  mutate(pileup = pileup / max(pileup))
#Fix "z" gene
temp$gene_symbol <- str_replace_all(temp$gene_symbol, "z", "") 



figX1 <- ggplot(filter(temp, #DM=="m",
                       stress=="None",
                       source=="cytosolic"),
                aes(x=gene_symbol, y=position, fill=pileup)) +
  geom_tile() +
  scale_y_reverse(breaks=c(9, 26, 37, 58), limits=c(89, 1) )+
  scale_fill_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="grey50")+
  theme(legend.position = "None",
        axis.text.x = element_text(angle = 90, hjust = 0, vjust=.5),
        axis.title.x = element_blank(),
        legend.text = element_text(angle=0, vjust=.5),
        plot.title = element_text(hjust = 0.5),
        axis.ticks.y = element_line(size=1),
        axis.ticks.length.y = unit(.2, "cm") )+ #element_text(hjust = 0.5)) +
  labs(#title="Mutation rate along tRNAs without demethylase treatment",
       fill="Normalize\npileup") +
  ylab("Position")  +
  labs(title="Nuclear-encoded") +
  theme(plot.title = element_text(hjust = 0) ) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")) +
  facet_grid(rows=vars(DM))
figX1


figX2 <- ggplot(filter(temp, #DM=="m",
                       stress=="None",
                       source=="mitochondrial"),
                aes(x=gene_symbol, y=position, fill=pileup)) +
  geom_tile() +
  scale_y_reverse(breaks=c(9, 26, 37, 58), limits=c(89, 0), position="right" )+
  scale_fill_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="grey50")+
  theme(legend.position = "right",
        axis.text.x = element_text(angle = 90, hjust = 0, vjust=.5),
        axis.title.x = element_blank(),
        legend.text = element_text(angle=0, vjust=.5),
        plot.title = element_text(hjust = 0.5),
        axis.ticks.y = element_line(size=1),
        axis.ticks.length.y = unit(.2, "cm") )+ #element_text(hjust = 0.5)) +
  labs(#title="Mutation rate along tRNAs without demethylase treatment",
       fill="Normalize\npileup") +
  ylab("Position")+
  labs(title="Mitochondrial-encoded") +
  theme(plot.title = element_text(hjust = 0) ) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")) +
  facet_grid(rows=vars(DM))
figX2


myplot1 <- arrangeGrob(figX1)
myplot2<- arrangeGrob(figX2)
lay = rbind(c(1,1,1,2,2))
figX <- grid.arrange(myplot1, myplot2, layout_matrix = lay)
figX



temp2 <- temp %>%
  filter(stress=="None") %>%
  group_by(gene_name, source, stress, position) %>%
  select(gene_name, gene_symbol, source, stress, position, pileup, DM) %>%
  group_by() %>%
  pivot_wider(names_from = DM, values_from = pileup) %>%
  mutate(difference = p - m)



figX1 <- ggplot(filter(temp2,
                       source=="cytosolic"),
                aes(x=gene_symbol, y=position, fill=difference)) +
  geom_tile() +
  scale_y_reverse(breaks=c(9, 26, 37, 58), limits=c(89, 1) )+
  scale_fill_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="grey50")+
  theme(legend.position = "None",
        axis.text.x = element_text(angle = 90, hjust = 0, vjust=.5),
        axis.title.x = element_blank(),
        legend.text = element_text(angle=0, vjust=.5),
        plot.title = element_text(hjust = 0.5),
        axis.ticks.y = element_line(size=1),
        axis.ticks.length.y = unit(.2, "cm") )+ #element_text(hjust = 0.5)) +
  labs(#title="Mutation rate along tRNAs without demethylase treatment",
       fill="Normalize\npileup") +
  ylab("Position")  +
  labs(title="Nuclear-encoded") +
  theme(plot.title = element_text(hjust = 0) ) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm"))+
  geom_hline(yintercept = 32, linetype=2, size=.2, alpha=0.8)

figX2 <- ggplot(filter(temp2,
                       source=="mitochondrial"),
                aes(x=gene_symbol, y=position, fill=difference)) +
  geom_tile() +
  scale_y_reverse(breaks=c(9, 26, 37, 58), limits=c(89, 0), position="right" )+
  scale_fill_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="grey50")+
  theme(legend.position = "right",
        axis.text.x = element_text(angle = 90, hjust = 0, vjust=.5),
        axis.title.x = element_blank(),
        legend.text = element_text(angle=0, vjust=.5),
        plot.title = element_text(hjust = 0.5),
        axis.ticks.y = element_line(size=1),
        axis.ticks.length.y = unit(.2, "cm") )+ #element_text(hjust = 0.5)) +
  labs(#title="Mutation rate along tRNAs without demethylase treatment",
       fill="Change in\nnormalized\npileup with DM") +
  ylab("Position")  +
  labs(title="Mitochondrial-encoded") +
  theme(plot.title = element_text(hjust = 0) ) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")) +
  geom_hline(yintercept = 32, linetype=2, size=.2, alpha=0.8)

myplot1 <- arrangeGrob(figX1)
myplot2<- arrangeGrob(figX2)
lay = rbind(c(1,1,1,2,2))
figX <- grid.arrange(myplot1, myplot2, layout_matrix = lay)
figX

DM_stops <- temp2 %>%
  filter(position == "34") %>%
  select(gene_symbol, stress, difference) 

```

```{r}

temp <- HEK_counts_data %>%
  filter(stress == "None") %>%
  group_by(rep, stress, DM, gene_biotype) %>%
  mutate(counts = counts/sum(counts)*1000000  ) %>%
  filter(gene_biotype %in% c("tRNA") )

temp <- temp %>%
  group_by(DM, stress, gene_symbol, gene_biotype) %>%
  filter(counts > 0)

temp <- temp %>%
  summarise(counts = mean(counts) )

temp$DM <- recode(temp$DM, "-"="minus", "+"="plus")

temp <- temp %>%
  filter(!grepl("Ecoli", gene_symbol)) %>%
  filter(!grepl("yeast", gene_symbol)) %>%
  filter(!grepl("spike", gene_symbol)) %>%
  group_by()

temp <- temp %>%
  pivot_wider(names_from = DM, values_from = counts) %>%
  filter(plus > -1, minus > -1)

r_squared = as.character(round(cor(temp$minus, temp$plus, method="spearman") **2,2))
temp$gene_symbol <- str_replace_all(temp$gene_symbol, ">mt", "MT-mt")

temp <- temp %>%
  separate(gene_symbol, sep="-", c("number", "gene_symbol"))


temp <- temp %>%
  full_join(., DM_stops, by=c("gene_symbol", "stress"))



figX3 <- ggplot(temp, aes(x=minus, y=plus, color=difference)) +
  geom_point() +
  geom_abline(linetype=2, slope=1, alpha=0.4) +
  scale_y_continuous(trans="log2", breaks=c(1,100,10000))+
  scale_x_continuous(trans="log2", breaks=c(1,100,10000))+
  theme(legend.position = "right",
        legend.background = element_rect(fill="transparent")) +
  ylab("Demethylase treated (RPM)")+
  xlab("Untreated (RPM)") +
  #labs(color="Transcript type") +
  coord_equal() +
  annotate(geom="text", label=bquote(~r^2==.(r_squared) ), x=5, y=10000) +
  theme(aspect.ratio = 1) +
  theme(plot.margin = unit(c(0.3, 0.3, 0.3, 0.3), "cm")) +
  scale_color_gradient2(low="blue", mid="grey70", midpoint=0, high="red", na.value="grey50") 
  #geom_point(data=filter(temp, gene_symbol %in% asdf$codon), color="blue")
  #geom_text_repel(data = filter(temp, temp$plus / temp$minus >7, temp$minus >5), aes(label=gene_symbol))


myplot1 <- arrangeGrob(figX1)
myplot2<- arrangeGrob(figX2)
myplot3<- arrangeGrob(figX3)
lay = rbind(c(1,1,1,2,2),
            c(3,3,3,3,3))
figX <- grid.arrange(myplot1, myplot2, myplot3, layout_matrix = lay)
figX

```

